{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Taking the coverage file from an alignment of all BS data to genome, will destrand."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2019-11-06T00:52:15.733763Z",
     "start_time": "2019-11-06T00:52:15.656154Z"
    }
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'/Users/sr320/Documents/GitHub/nb-2019/P_generosa'"
      ]
     },
     "execution_count": 1,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "pwd"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Pg_val_1_bismark_bt2_pe.deduplicated.bismark.cov\r\n"
     ]
    }
   ],
   "source": [
    "!ls /Volumes/Serine/wd/19-11-05/"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\r\n",
      "\r\n",
      "DESCRIPTION\r\n",
      "\r\n",
      "This script is supposed to convert a specified reference genome into two different bisulfite\r\n",
      "converted versions and index them for alignments with Bowtie 2 (default), or Bowtie 1. The first\r\n",
      "bisulfite genome will have all Cs converted to Ts (C->T), and the other one will have all Gs\r\n",
      "converted to As (G->A). Both bisulfite genomes will be stored in subfolders within the reference\r\n",
      "genome folder. Once the bisulfite conversion has been completed the program will fork and launch\r\n",
      "two simultaneous instances of the Bowtie 1 or 2 indexer (bowtie-build or bowtie2-build). Be aware\r\n",
      "that the indexing process can take up to several hours; this will mainly depend on genome size\r\n",
      "and system resources.\r\n",
      "\r\n",
      "\r\n",
      "\r\n",
      "The following is a brief description of command line options and arguments to control the\r\n",
      "Bismark Genome Preparation:\r\n",
      "\r\n",
      "\r\n",
      "USAGE: bismark_genome_preparation [options] <arguments>\r\n",
      "\r\n",
      "\r\n",
      "OPTIONS:\r\n",
      "\r\n",
      "--help/--man             Displays this help filea and exits.\r\n",
      "\r\n",
      "--version                Displays version information and exits.\r\n",
      "\r\n",
      "--verbose                Print verbose output for more details or debugging.\r\n",
      "\r\n",
      "--path_to_bowtie </../>  The full path to the Bowtie 1 or Bowtie 2 installation on your system\r\n",
      "                         (depending on which aligner/indexer you intend to use). Unless this path\r\n",
      "                         is specified it is assumed that Bowtie is in the PATH.\r\n",
      "\r\n",
      "--bowtie2                This will create bisulfite indexes for Bowtie 2. (Default: ON).\r\n",
      "\r\n",
      "--bowtie1                This will create bisulfite indexes for Bowtie 1. (Default: OFF).\r\n",
      "\r\n",
      "--single_fasta           Instruct the Bismark Indexer to write the converted genomes into\r\n",
      "                         single-entry FastA files instead of making one multi-FastA file (MFA)\r\n",
      "                         per chromosome. This might be useful if individual bisulfite converted\r\n",
      "                         chromosomes are needed (e.g. for debugging), however it can cause a\r\n",
      "                         problem with indexing if the number of chromosomes is vast (this is likely\r\n",
      "                         to be in the range of several thousand files; the operating system can\r\n",
      "                         only handle lists up to a certain length, and some newly assembled\r\n",
      "                         genomes may contain 20000-500000 contigs of scaffold files which do exceed\r\n",
      "                         this list length limit).\r\n",
      "\r\n",
      "--genomic_composition    Calculate and extract the genomic sequence composition for mono and di-nucleotides\r\n",
      "                         and write the genomic composition table 'genomic_nucleotide_frequencies.txt' to the\r\n",
      "                         genome folder. This may be useful later on when using bam2nuc or the Bismark option\r\n",
      "                         --nucleotide_coverage.\r\n",
      "\r\n",
      "\r\n",
      "ARGUMENTS:\r\n",
      "\r\n",
      "<path_to_genome_folder>  The path to the folder containing the genome to be bisulfite converted.\r\n",
      "                         The Bismark Genome Preparation expects one or more fastA files in the folder\r\n",
      "                         (with the file extension: .fa or .fasta (also ending in .gz)). Specifying this path is mandatory.\r\n",
      "\r\n",
      "\r\n",
      "This script was last modified on 26 April 2018.\r\n"
     ]
    }
   ],
   "source": [
    "!/Applications/bioinfo/Bismark_v0.20.0/bismark_genome_preparation -help"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Path to genome folder specified as: /Volumes/Serine/wd/19-11-05/\n",
      "Aligner to be used: Bowtie 2 (default)\n",
      "Writing bisulfite genomes out into a single MFA (multi FastA) file\n",
      "\n",
      "Bismark Genome Preparation - Step I: Preparing folders\n",
      "\n",
      "Path to Bowtie 2 specified: /Applications/bioinfo/bowtie2-2.3.4.1-macos-x86_64/\n",
      "Bisulfite Genome Indexer version v0.20.0 (last modified 26 April 2018)\n",
      "Created Bisulfite Genome folder /Volumes/Serine/wd/19-11-05/Bisulfite_Genome/\n",
      "Created Bisulfite Genome folder /Volumes/Serine/wd/19-11-05/Bisulfite_Genome/CT_conversion/\n",
      "Created Bisulfite Genome folder /Volumes/Serine/wd/19-11-05/Bisulfite_Genome/GA_conversion/\n",
      "\n",
      "Step I - Prepare genome folders - completed\n",
      "\n",
      "\n",
      "Bismark Genome Preparation - Step II: Bisulfite converting reference genome\n",
      "\n",
      "conversions performed:\n",
      "chromosome\tC->T\tG->A\n",
      "Scaffold_01\t12767429\t12765526\n",
      "Scaffold_02\t10088811\t10062918\n",
      "Scaffold_03\t8199278\t8194496\n",
      "Scaffold_04\t9161562\t9161987\n",
      "Scaffold_05\t9498414\t9497714\n",
      "Scaffold_06\t8458182\t8463483\n",
      "Scaffold_07\t6054340\t6047203\n",
      "Scaffold_08\t8517146\t8506733\n",
      "Scaffold_09\t5420328\t5445998\n",
      "Scaffold_10\t7547438\t7543051\n",
      "Scaffold_11\t7053984\t7054209\n",
      "Scaffold_12\t7145712\t7105651\n",
      "Scaffold_13\t6273846\t6290785\n",
      "Scaffold_14\t6515251\t6527291\n",
      "Scaffold_15\t6625508\t6624214\n",
      "Scaffold_16\t4528254\t4550625\n",
      "Scaffold_17\t4780809\t4787183\n",
      "Scaffold_18\t3756408\t3758031\n",
      "\n",
      "Total number of conversions performed:\n",
      "C->T:\t132392700\n",
      "G->A:\t132387098\n",
      "\n",
      "Step II - Genome bisulfite conversions - completed\n",
      "\n",
      "\n",
      "Bismark Genome Preparation - Step III: Launching the Bowtie 2 indexer\n",
      "Please be aware that this process can - depending on genome size - take several hours!\n",
      "\n",
      "Preparing indexing of CT converted genome in /Volumes/Serine/wd/19-11-05/Bisulfite_Genome/CT_conversion/\n",
      "Parent process: Starting to index C->T converted genome with the following command:\n",
      "\n",
      "/Applications/bioinfo/bowtie2-2.3.4.1-macos-x86_64/bowtie2-build -f genome_mfa.CT_conversion.fa BS_CT\n",
      "\n",
      "Settings:\n",
      "  Output files: \"BS_CT.*.bt2\"\n",
      "  Line rate: 6 (line is 64 bytes)\n",
      "  Lines per side: 1 (side is 64 bytes)\n",
      "  Offset rate: 4 (one in 16)\n",
      "  FTable chars: 10\n",
      "  Strings: unpacked\n",
      "  Max bucket size: default\n",
      "  Max bucket size, sqrt multiplier: default\n",
      "  Max bucket size, len divisor: 4\n",
      "  Difference-cover sample period: 1024\n",
      "  Endianness: little\n",
      "  Actual local endianness: little\n",
      "  Sanity checking: disabled\n",
      "  Assertions: disabled\n",
      "  Random seed: 0\n",
      "  Sizeofs: void*:8, int:4, long:8, size_t:8\n",
      "Input files DNA, FASTA:\n",
      "  genome_mfa.CT_conversion.fa\n",
      "Building a SMALL index\n",
      "Reading reference sizes\n",
      "Preparing indexing of GA converted genome in /Volumes/Serine/wd/19-11-05/Bisulfite_Genome/GA_conversion/\n",
      "Child process: Starting to index G->A converted genome with the following command:\n",
      "\n",
      "/Applications/bioinfo/bowtie2-2.3.4.1-macos-x86_64/bowtie2-build -f genome_mfa.GA_conversion.fa BS_GA\n",
      "\n",
      "Settings:\n",
      "  Output files: \"BS_GA.*.bt2\"\n",
      "  Line rate: 6 (line is 64 bytes)\n",
      "  Lines per side: 1 (side is 64 bytes)\n",
      "  Offset rate: 4 (one in 16)\n",
      "  FTable chars: 10\n",
      "  Strings: unpacked\n",
      "  Max bucket size: default\n",
      "  Max bucket size, sqrt multiplier: default\n",
      "  Max bucket size, len divisor: 4\n",
      "  Difference-cover sample period: 1024\n",
      "  Endianness: little\n",
      "  Actual local endianness: little\n",
      "  Sanity checking: disabled\n",
      "  Assertions: disabled\n",
      "  Random seed: 0\n",
      "  Sizeofs: void*:8, int:4, long:8, size_t:8\n",
      "Input files DNA, FASTA:\n",
      "  genome_mfa.GA_conversion.fa\n",
      "Building a SMALL index\n",
      "Reading reference sizes\n",
      "  Time reading reference sizes: 00:00:08\n",
      "Calculating joined length\n",
      "Writing header\n",
      "Reserving space for joined string\n",
      "Joining reference sequences\n",
      "  Time reading reference sizes: 00:00:08\n",
      "Calculating joined length\n",
      "Writing header\n",
      "Reserving space for joined string\n",
      "Joining reference sequences\n",
      "  Time to join reference sequences: 00:00:06\n",
      "bmax according to bmaxDivN setting: 195965290\n",
      "Using parameters --bmax 146973968 --dcv 1024\n",
      "  Doing ahead-of-time memory usage test\n",
      "  Passed!  Constructing with these parameters: --bmax 146973968 --dcv 1024\n",
      "Constructing suffix-array element generator\n",
      "Building DifferenceCoverSample\n",
      "  Building sPrime\n",
      "  Building sPrimeOrder\n",
      "  V-Sorting samples\n",
      "  Time to join reference sequences: 00:00:06\n",
      "bmax according to bmaxDivN setting: 195965290\n",
      "Using parameters --bmax 146973968 --dcv 1024\n",
      "  Doing ahead-of-time memory usage test\n",
      "  Passed!  Constructing with these parameters: --bmax 146973968 --dcv 1024\n",
      "Constructing suffix-array element generator\n",
      "Building DifferenceCoverSample\n",
      "  Building sPrime\n",
      "  Building sPrimeOrder\n",
      "  V-Sorting samples\n",
      "  V-Sorting samples time: 00:00:18\n",
      "  Allocating rank array\n",
      "  Ranking v-sort output\n",
      "  V-Sorting samples time: 00:00:18\n",
      "  Allocating rank array\n",
      "  Ranking v-sort output\n",
      "  Ranking v-sort output time: 00:00:06\n",
      "  Invoking Larsson-Sadakane on ranks\n",
      "  Ranking v-sort output time: 00:00:05\n",
      "  Invoking Larsson-Sadakane on ranks\n",
      "  Invoking Larsson-Sadakane on ranks time: 00:00:08\n",
      "  Sanity-checking and returning\n",
      "Building samples\n",
      "Reserving space for 12 sample suffixes\n",
      "Generating random suffixes\n",
      "QSorting 12 sample offsets, eliminating duplicates\n",
      "QSorting sample offsets, eliminating duplicates time: 00:00:00\n",
      "Multikey QSorting 12 samples\n",
      "  (Using difference cover)\n",
      "  Multikey QSorting samples time: 00:00:00\n",
      "Calculating bucket sizes\n",
      "Splitting and merging\n",
      "  Splitting and merging time: 00:00:00\n",
      "Avg bucket size: 7.83861e+08 (target: 146973967)\n",
      "Converting suffix-array elements to index image\n",
      "Allocating ftab, absorbFtab\n",
      "Entering Ebwt loop\n",
      "Getting block 1 of 1\n",
      "  No samples; assembling all-inclusive block\n",
      "  Invoking Larsson-Sadakane on ranks time: 00:00:08\n",
      "  Sanity-checking and returning\n",
      "Building samples\n",
      "Reserving space for 12 sample suffixes\n",
      "Generating random suffixes\n",
      "QSorting 12 sample offsets, eliminating duplicates\n",
      "QSorting sample offsets, eliminating duplicates time: 00:00:00\n",
      "Multikey QSorting 12 samples\n",
      "  (Using difference cover)\n",
      "  Multikey QSorting samples time: 00:00:00\n",
      "Calculating bucket sizes\n",
      "Splitting and merging\n",
      "  Splitting and merging time: 00:00:00\n",
      "Avg bucket size: 7.83861e+08 (target: 146973967)\n",
      "Converting suffix-array elements to index image\n",
      "Allocating ftab, absorbFtab\n",
      "Entering Ebwt loop\n",
      "Getting block 1 of 1\n",
      "  No samples; assembling all-inclusive block\n",
      "  Sorting block of length 783861161 for bucket 1\n",
      "  (Using difference cover)\n",
      "  Sorting block of length 783861161 for bucket 1\n",
      "  (Using difference cover)\n",
      "  Sorting block time: 00:07:31\n",
      "Returning block of 783861162 for bucket 1\n",
      "  Sorting block time: 00:09:24\n",
      "Returning block of 783861162 for bucket 1\n",
      "Exited Ebwt loop\n",
      "fchr[A]: 0\n",
      "fchr[C]: 391954798\n",
      "fchr[G]: 524347498\n",
      "fchr[T]: 524347498\n",
      "fchr[$]: 783861161\n",
      "Exiting Ebwt::buildToDisk()\n",
      "Returning from initFromVector\n",
      "Wrote 266840223 bytes to primary EBWT file: BS_GA.1.bt2\n",
      "Wrote 195965296 bytes to secondary EBWT file: BS_GA.2.bt2\n",
      "Re-opening _in1 and _in2 as input streams\n",
      "Returning from Ebwt constructor\n",
      "Headers:\n",
      "    len: 783861161\n",
      "    bwtLen: 783861162\n",
      "    sz: 195965291\n",
      "    bwtSz: 195965291\n",
      "    lineRate: 6\n",
      "    offRate: 4\n",
      "    offMask: 0xfffffff0\n",
      "    ftabChars: 10\n",
      "    eftabLen: 20\n",
      "    eftabSz: 80\n",
      "    ftabLen: 1048577\n",
      "    ftabSz: 4194308\n",
      "    offsLen: 48991323\n",
      "    offsSz: 195965292\n",
      "    lineSz: 64\n",
      "    sideSz: 64\n",
      "    sideBwtSz: 48\n",
      "    sideBwtLen: 192\n",
      "    numSides: 4082611\n",
      "    numLines: 4082611\n",
      "    ebwtTotLen: 261287104\n",
      "    ebwtTotSz: 261287104\n",
      "    color: 0\n",
      "    reverse: 0\n",
      "Total time for call to driver() for forward index: 00:10:18\n",
      "Reading reference sizes\n",
      "  Time reading reference sizes: 00:00:07\n",
      "Calculating joined length\n",
      "Writing header\n",
      "Reserving space for joined string\n",
      "Joining reference sequences\n",
      "  Time to join reference sequences: 00:00:06\n",
      "  Time to reverse reference sequence: 00:00:00\n",
      "bmax according to bmaxDivN setting: 195965290\n",
      "Using parameters --bmax 146973968 --dcv 1024\n",
      "  Doing ahead-of-time memory usage test\n",
      "  Passed!  Constructing with these parameters: --bmax 146973968 --dcv 1024\n",
      "Constructing suffix-array element generator\n",
      "Building DifferenceCoverSample\n",
      "  Building sPrime\n",
      "  Building sPrimeOrder\n",
      "  V-Sorting samples\n",
      "  V-Sorting samples time: 00:00:17\n",
      "  Allocating rank array\n",
      "  Ranking v-sort output\n",
      "  Ranking v-sort output time: 00:00:05\n",
      "  Invoking Larsson-Sadakane on ranks\n",
      "  Invoking Larsson-Sadakane on ranks time: 00:00:08\n",
      "  Sanity-checking and returning\n",
      "Building samples\n",
      "Reserving space for 12 sample suffixes\n",
      "Generating random suffixes\n",
      "QSorting 12 sample offsets, eliminating duplicates\n",
      "QSorting sample offsets, eliminating duplicates time: 00:00:00\n",
      "Multikey QSorting 12 samples\n",
      "  (Using difference cover)\n",
      "  Multikey QSorting samples time: 00:00:00\n",
      "Calculating bucket sizes\n",
      "Splitting and merging\n",
      "  Splitting and merging time: 00:00:00\n",
      "Avg bucket size: 7.83861e+08 (target: 146973967)\n",
      "Converting suffix-array elements to index image\n",
      "Allocating ftab, absorbFtab\n",
      "Entering Ebwt loop\n",
      "Getting block 1 of 1\n",
      "  No samples; assembling all-inclusive block\n",
      "  Sorting block of length 783861161 for bucket 1\n",
      "  (Using difference cover)\n",
      "Exited Ebwt loop\n",
      "fchr[A]: 0\n",
      "fchr[C]: 259567700\n",
      "fchr[G]: 259567700\n",
      "fchr[T]: 391954798\n",
      "fchr[$]: 783861161\n",
      "Exiting Ebwt::buildToDisk()\n",
      "Returning from initFromVector\n",
      "Wrote 266840223 bytes to primary EBWT file: BS_CT.1.bt2\n",
      "Wrote 195965296 bytes to secondary EBWT file: BS_CT.2.bt2\n",
      "Re-opening _in1 and _in2 as input streams\n",
      "Returning from Ebwt constructor\n",
      "Headers:\n",
      "    len: 783861161\n",
      "    bwtLen: 783861162\n",
      "    sz: 195965291\n",
      "    bwtSz: 195965291\n",
      "    lineRate: 6\n",
      "    offRate: 4\n",
      "    offMask: 0xfffffff0\n",
      "    ftabChars: 10\n",
      "    eftabLen: 20\n",
      "    eftabSz: 80\n",
      "    ftabLen: 1048577\n",
      "    ftabSz: 4194308\n",
      "    offsLen: 48991323\n",
      "    offsSz: 195965292\n",
      "    lineSz: 64\n",
      "    sideSz: 64\n",
      "    sideBwtSz: 48\n",
      "    sideBwtLen: 192\n",
      "    numSides: 4082611\n",
      "    numLines: 4082611\n",
      "    ebwtTotLen: 261287104\n",
      "    ebwtTotSz: 261287104\n",
      "    color: 0\n",
      "    reverse: 0\n",
      "Total time for call to driver() for forward index: 00:12:15\n",
      "Reading reference sizes\n",
      "  Time reading reference sizes: 00:00:07\n",
      "Calculating joined length\n",
      "Writing header\n",
      "Reserving space for joined string\n",
      "Joining reference sequences\n",
      "  Time to join reference sequences: 00:00:06\n",
      "  Time to reverse reference sequence: 00:00:01\n",
      "bmax according to bmaxDivN setting: 195965290\n",
      "Using parameters --bmax 146973968 --dcv 1024\n",
      "  Doing ahead-of-time memory usage test\n",
      "  Passed!  Constructing with these parameters: --bmax 146973968 --dcv 1024\n",
      "Constructing suffix-array element generator\n",
      "Building DifferenceCoverSample\n",
      "  Building sPrime\n",
      "  Building sPrimeOrder\n",
      "  V-Sorting samples\n",
      "  V-Sorting samples time: 00:00:18\n",
      "  Allocating rank array\n",
      "  Ranking v-sort output\n",
      "  Ranking v-sort output time: 00:00:06\n",
      "  Invoking Larsson-Sadakane on ranks\n",
      "  Invoking Larsson-Sadakane on ranks time: 00:00:08\n",
      "  Sanity-checking and returning\n",
      "Building samples\n",
      "Reserving space for 12 sample suffixes\n",
      "Generating random suffixes\n",
      "QSorting 12 sample offsets, eliminating duplicates\n",
      "QSorting sample offsets, eliminating duplicates time: 00:00:00\n",
      "Multikey QSorting 12 samples\n",
      "  (Using difference cover)\n",
      "  Multikey QSorting samples time: 00:00:00\n",
      "Calculating bucket sizes\n",
      "Splitting and merging\n",
      "  Splitting and merging time: 00:00:00\n",
      "Avg bucket size: 7.83861e+08 (target: 146973967)\n",
      "Converting suffix-array elements to index image\n",
      "Allocating ftab, absorbFtab\n",
      "Entering Ebwt loop\n",
      "Getting block 1 of 1\n",
      "  No samples; assembling all-inclusive block\n",
      "  Sorting block of length 783861161 for bucket 1\n",
      "  (Using difference cover)\n",
      "  Sorting block time: 00:07:37\n",
      "Returning block of 783861162 for bucket 1\n",
      "Exited Ebwt loop\n",
      "fchr[A]: 0\n",
      "fchr[C]: 391954798\n",
      "fchr[G]: 524347498\n",
      "fchr[T]: 524347498\n",
      "fchr[$]: 783861161\n",
      "Exiting Ebwt::buildToDisk()\n",
      "Returning from initFromVector\n",
      "Wrote 266840223 bytes to primary EBWT file: BS_GA.rev.1.bt2\n",
      "Wrote 195965296 bytes to secondary EBWT file: BS_GA.rev.2.bt2\n",
      "Re-opening _in1 and _in2 as input streams\n",
      "Returning from Ebwt constructor\n",
      "Headers:\n",
      "    len: 783861161\n",
      "    bwtLen: 783861162\n",
      "    sz: 195965291\n",
      "    bwtSz: 195965291\n",
      "    lineRate: 6\n",
      "    offRate: 4\n",
      "    offMask: 0xfffffff0\n",
      "    ftabChars: 10\n",
      "    eftabLen: 20\n",
      "    eftabSz: 80\n",
      "    ftabLen: 1048577\n",
      "    ftabSz: 4194308\n",
      "    offsLen: 48991323\n",
      "    offsSz: 195965292\n",
      "    lineSz: 64\n",
      "    sideSz: 64\n",
      "    sideBwtSz: 48\n",
      "    sideBwtLen: 192\n",
      "    numSides: 4082611\n",
      "    numLines: 4082611\n",
      "    ebwtTotLen: 261287104\n",
      "    ebwtTotSz: 261287104\n",
      "    color: 0\n",
      "    reverse: 1\n",
      "Total time for backward call to driver() for mirror index: 00:10:23\n",
      "  Sorting block time: 00:09:13\n",
      "Returning block of 783861162 for bucket 1\n",
      "Exited Ebwt loop\n",
      "fchr[A]: 0\n",
      "fchr[C]: 259567700\n",
      "fchr[G]: 259567700\n",
      "fchr[T]: 391954798\n",
      "fchr[$]: 783861161\n",
      "Exiting Ebwt::buildToDisk()\n",
      "Returning from initFromVector\n",
      "Wrote 266840223 bytes to primary EBWT file: BS_CT.rev.1.bt2\n",
      "Wrote 195965296 bytes to secondary EBWT file: BS_CT.rev.2.bt2\n",
      "Re-opening _in1 and _in2 as input streams\n",
      "Returning from Ebwt constructor\n",
      "Headers:\n",
      "    len: 783861161\n",
      "    bwtLen: 783861162\n",
      "    sz: 195965291\n",
      "    bwtSz: 195965291\n",
      "    lineRate: 6\n",
      "    offRate: 4\n",
      "    offMask: 0xfffffff0\n",
      "    ftabChars: 10\n",
      "    eftabLen: 20\n",
      "    eftabSz: 80\n",
      "    ftabLen: 1048577\n",
      "    ftabSz: 4194308\n",
      "    offsLen: 48991323\n",
      "    offsSz: 195965292\n",
      "    lineSz: 64\n",
      "    sideSz: 64\n",
      "    sideBwtSz: 48\n",
      "    sideBwtLen: 192\n",
      "    numSides: 4082611\n",
      "    numLines: 4082611\n",
      "    ebwtTotLen: 261287104\n",
      "    ebwtTotSz: 261287104\n",
      "    color: 0\n",
      "    reverse: 1\n",
      "Total time for backward call to driver() for mirror index: 00:11:49\n"
     ]
    }
   ],
   "source": [
    "!/Applications/bioinfo/Bismark_v0.20.0/bismark_genome_preparation \\\n",
    "--verbose \\\n",
    "--path_to_bowtie /Applications/bioinfo/bowtie2-2.3.4.1-macos-x86_64/ \\\n",
    "/Volumes/Serine/wd/19-11-05/"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Path to genome folder specified as: /Volumes/Serine/wd/19-11-05/v074/\n",
      "Aligner to be used: Bowtie 2 (default)\n",
      "Writing bisulfite genomes out into a single MFA (multi FastA) file\n",
      "\n",
      "Bismark Genome Preparation - Step I: Preparing folders\n",
      "\n",
      "Path to Bowtie 2 specified: /Applications/bioinfo/bowtie2-2.3.4.1-macos-x86_64/\n",
      "Bisulfite Genome Indexer version v0.20.0 (last modified 26 April 2018)\n",
      "Created Bisulfite Genome folder /Volumes/Serine/wd/19-11-05/v074/Bisulfite_Genome/\n",
      "Created Bisulfite Genome folder /Volumes/Serine/wd/19-11-05/v074/Bisulfite_Genome/CT_conversion/\n",
      "Created Bisulfite Genome folder /Volumes/Serine/wd/19-11-05/v074/Bisulfite_Genome/GA_conversion/\n",
      "\n",
      "Step I - Prepare genome folders - completed\n",
      "\n",
      "\n",
      "Bismark Genome Preparation - Step II: Bisulfite converting reference genome\n",
      "\n",
      "conversions performed:\n",
      "chromosome\tC->T\tG->A\n",
      "PGA_scaffold1__77_contigs__length_89643857\t12767429\t12765526\n",
      "PGA_scaffold2__36_contigs__length_69596280\t10088811\t10062918\n",
      "PGA_scaffold3__111_contigs__length_57743597\t8199278\t8194496\n",
      "PGA_scaffold4__129_contigs__length_65288255\t9161562\t9161987\n",
      "PGA_scaffold5__109_contigs__length_67248332\t9498414\t9497714\n",
      "PGA_scaffold6__104_contigs__length_61759565\t8458182\t8463483\n",
      "PGA_scaffold7__69_contigs__length_43120122\t6054340\t6047203\n",
      "PGA_scaffold8__63_contigs__length_61151155\t8517146\t8506733\n",
      "PGA_scaffold9__45_contigs__length_38581958\t5420328\t5445998\n",
      "PGA_scaffold10__49_contigs__length_53961475\t7547438\t7543051\n",
      "PGA_scaffold11__79_contigs__length_51449921\t7053984\t7054209\n",
      "PGA_scaffold12__71_contigs__length_50438331\t7145712\t7105651\n",
      "PGA_scaffold13__52_contigs__length_44396874\t6273846\t6290785\n",
      "PGA_scaffold14__91_contigs__length_45393038\t6515251\t6527291\n",
      "PGA_scaffold15__101_contigs__length_47938513\t6625508\t6624214\n",
      "PGA_scaffold16__33_contigs__length_31980953\t4528254\t4550625\n",
      "PGA_scaffold17__51_contigs__length_34923512\t4780809\t4787183\n",
      "PGA_scaffold18__69_contigs__length_27737463\t3756408\t3758031\n",
      "\n",
      "Total number of conversions performed:\n",
      "C->T:\t132392700\n",
      "G->A:\t132387098\n",
      "\n",
      "Step II - Genome bisulfite conversions - completed\n",
      "\n",
      "\n",
      "Bismark Genome Preparation - Step III: Launching the Bowtie 2 indexer\n",
      "Please be aware that this process can - depending on genome size - take several hours!\n",
      "\n",
      "Preparing indexing of CT converted genome in /Volumes/Serine/wd/19-11-05/v074/Bisulfite_Genome/CT_conversion/\n",
      "Parent process: Starting to index C->T converted genome with the following command:\n",
      "\n",
      "/Applications/bioinfo/bowtie2-2.3.4.1-macos-x86_64/bowtie2-build -f genome_mfa.CT_conversion.fa BS_CT\n",
      "\n",
      "Settings:\n",
      "  Output files: \"BS_CT.*.bt2\"\n",
      "  Line rate: 6 (line is 64 bytes)\n",
      "  Lines per side: 1 (side is 64 bytes)\n",
      "  Offset rate: 4 (one in 16)\n",
      "  FTable chars: 10\n",
      "  Strings: unpacked\n",
      "  Max bucket size: default\n",
      "  Max bucket size, sqrt multiplier: default\n",
      "  Max bucket size, len divisor: 4\n",
      "  Difference-cover sample period: 1024\n",
      "  Endianness: little\n",
      "  Actual local endianness: little\n",
      "  Sanity checking: disabled\n",
      "  Assertions: disabled\n",
      "  Random seed: 0\n",
      "  Sizeofs: void*:8, int:4, long:8, size_t:8\n",
      "Input files DNA, FASTA:\n",
      "  genome_mfa.CT_conversion.fa\n",
      "Building a SMALL index\n",
      "Reading reference sizes\n",
      "Preparing indexing of GA converted genome in /Volumes/Serine/wd/19-11-05/v074/Bisulfite_Genome/GA_conversion/\n",
      "Child process: Starting to index G->A converted genome with the following command:\n",
      "\n",
      "/Applications/bioinfo/bowtie2-2.3.4.1-macos-x86_64/bowtie2-build -f genome_mfa.GA_conversion.fa BS_GA\n",
      "\n",
      "Settings:\n",
      "  Output files: \"BS_GA.*.bt2\"\n",
      "  Line rate: 6 (line is 64 bytes)\n",
      "  Lines per side: 1 (side is 64 bytes)\n",
      "  Offset rate: 4 (one in 16)\n",
      "  FTable chars: 10\n",
      "  Strings: unpacked\n",
      "  Max bucket size: default\n",
      "  Max bucket size, sqrt multiplier: default\n",
      "  Max bucket size, len divisor: 4\n",
      "  Difference-cover sample period: 1024\n",
      "  Endianness: little\n",
      "  Actual local endianness: little\n",
      "  Sanity checking: disabled\n",
      "  Assertions: disabled\n",
      "  Random seed: 0\n",
      "  Sizeofs: void*:8, int:4, long:8, size_t:8\n",
      "Input files DNA, FASTA:\n",
      "  genome_mfa.GA_conversion.fa\n",
      "Building a SMALL index\n",
      "Reading reference sizes\n",
      "  Time reading reference sizes: 00:00:08\n",
      "Calculating joined length\n",
      "Writing header\n",
      "Reserving space for joined string\n",
      "Joining reference sequences\n",
      "  Time reading reference sizes: 00:00:08\n",
      "Calculating joined length\n",
      "Writing header\n",
      "Reserving space for joined string\n",
      "Joining reference sequences\n",
      "  Time to join reference sequences: 00:00:06\n",
      "bmax according to bmaxDivN setting: 195965290\n",
      "Using parameters --bmax 146973968 --dcv 1024\n",
      "  Doing ahead-of-time memory usage test\n",
      "  Passed!  Constructing with these parameters: --bmax 146973968 --dcv 1024\n",
      "Constructing suffix-array element generator\n",
      "Building DifferenceCoverSample\n",
      "  Building sPrime\n",
      "  Building sPrimeOrder\n",
      "  V-Sorting samples\n",
      "  Time to join reference sequences: 00:00:06\n",
      "bmax according to bmaxDivN setting: 195965290\n",
      "Using parameters --bmax 146973968 --dcv 1024\n",
      "  Doing ahead-of-time memory usage test\n",
      "  Passed!  Constructing with these parameters: --bmax 146973968 --dcv 1024\n",
      "Constructing suffix-array element generator\n",
      "Building DifferenceCoverSample\n",
      "  Building sPrime\n",
      "  Building sPrimeOrder\n",
      "  V-Sorting samples\n",
      "  V-Sorting samples time: 00:00:19\n",
      "  Allocating rank array\n",
      "  Ranking v-sort output\n",
      "  V-Sorting samples time: 00:00:18\n",
      "  Allocating rank array\n",
      "  Ranking v-sort output\n",
      "  Ranking v-sort output time: 00:00:05\n",
      "  Invoking Larsson-Sadakane on ranks\n",
      "  Ranking v-sort output time: 00:00:06\n",
      "  Invoking Larsson-Sadakane on ranks\n",
      "  Invoking Larsson-Sadakane on ranks time: 00:00:08\n",
      "  Sanity-checking and returning\n",
      "Building samples\n",
      "Reserving space for 12 sample suffixes\n",
      "Generating random suffixes\n",
      "QSorting 12 sample offsets, eliminating duplicates\n",
      "QSorting sample offsets, eliminating duplicates time: 00:00:00\n",
      "Multikey QSorting 12 samples\n",
      "  (Using difference cover)\n",
      "  Multikey QSorting samples time: 00:00:00\n",
      "Calculating bucket sizes\n",
      "Splitting and merging\n",
      "  Splitting and merging time: 00:00:00\n",
      "Avg bucket size: 7.83861e+08 (target: 146973967)\n",
      "Converting suffix-array elements to index image\n",
      "Allocating ftab, absorbFtab\n",
      "Entering Ebwt loop\n",
      "Getting block 1 of 1\n",
      "  No samples; assembling all-inclusive block\n",
      "  Invoking Larsson-Sadakane on ranks time: 00:00:08\n",
      "  Sanity-checking and returning\n",
      "Building samples\n",
      "Reserving space for 12 sample suffixes\n",
      "Generating random suffixes\n",
      "QSorting 12 sample offsets, eliminating duplicates\n",
      "QSorting sample offsets, eliminating duplicates time: 00:00:00\n",
      "Multikey QSorting 12 samples\n",
      "  (Using difference cover)\n",
      "  Multikey QSorting samples time: 00:00:00\n",
      "Calculating bucket sizes\n",
      "Splitting and merging\n",
      "  Splitting and merging time: 00:00:00\n",
      "Avg bucket size: 7.83861e+08 (target: 146973967)\n",
      "Converting suffix-array elements to index image\n",
      "Allocating ftab, absorbFtab\n",
      "Entering Ebwt loop\n",
      "Getting block 1 of 1\n",
      "  No samples; assembling all-inclusive block\n",
      "  Sorting block of length 783861161 for bucket 1\n",
      "  (Using difference cover)\n",
      "  Sorting block of length 783861161 for bucket 1\n",
      "  (Using difference cover)\n",
      "  Sorting block time: 00:07:32\n",
      "Returning block of 783861162 for bucket 1\n",
      "  Sorting block time: 00:09:31\n",
      "Returning block of 783861162 for bucket 1\n",
      "Exited Ebwt loop\n",
      "fchr[A]: 0\n",
      "fchr[C]: 391954798\n",
      "fchr[G]: 524347498\n",
      "fchr[T]: 524347498\n",
      "fchr[$]: 783861161\n",
      "Exiting Ebwt::buildToDisk()\n",
      "Returning from initFromVector\n",
      "Wrote 266840795 bytes to primary EBWT file: BS_GA.1.bt2\n",
      "Wrote 195965296 bytes to secondary EBWT file: BS_GA.2.bt2\n",
      "Re-opening _in1 and _in2 as input streams\n",
      "Returning from Ebwt constructor\n",
      "Headers:\n",
      "    len: 783861161\n",
      "    bwtLen: 783861162\n",
      "    sz: 195965291\n",
      "    bwtSz: 195965291\n",
      "    lineRate: 6\n",
      "    offRate: 4\n",
      "    offMask: 0xfffffff0\n",
      "    ftabChars: 10\n",
      "    eftabLen: 20\n",
      "    eftabSz: 80\n",
      "    ftabLen: 1048577\n",
      "    ftabSz: 4194308\n",
      "    offsLen: 48991323\n",
      "    offsSz: 195965292\n",
      "    lineSz: 64\n",
      "    sideSz: 64\n",
      "    sideBwtSz: 48\n",
      "    sideBwtLen: 192\n",
      "    numSides: 4082611\n",
      "    numLines: 4082611\n",
      "    ebwtTotLen: 261287104\n",
      "    ebwtTotSz: 261287104\n",
      "    color: 0\n",
      "    reverse: 0\n",
      "Total time for call to driver() for forward index: 00:10:25\n",
      "Reading reference sizes\n",
      "  Time reading reference sizes: 00:00:06\n",
      "Calculating joined length\n",
      "Writing header\n",
      "Reserving space for joined string\n",
      "Joining reference sequences\n",
      "  Time to join reference sequences: 00:00:06\n",
      "  Time to reverse reference sequence: 00:00:01\n",
      "bmax according to bmaxDivN setting: 195965290\n",
      "Using parameters --bmax 146973968 --dcv 1024\n",
      "  Doing ahead-of-time memory usage test\n",
      "  Passed!  Constructing with these parameters: --bmax 146973968 --dcv 1024\n",
      "Constructing suffix-array element generator\n",
      "Building DifferenceCoverSample\n",
      "  Building sPrime\n",
      "  Building sPrimeOrder\n",
      "  V-Sorting samples\n",
      "  V-Sorting samples time: 00:00:17\n",
      "  Allocating rank array\n",
      "  Ranking v-sort output\n",
      "  Ranking v-sort output time: 00:00:06\n",
      "  Invoking Larsson-Sadakane on ranks\n",
      "  Invoking Larsson-Sadakane on ranks time: 00:00:07\n",
      "  Sanity-checking and returning\n",
      "Building samples\n",
      "Reserving space for 12 sample suffixes\n",
      "Generating random suffixes\n",
      "QSorting 12 sample offsets, eliminating duplicates\n",
      "QSorting sample offsets, eliminating duplicates time: 00:00:00\n",
      "Multikey QSorting 12 samples\n",
      "  (Using difference cover)\n",
      "  Multikey QSorting samples time: 00:00:00\n",
      "Calculating bucket sizes\n",
      "Splitting and merging\n",
      "  Splitting and merging time: 00:00:00\n",
      "Avg bucket size: 7.83861e+08 (target: 146973967)\n",
      "Converting suffix-array elements to index image\n",
      "Allocating ftab, absorbFtab\n",
      "Entering Ebwt loop\n",
      "Getting block 1 of 1\n",
      "  No samples; assembling all-inclusive block\n",
      "  Sorting block of length 783861161 for bucket 1\n",
      "  (Using difference cover)\n",
      "Exited Ebwt loop\n",
      "fchr[A]: 0\n",
      "fchr[C]: 259567700\n",
      "fchr[G]: 259567700\n",
      "fchr[T]: 391954798\n",
      "fchr[$]: 783861161\n",
      "Exiting Ebwt::buildToDisk()\n",
      "Returning from initFromVector\n",
      "Wrote 266840795 bytes to primary EBWT file: BS_CT.1.bt2\n",
      "Wrote 195965296 bytes to secondary EBWT file: BS_CT.2.bt2\n",
      "Re-opening _in1 and _in2 as input streams\n",
      "Returning from Ebwt constructor\n",
      "Headers:\n",
      "    len: 783861161\n",
      "    bwtLen: 783861162\n",
      "    sz: 195965291\n",
      "    bwtSz: 195965291\n",
      "    lineRate: 6\n",
      "    offRate: 4\n",
      "    offMask: 0xfffffff0\n",
      "    ftabChars: 10\n",
      "    eftabLen: 20\n",
      "    eftabSz: 80\n",
      "    ftabLen: 1048577\n",
      "    ftabSz: 4194308\n",
      "    offsLen: 48991323\n",
      "    offsSz: 195965292\n",
      "    lineSz: 64\n",
      "    sideSz: 64\n",
      "    sideBwtSz: 48\n",
      "    sideBwtLen: 192\n",
      "    numSides: 4082611\n",
      "    numLines: 4082611\n",
      "    ebwtTotLen: 261287104\n",
      "    ebwtTotSz: 261287104\n",
      "    color: 0\n",
      "    reverse: 0\n",
      "Total time for call to driver() for forward index: 00:12:24\n",
      "Reading reference sizes\n",
      "  Time reading reference sizes: 00:00:06\n",
      "Calculating joined length\n",
      "Writing header\n",
      "Reserving space for joined string\n",
      "Joining reference sequences\n",
      "  Time to join reference sequences: 00:00:06\n",
      "  Time to reverse reference sequence: 00:00:00\n",
      "bmax according to bmaxDivN setting: 195965290\n",
      "Using parameters --bmax 146973968 --dcv 1024\n",
      "  Doing ahead-of-time memory usage test\n",
      "  Passed!  Constructing with these parameters: --bmax 146973968 --dcv 1024\n",
      "Constructing suffix-array element generator\n",
      "Building DifferenceCoverSample\n",
      "  Building sPrime\n",
      "  Building sPrimeOrder\n",
      "  V-Sorting samples\n",
      "  V-Sorting samples time: 00:00:17\n",
      "  Allocating rank array\n",
      "  Ranking v-sort output\n",
      "  Ranking v-sort output time: 00:00:06\n",
      "  Invoking Larsson-Sadakane on ranks\n",
      "  Invoking Larsson-Sadakane on ranks time: 00:00:07\n",
      "  Sanity-checking and returning\n",
      "Building samples\n",
      "Reserving space for 12 sample suffixes\n",
      "Generating random suffixes\n",
      "QSorting 12 sample offsets, eliminating duplicates\n",
      "QSorting sample offsets, eliminating duplicates time: 00:00:00\n",
      "Multikey QSorting 12 samples\n",
      "  (Using difference cover)\n",
      "  Multikey QSorting samples time: 00:00:00\n",
      "Calculating bucket sizes\n",
      "Splitting and merging\n",
      "  Splitting and merging time: 00:00:00\n",
      "Avg bucket size: 7.83861e+08 (target: 146973967)\n",
      "Converting suffix-array elements to index image\n",
      "Allocating ftab, absorbFtab\n",
      "Entering Ebwt loop\n",
      "Getting block 1 of 1\n",
      "  No samples; assembling all-inclusive block\n",
      "  Sorting block of length 783861161 for bucket 1\n",
      "  (Using difference cover)\n",
      "  Sorting block time: 00:07:37\n",
      "Returning block of 783861162 for bucket 1\n",
      "Exited Ebwt loop\n",
      "fchr[A]: 0\n",
      "fchr[C]: 391954798\n",
      "fchr[G]: 524347498\n",
      "fchr[T]: 524347498\n",
      "fchr[$]: 783861161\n",
      "Exiting Ebwt::buildToDisk()\n",
      "Returning from initFromVector\n",
      "Wrote 266840795 bytes to primary EBWT file: BS_GA.rev.1.bt2\n",
      "Wrote 195965296 bytes to secondary EBWT file: BS_GA.rev.2.bt2\n",
      "Re-opening _in1 and _in2 as input streams\n",
      "Returning from Ebwt constructor\n",
      "Headers:\n",
      "    len: 783861161\n",
      "    bwtLen: 783861162\n",
      "    sz: 195965291\n",
      "    bwtSz: 195965291\n",
      "    lineRate: 6\n",
      "    offRate: 4\n",
      "    offMask: 0xfffffff0\n",
      "    ftabChars: 10\n",
      "    eftabLen: 20\n",
      "    eftabSz: 80\n",
      "    ftabLen: 1048577\n",
      "    ftabSz: 4194308\n",
      "    offsLen: 48991323\n",
      "    offsSz: 195965292\n",
      "    lineSz: 64\n",
      "    sideSz: 64\n",
      "    sideBwtSz: 48\n",
      "    sideBwtLen: 192\n",
      "    numSides: 4082611\n",
      "    numLines: 4082611\n",
      "    ebwtTotLen: 261287104\n",
      "    ebwtTotSz: 261287104\n",
      "    color: 0\n",
      "    reverse: 1\n",
      "Total time for backward call to driver() for mirror index: 00:10:22\n",
      "  Sorting block time: 00:09:32\n",
      "Returning block of 783861162 for bucket 1\n",
      "Exited Ebwt loop\n",
      "fchr[A]: 0\n",
      "fchr[C]: 259567700\n",
      "fchr[G]: 259567700\n",
      "fchr[T]: 391954798\n",
      "fchr[$]: 783861161\n",
      "Exiting Ebwt::buildToDisk()\n",
      "Returning from initFromVector\n",
      "Wrote 266840795 bytes to primary EBWT file: BS_CT.rev.1.bt2\n",
      "Wrote 195965296 bytes to secondary EBWT file: BS_CT.rev.2.bt2\n",
      "Re-opening _in1 and _in2 as input streams\n",
      "Returning from Ebwt constructor\n",
      "Headers:\n",
      "    len: 783861161\n",
      "    bwtLen: 783861162\n",
      "    sz: 195965291\n",
      "    bwtSz: 195965291\n",
      "    lineRate: 6\n",
      "    offRate: 4\n",
      "    offMask: 0xfffffff0\n",
      "    ftabChars: 10\n",
      "    eftabLen: 20\n",
      "    eftabSz: 80\n",
      "    ftabLen: 1048577\n",
      "    ftabSz: 4194308\n",
      "    offsLen: 48991323\n",
      "    offsSz: 195965292\n",
      "    lineSz: 64\n",
      "    sideSz: 64\n",
      "    sideBwtSz: 48\n",
      "    sideBwtLen: 192\n",
      "    numSides: 4082611\n",
      "    numLines: 4082611\n",
      "    ebwtTotLen: 261287104\n",
      "    ebwtTotSz: 261287104\n",
      "    color: 0\n",
      "    reverse: 1\n",
      "Total time for backward call to driver() for mirror index: 00:12:01\n"
     ]
    }
   ],
   "source": [
    "!/Applications/bioinfo/Bismark_v0.20.0/bismark_genome_preparation \\\n",
    "--verbose \\\n",
    "--path_to_bowtie /Applications/bioinfo/bowtie2-2.3.4.1-macos-x86_64/ \\\n",
    "/Volumes/Serine/wd/19-11-05/v074/"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Summary of parameters for genome-wide cytosine report:\n",
      "==============================================================================\n",
      "Coverage infile:\t\t\t\t/Volumes/Serine/wd/19-11-05/Pg_val_1_bismark_bt2_pe.deduplicated.bismark.cov\n",
      "Output directory:\t\t\t\t><\n",
      "Parent directory:\t\t\t\t>/Users/sr320/git-repos/nb-2019/P_generosa/<\n",
      "Genome directory:\t\t\t\t>/Volumes/Serine/wd/19-11-05/genome/<\n",
      "CX context:\t\t\t\t\tno (CpG context only, default)\n",
      "Pooling CpG top/bottom strand evidence:\t\tyes\n",
      "Genome coordinates used:\t\t\t1-based (default)\n",
      "GZIP compression:\t\t\t\tno\n",
      "Split by chromosome:\t\t\t\tno\n",
      "\n",
      "\n",
      "Now reading in and storing sequence information of the genome specified in: /Volumes/Serine/wd/19-11-05/genome/\n",
      "\n",
      "chr Scaffold_01 (89643857 bp)\n",
      "chr Scaffold_02 (69596280 bp)\n",
      "chr Scaffold_03 (57743597 bp)\n",
      "chr Scaffold_04 (65288255 bp)\n",
      "chr Scaffold_05 (67248332 bp)\n",
      "chr Scaffold_06 (61759565 bp)\n",
      "chr Scaffold_07 (43120122 bp)\n",
      "chr Scaffold_08 (61151155 bp)\n",
      "chr Scaffold_09 (38581958 bp)\n",
      "chr Scaffold_10 (53961475 bp)\n",
      "chr Scaffold_11 (51449921 bp)\n",
      "chr Scaffold_12 (50438331 bp)\n",
      "chr Scaffold_13 (44396874 bp)\n",
      "chr Scaffold_14 (45393038 bp)\n",
      "chr Scaffold_15 (47938513 bp)\n",
      "chr Scaffold_16 (31980953 bp)\n",
      "chr Scaffold_17 (34923512 bp)\n",
      "chr Scaffold_18 (27737463 bp)\n",
      "\n",
      "Stored sequence information of 18 chromosomes/scaffolds in total\n",
      "\n",
      "==============================================================================\n",
      "Methylation information will now be written into a genome-wide cytosine report\n",
      "==============================================================================\n",
      "\n",
      ">>> Writing genome-wide cytosine report to: /Volumes/Serine/wd/19-11-05/PgALL.CpG_report.txt <<<\n",
      "\n",
      "Storing all covered cytosine positions for chromosome: PGA_scaffold1__77_contigs__length_89643857\n",
      "Writing cytosine report for chromosome PGA_scaffold1__77_contigs__length_89643857 (stored 2673624 different covered positions)\n",
      "Use of uninitialized value within %chromosomes in pattern match (m//) at /Applications/bioinfo/Bismark_v0.20.0/coverage2cytosine line 194, <IN> line 2673625.\n",
      "Writing cytosine report for chromosome PGA_scaffold2__36_contigs__length_69596280 (stored 2102612 different covered positions)\n",
      "Use of uninitialized value within %chromosomes in pattern match (m//) at /Applications/bioinfo/Bismark_v0.20.0/coverage2cytosine line 194, <IN> line 4776237.\n",
      "Writing cytosine report for chromosome PGA_scaffold3__111_contigs__length_57743597 (stored 1720926 different covered positions)\n",
      "Use of uninitialized value within %chromosomes in pattern match (m//) at /Applications/bioinfo/Bismark_v0.20.0/coverage2cytosine line 194, <IN> line 6497163.\n",
      "Writing cytosine report for chromosome PGA_scaffold4__129_contigs__length_65288255 (stored 2016884 different covered positions)\n",
      "Use of uninitialized value within %chromosomes in pattern match (m//) at /Applications/bioinfo/Bismark_v0.20.0/coverage2cytosine line 194, <IN> line 8514047.\n",
      "Writing cytosine report for chromosome PGA_scaffold5__109_contigs__length_67248332 (stored 1946117 different covered positions)\n",
      "Use of uninitialized value within %chromosomes in pattern match (m//) at /Applications/bioinfo/Bismark_v0.20.0/coverage2cytosine line 194, <IN> line 10460164.\n",
      "Writing cytosine report for chromosome PGA_scaffold6__104_contigs__length_61759565 (stored 1765092 different covered positions)\n",
      "Use of uninitialized value within %chromosomes in pattern match (m//) at /Applications/bioinfo/Bismark_v0.20.0/coverage2cytosine line 194, <IN> line 12225256.\n",
      "Writing cytosine report for chromosome PGA_scaffold7__69_contigs__length_43120122 (stored 1288185 different covered positions)\n",
      "Use of uninitialized value within %chromosomes in pattern match (m//) at /Applications/bioinfo/Bismark_v0.20.0/coverage2cytosine line 194, <IN> line 13513441.\n",
      "Writing cytosine report for chromosome PGA_scaffold8__63_contigs__length_61151155 (stored 1780075 different covered positions)\n",
      "Use of uninitialized value within %chromosomes in pattern match (m//) at /Applications/bioinfo/Bismark_v0.20.0/coverage2cytosine line 194, <IN> line 15293516.\n",
      "Writing cytosine report for chromosome PGA_scaffold9__45_contigs__length_38581958 (stored 1136392 different covered positions)\n",
      "Use of uninitialized value within %chromosomes in pattern match (m//) at /Applications/bioinfo/Bismark_v0.20.0/coverage2cytosine line 194, <IN> line 16429908.\n",
      "Writing cytosine report for chromosome PGA_scaffold10__49_contigs__length_53961475 (stored 1498339 different covered positions)\n",
      "Use of uninitialized value within %chromosomes in pattern match (m//) at /Applications/bioinfo/Bismark_v0.20.0/coverage2cytosine line 194, <IN> line 17928247.\n",
      "Writing cytosine report for chromosome PGA_scaffold11__79_contigs__length_51449921 (stored 1532901 different covered positions)\n",
      "Use of uninitialized value within %chromosomes in pattern match (m//) at /Applications/bioinfo/Bismark_v0.20.0/coverage2cytosine line 194, <IN> line 19461148.\n",
      "Writing cytosine report for chromosome PGA_scaffold12__71_contigs__length_50438331 (stored 1529835 different covered positions)\n",
      "Use of uninitialized value within %chromosomes in pattern match (m//) at /Applications/bioinfo/Bismark_v0.20.0/coverage2cytosine line 194, <IN> line 20990983.\n",
      "Writing cytosine report for chromosome PGA_scaffold13__52_contigs__length_44396874 (stored 1285005 different covered positions)\n",
      "Use of uninitialized value within %chromosomes in pattern match (m//) at /Applications/bioinfo/Bismark_v0.20.0/coverage2cytosine line 194, <IN> line 22275988.\n",
      "Writing cytosine report for chromosome PGA_scaffold14__91_contigs__length_45393038 (stored 1503871 different covered positions)\n",
      "Use of uninitialized value within %chromosomes in pattern match (m//) at /Applications/bioinfo/Bismark_v0.20.0/coverage2cytosine line 194, <IN> line 23779859.\n",
      "Writing cytosine report for chromosome PGA_scaffold15__101_contigs__length_47938513 (stored 1391061 different covered positions)\n",
      "Use of uninitialized value within %chromosomes in pattern match (m//) at /Applications/bioinfo/Bismark_v0.20.0/coverage2cytosine line 194, <IN> line 25170920.\n",
      "Writing cytosine report for chromosome PGA_scaffold16__33_contigs__length_31980953 (stored 1020688 different covered positions)\n",
      "Use of uninitialized value within %chromosomes in pattern match (m//) at /Applications/bioinfo/Bismark_v0.20.0/coverage2cytosine line 194, <IN> line 26191608.\n",
      "Writing cytosine report for chromosome PGA_scaffold17__51_contigs__length_34923512 (stored 1071550 different covered positions)\n",
      "Use of uninitialized value within %chromosomes in pattern match (m//) at /Applications/bioinfo/Bismark_v0.20.0/coverage2cytosine line 194, <IN> line 27263158.\n",
      "Writing cytosine report for last chromosome PGA_scaffold18__69_contigs__length_27737463 (stored 818843 different covered positions)\n",
      "Use of uninitialized value within %chromosomes in pattern match (m//) at /Applications/bioinfo/Bismark_v0.20.0/coverage2cytosine line 413, <IN> line 28082000.\n",
      "Finished writing out cytosine report for covered chromosomes (processed 18 chromosomes/scaffolds in total)\n",
      "\n",
      "Now processing chromosomes that were not covered by any methylation calls in the coverage file...\n",
      "Writing cytosine report for not covered chromosome Scaffold_01\n",
      "Writing cytosine report for not covered chromosome Scaffold_02\n",
      "Writing cytosine report for not covered chromosome Scaffold_03\n",
      "Writing cytosine report for not covered chromosome Scaffold_04\n",
      "Writing cytosine report for not covered chromosome Scaffold_05\n",
      "Writing cytosine report for not covered chromosome Scaffold_06\n",
      "Writing cytosine report for not covered chromosome Scaffold_07\n",
      "Writing cytosine report for not covered chromosome Scaffold_08\n",
      "Writing cytosine report for not covered chromosome Scaffold_09\n",
      "Writing cytosine report for not covered chromosome Scaffold_10\n",
      "Writing cytosine report for not covered chromosome Scaffold_11\n",
      "Writing cytosine report for not covered chromosome Scaffold_12\n",
      "Writing cytosine report for not covered chromosome Scaffold_13\n",
      "Writing cytosine report for not covered chromosome Scaffold_14\n",
      "Writing cytosine report for not covered chromosome Scaffold_15\n",
      "Writing cytosine report for not covered chromosome Scaffold_16\n",
      "Writing cytosine report for not covered chromosome Scaffold_17\n",
      "Writing cytosine report for not covered chromosome Scaffold_18\n",
      "Finished writing out cytosine report (processed 36 chromosomes/scaffolds in total). coverage2cytosine processing complete.\n",
      "\n",
      "Now merging top and bottom strand CpGs into a single CG dinucleotide entity (reading from file >>/Volumes/Serine/wd/19-11-05/PgALL.CpG_report.txt<<, in output directory '')\n",
      ">>> Writing a new coverage file with top and bottom strand CpG methylation evidence merged to /Volumes/Serine/wd/19-11-05/PgALL.CpG_report.merged_CpG_evidence.cov <<<\n",
      "\n",
      "CpG merging complete. Good luck finding DMRs with bsseq!\n",
      "\n"
     ]
    }
   ],
   "source": [
    "!/Applications/bioinfo/Bismark_v0.20.0/coverage2cytosine \\\n",
    "--genome_folder /Volumes/Serine/wd/19-11-05/genome/ \\\n",
    "-o /Volumes/Serine/wd/19-11-05/PgALL \\\n",
    "--merge_CpG \\\n",
    "/Volumes/Serine/wd/19-11-05/Pg_val_1_bismark_bt2_pe.deduplicated.bismark.cov"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "total 6055968\r\n",
      "-rw-r--r--  1 sr320  staff     0B Nov  5 20:18 PgALL.CpG_report.merged_CpG_evidence.cov\r\n",
      "-rw-r--r--  1 sr320  staff   1.0G Nov  5 20:18 PgALL.CpG_report.txt\r\n",
      "-rw-r--r--@ 1 sr320  staff   1.9G Nov  5 19:10 Pg_val_1_bismark_bt2_pe.deduplicated.bismark.cov\r\n",
      "drwxr-xr-x  5 sr320  staff   170B Nov  5 20:07 \u001b[34mgenome\u001b[m\u001b[m\r\n",
      "drwxr-xr-x  2 sr320  staff    68B Nov  5 20:09 \u001b[34mv074\u001b[m\u001b[m\r\n"
     ]
    }
   ],
   "source": [
    "!ls -lh /Volumes/Serine/wd/19-11-05/ "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Scaffold_01\t4\t+\t0\t0\tCG\tCGC\r\n",
      "Scaffold_01\t5\t-\t0\t0\tCG\tCGC\r\n",
      "Scaffold_01\t18\t+\t0\t0\tCG\tCGG\r\n",
      "Scaffold_01\t19\t-\t0\t0\tCG\tCGG\r\n",
      "Scaffold_01\t54\t+\t0\t0\tCG\tCGT\r\n",
      "Scaffold_01\t55\t-\t0\t0\tCG\tCGG\r\n",
      "Scaffold_01\t72\t+\t0\t0\tCG\tCGA\r\n",
      "Scaffold_01\t73\t-\t0\t0\tCG\tCGC\r\n",
      "Scaffold_01\t96\t+\t0\t0\tCG\tCGC\r\n",
      "Scaffold_01\t97\t-\t0\t0\tCG\tCGG\r\n"
     ]
    }
   ],
   "source": [
    "!head /Volumes/Serine/wd/19-11-05/PgALL.CpG_report.txt\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Summary of parameters for genome-wide cytosine report:\n",
      "==============================================================================\n",
      "Coverage infile:\t\t\t\t/Volumes/Serine/wd/19-11-05/Pg_val_1_bismark_bt2_pe.deduplicated.bismark.cov\n",
      "Output directory:\t\t\t\t><\n",
      "Parent directory:\t\t\t\t>/Users/sr320/git-repos/nb-2019/P_generosa/<\n",
      "Genome directory:\t\t\t\t>/Volumes/Serine/wd/19-11-05/v074/<\n",
      "CX context:\t\t\t\t\tno (CpG context only, default)\n",
      "Pooling CpG top/bottom strand evidence:\t\tyes\n",
      "Genome coordinates used:\t\t\t1-based (default)\n",
      "GZIP compression:\t\t\t\tno\n",
      "Split by chromosome:\t\t\t\tno\n",
      "\n",
      "\n",
      "Now reading in and storing sequence information of the genome specified in: /Volumes/Serine/wd/19-11-05/v074/\n",
      "\n",
      "chr PGA_scaffold1__77_contigs__length_89643857 (89643857 bp)\n",
      "chr PGA_scaffold2__36_contigs__length_69596280 (69596280 bp)\n",
      "chr PGA_scaffold3__111_contigs__length_57743597 (57743597 bp)\n",
      "chr PGA_scaffold4__129_contigs__length_65288255 (65288255 bp)\n",
      "chr PGA_scaffold5__109_contigs__length_67248332 (67248332 bp)\n",
      "chr PGA_scaffold6__104_contigs__length_61759565 (61759565 bp)\n",
      "chr PGA_scaffold7__69_contigs__length_43120122 (43120122 bp)\n",
      "chr PGA_scaffold8__63_contigs__length_61151155 (61151155 bp)\n",
      "chr PGA_scaffold9__45_contigs__length_38581958 (38581958 bp)\n",
      "chr PGA_scaffold10__49_contigs__length_53961475 (53961475 bp)\n",
      "chr PGA_scaffold11__79_contigs__length_51449921 (51449921 bp)\n",
      "chr PGA_scaffold12__71_contigs__length_50438331 (50438331 bp)\n",
      "chr PGA_scaffold13__52_contigs__length_44396874 (44396874 bp)\n",
      "chr PGA_scaffold14__91_contigs__length_45393038 (45393038 bp)\n",
      "chr PGA_scaffold15__101_contigs__length_47938513 (47938513 bp)\n",
      "chr PGA_scaffold16__33_contigs__length_31980953 (31980953 bp)\n",
      "chr PGA_scaffold17__51_contigs__length_34923512 (34923512 bp)\n",
      "chr PGA_scaffold18__69_contigs__length_27737463 (27737463 bp)\n",
      "\n",
      "Stored sequence information of 18 chromosomes/scaffolds in total\n",
      "\n",
      "==============================================================================\n",
      "Methylation information will now be written into a genome-wide cytosine report\n",
      "==============================================================================\n",
      "\n",
      ">>> Writing genome-wide cytosine report to: /Volumes/Serine/wd/19-11-05/PgALL.74.CpG_report.txt <<<\n",
      "\n",
      "Storing all covered cytosine positions for chromosome: PGA_scaffold1__77_contigs__length_89643857\n",
      "Writing cytosine report for chromosome PGA_scaffold1__77_contigs__length_89643857 (stored 2673624 different covered positions)\n",
      "Writing cytosine report for chromosome PGA_scaffold2__36_contigs__length_69596280 (stored 2102612 different covered positions)\n",
      "Writing cytosine report for chromosome PGA_scaffold3__111_contigs__length_57743597 (stored 1720926 different covered positions)\n",
      "Writing cytosine report for chromosome PGA_scaffold4__129_contigs__length_65288255 (stored 2016884 different covered positions)\n",
      "Writing cytosine report for chromosome PGA_scaffold5__109_contigs__length_67248332 (stored 1946117 different covered positions)\n",
      "Writing cytosine report for chromosome PGA_scaffold6__104_contigs__length_61759565 (stored 1765092 different covered positions)\n",
      "Writing cytosine report for chromosome PGA_scaffold7__69_contigs__length_43120122 (stored 1288185 different covered positions)\n",
      "Writing cytosine report for chromosome PGA_scaffold8__63_contigs__length_61151155 (stored 1780075 different covered positions)\n",
      "Writing cytosine report for chromosome PGA_scaffold9__45_contigs__length_38581958 (stored 1136392 different covered positions)\n",
      "Writing cytosine report for chromosome PGA_scaffold10__49_contigs__length_53961475 (stored 1498339 different covered positions)\n",
      "Writing cytosine report for chromosome PGA_scaffold11__79_contigs__length_51449921 (stored 1532901 different covered positions)\n",
      "Writing cytosine report for chromosome PGA_scaffold12__71_contigs__length_50438331 (stored 1529835 different covered positions)\n",
      "Writing cytosine report for chromosome PGA_scaffold13__52_contigs__length_44396874 (stored 1285005 different covered positions)\n",
      "Writing cytosine report for chromosome PGA_scaffold14__91_contigs__length_45393038 (stored 1503871 different covered positions)\n",
      "Writing cytosine report for chromosome PGA_scaffold15__101_contigs__length_47938513 (stored 1391061 different covered positions)\n",
      "Writing cytosine report for chromosome PGA_scaffold16__33_contigs__length_31980953 (stored 1020688 different covered positions)\n",
      "Writing cytosine report for chromosome PGA_scaffold17__51_contigs__length_34923512 (stored 1071550 different covered positions)\n",
      "Writing cytosine report for last chromosome PGA_scaffold18__69_contigs__length_27737463 (stored 818843 different covered positions)\n",
      "Finished writing out cytosine report for covered chromosomes (processed 18 chromosomes/scaffolds in total)\n",
      "\n",
      "Now processing chromosomes that were not covered by any methylation calls in the coverage file...\n",
      "All chromosomes in the genome were covered by at least some reads. coverage2cytosine processing complete.\n",
      "\n",
      "Now merging top and bottom strand CpGs into a single CG dinucleotide entity (reading from file >>/Volumes/Serine/wd/19-11-05/PgALL.74.CpG_report.txt<<, in output directory '')\n",
      ">>> Writing a new coverage file with top and bottom strand CpG methylation evidence merged to /Volumes/Serine/wd/19-11-05/PgALL.74.CpG_report.merged_CpG_evidence.cov <<<\n",
      "\n",
      "CpG merging complete. Good luck finding DMRs with bsseq!\n",
      "\n"
     ]
    }
   ],
   "source": [
    "!/Applications/bioinfo/Bismark_v0.20.0/coverage2cytosine \\\n",
    "--genome_folder /Volumes/Serine/wd/19-11-05/v074 \\\n",
    "-o /Volumes/Serine/wd/19-11-05/PgALL.74 \\\n",
    "--merge_CpG \\\n",
    "/Volumes/Serine/wd/19-11-05/Pg_val_1_bismark_bt2_pe.deduplicated.bismark.cov"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "total 12338208\r\n",
      "-rw-r--r--  1 sr320  staff   1.1G Nov  5 21:00 PgALL.74.CpG_report.merged_CpG_evidence.cov\r\n",
      "-rw-r--r--  1 sr320  staff   1.9G Nov  5 20:58 PgALL.74.CpG_report.txt\r\n",
      "-rw-r--r--  1 sr320  staff     0B Nov  5 20:18 PgALL.CpG_report.merged_CpG_evidence.cov\r\n",
      "-rw-r--r--  1 sr320  staff   1.0G Nov  5 20:18 PgALL.CpG_report.txt\r\n",
      "-rw-r--r--@ 1 sr320  staff   1.9G Nov  5 19:10 Pg_val_1_bismark_bt2_pe.deduplicated.bismark.cov\r\n",
      "drwxr-xr-x  5 sr320  staff   170B Nov  5 20:07 \u001b[34mgenome\u001b[m\u001b[m\r\n",
      "drwxr-xr-x  4 sr320  staff   136B Nov  5 20:20 \u001b[34mv074\u001b[m\u001b[m\r\n"
     ]
    }
   ],
   "source": [
    "!ls -lh /Volumes/Serine/wd/19-11-05/ "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PGA_scaffold1__77_contigs__length_89643857\t4\t5\t50.000000\t1\t1\r\n",
      "PGA_scaffold1__77_contigs__length_89643857\t18\t19\t11.111111\t1\t8\r\n",
      "PGA_scaffold1__77_contigs__length_89643857\t54\t55\t3.125000\t1\t31\r\n",
      "PGA_scaffold1__77_contigs__length_89643857\t72\t73\t2.777778\t1\t35\r\n",
      "PGA_scaffold1__77_contigs__length_89643857\t96\t97\t2.040816\t1\t48\r\n",
      "PGA_scaffold1__77_contigs__length_89643857\t119\t120\t0.000000\t0\t48\r\n",
      "PGA_scaffold1__77_contigs__length_89643857\t193\t194\t0.000000\t0\t43\r\n",
      "PGA_scaffold1__77_contigs__length_89643857\t202\t203\t0.000000\t0\t48\r\n",
      "PGA_scaffold1__77_contigs__length_89643857\t209\t210\t0.000000\t0\t54\r\n",
      "PGA_scaffold1__77_contigs__length_89643857\t213\t214\t0.000000\t0\t57\r\n"
     ]
    }
   ],
   "source": [
    "!head /Volumes/Serine/wd/19-11-05/PgALL.74.CpG_report.merged_CpG_evidence.cov"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PGA_scaffold1__77_contigs__length_89643857 54 55 3.125000 1 31\r\n",
      "PGA_scaffold1__77_contigs__length_89643857 72 73 2.777778 1 35\r\n",
      "PGA_scaffold1__77_contigs__length_89643857 96 97 2.040816 1 48\r\n",
      "PGA_scaffold1__77_contigs__length_89643857 119 120 0.000000 0 48\r\n",
      "PGA_scaffold1__77_contigs__length_89643857 193 194 0.000000 0 43\r\n",
      "PGA_scaffold1__77_contigs__length_89643857 202 203 0.000000 0 48\r\n",
      "PGA_scaffold1__77_contigs__length_89643857 209 210 0.000000 0 54\r\n",
      "PGA_scaffold1__77_contigs__length_89643857 213 214 0.000000 0 57\r\n",
      "PGA_scaffold1__77_contigs__length_89643857 221 222 0.000000 0 62\r\n",
      "PGA_scaffold1__77_contigs__length_89643857 238 239 1.315789 1 75\r\n",
      "awk: write error on /dev/stdout\r\n",
      " input record number 605, file \r\n",
      " source line number 1\r\n",
      "cat: stdout: Broken pipe\r\n"
     ]
    }
   ],
   "source": [
    "!cat /Volumes/Serine/wd/19-11-05/PgALL.74.CpG_report.merged_CpG_evidence.cov \\\n",
    "| awk '{if ($5+$6 >= 10) { print $1, $2, $3, $4, $5, $6 }}' | head"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PGA_scaffold10__49_contigs__length_53961475\n",
      "PGA_scaffold11__79_contigs__length_51449921\n",
      "PGA_scaffold12__71_contigs__length_50438331\n",
      "PGA_scaffold13__52_contigs__length_44396874\n",
      "PGA_scaffold14__91_contigs__length_45393038\n",
      "PGA_scaffold15__101_contigs__length_47938513\n",
      "PGA_scaffold16__33_contigs__length_31980953\n",
      "PGA_scaffold17__51_contigs__length_34923512\n",
      "PGA_scaffold18__69_contigs__length_27737463\n",
      "PGA_scaffold1__77_contigs__length_89643857\n",
      "PGA_scaffold2__36_contigs__length_69596280\n",
      "PGA_scaffold3__111_contigs__length_57743597\n",
      "PGA_scaffold4__129_contigs__length_65288255\n",
      "PGA_scaffold5__109_contigs__length_67248332\n",
      "PGA_scaffold6__104_contigs__length_61759565\n",
      "PGA_scaffold7__69_contigs__length_43120122\n",
      "PGA_scaffold8__63_contigs__length_61151155\n",
      "PGA_scaffold9__45_contigs__length_38581958\n"
     ]
    }
   ],
   "source": [
    "!cat /Volumes/Serine/wd/19-11-05/PgALL.74.CpG_report.merged_CpG_evidence.cov | cut -f 1 | sort | uniq"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Scaffold_01\n",
      "Scaffold_02\n",
      "Scaffold_03\n",
      "Scaffold_04\n",
      "Scaffold_05\n",
      "Scaffold_06\n",
      "Scaffold_07\n",
      "Scaffold_08\n",
      "Scaffold_09\n",
      "Scaffold_10\n",
      "Scaffold_11\n",
      "Scaffold_12\n",
      "Scaffold_13\n",
      "Scaffold_14\n",
      "Scaffold_15\n",
      "Scaffold_16\n",
      "Scaffold_17\n",
      "Scaffold_18\n"
     ]
    }
   ],
   "source": [
    "!cat /Volumes/Serine/wd/19-11-05/PgALL.74.CpG_report.merged_CpG_evidence.cov \\\n",
    "| sed 's/PGA_scaffold10__49_contigs__length_53961475/Scaffold_10/g' \\\n",
    "| sed 's/PGA_scaffold11__79_contigs__length_51449921/Scaffold_11/g' \\\n",
    "| sed 's/PGA_scaffold12__71_contigs__length_50438331/Scaffold_12/g' \\\n",
    "| sed 's/PGA_scaffold13__52_contigs__length_44396874/Scaffold_13/g' \\\n",
    "| sed 's/PGA_scaffold14__91_contigs__length_45393038/Scaffold_14/g' \\\n",
    "| sed 's/PGA_scaffold15__101_contigs__length_47938513/Scaffold_15/g' \\\n",
    "| sed 's/PGA_scaffold16__33_contigs__length_31980953/Scaffold_16/g' \\\n",
    "| sed 's/PGA_scaffold17__51_contigs__length_34923512/Scaffold_17/g' \\\n",
    "| sed 's/PGA_scaffold18__69_contigs__length_27737463/Scaffold_18/g' \\\n",
    "| sed 's/PGA_scaffold1__77_contigs__length_89643857/Scaffold_01/g' \\\n",
    "| sed 's/PGA_scaffold2__36_contigs__length_69596280/Scaffold_02/g' \\\n",
    "| sed 's/PGA_scaffold3__111_contigs__length_57743597/Scaffold_03/g' \\\n",
    "| sed 's/PGA_scaffold4__129_contigs__length_65288255/Scaffold_04/g' \\\n",
    "| sed 's/PGA_scaffold5__109_contigs__length_67248332/Scaffold_05/g' \\\n",
    "| sed 's/PGA_scaffold6__104_contigs__length_61759565/Scaffold_06/g' \\\n",
    "| sed 's/PGA_scaffold7__69_contigs__length_43120122/Scaffold_07/g' \\\n",
    "| sed 's/PGA_scaffold8__63_contigs__length_61151155/Scaffold_08/g' \\\n",
    "| sed 's/PGA_scaffold9__45_contigs__length_38581958/Scaffold_09/g' \\\n",
    "| cut -f 1 | sort | uniq"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 32,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Scaffold_01 \t 54 \t 55 \t 3.125000\r\n",
      "Scaffold_01 \t 72 \t 73 \t 2.777778\r\n",
      "Scaffold_01 \t 96 \t 97 \t 2.040816\r\n",
      "Scaffold_01 \t 119 \t 120 \t 0.000000\r\n",
      "Scaffold_01 \t 193 \t 194 \t 0.000000\r\n",
      "Scaffold_01 \t 202 \t 203 \t 0.000000\r\n",
      "Scaffold_01 \t 209 \t 210 \t 0.000000\r\n",
      "Scaffold_01 \t 213 \t 214 \t 0.000000\r\n",
      "Scaffold_01 \t 221 \t 222 \t 0.000000\r\n",
      "Scaffold_01 \t 238 \t 239 \t 1.315789\r\n",
      "awk: write error on /dev/stdout\r\n",
      " input record number 1533, file \r\n",
      " source line number 1\r\n",
      "sed: stdout: Broken pipe\r\n",
      "sed: stdout: Broken pipe\r\n",
      "sed: stdout: Broken pipe\r\n",
      "sed: stdout: Broken pipe\r\n",
      "sed: stdout: Broken pipe\r\n",
      "sed: stdout: Broken pipe\r\n",
      "sed: stdout: Broken pipe\r\n",
      "sed: stdout: Broken pipe\r\n",
      "sed: stdout: Broken pipe\r\n",
      "sed: stdout: Broken pipe\r\n",
      "sed: stdout: Broken pipe\r\n",
      "sed: stdout: Broken pipe\r\n",
      "sed: stdout: Broken pipe\r\n",
      "sed: stdout: Broken pipe\r\n",
      "sed: stdout: Broken pipe\r\n",
      "sed: stdout: Broken pipe\r\n",
      "sed: stdout: Broken pipe\r\n",
      "sed: stdout: Broken pipe\r\n",
      "cat: stdout: Broken pipe\r\n"
     ]
    }
   ],
   "source": [
    "!cat /Volumes/Serine/wd/19-11-05/PgALL.74.CpG_report.merged_CpG_evidence.cov \\\n",
    "| sed 's/PGA_scaffold10__49_contigs__length_53961475/Scaffold_10/g' \\\n",
    "| sed 's/PGA_scaffold11__79_contigs__length_51449921/Scaffold_11/g' \\\n",
    "| sed 's/PGA_scaffold12__71_contigs__length_50438331/Scaffold_12/g' \\\n",
    "| sed 's/PGA_scaffold13__52_contigs__length_44396874/Scaffold_13/g' \\\n",
    "| sed 's/PGA_scaffold14__91_contigs__length_45393038/Scaffold_14/g' \\\n",
    "| sed 's/PGA_scaffold15__101_contigs__length_47938513/Scaffold_15/g' \\\n",
    "| sed 's/PGA_scaffold16__33_contigs__length_31980953/Scaffold_16/g' \\\n",
    "| sed 's/PGA_scaffold17__51_contigs__length_34923512/Scaffold_17/g' \\\n",
    "| sed 's/PGA_scaffold18__69_contigs__length_27737463/Scaffold_18/g' \\\n",
    "| sed 's/PGA_scaffold1__77_contigs__length_89643857/Scaffold_01/g' \\\n",
    "| sed 's/PGA_scaffold2__36_contigs__length_69596280/Scaffold_02/g' \\\n",
    "| sed 's/PGA_scaffold3__111_contigs__length_57743597/Scaffold_03/g' \\\n",
    "| sed 's/PGA_scaffold4__129_contigs__length_65288255/Scaffold_04/g' \\\n",
    "| sed 's/PGA_scaffold5__109_contigs__length_67248332/Scaffold_05/g' \\\n",
    "| sed 's/PGA_scaffold6__104_contigs__length_61759565/Scaffold_06/g' \\\n",
    "| sed 's/PGA_scaffold7__69_contigs__length_43120122/Scaffold_07/g' \\\n",
    "| sed 's/PGA_scaffold8__63_contigs__length_61151155/Scaffold_08/g' \\\n",
    "| sed 's/PGA_scaffold9__45_contigs__length_38581958/Scaffold_09/g' \\\n",
    "| awk '{if ($5+$6 >= 10) { print $1, \"\\t\", $2, \"\\t\", $3, \"\\t\", $4}}' | head"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 33,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "!cat /Volumes/Serine/wd/19-11-05/PgALL.74.CpG_report.merged_CpG_evidence.cov \\\n",
    "| sed 's/PGA_scaffold10__49_contigs__length_53961475/Scaffold_10/g' \\\n",
    "| sed 's/PGA_scaffold11__79_contigs__length_51449921/Scaffold_11/g' \\\n",
    "| sed 's/PGA_scaffold12__71_contigs__length_50438331/Scaffold_12/g' \\\n",
    "| sed 's/PGA_scaffold13__52_contigs__length_44396874/Scaffold_13/g' \\\n",
    "| sed 's/PGA_scaffold14__91_contigs__length_45393038/Scaffold_14/g' \\\n",
    "| sed 's/PGA_scaffold15__101_contigs__length_47938513/Scaffold_15/g' \\\n",
    "| sed 's/PGA_scaffold16__33_contigs__length_31980953/Scaffold_16/g' \\\n",
    "| sed 's/PGA_scaffold17__51_contigs__length_34923512/Scaffold_17/g' \\\n",
    "| sed 's/PGA_scaffold18__69_contigs__length_27737463/Scaffold_18/g' \\\n",
    "| sed 's/PGA_scaffold1__77_contigs__length_89643857/Scaffold_01/g' \\\n",
    "| sed 's/PGA_scaffold2__36_contigs__length_69596280/Scaffold_02/g' \\\n",
    "| sed 's/PGA_scaffold3__111_contigs__length_57743597/Scaffold_03/g' \\\n",
    "| sed 's/PGA_scaffold4__129_contigs__length_65288255/Scaffold_04/g' \\\n",
    "| sed 's/PGA_scaffold5__109_contigs__length_67248332/Scaffold_05/g' \\\n",
    "| sed 's/PGA_scaffold6__104_contigs__length_61759565/Scaffold_06/g' \\\n",
    "| sed 's/PGA_scaffold7__69_contigs__length_43120122/Scaffold_07/g' \\\n",
    "| sed 's/PGA_scaffold8__63_contigs__length_61151155/Scaffold_08/g' \\\n",
    "| sed 's/PGA_scaffold9__45_contigs__length_38581958/Scaffold_09/g' \\\n",
    "| awk '{if ($5+$6 >= 10) { print $1, \"\\t\", $2, \"\\t\", $3, \"\\t\", $4}}' \\\n",
    "> analyses/PgALL.bed"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2019-11-13T15:25:21.360448Z",
     "start_time": "2019-11-13T15:25:21.198470Z"
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "0225_igv_session.xml\r\n",
      "0817_GO-annot.tab\r\n",
      "0821_GO-annot.tab\r\n",
      "1023_GO-annot.tab\r\n",
      "50p_intersect_exon_u.txt\r\n",
      "50p_intersect_intron_u.txt\r\n",
      "50p_intersect_rm_u.txt\r\n",
      "5xbedgraph.zip\r\n",
      "EPI-103_S27_L005_R1_001_val_1_bismark_bt2_pe.deduplicated.bismark.cov\r\n",
      "EPI-103_S27_L005_R1_001_val_1_bismark_bt2_pe.deduplicated.bismark.cov.destrand.10x.bedgraph\r\n",
      "EPI-103_S27_L005_R1_001_val_1_bismark_bt2_pe.deduplicated.bismark.cov.destrand.5x.bedgraph\r\n",
      "EPI-103_S27_L005_R1_001_val_1_bismark_bt2_pe.deduplicated.bismark.cov_10x.bedgraph\r\n",
      "EPI-104_S28_L005_R1_001_val_1_bismark_bt2_pe.deduplicated.bismark.cov\r\n",
      "EPI-104_S28_L005_R1_001_val_1_bismark_bt2_pe.deduplicated.bismark.cov.destrand.10x.bedgraph\r\n",
      "EPI-104_S28_L005_R1_001_val_1_bismark_bt2_pe.deduplicated.bismark.cov.destrand.5x.bedgraph\r\n",
      "EPI-104_S28_L005_R1_001_val_1_bismark_bt2_pe.deduplicated.bismark.cov_10x.bedgraph\r\n",
      "EPI-111_S29_L005_R1_001_val_1_bismark_bt2_pe.deduplicated.bismark.cov\r\n",
      "EPI-111_S29_L005_R1_001_val_1_bismark_bt2_pe.deduplicated.bismark.cov.destrand.10x.bedgraph\r\n",
      "EPI-111_S29_L005_R1_001_val_1_bismark_bt2_pe.deduplicated.bismark.cov.destrand.5x.bedgraph\r\n",
      "EPI-111_S29_L005_R1_001_val_1_bismark_bt2_pe.deduplicated.bismark.cov_10x.bedgraph\r\n",
      "EPI-113_S30_L005_R1_001_val_1_bismark_bt2_pe.deduplicated.bismark.cov\r\n",
      "EPI-113_S30_L005_R1_001_val_1_bismark_bt2_pe.deduplicated.bismark.cov.destrand.10x.bedgraph\r\n",
      "EPI-113_S30_L005_R1_001_val_1_bismark_bt2_pe.deduplicated.bismark.cov.destrand.5x.bedgraph\r\n",
      "EPI-113_S30_L005_R1_001_val_1_bismark_bt2_pe.deduplicated.bismark.cov_10x.bedgraph\r\n",
      "EPI-119_S31_L005_R1_001_val_1_bismark_bt2_pe.deduplicated.bismark.cov\r\n",
      "EPI-119_S31_L005_R1_001_val_1_bismark_bt2_pe.deduplicated.bismark.cov.destrand.10x.bedgraph\r\n",
      "EPI-119_S31_L005_R1_001_val_1_bismark_bt2_pe.deduplicated.bismark.cov.destrand.5x.bedgraph\r\n",
      "EPI-119_S31_L005_R1_001_val_1_bismark_bt2_pe.deduplicated.bismark.cov_10x.bedgraph\r\n",
      "EPI-120_S32_L005_R1_001_val_1_bismark_bt2_pe.deduplicated.bismark.cov\r\n",
      "EPI-120_S32_L005_R1_001_val_1_bismark_bt2_pe.deduplicated.bismark.cov.destrand.10x.bedgraph\r\n",
      "EPI-120_S32_L005_R1_001_val_1_bismark_bt2_pe.deduplicated.bismark.cov.destrand.5x.bedgraph\r\n",
      "EPI-120_S32_L005_R1_001_val_1_bismark_bt2_pe.deduplicated.bismark.cov_10x.bedgraph\r\n",
      "EPI-127_S33_L005_R1_001_val_1_bismark_bt2_pe.deduplicated.bismark.cov\r\n",
      "EPI-127_S33_L005_R1_001_val_1_bismark_bt2_pe.deduplicated.bismark.cov.destrand.10x.bedgraph\r\n",
      "EPI-127_S33_L005_R1_001_val_1_bismark_bt2_pe.deduplicated.bismark.cov.destrand.5x.bedgraph\r\n",
      "EPI-127_S33_L005_R1_001_val_1_bismark_bt2_pe.deduplicated.bismark.cov_10x.bedgraph\r\n",
      "EPI-128_S34_L005_R1_001_val_1_bismark_bt2_pe.deduplicated.bismark.cov\r\n",
      "EPI-128_S34_L005_R1_001_val_1_bismark_bt2_pe.deduplicated.bismark.cov.destrand.10x.bedgraph\r\n",
      "EPI-128_S34_L005_R1_001_val_1_bismark_bt2_pe.deduplicated.bismark.cov.destrand.5x.bedgraph\r\n",
      "EPI-128_S34_L005_R1_001_val_1_bismark_bt2_pe.deduplicated.bismark.cov_10x.bedgraph\r\n",
      "EPI-135_S35_L005_R1_001_val_1_bismark_bt2_pe.deduplicated.bismark.cov\r\n",
      "EPI-135_S35_L005_R1_001_val_1_bismark_bt2_pe.deduplicated.bismark.cov.destrand.10x.bedgraph\r\n",
      "EPI-135_S35_L005_R1_001_val_1_bismark_bt2_pe.deduplicated.bismark.cov.destrand.5x.bedgraph\r\n",
      "EPI-135_S35_L005_R1_001_val_1_bismark_bt2_pe.deduplicated.bismark.cov_10x.bedgraph\r\n",
      "EPI-136_S36_L005_R1_001_val_1_bismark_bt2_pe.deduplicated.bismark.cov\r\n",
      "EPI-136_S36_L005_R1_001_val_1_bismark_bt2_pe.deduplicated.bismark.cov.destrand.10x.bedgraph\r\n",
      "EPI-136_S36_L005_R1_001_val_1_bismark_bt2_pe.deduplicated.bismark.cov.destrand.5x.bedgraph\r\n",
      "EPI-136_S36_L005_R1_001_val_1_bismark_bt2_pe.deduplicated.bismark.cov_10x.bedgraph\r\n",
      "EPI-143_S37_L005_R1_001_val_1_bismark_bt2_pe.deduplicated.bismark.cov\r\n",
      "EPI-143_S37_L005_R1_001_val_1_bismark_bt2_pe.deduplicated.bismark.cov.destrand.10x.bedgraph\r\n",
      "EPI-143_S37_L005_R1_001_val_1_bismark_bt2_pe.deduplicated.bismark.cov.destrand.5x.bedgraph\r\n",
      "EPI-143_S37_L005_R1_001_val_1_bismark_bt2_pe.deduplicated.bismark.cov_10x.bedgraph\r\n",
      "EPI-145_S38_L005_R1_001_val_1_bismark_bt2_pe.deduplicated.bismark.cov\r\n",
      "EPI-145_S38_L005_R1_001_val_1_bismark_bt2_pe.deduplicated.bismark.cov.destrand\r\n",
      "EPI-145_S38_L005_R1_001_val_1_bismark_bt2_pe.deduplicated.bismark.cov.destrand.10x.bedgraph\r\n",
      "EPI-145_S38_L005_R1_001_val_1_bismark_bt2_pe.deduplicated.bismark.cov.destrand.5x.bedgraph\r\n",
      "EPI-151_S2_L002_R1_001_val_1_bismark_bt2_pe.deduplicated.bismark.cov\r\n",
      "EPI-151_S2_L002_R1_001_val_1_bismark_bt2_pe.deduplicated.bismark.cov.destrand.10x.bedgraph\r\n",
      "EPI-151_S2_L002_R1_001_val_1_bismark_bt2_pe.deduplicated.bismark.cov.destrand.5x.bedgraph\r\n",
      "EPI-151_S2_L002_R1_001_val_1_bismark_bt2_pe.deduplicated.bismark.cov_10x.bedgraph\r\n",
      "EPI-152_S3_L002_R1_001_val_1_bismark_bt2_pe.deduplicated.bismark.cov\r\n",
      "EPI-152_S3_L002_R1_001_val_1_bismark_bt2_pe.deduplicated.bismark.cov.destrand.10x.bedgraph\r\n",
      "EPI-152_S3_L002_R1_001_val_1_bismark_bt2_pe.deduplicated.bismark.cov.destrand.5x.bedgraph\r\n",
      "EPI-152_S3_L002_R1_001_val_1_bismark_bt2_pe.deduplicated.bismark.cov_10x.bedgraph\r\n",
      "EPI-153_S4_L002_R1_001_val_1_bismark_bt2_pe.deduplicated.bismark.cov\r\n",
      "EPI-153_S4_L002_R1_001_val_1_bismark_bt2_pe.deduplicated.bismark.cov.destrand.10x.bedgraph\r\n",
      "EPI-153_S4_L002_R1_001_val_1_bismark_bt2_pe.deduplicated.bismark.cov.destrand.5x.bedgraph\r\n",
      "EPI-153_S4_L002_R1_001_val_1_bismark_bt2_pe.deduplicated.bismark.cov_10x.bedgraph\r\n",
      "EPI-154_S5_L002_R1_001_val_1_bismark_bt2_pe.deduplicated.bismark.cov\r\n",
      "EPI-154_S5_L002_R1_001_val_1_bismark_bt2_pe.deduplicated.bismark.cov.destrand.10x.bedgraph\r\n",
      "EPI-154_S5_L002_R1_001_val_1_bismark_bt2_pe.deduplicated.bismark.cov.destrand.5x.bedgraph\r\n",
      "EPI-154_S5_L002_R1_001_val_1_bismark_bt2_pe.deduplicated.bismark.cov_10x.bedgraph\r\n",
      "EPI-159_S6_L002_R1_001_val_1_bismark_bt2_pe.deduplicated.bismark.cov\r\n",
      "EPI-159_S6_L002_R1_001_val_1_bismark_bt2_pe.deduplicated.bismark.cov.destrand.10x.bedgraph\r\n",
      "EPI-159_S6_L002_R1_001_val_1_bismark_bt2_pe.deduplicated.bismark.cov.destrand.5x.bedgraph\r\n",
      "EPI-159_S6_L002_R1_001_val_1_bismark_bt2_pe.deduplicated.bismark.cov_10x.bedgraph\r\n",
      "EPI-160_S7_L002_R1_001_val_1_bismark_bt2_pe.deduplicated.bismark.cov\r\n",
      "EPI-160_S7_L002_R1_001_val_1_bismark_bt2_pe.deduplicated.bismark.cov.destrand.10x.bedgraph\r\n",
      "EPI-160_S7_L002_R1_001_val_1_bismark_bt2_pe.deduplicated.bismark.cov.destrand.5x.bedgraph\r\n",
      "EPI-160_S7_L002_R1_001_val_1_bismark_bt2_pe.deduplicated.bismark.cov_10x.bedgraph\r\n",
      "EPI-161_S8_L002_R1_001_val_1_bismark_bt2_pe.deduplicated.bismark.cov\r\n",
      "EPI-161_S8_L002_R1_001_val_1_bismark_bt2_pe.deduplicated.bismark.cov.destrand.10x.bedgraph\r\n",
      "EPI-161_S8_L002_R1_001_val_1_bismark_bt2_pe.deduplicated.bismark.cov.destrand.5x.bedgraph\r\n",
      "EPI-161_S8_L002_R1_001_val_1_bismark_bt2_pe.deduplicated.bismark.cov_10x.bedgraph\r\n",
      "EPI-162_S9_L002_R1_001_val_1_bismark_bt2_pe.deduplicated.bismark.cov\r\n",
      "EPI-162_S9_L002_R1_001_val_1_bismark_bt2_pe.deduplicated.bismark.cov.destrand.10x.bedgraph\r\n",
      "EPI-162_S9_L002_R1_001_val_1_bismark_bt2_pe.deduplicated.bismark.cov.destrand.5x.bedgraph\r\n",
      "EPI-162_S9_L002_R1_001_val_1_bismark_bt2_pe.deduplicated.bismark.cov_10x.bedgraph\r\n",
      "EPI-167_S10_L002_R1_001_val_1_bismark_bt2_pe.deduplicated.bismark.cov\r\n",
      "EPI-167_S10_L002_R1_001_val_1_bismark_bt2_pe.deduplicated.bismark.cov.destrand.10x.bedgraph\r\n",
      "EPI-167_S10_L002_R1_001_val_1_bismark_bt2_pe.deduplicated.bismark.cov.destrand.5x.bedgraph\r\n",
      "EPI-167_S10_L002_R1_001_val_1_bismark_bt2_pe.deduplicated.bismark.cov_10x.bedgraph\r\n",
      "EPI-168_S11_L002_R1_001_val_1_bismark_bt2_pe.deduplicated.bismark.cov\r\n",
      "EPI-168_S11_L002_R1_001_val_1_bismark_bt2_pe.deduplicated.bismark.cov.destrand.10x.bedgraph\r\n",
      "EPI-168_S11_L002_R1_001_val_1_bismark_bt2_pe.deduplicated.bismark.cov.destrand.5x.bedgraph\r\n",
      "EPI-168_S11_L002_R1_001_val_1_bismark_bt2_pe.deduplicated.bismark.cov_10x.bedgraph\r\n",
      "EPI-169_S12_L002_R1_001_val_1_bismark_bt2_pe.deduplicated.bismark.cov\r\n",
      "EPI-169_S12_L002_R1_001_val_1_bismark_bt2_pe.deduplicated.bismark.cov.destrand.10x.bedgraph\r\n",
      "EPI-169_S12_L002_R1_001_val_1_bismark_bt2_pe.deduplicated.bismark.cov.destrand.5x.bedgraph\r\n",
      "EPI-169_S12_L002_R1_001_val_1_bismark_bt2_pe.deduplicated.bismark.cov_10x.bedgraph\r\n",
      "EPI-170_S13_L002_R1_001_val_1_bismark_bt2_pe.deduplicated.bismark.cov\r\n",
      "EPI-170_S13_L002_R1_001_val_1_bismark_bt2_pe.deduplicated.bismark.cov.destrand.10x.bedgraph\r\n",
      "EPI-170_S13_L002_R1_001_val_1_bismark_bt2_pe.deduplicated.bismark.cov.destrand.5x.bedgraph\r\n",
      "EPI-170_S13_L002_R1_001_val_1_bismark_bt2_pe.deduplicated.bismark.cov_10x.bedgraph\r\n",
      "EPI-175_S14_L003_R1_001_val_1_bismark_bt2_pe.deduplicated.bismark.cov\r\n",
      "EPI-175_S14_L003_R1_001_val_1_bismark_bt2_pe.deduplicated.bismark.cov.destrand.10x.bedgraph\r\n",
      "EPI-175_S14_L003_R1_001_val_1_bismark_bt2_pe.deduplicated.bismark.cov.destrand.5x.bedgraph\r\n",
      "EPI-175_S14_L003_R1_001_val_1_bismark_bt2_pe.deduplicated.bismark.cov_10x.bedgraph\r\n",
      "EPI-176_S15_L003_R1_001_val_1_bismark_bt2_pe.deduplicated.bismark.cov\r\n",
      "EPI-176_S15_L003_R1_001_val_1_bismark_bt2_pe.deduplicated.bismark.cov.destrand.10x.bedgraph\r\n",
      "EPI-176_S15_L003_R1_001_val_1_bismark_bt2_pe.deduplicated.bismark.cov.destrand.5x.bedgraph\r\n",
      "EPI-176_S15_L003_R1_001_val_1_bismark_bt2_pe.deduplicated.bismark.cov_10x.bedgraph\r\n",
      "EPI-181_S16_L003_R1_001_val_1_bismark_bt2_pe.deduplicated.bismark.cov\r\n",
      "EPI-181_S16_L003_R1_001_val_1_bismark_bt2_pe.deduplicated.bismark.cov.destrand.10x.bedgraph\r\n",
      "EPI-181_S16_L003_R1_001_val_1_bismark_bt2_pe.deduplicated.bismark.cov.destrand.5x.bedgraph\r\n",
      "EPI-181_S16_L003_R1_001_val_1_bismark_bt2_pe.deduplicated.bismark.cov_10x.bedgraph\r\n",
      "EPI-182_S17_L003_R1_001_val_1_bismark_bt2_pe.deduplicated.bismark.cov\r\n",
      "EPI-182_S17_L003_R1_001_val_1_bismark_bt2_pe.deduplicated.bismark.cov.destrand.10x.bedgraph\r\n",
      "EPI-182_S17_L003_R1_001_val_1_bismark_bt2_pe.deduplicated.bismark.cov.destrand.5x.bedgraph\r\n",
      "EPI-182_S17_L003_R1_001_val_1_bismark_bt2_pe.deduplicated.bismark.cov_10x.bedgraph\r\n",
      "EPI-184_S18_L003_R1_001_val_1_bismark_bt2_pe.deduplicated.bismark.cov\r\n",
      "EPI-184_S18_L003_R1_001_val_1_bismark_bt2_pe.deduplicated.bismark.cov.destrand.10x.bedgraph\r\n",
      "EPI-184_S18_L003_R1_001_val_1_bismark_bt2_pe.deduplicated.bismark.cov.destrand.5x.bedgraph\r\n",
      "EPI-184_S18_L003_R1_001_val_1_bismark_bt2_pe.deduplicated.bismark.cov_10x.bedgraph\r\n",
      "EPI-185_S19_L003_R1_001_val_1_bismark_bt2_pe.deduplicated.bismark.cov\r\n",
      "EPI-185_S19_L003_R1_001_val_1_bismark_bt2_pe.deduplicated.bismark.cov.destrand.10x.bedgraph\r\n",
      "EPI-185_S19_L003_R1_001_val_1_bismark_bt2_pe.deduplicated.bismark.cov.destrand.5x.bedgraph\r\n",
      "EPI-185_S19_L003_R1_001_val_1_bismark_bt2_pe.deduplicated.bismark.cov_10x.bedgraph\r\n",
      "EPI-187_S20_L003_R1_001_val_1_bismark_bt2_pe.deduplicated.bismark.cov\r\n",
      "EPI-187_S20_L003_R1_001_val_1_bismark_bt2_pe.deduplicated.bismark.cov.destrand.10x.bedgraph\r\n",
      "EPI-187_S20_L003_R1_001_val_1_bismark_bt2_pe.deduplicated.bismark.cov.destrand.5x.bedgraph\r\n",
      "EPI-187_S20_L003_R1_001_val_1_bismark_bt2_pe.deduplicated.bismark.cov_10x.bedgraph\r\n",
      "EPI-188_S21_L003_R1_001_val_1_bismark_bt2_pe.deduplicated.bismark.cov\r\n",
      "EPI-188_S21_L003_R1_001_val_1_bismark_bt2_pe.deduplicated.bismark.cov.destrand.10x.bedgraph\r\n",
      "EPI-188_S21_L003_R1_001_val_1_bismark_bt2_pe.deduplicated.bismark.cov.destrand.5x.bedgraph\r\n",
      "EPI-188_S21_L003_R1_001_val_1_bismark_bt2_pe.deduplicated.bismark.cov_10x.bedgraph\r\n",
      "EPI-193_S22_L003_R1_001_val_1_bismark_bt2_pe.deduplicated.bismark.cov\r\n",
      "EPI-193_S22_L003_R1_001_val_1_bismark_bt2_pe.deduplicated.bismark.cov.destrand.10x.bedgraph\r\n",
      "EPI-193_S22_L003_R1_001_val_1_bismark_bt2_pe.deduplicated.bismark.cov.destrand.5x.bedgraph\r\n",
      "EPI-193_S22_L003_R1_001_val_1_bismark_bt2_pe.deduplicated.bismark.cov_10x.bedgraph\r\n",
      "EPI-194_S23_L003_R1_001_val_1_bismark_bt2_pe.deduplicated.bismark.cov\r\n",
      "EPI-194_S23_L003_R1_001_val_1_bismark_bt2_pe.deduplicated.bismark.cov.destrand.10x.bedgraph\r\n",
      "EPI-194_S23_L003_R1_001_val_1_bismark_bt2_pe.deduplicated.bismark.cov.destrand.5x.bedgraph\r\n",
      "EPI-194_S23_L003_R1_001_val_1_bismark_bt2_pe.deduplicated.bismark.cov_10x.bedgraph\r\n",
      "EPI-199_S24_L003_R1_001_val_1_bismark_bt2_pe.deduplicated.bismark.cov\r\n",
      "EPI-199_S24_L003_R1_001_val_1_bismark_bt2_pe.deduplicated.bismark.cov.destrand.10x.bedgraph\r\n",
      "EPI-199_S24_L003_R1_001_val_1_bismark_bt2_pe.deduplicated.bismark.cov.destrand.5x.bedgraph\r\n",
      "EPI-199_S24_L003_R1_001_val_1_bismark_bt2_pe.deduplicated.bismark.cov_10x.bedgraph\r\n",
      "EPI-200_S25_L003_R1_001_val_1_bismark_bt2_pe.deduplicated.bismark.cov\r\n",
      "EPI-200_S25_L003_R1_001_val_1_bismark_bt2_pe.deduplicated.bismark.cov.destrand.10x.bedgraph\r\n",
      "EPI-200_S25_L003_R1_001_val_1_bismark_bt2_pe.deduplicated.bismark.cov.destrand.5x.bedgraph\r\n",
      "EPI-200_S25_L003_R1_001_val_1_bismark_bt2_pe.deduplicated.bismark.cov_10x.bedgraph\r\n",
      "EPI-205_S26_L004_R1_001_val_1_bismark_bt2_pe.deduplicated.bismark.cov\r\n",
      "EPI-205_S26_L004_R1_001_val_1_bismark_bt2_pe.deduplicated.bismark.cov.destrand.10x.bedgraph\r\n",
      "EPI-205_S26_L004_R1_001_val_1_bismark_bt2_pe.deduplicated.bismark.cov.destrand.5x.bedgraph\r\n",
      "EPI-205_S26_L004_R1_001_val_1_bismark_bt2_pe.deduplicated.bismark.cov_10x.bedgraph\r\n",
      "EPI-206_S27_L004_R1_001_val_1_bismark_bt2_pe.deduplicated.bismark.cov\r\n",
      "EPI-206_S27_L004_R1_001_val_1_bismark_bt2_pe.deduplicated.bismark.cov.destrand.10x.bedgraph\r\n",
      "EPI-206_S27_L004_R1_001_val_1_bismark_bt2_pe.deduplicated.bismark.cov.destrand.5x.bedgraph\r\n",
      "EPI-206_S27_L004_R1_001_val_1_bismark_bt2_pe.deduplicated.bismark.cov_10x.bedgraph\r\n",
      "EPI-208_S28_L004_R1_001_val_1_bismark_bt2_pe.deduplicated.bismark.cov\r\n",
      "EPI-208_S28_L004_R1_001_val_1_bismark_bt2_pe.deduplicated.bismark.cov.destrand.10x.bedgraph\r\n",
      "EPI-208_S28_L004_R1_001_val_1_bismark_bt2_pe.deduplicated.bismark.cov.destrand.5x.bedgraph\r",
      "\r\n",
      "EPI-208_S28_L004_R1_001_val_1_bismark_bt2_pe.deduplicated.bismark.cov_10x.bedgraph\r\n",
      "EPI-209_S29_L004_R1_001_val_1_bismark_bt2_pe.deduplicated.bismark.cov\r\n",
      "EPI-209_S29_L004_R1_001_val_1_bismark_bt2_pe.deduplicated.bismark.cov.destrand.10x.bedgraph\r\n",
      "EPI-209_S29_L004_R1_001_val_1_bismark_bt2_pe.deduplicated.bismark.cov.destrand.5x.bedgraph\r\n",
      "EPI-209_S29_L004_R1_001_val_1_bismark_bt2_pe.deduplicated.bismark.cov_10x.bedgraph\r\n",
      "EPI-214_S30_L004_R1_001_val_1_bismark_bt2_pe.deduplicated.bismark.cov\r\n",
      "EPI-214_S30_L004_R1_001_val_1_bismark_bt2_pe.deduplicated.bismark.cov.destrand.10x.bedgraph\r\n",
      "EPI-214_S30_L004_R1_001_val_1_bismark_bt2_pe.deduplicated.bismark.cov.destrand.5x.bedgraph\r\n",
      "EPI-214_S30_L004_R1_001_val_1_bismark_bt2_pe.deduplicated.bismark.cov_10x.bedgraph\r\n",
      "EPI-215_S31_L004_R1_001_val_1_bismark_bt2_pe.deduplicated.bismark.cov\r\n",
      "EPI-215_S31_L004_R1_001_val_1_bismark_bt2_pe.deduplicated.bismark.cov.destrand.10x.bedgraph\r\n",
      "EPI-215_S31_L004_R1_001_val_1_bismark_bt2_pe.deduplicated.bismark.cov.destrand.5x.bedgraph\r\n",
      "EPI-215_S31_L004_R1_001_val_1_bismark_bt2_pe.deduplicated.bismark.cov_10x.bedgraph\r\n",
      "EPI-220_S32_L004_R1_001_val_1_bismark_bt2_pe.deduplicated.bismark.cov\r\n",
      "EPI-220_S32_L004_R1_001_val_1_bismark_bt2_pe.deduplicated.bismark.cov.destrand.10x.bedgraph\r\n",
      "EPI-220_S32_L004_R1_001_val_1_bismark_bt2_pe.deduplicated.bismark.cov.destrand.5x.bedgraph\r\n",
      "EPI-220_S32_L004_R1_001_val_1_bismark_bt2_pe.deduplicated.bismark.cov_10x.bedgraph\r\n",
      "EPI-221_S33_L004_R1_001_val_1_bismark_bt2_pe.deduplicated.bismark.cov\r\n",
      "EPI-221_S33_L004_R1_001_val_1_bismark_bt2_pe.deduplicated.bismark.cov.destrand.10x.bedgraph\r\n",
      "EPI-221_S33_L004_R1_001_val_1_bismark_bt2_pe.deduplicated.bismark.cov.destrand.5x.bedgraph\r\n",
      "EPI-221_S33_L004_R1_001_val_1_bismark_bt2_pe.deduplicated.bismark.cov_10x.bedgraph\r\n",
      "EPI-226_S34_L004_R1_001_val_1_bismark_bt2_pe.deduplicated.bismark.cov\r\n",
      "EPI-226_S34_L004_R1_001_val_1_bismark_bt2_pe.deduplicated.bismark.cov.destrand.10x.bedgraph\r\n",
      "EPI-226_S34_L004_R1_001_val_1_bismark_bt2_pe.deduplicated.bismark.cov.destrand.5x.bedgraph\r\n",
      "EPI-226_S34_L004_R1_001_val_1_bismark_bt2_pe.deduplicated.bismark.cov_10x.bedgraph\r\n",
      "EPI-227_S35_L004_R1_001_val_1_bismark_bt2_pe.deduplicated.bismark.cov\r\n",
      "EPI-227_S35_L004_R1_001_val_1_bismark_bt2_pe.deduplicated.bismark.cov.destrand.10x.bedgraph\r\n",
      "EPI-227_S35_L004_R1_001_val_1_bismark_bt2_pe.deduplicated.bismark.cov.destrand.5x.bedgraph\r\n",
      "EPI-227_S35_L004_R1_001_val_1_bismark_bt2_pe.deduplicated.bismark.cov_10x.bedgraph\r\n",
      "EPI-229_S36_L004_R1_001_val_1_bismark_bt2_pe.deduplicated.bismark.cov\r\n",
      "EPI-229_S36_L004_R1_001_val_1_bismark_bt2_pe.deduplicated.bismark.cov.destrand.10x.bedgraph\r\n",
      "EPI-229_S36_L004_R1_001_val_1_bismark_bt2_pe.deduplicated.bismark.cov.destrand.5x.bedgraph\r\n",
      "EPI-229_S36_L004_R1_001_val_1_bismark_bt2_pe.deduplicated.bismark.cov_10x.bedgraph\r\n",
      "EPI-230_S37_L004_R1_001_val_1_bismark_bt2_pe.deduplicated.bismark.cov\r\n",
      "EPI-230_S37_L004_R1_001_val_1_bismark_bt2_pe.deduplicated.bismark.cov.destrand.10x.bedgraph\r\n",
      "EPI-230_S37_L004_R1_001_val_1_bismark_bt2_pe.deduplicated.bismark.cov.destrand.5x.bedgraph\r\n",
      "EPI-230_S37_L004_R1_001_val_1_bismark_bt2_pe.deduplicated.bismark.cov_10x.bedgraph\r\n",
      "EPI-41_S38_L005_R1_001_val_1_bismark_bt2_pe.deduplicated.bismark.cov\r\n",
      "EPI-41_S38_L005_R1_001_val_1_bismark_bt2_pe.deduplicated.bismark.cov.destrand.10x.bedgraph\r\n",
      "EPI-41_S38_L005_R1_001_val_1_bismark_bt2_pe.deduplicated.bismark.cov.destrand.5x.bedgraph\r\n",
      "EPI-41_S38_L005_R1_001_val_1_bismark_bt2_pe.deduplicated.bismark.cov_10x.bedgraph\r\n",
      "EPI-42_S39_L005_R1_001_val_1_bismark_bt2_pe.deduplicated.bismark.cov\r\n",
      "EPI-42_S39_L005_R1_001_val_1_bismark_bt2_pe.deduplicated.bismark.cov.destrand.10x.bedgraph\r\n",
      "EPI-42_S39_L005_R1_001_val_1_bismark_bt2_pe.deduplicated.bismark.cov.destrand.5x.bedgraph\r\n",
      "EPI-42_S39_L005_R1_001_val_1_bismark_bt2_pe.deduplicated.bismark.cov_10x.bedgraph\r\n",
      "EPI-43_S40_L005_R1_001_val_1_bismark_bt2_pe.deduplicated.bismark.cov\r\n",
      "EPI-43_S40_L005_R1_001_val_1_bismark_bt2_pe.deduplicated.bismark.cov.destrand.10x.bedgraph\r\n",
      "EPI-43_S40_L005_R1_001_val_1_bismark_bt2_pe.deduplicated.bismark.cov.destrand.5x.bedgraph\r\n",
      "EPI-43_S40_L005_R1_001_val_1_bismark_bt2_pe.deduplicated.bismark.cov_10x.bedgraph\r\n",
      "EPI-44.2.destrand.bedgraph\r\n",
      "EPI-44.3.destrand.bedgraph\r\n",
      "EPI-44.4.destrand.bedgraph\r\n",
      "EPI-44.5.destrand.bedgraph\r\n",
      "EPI-44.destrand\r\n",
      "EPI-44.destrand.bedgraph\r\n",
      "EPI-44_S41_L005_R1_001_val_1_bismark_bt2_pe.deduplicated.bismark.cov\r\n",
      "EPI-44_S41_L005_R1_001_val_1_bismark_bt2_pe.deduplicated.bismark.cov.destrand.10x.bedgraph\r\n",
      "EPI-44_S41_L005_R1_001_val_1_bismark_bt2_pe.deduplicated.bismark.cov.destrand.5x.bedgraph\r\n",
      "EPI-44_S41_L005_R1_001_val_1_bismark_bt2_pe.deduplicated.bismark.cov_10x.bedgraph\r\n",
      "PGA_scaffold1_89643857.fa\r\n",
      "PGA_scaffold1__77_contigs__length_89643857.fa\r\n",
      "PGA_scaffold1_splits_2000.fasta\r\n",
      "Panopea-generosa-vv0.74.a3.exon.comp.bed\r\n",
      "Panopea-generosa-vv0.74.a3.exon.gff3.comp\r\n",
      "Pg-gensas-blastx.sorted\r\n",
      "Pg-gensas-blastx.tab\r\n",
      "Pg_genome.tab\r\n",
      "Pg_sc1-uniprot_blastx.tab\r\n",
      "Pg_val_destrand.10x.50p.bed\r\n",
      "Pg_val_destrand.10x.50p.bedgraph\r\n",
      "Pg_val_destrand.10x.bedgraph\r\n",
      "Pg_val_destrand.10x.bedgraph.tdf\r\n",
      "Pg_val_destrand.10x.sparse.bed\r\n",
      "Pg_val_destrand.10x.sparse.bedgraph\r\n",
      "Pg_val_destrand.10x.zero.bed\r\n",
      "Pg_val_destrand.10x.zero.bedgraph\r\n",
      "Pg_val_destrand.5x.50p.bed\r\n",
      "Pg_val_destrand.5x.50p.bed.idx\r\n",
      "Pg_val_destrand.5x.50p.bedgraph\r\n",
      "Pg_val_destrand.5x.bedgraph\r\n",
      "Pg_val_destrand.5x.sparse.bed\r\n",
      "Pg_val_destrand.5x.sparse.bedgraph\r\n",
      "Pg_val_destrand.5x.zero.bed\r\n",
      "Pg_val_destrand.5x.zero.bedgraph\r\n",
      "_test.1.bed\r\n",
      "_test.1.intron\r\n",
      "_test.1.tab\r\n",
      "_test.bedgraph\r\n",
      "_test.gene.1.bed\r\n",
      "_testout.merge.tab.CpG_report.merged_CpG_evidence.cov\r\n",
      "_testout.merge.tab.CpG_report.txt\r\n",
      "_testout.tab.CpG_report.txt\r\n",
      "_testv2.bedGraph\r\n",
      "blastp.outfmt6\r\n",
      "blastp_v074.outfmt6\r\n",
      "blastp_v074.sorted\r\n",
      "dml_d10.bed\r\n",
      "dmr.bed\r\n",
      "gene_blastp.col4\r\n",
      "gene_blastp.col5\r\n",
      "hp_blastp.col3\r\n",
      "hp_blastp.outfmt6\r\n",
      "hp_blastp.sorted\r\n",
      "hp_blastp_v074.col3\r\n",
      "hp_blastp_v074.outfmt6\r\n",
      "hp_blastp_v074.sorted\r\n",
      "myDiff50p_d10.tab\r\n",
      "myDiff75p_d10.tab\r\n",
      "myobj_145\r\n",
      "myobj_d10\r\n",
      "sparse_intersect_exon_u.txt\r\n",
      "sparse_intersect_intron_u.txt\r\n",
      "sparse_intersect_rm_u.txt\r\n",
      "\u001b[34mtest\u001b[m\u001b[m/\r\n",
      "testv3.bedGraph\r\n",
      "tmp.bed\r\n",
      "tmp_002_cov.txt\r\n",
      "uniprot-E123954914.sorted\r\n",
      "uniprot-yourlist_M2019.sorted\r\n",
      "uniprot-yourlist_M201908176746803381A1F0E0DB47453E0216320D908FE8J.tab\r\n",
      "uniprot-yourlist_M201908218471C63D39733769F8E060B506551E123954914.tab\r\n",
      "uniprot-yourlist_M201908218471C63D39733769F8E060B506551E123954914.tab.gz\r\n",
      "uniprot-yourlist_M20191023216DA2B77BFBD2E6699CA9B6D1C41EB24D82ED2.tab\r\n",
      "uniprot-yourlist_M2019_82ED2.sorted\r\n",
      "zero_intersect_exon_u.txt\r\n",
      "zero_intersect_intron_u.txt\r\n",
      "zero_intersect_rm_u.txt\r\n"
     ]
    }
   ],
   "source": [
    "ls analyses/"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2019-11-13T15:24:43.857110Z",
     "start_time": "2019-11-13T15:24:43.692760Z"
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "ls: analyses/PgALL.bed: No such file or directory\r\n"
     ]
    }
   ],
   "source": [
    "!ls -l -h analyses/PgALL.bed"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "https://d.pr/f/O5Dg5b"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.6.8"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}