{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {
    "deletable": true,
    "editable": true
   },
   "source": [
    "## Taking the coverage file from an alignments -destrand."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2019-11-06T00:52:15.733763Z",
     "start_time": "2019-11-06T00:52:15.656154Z"
    },
    "collapsed": false,
    "deletable": true,
    "editable": true
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "u'/Users/sr320/git-repos/nb-2019/P_generosa'"
      ]
     },
     "execution_count": 1,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "pwd"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {
    "collapsed": false,
    "deletable": true,
    "editable": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PgALL.74.CpG_report.merged_CpG_evidence.cov      Pg_val_1_bismark_bt2_pe.deduplicated.bismark.cov\r\n",
      "PgALL.74.CpG_report.txt                          \u001b[34mgenome\u001b[m\u001b[m\r\n",
      "PgALL.CpG_report.merged_CpG_evidence.cov         \u001b[34mv074\u001b[m\u001b[m\r\n",
      "PgALL.CpG_report.txt\r\n"
     ]
    }
   ],
   "source": [
    "!ls /Volumes/Serine/wd/19-11-05/"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\u001b[34mbismark_coverage_files\u001b[m\u001b[m/     bismark_coverage_files.zip\r\n"
     ]
    }
   ],
   "source": [
    "ls /Volumes/Serine/wd/19-12-17/"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Scaffold_01\t54\t54\t0\t0\t1\r\n",
      "Scaffold_01\t72\t72\t0\t0\t1\r\n",
      "Scaffold_01\t96\t96\t0\t0\t1\r\n",
      "Scaffold_01\t2084\t2084\t0\t0\t1\r\n",
      "Scaffold_01\t2154\t2154\t0\t0\t1\r\n",
      "Scaffold_01\t2187\t2187\t0\t0\t1\r\n",
      "Scaffold_01\t2196\t2196\t0\t0\t1\r\n",
      "Scaffold_01\t2509\t2509\t0\t0\t2\r\n",
      "Scaffold_01\t2554\t2554\t0\t0\t1\r\n",
      "Scaffold_01\t2588\t2588\t0\t0\t1\r\n"
     ]
    }
   ],
   "source": [
    "!head /Volumes/Serine/wd/19-12-17/EPI-41_S38_L005_R1_001_val_1_bismark_bt2_pe.deduplicated.bismark.cov"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## have 2 prepared genomes "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {
    "collapsed": false,
    "deletable": true,
    "editable": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Path to genome folder specified as: /Volumes/Serine/wd/19-11-05/\n",
      "Aligner to be used: Bowtie 2 (default)\n",
      "Writing bisulfite genomes out into a single MFA (multi FastA) file\n",
      "\n",
      "Bismark Genome Preparation - Step I: Preparing folders\n",
      "\n",
      "Path to Bowtie 2 specified: /Applications/bioinfo/bowtie2-2.3.4.1-macos-x86_64/\n",
      "Bisulfite Genome Indexer version v0.20.0 (last modified 26 April 2018)\n",
      "Created Bisulfite Genome folder /Volumes/Serine/wd/19-11-05/Bisulfite_Genome/\n",
      "Created Bisulfite Genome folder /Volumes/Serine/wd/19-11-05/Bisulfite_Genome/CT_conversion/\n",
      "Created Bisulfite Genome folder /Volumes/Serine/wd/19-11-05/Bisulfite_Genome/GA_conversion/\n",
      "\n",
      "Step I - Prepare genome folders - completed\n",
      "\n",
      "\n",
      "Bismark Genome Preparation - Step II: Bisulfite converting reference genome\n",
      "\n",
      "conversions performed:\n",
      "chromosome\tC->T\tG->A\n",
      "Scaffold_01\t12767429\t12765526\n",
      "Scaffold_02\t10088811\t10062918\n",
      "Scaffold_03\t8199278\t8194496\n",
      "Scaffold_04\t9161562\t9161987\n",
      "Scaffold_05\t9498414\t9497714\n",
      "Scaffold_06\t8458182\t8463483\n",
      "Scaffold_07\t6054340\t6047203\n",
      "Scaffold_08\t8517146\t8506733\n",
      "Scaffold_09\t5420328\t5445998\n",
      "Scaffold_10\t7547438\t7543051\n",
      "Scaffold_11\t7053984\t7054209\n",
      "Scaffold_12\t7145712\t7105651\n",
      "Scaffold_13\t6273846\t6290785\n",
      "Scaffold_14\t6515251\t6527291\n",
      "Scaffold_15\t6625508\t6624214\n",
      "Scaffold_16\t4528254\t4550625\n",
      "Scaffold_17\t4780809\t4787183\n",
      "Scaffold_18\t3756408\t3758031\n",
      "\n",
      "Total number of conversions performed:\n",
      "C->T:\t132392700\n",
      "G->A:\t132387098\n",
      "\n",
      "Step II - Genome bisulfite conversions - completed\n",
      "\n",
      "\n",
      "Bismark Genome Preparation - Step III: Launching the Bowtie 2 indexer\n",
      "Please be aware that this process can - depending on genome size - take several hours!\n",
      "\n",
      "Preparing indexing of CT converted genome in /Volumes/Serine/wd/19-11-05/Bisulfite_Genome/CT_conversion/\n",
      "Parent process: Starting to index C->T converted genome with the following command:\n",
      "\n",
      "/Applications/bioinfo/bowtie2-2.3.4.1-macos-x86_64/bowtie2-build -f genome_mfa.CT_conversion.fa BS_CT\n",
      "\n",
      "Settings:\n",
      "  Output files: \"BS_CT.*.bt2\"\n",
      "  Line rate: 6 (line is 64 bytes)\n",
      "  Lines per side: 1 (side is 64 bytes)\n",
      "  Offset rate: 4 (one in 16)\n",
      "  FTable chars: 10\n",
      "  Strings: unpacked\n",
      "  Max bucket size: default\n",
      "  Max bucket size, sqrt multiplier: default\n",
      "  Max bucket size, len divisor: 4\n",
      "  Difference-cover sample period: 1024\n",
      "  Endianness: little\n",
      "  Actual local endianness: little\n",
      "  Sanity checking: disabled\n",
      "  Assertions: disabled\n",
      "  Random seed: 0\n",
      "  Sizeofs: void*:8, int:4, long:8, size_t:8\n",
      "Input files DNA, FASTA:\n",
      "  genome_mfa.CT_conversion.fa\n",
      "Building a SMALL index\n",
      "Reading reference sizes\n",
      "Preparing indexing of GA converted genome in /Volumes/Serine/wd/19-11-05/Bisulfite_Genome/GA_conversion/\n",
      "Child process: Starting to index G->A converted genome with the following command:\n",
      "\n",
      "/Applications/bioinfo/bowtie2-2.3.4.1-macos-x86_64/bowtie2-build -f genome_mfa.GA_conversion.fa BS_GA\n",
      "\n",
      "Settings:\n",
      "  Output files: \"BS_GA.*.bt2\"\n",
      "  Line rate: 6 (line is 64 bytes)\n",
      "  Lines per side: 1 (side is 64 bytes)\n",
      "  Offset rate: 4 (one in 16)\n",
      "  FTable chars: 10\n",
      "  Strings: unpacked\n",
      "  Max bucket size: default\n",
      "  Max bucket size, sqrt multiplier: default\n",
      "  Max bucket size, len divisor: 4\n",
      "  Difference-cover sample period: 1024\n",
      "  Endianness: little\n",
      "  Actual local endianness: little\n",
      "  Sanity checking: disabled\n",
      "  Assertions: disabled\n",
      "  Random seed: 0\n",
      "  Sizeofs: void*:8, int:4, long:8, size_t:8\n",
      "Input files DNA, FASTA:\n",
      "  genome_mfa.GA_conversion.fa\n",
      "Building a SMALL index\n",
      "Reading reference sizes\n",
      "  Time reading reference sizes: 00:00:08\n",
      "Calculating joined length\n",
      "Writing header\n",
      "Reserving space for joined string\n",
      "Joining reference sequences\n",
      "  Time reading reference sizes: 00:00:08\n",
      "Calculating joined length\n",
      "Writing header\n",
      "Reserving space for joined string\n",
      "Joining reference sequences\n",
      "  Time to join reference sequences: 00:00:06\n",
      "bmax according to bmaxDivN setting: 195965290\n",
      "Using parameters --bmax 146973968 --dcv 1024\n",
      "  Doing ahead-of-time memory usage test\n",
      "  Passed!  Constructing with these parameters: --bmax 146973968 --dcv 1024\n",
      "Constructing suffix-array element generator\n",
      "Building DifferenceCoverSample\n",
      "  Building sPrime\n",
      "  Building sPrimeOrder\n",
      "  V-Sorting samples\n",
      "  Time to join reference sequences: 00:00:06\n",
      "bmax according to bmaxDivN setting: 195965290\n",
      "Using parameters --bmax 146973968 --dcv 1024\n",
      "  Doing ahead-of-time memory usage test\n",
      "  Passed!  Constructing with these parameters: --bmax 146973968 --dcv 1024\n",
      "Constructing suffix-array element generator\n",
      "Building DifferenceCoverSample\n",
      "  Building sPrime\n",
      "  Building sPrimeOrder\n",
      "  V-Sorting samples\n",
      "  V-Sorting samples time: 00:00:18\n",
      "  Allocating rank array\n",
      "  Ranking v-sort output\n",
      "  V-Sorting samples time: 00:00:18\n",
      "  Allocating rank array\n",
      "  Ranking v-sort output\n",
      "  Ranking v-sort output time: 00:00:06\n",
      "  Invoking Larsson-Sadakane on ranks\n",
      "  Ranking v-sort output time: 00:00:05\n",
      "  Invoking Larsson-Sadakane on ranks\n",
      "  Invoking Larsson-Sadakane on ranks time: 00:00:08\n",
      "  Sanity-checking and returning\n",
      "Building samples\n",
      "Reserving space for 12 sample suffixes\n",
      "Generating random suffixes\n",
      "QSorting 12 sample offsets, eliminating duplicates\n",
      "QSorting sample offsets, eliminating duplicates time: 00:00:00\n",
      "Multikey QSorting 12 samples\n",
      "  (Using difference cover)\n",
      "  Multikey QSorting samples time: 00:00:00\n",
      "Calculating bucket sizes\n",
      "Splitting and merging\n",
      "  Splitting and merging time: 00:00:00\n",
      "Avg bucket size: 7.83861e+08 (target: 146973967)\n",
      "Converting suffix-array elements to index image\n",
      "Allocating ftab, absorbFtab\n",
      "Entering Ebwt loop\n",
      "Getting block 1 of 1\n",
      "  No samples; assembling all-inclusive block\n",
      "  Invoking Larsson-Sadakane on ranks time: 00:00:08\n",
      "  Sanity-checking and returning\n",
      "Building samples\n",
      "Reserving space for 12 sample suffixes\n",
      "Generating random suffixes\n",
      "QSorting 12 sample offsets, eliminating duplicates\n",
      "QSorting sample offsets, eliminating duplicates time: 00:00:00\n",
      "Multikey QSorting 12 samples\n",
      "  (Using difference cover)\n",
      "  Multikey QSorting samples time: 00:00:00\n",
      "Calculating bucket sizes\n",
      "Splitting and merging\n",
      "  Splitting and merging time: 00:00:00\n",
      "Avg bucket size: 7.83861e+08 (target: 146973967)\n",
      "Converting suffix-array elements to index image\n",
      "Allocating ftab, absorbFtab\n",
      "Entering Ebwt loop\n",
      "Getting block 1 of 1\n",
      "  No samples; assembling all-inclusive block\n",
      "  Sorting block of length 783861161 for bucket 1\n",
      "  (Using difference cover)\n",
      "  Sorting block of length 783861161 for bucket 1\n",
      "  (Using difference cover)\n",
      "  Sorting block time: 00:07:31\n",
      "Returning block of 783861162 for bucket 1\n",
      "  Sorting block time: 00:09:24\n",
      "Returning block of 783861162 for bucket 1\n",
      "Exited Ebwt loop\n",
      "fchr[A]: 0\n",
      "fchr[C]: 391954798\n",
      "fchr[G]: 524347498\n",
      "fchr[T]: 524347498\n",
      "fchr[$]: 783861161\n",
      "Exiting Ebwt::buildToDisk()\n",
      "Returning from initFromVector\n",
      "Wrote 266840223 bytes to primary EBWT file: BS_GA.1.bt2\n",
      "Wrote 195965296 bytes to secondary EBWT file: BS_GA.2.bt2\n",
      "Re-opening _in1 and _in2 as input streams\n",
      "Returning from Ebwt constructor\n",
      "Headers:\n",
      "    len: 783861161\n",
      "    bwtLen: 783861162\n",
      "    sz: 195965291\n",
      "    bwtSz: 195965291\n",
      "    lineRate: 6\n",
      "    offRate: 4\n",
      "    offMask: 0xfffffff0\n",
      "    ftabChars: 10\n",
      "    eftabLen: 20\n",
      "    eftabSz: 80\n",
      "    ftabLen: 1048577\n",
      "    ftabSz: 4194308\n",
      "    offsLen: 48991323\n",
      "    offsSz: 195965292\n",
      "    lineSz: 64\n",
      "    sideSz: 64\n",
      "    sideBwtSz: 48\n",
      "    sideBwtLen: 192\n",
      "    numSides: 4082611\n",
      "    numLines: 4082611\n",
      "    ebwtTotLen: 261287104\n",
      "    ebwtTotSz: 261287104\n",
      "    color: 0\n",
      "    reverse: 0\n",
      "Total time for call to driver() for forward index: 00:10:18\n",
      "Reading reference sizes\n",
      "  Time reading reference sizes: 00:00:07\n",
      "Calculating joined length\n",
      "Writing header\n",
      "Reserving space for joined string\n",
      "Joining reference sequences\n",
      "  Time to join reference sequences: 00:00:06\n",
      "  Time to reverse reference sequence: 00:00:00\n",
      "bmax according to bmaxDivN setting: 195965290\n",
      "Using parameters --bmax 146973968 --dcv 1024\n",
      "  Doing ahead-of-time memory usage test\n",
      "  Passed!  Constructing with these parameters: --bmax 146973968 --dcv 1024\n",
      "Constructing suffix-array element generator\n",
      "Building DifferenceCoverSample\n",
      "  Building sPrime\n",
      "  Building sPrimeOrder\n",
      "  V-Sorting samples\n",
      "  V-Sorting samples time: 00:00:17\n",
      "  Allocating rank array\n",
      "  Ranking v-sort output\n",
      "  Ranking v-sort output time: 00:00:05\n",
      "  Invoking Larsson-Sadakane on ranks\n",
      "  Invoking Larsson-Sadakane on ranks time: 00:00:08\n",
      "  Sanity-checking and returning\n",
      "Building samples\n",
      "Reserving space for 12 sample suffixes\n",
      "Generating random suffixes\n",
      "QSorting 12 sample offsets, eliminating duplicates\n",
      "QSorting sample offsets, eliminating duplicates time: 00:00:00\n",
      "Multikey QSorting 12 samples\n",
      "  (Using difference cover)\n",
      "  Multikey QSorting samples time: 00:00:00\n",
      "Calculating bucket sizes\n",
      "Splitting and merging\n",
      "  Splitting and merging time: 00:00:00\n",
      "Avg bucket size: 7.83861e+08 (target: 146973967)\n",
      "Converting suffix-array elements to index image\n",
      "Allocating ftab, absorbFtab\n",
      "Entering Ebwt loop\n",
      "Getting block 1 of 1\n",
      "  No samples; assembling all-inclusive block\n",
      "  Sorting block of length 783861161 for bucket 1\n",
      "  (Using difference cover)\n",
      "Exited Ebwt loop\n",
      "fchr[A]: 0\n",
      "fchr[C]: 259567700\n",
      "fchr[G]: 259567700\n",
      "fchr[T]: 391954798\n",
      "fchr[$]: 783861161\n",
      "Exiting Ebwt::buildToDisk()\n",
      "Returning from initFromVector\n",
      "Wrote 266840223 bytes to primary EBWT file: BS_CT.1.bt2\n",
      "Wrote 195965296 bytes to secondary EBWT file: BS_CT.2.bt2\n",
      "Re-opening _in1 and _in2 as input streams\n",
      "Returning from Ebwt constructor\n",
      "Headers:\n",
      "    len: 783861161\n",
      "    bwtLen: 783861162\n",
      "    sz: 195965291\n",
      "    bwtSz: 195965291\n",
      "    lineRate: 6\n",
      "    offRate: 4\n",
      "    offMask: 0xfffffff0\n",
      "    ftabChars: 10\n",
      "    eftabLen: 20\n",
      "    eftabSz: 80\n",
      "    ftabLen: 1048577\n",
      "    ftabSz: 4194308\n",
      "    offsLen: 48991323\n",
      "    offsSz: 195965292\n",
      "    lineSz: 64\n",
      "    sideSz: 64\n",
      "    sideBwtSz: 48\n",
      "    sideBwtLen: 192\n",
      "    numSides: 4082611\n",
      "    numLines: 4082611\n",
      "    ebwtTotLen: 261287104\n",
      "    ebwtTotSz: 261287104\n",
      "    color: 0\n",
      "    reverse: 0\n",
      "Total time for call to driver() for forward index: 00:12:15\n",
      "Reading reference sizes\n",
      "  Time reading reference sizes: 00:00:07\n",
      "Calculating joined length\n",
      "Writing header\n",
      "Reserving space for joined string\n",
      "Joining reference sequences\n",
      "  Time to join reference sequences: 00:00:06\n",
      "  Time to reverse reference sequence: 00:00:01\n",
      "bmax according to bmaxDivN setting: 195965290\n",
      "Using parameters --bmax 146973968 --dcv 1024\n",
      "  Doing ahead-of-time memory usage test\n",
      "  Passed!  Constructing with these parameters: --bmax 146973968 --dcv 1024\n",
      "Constructing suffix-array element generator\n",
      "Building DifferenceCoverSample\n",
      "  Building sPrime\n",
      "  Building sPrimeOrder\n",
      "  V-Sorting samples\n",
      "  V-Sorting samples time: 00:00:18\n",
      "  Allocating rank array\n",
      "  Ranking v-sort output\n",
      "  Ranking v-sort output time: 00:00:06\n",
      "  Invoking Larsson-Sadakane on ranks\n",
      "  Invoking Larsson-Sadakane on ranks time: 00:00:08\n",
      "  Sanity-checking and returning\n",
      "Building samples\n",
      "Reserving space for 12 sample suffixes\n",
      "Generating random suffixes\n",
      "QSorting 12 sample offsets, eliminating duplicates\n",
      "QSorting sample offsets, eliminating duplicates time: 00:00:00\n",
      "Multikey QSorting 12 samples\n",
      "  (Using difference cover)\n",
      "  Multikey QSorting samples time: 00:00:00\n",
      "Calculating bucket sizes\n",
      "Splitting and merging\n",
      "  Splitting and merging time: 00:00:00\n",
      "Avg bucket size: 7.83861e+08 (target: 146973967)\n",
      "Converting suffix-array elements to index image\n",
      "Allocating ftab, absorbFtab\n",
      "Entering Ebwt loop\n",
      "Getting block 1 of 1\n",
      "  No samples; assembling all-inclusive block\n",
      "  Sorting block of length 783861161 for bucket 1\n",
      "  (Using difference cover)\n",
      "  Sorting block time: 00:07:37\n",
      "Returning block of 783861162 for bucket 1\n",
      "Exited Ebwt loop\n",
      "fchr[A]: 0\n",
      "fchr[C]: 391954798\n",
      "fchr[G]: 524347498\n",
      "fchr[T]: 524347498\n",
      "fchr[$]: 783861161\n",
      "Exiting Ebwt::buildToDisk()\n",
      "Returning from initFromVector\n",
      "Wrote 266840223 bytes to primary EBWT file: BS_GA.rev.1.bt2\n",
      "Wrote 195965296 bytes to secondary EBWT file: BS_GA.rev.2.bt2\n",
      "Re-opening _in1 and _in2 as input streams\n",
      "Returning from Ebwt constructor\n",
      "Headers:\n",
      "    len: 783861161\n",
      "    bwtLen: 783861162\n",
      "    sz: 195965291\n",
      "    bwtSz: 195965291\n",
      "    lineRate: 6\n",
      "    offRate: 4\n",
      "    offMask: 0xfffffff0\n",
      "    ftabChars: 10\n",
      "    eftabLen: 20\n",
      "    eftabSz: 80\n",
      "    ftabLen: 1048577\n",
      "    ftabSz: 4194308\n",
      "    offsLen: 48991323\n",
      "    offsSz: 195965292\n",
      "    lineSz: 64\n",
      "    sideSz: 64\n",
      "    sideBwtSz: 48\n",
      "    sideBwtLen: 192\n",
      "    numSides: 4082611\n",
      "    numLines: 4082611\n",
      "    ebwtTotLen: 261287104\n",
      "    ebwtTotSz: 261287104\n",
      "    color: 0\n",
      "    reverse: 1\n",
      "Total time for backward call to driver() for mirror index: 00:10:23\n",
      "  Sorting block time: 00:09:13\n",
      "Returning block of 783861162 for bucket 1\n",
      "Exited Ebwt loop\n",
      "fchr[A]: 0\n",
      "fchr[C]: 259567700\n",
      "fchr[G]: 259567700\n",
      "fchr[T]: 391954798\n",
      "fchr[$]: 783861161\n",
      "Exiting Ebwt::buildToDisk()\n",
      "Returning from initFromVector\n",
      "Wrote 266840223 bytes to primary EBWT file: BS_CT.rev.1.bt2\n",
      "Wrote 195965296 bytes to secondary EBWT file: BS_CT.rev.2.bt2\n",
      "Re-opening _in1 and _in2 as input streams\n",
      "Returning from Ebwt constructor\n",
      "Headers:\n",
      "    len: 783861161\n",
      "    bwtLen: 783861162\n",
      "    sz: 195965291\n",
      "    bwtSz: 195965291\n",
      "    lineRate: 6\n",
      "    offRate: 4\n",
      "    offMask: 0xfffffff0\n",
      "    ftabChars: 10\n",
      "    eftabLen: 20\n",
      "    eftabSz: 80\n",
      "    ftabLen: 1048577\n",
      "    ftabSz: 4194308\n",
      "    offsLen: 48991323\n",
      "    offsSz: 195965292\n",
      "    lineSz: 64\n",
      "    sideSz: 64\n",
      "    sideBwtSz: 48\n",
      "    sideBwtLen: 192\n",
      "    numSides: 4082611\n",
      "    numLines: 4082611\n",
      "    ebwtTotLen: 261287104\n",
      "    ebwtTotSz: 261287104\n",
      "    color: 0\n",
      "    reverse: 1\n",
      "Total time for backward call to driver() for mirror index: 00:11:49\n"
     ]
    }
   ],
   "source": [
    "!/Applications/bioinfo/Bismark_v0.20.0/bismark_genome_preparation \\\n",
    "--verbose \\\n",
    "--path_to_bowtie /Applications/bioinfo/bowtie2-2.3.4.1-macos-x86_64/ \\\n",
    "/Volumes/Serine/wd/19-11-05/"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {
    "collapsed": true,
    "deletable": true,
    "editable": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Path to genome folder specified as: /Volumes/Serine/wd/19-11-05/v074/\n",
      "Aligner to be used: Bowtie 2 (default)\n",
      "Writing bisulfite genomes out into a single MFA (multi FastA) file\n",
      "\n",
      "Bismark Genome Preparation - Step I: Preparing folders\n",
      "\n",
      "Path to Bowtie 2 specified: /Applications/bioinfo/bowtie2-2.3.4.1-macos-x86_64/\n",
      "Bisulfite Genome Indexer version v0.20.0 (last modified 26 April 2018)\n",
      "Created Bisulfite Genome folder /Volumes/Serine/wd/19-11-05/v074/Bisulfite_Genome/\n",
      "Created Bisulfite Genome folder /Volumes/Serine/wd/19-11-05/v074/Bisulfite_Genome/CT_conversion/\n",
      "Created Bisulfite Genome folder /Volumes/Serine/wd/19-11-05/v074/Bisulfite_Genome/GA_conversion/\n",
      "\n",
      "Step I - Prepare genome folders - completed\n",
      "\n",
      "\n",
      "Bismark Genome Preparation - Step II: Bisulfite converting reference genome\n",
      "\n",
      "conversions performed:\n",
      "chromosome\tC->T\tG->A\n",
      "PGA_scaffold1__77_contigs__length_89643857\t12767429\t12765526\n",
      "PGA_scaffold2__36_contigs__length_69596280\t10088811\t10062918\n",
      "PGA_scaffold3__111_contigs__length_57743597\t8199278\t8194496\n",
      "PGA_scaffold4__129_contigs__length_65288255\t9161562\t9161987\n",
      "PGA_scaffold5__109_contigs__length_67248332\t9498414\t9497714\n",
      "PGA_scaffold6__104_contigs__length_61759565\t8458182\t8463483\n",
      "PGA_scaffold7__69_contigs__length_43120122\t6054340\t6047203\n",
      "PGA_scaffold8__63_contigs__length_61151155\t8517146\t8506733\n",
      "PGA_scaffold9__45_contigs__length_38581958\t5420328\t5445998\n",
      "PGA_scaffold10__49_contigs__length_53961475\t7547438\t7543051\n",
      "PGA_scaffold11__79_contigs__length_51449921\t7053984\t7054209\n",
      "PGA_scaffold12__71_contigs__length_50438331\t7145712\t7105651\n",
      "PGA_scaffold13__52_contigs__length_44396874\t6273846\t6290785\n",
      "PGA_scaffold14__91_contigs__length_45393038\t6515251\t6527291\n",
      "PGA_scaffold15__101_contigs__length_47938513\t6625508\t6624214\n",
      "PGA_scaffold16__33_contigs__length_31980953\t4528254\t4550625\n",
      "PGA_scaffold17__51_contigs__length_34923512\t4780809\t4787183\n",
      "PGA_scaffold18__69_contigs__length_27737463\t3756408\t3758031\n",
      "\n",
      "Total number of conversions performed:\n",
      "C->T:\t132392700\n",
      "G->A:\t132387098\n",
      "\n",
      "Step II - Genome bisulfite conversions - completed\n",
      "\n",
      "\n",
      "Bismark Genome Preparation - Step III: Launching the Bowtie 2 indexer\n",
      "Please be aware that this process can - depending on genome size - take several hours!\n",
      "\n",
      "Preparing indexing of CT converted genome in /Volumes/Serine/wd/19-11-05/v074/Bisulfite_Genome/CT_conversion/\n",
      "Parent process: Starting to index C->T converted genome with the following command:\n",
      "\n",
      "/Applications/bioinfo/bowtie2-2.3.4.1-macos-x86_64/bowtie2-build -f genome_mfa.CT_conversion.fa BS_CT\n",
      "\n",
      "Settings:\n",
      "  Output files: \"BS_CT.*.bt2\"\n",
      "  Line rate: 6 (line is 64 bytes)\n",
      "  Lines per side: 1 (side is 64 bytes)\n",
      "  Offset rate: 4 (one in 16)\n",
      "  FTable chars: 10\n",
      "  Strings: unpacked\n",
      "  Max bucket size: default\n",
      "  Max bucket size, sqrt multiplier: default\n",
      "  Max bucket size, len divisor: 4\n",
      "  Difference-cover sample period: 1024\n",
      "  Endianness: little\n",
      "  Actual local endianness: little\n",
      "  Sanity checking: disabled\n",
      "  Assertions: disabled\n",
      "  Random seed: 0\n",
      "  Sizeofs: void*:8, int:4, long:8, size_t:8\n",
      "Input files DNA, FASTA:\n",
      "  genome_mfa.CT_conversion.fa\n",
      "Building a SMALL index\n",
      "Reading reference sizes\n",
      "Preparing indexing of GA converted genome in /Volumes/Serine/wd/19-11-05/v074/Bisulfite_Genome/GA_conversion/\n",
      "Child process: Starting to index G->A converted genome with the following command:\n",
      "\n",
      "/Applications/bioinfo/bowtie2-2.3.4.1-macos-x86_64/bowtie2-build -f genome_mfa.GA_conversion.fa BS_GA\n",
      "\n",
      "Settings:\n",
      "  Output files: \"BS_GA.*.bt2\"\n",
      "  Line rate: 6 (line is 64 bytes)\n",
      "  Lines per side: 1 (side is 64 bytes)\n",
      "  Offset rate: 4 (one in 16)\n",
      "  FTable chars: 10\n",
      "  Strings: unpacked\n",
      "  Max bucket size: default\n",
      "  Max bucket size, sqrt multiplier: default\n",
      "  Max bucket size, len divisor: 4\n",
      "  Difference-cover sample period: 1024\n",
      "  Endianness: little\n",
      "  Actual local endianness: little\n",
      "  Sanity checking: disabled\n",
      "  Assertions: disabled\n",
      "  Random seed: 0\n",
      "  Sizeofs: void*:8, int:4, long:8, size_t:8\n",
      "Input files DNA, FASTA:\n",
      "  genome_mfa.GA_conversion.fa\n",
      "Building a SMALL index\n",
      "Reading reference sizes\n",
      "  Time reading reference sizes: 00:00:08\n",
      "Calculating joined length\n",
      "Writing header\n",
      "Reserving space for joined string\n",
      "Joining reference sequences\n",
      "  Time reading reference sizes: 00:00:08\n",
      "Calculating joined length\n",
      "Writing header\n",
      "Reserving space for joined string\n",
      "Joining reference sequences\n",
      "  Time to join reference sequences: 00:00:06\n",
      "bmax according to bmaxDivN setting: 195965290\n",
      "Using parameters --bmax 146973968 --dcv 1024\n",
      "  Doing ahead-of-time memory usage test\n",
      "  Passed!  Constructing with these parameters: --bmax 146973968 --dcv 1024\n",
      "Constructing suffix-array element generator\n",
      "Building DifferenceCoverSample\n",
      "  Building sPrime\n",
      "  Building sPrimeOrder\n",
      "  V-Sorting samples\n",
      "  Time to join reference sequences: 00:00:06\n",
      "bmax according to bmaxDivN setting: 195965290\n",
      "Using parameters --bmax 146973968 --dcv 1024\n",
      "  Doing ahead-of-time memory usage test\n",
      "  Passed!  Constructing with these parameters: --bmax 146973968 --dcv 1024\n",
      "Constructing suffix-array element generator\n",
      "Building DifferenceCoverSample\n",
      "  Building sPrime\n",
      "  Building sPrimeOrder\n",
      "  V-Sorting samples\n",
      "  V-Sorting samples time: 00:00:19\n",
      "  Allocating rank array\n",
      "  Ranking v-sort output\n",
      "  V-Sorting samples time: 00:00:18\n",
      "  Allocating rank array\n",
      "  Ranking v-sort output\n",
      "  Ranking v-sort output time: 00:00:05\n",
      "  Invoking Larsson-Sadakane on ranks\n",
      "  Ranking v-sort output time: 00:00:06\n",
      "  Invoking Larsson-Sadakane on ranks\n",
      "  Invoking Larsson-Sadakane on ranks time: 00:00:08\n",
      "  Sanity-checking and returning\n",
      "Building samples\n",
      "Reserving space for 12 sample suffixes\n",
      "Generating random suffixes\n",
      "QSorting 12 sample offsets, eliminating duplicates\n",
      "QSorting sample offsets, eliminating duplicates time: 00:00:00\n",
      "Multikey QSorting 12 samples\n",
      "  (Using difference cover)\n",
      "  Multikey QSorting samples time: 00:00:00\n",
      "Calculating bucket sizes\n",
      "Splitting and merging\n",
      "  Splitting and merging time: 00:00:00\n",
      "Avg bucket size: 7.83861e+08 (target: 146973967)\n",
      "Converting suffix-array elements to index image\n",
      "Allocating ftab, absorbFtab\n",
      "Entering Ebwt loop\n",
      "Getting block 1 of 1\n",
      "  No samples; assembling all-inclusive block\n",
      "  Invoking Larsson-Sadakane on ranks time: 00:00:08\n",
      "  Sanity-checking and returning\n",
      "Building samples\n",
      "Reserving space for 12 sample suffixes\n",
      "Generating random suffixes\n",
      "QSorting 12 sample offsets, eliminating duplicates\n",
      "QSorting sample offsets, eliminating duplicates time: 00:00:00\n",
      "Multikey QSorting 12 samples\n",
      "  (Using difference cover)\n",
      "  Multikey QSorting samples time: 00:00:00\n",
      "Calculating bucket sizes\n",
      "Splitting and merging\n",
      "  Splitting and merging time: 00:00:00\n",
      "Avg bucket size: 7.83861e+08 (target: 146973967)\n",
      "Converting suffix-array elements to index image\n",
      "Allocating ftab, absorbFtab\n",
      "Entering Ebwt loop\n",
      "Getting block 1 of 1\n",
      "  No samples; assembling all-inclusive block\n",
      "  Sorting block of length 783861161 for bucket 1\n",
      "  (Using difference cover)\n",
      "  Sorting block of length 783861161 for bucket 1\n",
      "  (Using difference cover)\n",
      "  Sorting block time: 00:07:32\n",
      "Returning block of 783861162 for bucket 1\n",
      "  Sorting block time: 00:09:31\n",
      "Returning block of 783861162 for bucket 1\n",
      "Exited Ebwt loop\n",
      "fchr[A]: 0\n",
      "fchr[C]: 391954798\n",
      "fchr[G]: 524347498\n",
      "fchr[T]: 524347498\n",
      "fchr[$]: 783861161\n",
      "Exiting Ebwt::buildToDisk()\n",
      "Returning from initFromVector\n",
      "Wrote 266840795 bytes to primary EBWT file: BS_GA.1.bt2\n",
      "Wrote 195965296 bytes to secondary EBWT file: BS_GA.2.bt2\n",
      "Re-opening _in1 and _in2 as input streams\n",
      "Returning from Ebwt constructor\n",
      "Headers:\n",
      "    len: 783861161\n",
      "    bwtLen: 783861162\n",
      "    sz: 195965291\n",
      "    bwtSz: 195965291\n",
      "    lineRate: 6\n",
      "    offRate: 4\n",
      "    offMask: 0xfffffff0\n",
      "    ftabChars: 10\n",
      "    eftabLen: 20\n",
      "    eftabSz: 80\n",
      "    ftabLen: 1048577\n",
      "    ftabSz: 4194308\n",
      "    offsLen: 48991323\n",
      "    offsSz: 195965292\n",
      "    lineSz: 64\n",
      "    sideSz: 64\n",
      "    sideBwtSz: 48\n",
      "    sideBwtLen: 192\n",
      "    numSides: 4082611\n",
      "    numLines: 4082611\n",
      "    ebwtTotLen: 261287104\n",
      "    ebwtTotSz: 261287104\n",
      "    color: 0\n",
      "    reverse: 0\n",
      "Total time for call to driver() for forward index: 00:10:25\n",
      "Reading reference sizes\n",
      "  Time reading reference sizes: 00:00:06\n",
      "Calculating joined length\n",
      "Writing header\n",
      "Reserving space for joined string\n",
      "Joining reference sequences\n",
      "  Time to join reference sequences: 00:00:06\n",
      "  Time to reverse reference sequence: 00:00:01\n",
      "bmax according to bmaxDivN setting: 195965290\n",
      "Using parameters --bmax 146973968 --dcv 1024\n",
      "  Doing ahead-of-time memory usage test\n",
      "  Passed!  Constructing with these parameters: --bmax 146973968 --dcv 1024\n",
      "Constructing suffix-array element generator\n",
      "Building DifferenceCoverSample\n",
      "  Building sPrime\n",
      "  Building sPrimeOrder\n",
      "  V-Sorting samples\n",
      "  V-Sorting samples time: 00:00:17\n",
      "  Allocating rank array\n",
      "  Ranking v-sort output\n",
      "  Ranking v-sort output time: 00:00:06\n",
      "  Invoking Larsson-Sadakane on ranks\n",
      "  Invoking Larsson-Sadakane on ranks time: 00:00:07\n",
      "  Sanity-checking and returning\n",
      "Building samples\n",
      "Reserving space for 12 sample suffixes\n",
      "Generating random suffixes\n",
      "QSorting 12 sample offsets, eliminating duplicates\n",
      "QSorting sample offsets, eliminating duplicates time: 00:00:00\n",
      "Multikey QSorting 12 samples\n",
      "  (Using difference cover)\n",
      "  Multikey QSorting samples time: 00:00:00\n",
      "Calculating bucket sizes\n",
      "Splitting and merging\n",
      "  Splitting and merging time: 00:00:00\n",
      "Avg bucket size: 7.83861e+08 (target: 146973967)\n",
      "Converting suffix-array elements to index image\n",
      "Allocating ftab, absorbFtab\n",
      "Entering Ebwt loop\n",
      "Getting block 1 of 1\n",
      "  No samples; assembling all-inclusive block\n",
      "  Sorting block of length 783861161 for bucket 1\n",
      "  (Using difference cover)\n",
      "Exited Ebwt loop\n",
      "fchr[A]: 0\n",
      "fchr[C]: 259567700\n",
      "fchr[G]: 259567700\n",
      "fchr[T]: 391954798\n",
      "fchr[$]: 783861161\n",
      "Exiting Ebwt::buildToDisk()\n",
      "Returning from initFromVector\n",
      "Wrote 266840795 bytes to primary EBWT file: BS_CT.1.bt2\n",
      "Wrote 195965296 bytes to secondary EBWT file: BS_CT.2.bt2\n",
      "Re-opening _in1 and _in2 as input streams\n",
      "Returning from Ebwt constructor\n",
      "Headers:\n",
      "    len: 783861161\n",
      "    bwtLen: 783861162\n",
      "    sz: 195965291\n",
      "    bwtSz: 195965291\n",
      "    lineRate: 6\n",
      "    offRate: 4\n",
      "    offMask: 0xfffffff0\n",
      "    ftabChars: 10\n",
      "    eftabLen: 20\n",
      "    eftabSz: 80\n",
      "    ftabLen: 1048577\n",
      "    ftabSz: 4194308\n",
      "    offsLen: 48991323\n",
      "    offsSz: 195965292\n",
      "    lineSz: 64\n",
      "    sideSz: 64\n",
      "    sideBwtSz: 48\n",
      "    sideBwtLen: 192\n",
      "    numSides: 4082611\n",
      "    numLines: 4082611\n",
      "    ebwtTotLen: 261287104\n",
      "    ebwtTotSz: 261287104\n",
      "    color: 0\n",
      "    reverse: 0\n",
      "Total time for call to driver() for forward index: 00:12:24\n",
      "Reading reference sizes\n",
      "  Time reading reference sizes: 00:00:06\n",
      "Calculating joined length\n",
      "Writing header\n",
      "Reserving space for joined string\n",
      "Joining reference sequences\n",
      "  Time to join reference sequences: 00:00:06\n",
      "  Time to reverse reference sequence: 00:00:00\n",
      "bmax according to bmaxDivN setting: 195965290\n",
      "Using parameters --bmax 146973968 --dcv 1024\n",
      "  Doing ahead-of-time memory usage test\n",
      "  Passed!  Constructing with these parameters: --bmax 146973968 --dcv 1024\n",
      "Constructing suffix-array element generator\n",
      "Building DifferenceCoverSample\n",
      "  Building sPrime\n",
      "  Building sPrimeOrder\n",
      "  V-Sorting samples\n",
      "  V-Sorting samples time: 00:00:17\n",
      "  Allocating rank array\n",
      "  Ranking v-sort output\n",
      "  Ranking v-sort output time: 00:00:06\n",
      "  Invoking Larsson-Sadakane on ranks\n",
      "  Invoking Larsson-Sadakane on ranks time: 00:00:07\n",
      "  Sanity-checking and returning\n",
      "Building samples\n",
      "Reserving space for 12 sample suffixes\n",
      "Generating random suffixes\n",
      "QSorting 12 sample offsets, eliminating duplicates\n",
      "QSorting sample offsets, eliminating duplicates time: 00:00:00\n",
      "Multikey QSorting 12 samples\n",
      "  (Using difference cover)\n",
      "  Multikey QSorting samples time: 00:00:00\n",
      "Calculating bucket sizes\n",
      "Splitting and merging\n",
      "  Splitting and merging time: 00:00:00\n",
      "Avg bucket size: 7.83861e+08 (target: 146973967)\n",
      "Converting suffix-array elements to index image\n",
      "Allocating ftab, absorbFtab\n",
      "Entering Ebwt loop\n",
      "Getting block 1 of 1\n",
      "  No samples; assembling all-inclusive block\n",
      "  Sorting block of length 783861161 for bucket 1\n",
      "  (Using difference cover)\n",
      "  Sorting block time: 00:07:37\n",
      "Returning block of 783861162 for bucket 1\n",
      "Exited Ebwt loop\n",
      "fchr[A]: 0\n",
      "fchr[C]: 391954798\n",
      "fchr[G]: 524347498\n",
      "fchr[T]: 524347498\n",
      "fchr[$]: 783861161\n",
      "Exiting Ebwt::buildToDisk()\n",
      "Returning from initFromVector\n",
      "Wrote 266840795 bytes to primary EBWT file: BS_GA.rev.1.bt2\n",
      "Wrote 195965296 bytes to secondary EBWT file: BS_GA.rev.2.bt2\n",
      "Re-opening _in1 and _in2 as input streams\n",
      "Returning from Ebwt constructor\n",
      "Headers:\n",
      "    len: 783861161\n",
      "    bwtLen: 783861162\n",
      "    sz: 195965291\n",
      "    bwtSz: 195965291\n",
      "    lineRate: 6\n",
      "    offRate: 4\n",
      "    offMask: 0xfffffff0\n",
      "    ftabChars: 10\n",
      "    eftabLen: 20\n",
      "    eftabSz: 80\n",
      "    ftabLen: 1048577\n",
      "    ftabSz: 4194308\n",
      "    offsLen: 48991323\n",
      "    offsSz: 195965292\n",
      "    lineSz: 64\n",
      "    sideSz: 64\n",
      "    sideBwtSz: 48\n",
      "    sideBwtLen: 192\n",
      "    numSides: 4082611\n",
      "    numLines: 4082611\n",
      "    ebwtTotLen: 261287104\n",
      "    ebwtTotSz: 261287104\n",
      "    color: 0\n",
      "    reverse: 1\n",
      "Total time for backward call to driver() for mirror index: 00:10:22\n",
      "  Sorting block time: 00:09:32\n",
      "Returning block of 783861162 for bucket 1\n",
      "Exited Ebwt loop\n",
      "fchr[A]: 0\n",
      "fchr[C]: 259567700\n",
      "fchr[G]: 259567700\n",
      "fchr[T]: 391954798\n",
      "fchr[$]: 783861161\n",
      "Exiting Ebwt::buildToDisk()\n",
      "Returning from initFromVector\n",
      "Wrote 266840795 bytes to primary EBWT file: BS_CT.rev.1.bt2\n",
      "Wrote 195965296 bytes to secondary EBWT file: BS_CT.rev.2.bt2\n",
      "Re-opening _in1 and _in2 as input streams\n",
      "Returning from Ebwt constructor\n",
      "Headers:\n",
      "    len: 783861161\n",
      "    bwtLen: 783861162\n",
      "    sz: 195965291\n",
      "    bwtSz: 195965291\n",
      "    lineRate: 6\n",
      "    offRate: 4\n",
      "    offMask: 0xfffffff0\n",
      "    ftabChars: 10\n",
      "    eftabLen: 20\n",
      "    eftabSz: 80\n",
      "    ftabLen: 1048577\n",
      "    ftabSz: 4194308\n",
      "    offsLen: 48991323\n",
      "    offsSz: 195965292\n",
      "    lineSz: 64\n",
      "    sideSz: 64\n",
      "    sideBwtSz: 48\n",
      "    sideBwtLen: 192\n",
      "    numSides: 4082611\n",
      "    numLines: 4082611\n",
      "    ebwtTotLen: 261287104\n",
      "    ebwtTotSz: 261287104\n",
      "    color: 0\n",
      "    reverse: 1\n",
      "Total time for backward call to driver() for mirror index: 00:12:01\n"
     ]
    }
   ],
   "source": [
    "!/Applications/bioinfo/Bismark_v0.20.0/bismark_genome_preparation \\\n",
    "--verbose \\\n",
    "--path_to_bowtie /Applications/bioinfo/bowtie2-2.3.4.1-macos-x86_64/ \\\n",
    "/Volumes/Serine/wd/19-11-05/v074/"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "collapsed": true,
    "deletable": true,
    "editable": true
   },
   "source": [
    "## Cov2cyt"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {
    "collapsed": false,
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Summary of parameters for genome-wide cytosine report:\n",
      "==============================================================================\n",
      "Coverage infile:\t\t\t\t/Volumes/Serine/wd/19-12-17/bismark_coverage_files/EPI-103_S27_L005_R1_001_val_1_bismark_bt2_pe.deduplicated.bismark.cov.gz\n",
      "Output directory:\t\t\t\t><\n",
      "Parent directory:\t\t\t\t>/Users/sr320/git-repos/nb-2019/P_generosa/<\n",
      "Genome directory:\t\t\t\t>/Volumes/Serine/wd/19-11-05/genome/<\n",
      "CX context:\t\t\t\t\tno (CpG context only, default)\n",
      "Pooling CpG top/bottom strand evidence:\t\tyes\n",
      "Genome coordinates used:\t\t\t1-based (default)\n",
      "GZIP compression:\t\t\t\tno\n",
      "Split by chromosome:\t\t\t\tno\n",
      "\n",
      "\n",
      "Now reading in and storing sequence information of the genome specified in: /Volumes/Serine/wd/19-11-05/genome/\n",
      "\n",
      "chr Scaffold_01 (89643857 bp)\n",
      "chr Scaffold_02 (69596280 bp)\n",
      "chr Scaffold_03 (57743597 bp)\n",
      "chr Scaffold_04 (65288255 bp)\n",
      "chr Scaffold_05 (67248332 bp)\n",
      "chr Scaffold_06 (61759565 bp)\n",
      "chr Scaffold_07 (43120122 bp)\n",
      "chr Scaffold_08 (61151155 bp)\n",
      "chr Scaffold_09 (38581958 bp)\n",
      "chr Scaffold_10 (53961475 bp)\n",
      "chr Scaffold_11 (51449921 bp)\n",
      "chr Scaffold_12 (50438331 bp)\n",
      "chr Scaffold_13 (44396874 bp)\n",
      "chr Scaffold_14 (45393038 bp)\n",
      "chr Scaffold_15 (47938513 bp)\n",
      "chr Scaffold_16 (31980953 bp)\n",
      "chr Scaffold_17 (34923512 bp)\n",
      "chr Scaffold_18 (27737463 bp)\n",
      "\n",
      "Stored sequence information of 18 chromosomes/scaffolds in total\n",
      "\n",
      "==============================================================================\n",
      "Methylation information will now be written into a genome-wide cytosine report\n",
      "==============================================================================\n",
      "\n",
      ">>> Writing genome-wide cytosine report to: /Volumes/Serine/wd/19-12-17/EPI-103_S27_L005.CpG_report.txt <<<\n",
      "\n",
      "Storing all covered cytosine positions for chromosome: Scaffold_01\n",
      "Writing cytosine report for chromosome Scaffold_01 (stored 1084861 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_02 (stored 838867 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_03 (stored 675348 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_04 (stored 771355 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_05 (stored 694256 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_06 (stored 676942 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_07 (stored 495336 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_08 (stored 695303 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_09 (stored 438486 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_10 (stored 575576 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_11 (stored 570579 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_12 (stored 606443 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_13 (stored 489725 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_14 (stored 626011 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_15 (stored 499547 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_16 (stored 395422 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_17 (stored 420528 different covered positions)\n",
      "Writing cytosine report for last chromosome Scaffold_18 (stored 304048 different covered positions)\n",
      "Finished writing out cytosine report for covered chromosomes (processed 18 chromosomes/scaffolds in total)\n",
      "\n",
      "Now processing chromosomes that were not covered by any methylation calls in the coverage file...\n",
      "All chromosomes in the genome were covered by at least some reads. coverage2cytosine processing complete.\n",
      "\n",
      "Now merging top and bottom strand CpGs into a single CG dinucleotide entity (reading from file >>/Volumes/Serine/wd/19-12-17/EPI-103_S27_L005.CpG_report.txt<<, in output directory '')\n",
      ">>> Writing a new coverage file with top and bottom strand CpG methylation evidence merged to /Volumes/Serine/wd/19-12-17/EPI-103_S27_L005.CpG_report.merged_CpG_evidence.cov <<<\n",
      "\n",
      "CpG merging complete. Good luck finding DMRs with bsseq!\n",
      "\n",
      "Summary of parameters for genome-wide cytosine report:\n",
      "==============================================================================\n",
      "Coverage infile:\t\t\t\t/Volumes/Serine/wd/19-12-17/bismark_coverage_files/EPI-104_S28_L005_R1_001_val_1_bismark_bt2_pe.deduplicated.bismark.cov.gz\n",
      "Output directory:\t\t\t\t><\n",
      "Parent directory:\t\t\t\t>/Users/sr320/git-repos/nb-2019/P_generosa/<\n",
      "Genome directory:\t\t\t\t>/Volumes/Serine/wd/19-11-05/genome/<\n",
      "CX context:\t\t\t\t\tno (CpG context only, default)\n",
      "Pooling CpG top/bottom strand evidence:\t\tyes\n",
      "Genome coordinates used:\t\t\t1-based (default)\n",
      "GZIP compression:\t\t\t\tno\n",
      "Split by chromosome:\t\t\t\tno\n",
      "\n",
      "\n",
      "Now reading in and storing sequence information of the genome specified in: /Volumes/Serine/wd/19-11-05/genome/\n",
      "\n",
      "chr Scaffold_01 (89643857 bp)\n",
      "chr Scaffold_02 (69596280 bp)\n",
      "chr Scaffold_03 (57743597 bp)\n",
      "chr Scaffold_04 (65288255 bp)\n",
      "chr Scaffold_05 (67248332 bp)\n",
      "chr Scaffold_06 (61759565 bp)\n",
      "chr Scaffold_07 (43120122 bp)\n",
      "chr Scaffold_08 (61151155 bp)\n",
      "chr Scaffold_09 (38581958 bp)\n",
      "chr Scaffold_10 (53961475 bp)\n",
      "chr Scaffold_11 (51449921 bp)\n",
      "chr Scaffold_12 (50438331 bp)\n",
      "chr Scaffold_13 (44396874 bp)\n",
      "chr Scaffold_14 (45393038 bp)\n",
      "chr Scaffold_15 (47938513 bp)\n",
      "chr Scaffold_16 (31980953 bp)\n",
      "chr Scaffold_17 (34923512 bp)\n",
      "chr Scaffold_18 (27737463 bp)\n",
      "\n",
      "Stored sequence information of 18 chromosomes/scaffolds in total\n",
      "\n",
      "==============================================================================\n",
      "Methylation information will now be written into a genome-wide cytosine report\n",
      "==============================================================================\n",
      "\n",
      ">>> Writing genome-wide cytosine report to: /Volumes/Serine/wd/19-12-17/EPI-104_S28_L005.CpG_report.txt <<<\n",
      "\n",
      "Storing all covered cytosine positions for chromosome: Scaffold_01\n",
      "Writing cytosine report for chromosome Scaffold_01 (stored 1245587 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_02 (stored 972582 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_03 (stored 779148 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_04 (stored 885398 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_05 (stored 806823 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_06 (stored 781504 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_07 (stored 574871 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_08 (stored 805014 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_09 (stored 508231 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_10 (stored 670718 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_11 (stored 654979 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_12 (stored 701313 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_13 (stored 568484 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_14 (stored 707519 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_15 (stored 574717 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_16 (stored 454207 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_17 (stored 481518 different covered positions)\n",
      "Writing cytosine report for last chromosome Scaffold_18 (stored 346412 different covered positions)\n",
      "Finished writing out cytosine report for covered chromosomes (processed 18 chromosomes/scaffolds in total)\n",
      "\n",
      "Now processing chromosomes that were not covered by any methylation calls in the coverage file...\n",
      "All chromosomes in the genome were covered by at least some reads. coverage2cytosine processing complete.\n",
      "\n",
      "Now merging top and bottom strand CpGs into a single CG dinucleotide entity (reading from file >>/Volumes/Serine/wd/19-12-17/EPI-104_S28_L005.CpG_report.txt<<, in output directory '')\n",
      ">>> Writing a new coverage file with top and bottom strand CpG methylation evidence merged to /Volumes/Serine/wd/19-12-17/EPI-104_S28_L005.CpG_report.merged_CpG_evidence.cov <<<\n",
      "\n",
      "CpG merging complete. Good luck finding DMRs with bsseq!\n",
      "\n",
      "Summary of parameters for genome-wide cytosine report:\n",
      "==============================================================================\n",
      "Coverage infile:\t\t\t\t/Volumes/Serine/wd/19-12-17/bismark_coverage_files/EPI-111_S29_L005_R1_001_val_1_bismark_bt2_pe.deduplicated.bismark.cov.gz\n",
      "Output directory:\t\t\t\t><\n",
      "Parent directory:\t\t\t\t>/Users/sr320/git-repos/nb-2019/P_generosa/<\n",
      "Genome directory:\t\t\t\t>/Volumes/Serine/wd/19-11-05/genome/<\n",
      "CX context:\t\t\t\t\tno (CpG context only, default)\n",
      "Pooling CpG top/bottom strand evidence:\t\tyes\n",
      "Genome coordinates used:\t\t\t1-based (default)\n",
      "GZIP compression:\t\t\t\tno\n",
      "Split by chromosome:\t\t\t\tno\n",
      "\n",
      "\n",
      "Now reading in and storing sequence information of the genome specified in: /Volumes/Serine/wd/19-11-05/genome/\n",
      "\n",
      "chr Scaffold_01 (89643857 bp)\n",
      "chr Scaffold_02 (69596280 bp)\n",
      "chr Scaffold_03 (57743597 bp)\n",
      "chr Scaffold_04 (65288255 bp)\n",
      "chr Scaffold_05 (67248332 bp)\n",
      "chr Scaffold_06 (61759565 bp)\n",
      "chr Scaffold_07 (43120122 bp)\n",
      "chr Scaffold_08 (61151155 bp)\n",
      "chr Scaffold_09 (38581958 bp)\n",
      "chr Scaffold_10 (53961475 bp)\n",
      "chr Scaffold_11 (51449921 bp)\n",
      "chr Scaffold_12 (50438331 bp)\n",
      "chr Scaffold_13 (44396874 bp)\n",
      "chr Scaffold_14 (45393038 bp)\n",
      "chr Scaffold_15 (47938513 bp)\n",
      "chr Scaffold_16 (31980953 bp)\n",
      "chr Scaffold_17 (34923512 bp)\n",
      "chr Scaffold_18 (27737463 bp)\n",
      "\n",
      "Stored sequence information of 18 chromosomes/scaffolds in total\n",
      "\n",
      "==============================================================================\n",
      "Methylation information will now be written into a genome-wide cytosine report\n",
      "==============================================================================\n",
      "\n",
      ">>> Writing genome-wide cytosine report to: /Volumes/Serine/wd/19-12-17/EPI-111_S29_L005.CpG_report.txt <<<\n",
      "\n",
      "Storing all covered cytosine positions for chromosome: Scaffold_01\n",
      "Writing cytosine report for chromosome Scaffold_01 (stored 1191107 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_02 (stored 919576 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_03 (stored 734661 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_04 (stored 845147 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_05 (stored 756515 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_06 (stored 738196 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_07 (stored 545035 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_08 (stored 758039 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_09 (stored 480139 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_10 (stored 634255 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_11 (stored 620582 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_12 (stored 664082 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_13 (stored 538729 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_14 (stored 676868 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_15 (stored 545617 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_16 (stored 432366 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_17 (stored 462079 different covered positions)\n",
      "Writing cytosine report for last chromosome Scaffold_18 (stored 328702 different covered positions)\n",
      "Finished writing out cytosine report for covered chromosomes (processed 18 chromosomes/scaffolds in total)\n",
      "\n",
      "Now processing chromosomes that were not covered by any methylation calls in the coverage file...\n",
      "All chromosomes in the genome were covered by at least some reads. coverage2cytosine processing complete.\n",
      "\n",
      "Now merging top and bottom strand CpGs into a single CG dinucleotide entity (reading from file >>/Volumes/Serine/wd/19-12-17/EPI-111_S29_L005.CpG_report.txt<<, in output directory '')\n",
      ">>> Writing a new coverage file with top and bottom strand CpG methylation evidence merged to /Volumes/Serine/wd/19-12-17/EPI-111_S29_L005.CpG_report.merged_CpG_evidence.cov <<<\n",
      "\n",
      "CpG merging complete. Good luck finding DMRs with bsseq!\n",
      "\n",
      "Summary of parameters for genome-wide cytosine report:\n",
      "==============================================================================\n",
      "Coverage infile:\t\t\t\t/Volumes/Serine/wd/19-12-17/bismark_coverage_files/EPI-113_S30_L005_R1_001_val_1_bismark_bt2_pe.deduplicated.bismark.cov.gz\n",
      "Output directory:\t\t\t\t><\n",
      "Parent directory:\t\t\t\t>/Users/sr320/git-repos/nb-2019/P_generosa/<\n",
      "Genome directory:\t\t\t\t>/Volumes/Serine/wd/19-11-05/genome/<\n",
      "CX context:\t\t\t\t\tno (CpG context only, default)\n",
      "Pooling CpG top/bottom strand evidence:\t\tyes\n",
      "Genome coordinates used:\t\t\t1-based (default)\n",
      "GZIP compression:\t\t\t\tno\n",
      "Split by chromosome:\t\t\t\tno\n",
      "\n",
      "\n",
      "Now reading in and storing sequence information of the genome specified in: /Volumes/Serine/wd/19-11-05/genome/\n",
      "\n",
      "chr Scaffold_01 (89643857 bp)\n",
      "chr Scaffold_02 (69596280 bp)\n",
      "chr Scaffold_03 (57743597 bp)\n",
      "chr Scaffold_04 (65288255 bp)\n",
      "chr Scaffold_05 (67248332 bp)\n",
      "chr Scaffold_06 (61759565 bp)\n",
      "chr Scaffold_07 (43120122 bp)\n",
      "chr Scaffold_08 (61151155 bp)\n",
      "chr Scaffold_09 (38581958 bp)\n",
      "chr Scaffold_10 (53961475 bp)\n",
      "chr Scaffold_11 (51449921 bp)\n",
      "chr Scaffold_12 (50438331 bp)\n",
      "chr Scaffold_13 (44396874 bp)\n",
      "chr Scaffold_14 (45393038 bp)\n",
      "chr Scaffold_15 (47938513 bp)\n",
      "chr Scaffold_16 (31980953 bp)\n",
      "chr Scaffold_17 (34923512 bp)\n",
      "chr Scaffold_18 (27737463 bp)\n",
      "\n",
      "Stored sequence information of 18 chromosomes/scaffolds in total\n",
      "\n",
      "==============================================================================\n",
      "Methylation information will now be written into a genome-wide cytosine report\n",
      "==============================================================================\n",
      "\n",
      ">>> Writing genome-wide cytosine report to: /Volumes/Serine/wd/19-12-17/EPI-113_S30_L005.CpG_report.txt <<<\n",
      "\n",
      "Storing all covered cytosine positions for chromosome: Scaffold_01\n",
      "Writing cytosine report for chromosome Scaffold_01 (stored 1144955 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_02 (stored 887443 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_03 (stored 713156 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_04 (stored 820426 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_05 (stored 742515 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_06 (stored 722827 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_07 (stored 533945 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_08 (stored 737135 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_09 (stored 466646 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_10 (stored 616450 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_11 (stored 600686 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_12 (stored 641027 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_13 (stored 521484 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_14 (stored 651840 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_15 (stored 530329 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_16 (stored 417711 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_17 (stored 446720 different covered positions)\n",
      "Writing cytosine report for last chromosome Scaffold_18 (stored 324079 different covered positions)\n",
      "Finished writing out cytosine report for covered chromosomes (processed 18 chromosomes/scaffolds in total)\n",
      "\n",
      "Now processing chromosomes that were not covered by any methylation calls in the coverage file...\n",
      "All chromosomes in the genome were covered by at least some reads. coverage2cytosine processing complete.\n",
      "\n",
      "Now merging top and bottom strand CpGs into a single CG dinucleotide entity (reading from file >>/Volumes/Serine/wd/19-12-17/EPI-113_S30_L005.CpG_report.txt<<, in output directory '')\n",
      ">>> Writing a new coverage file with top and bottom strand CpG methylation evidence merged to /Volumes/Serine/wd/19-12-17/EPI-113_S30_L005.CpG_report.merged_CpG_evidence.cov <<<\n",
      "\n",
      "CpG merging complete. Good luck finding DMRs with bsseq!\n",
      "\n",
      "Summary of parameters for genome-wide cytosine report:\n",
      "==============================================================================\n",
      "Coverage infile:\t\t\t\t/Volumes/Serine/wd/19-12-17/bismark_coverage_files/EPI-119_S31_L005_R1_001_val_1_bismark_bt2_pe.deduplicated.bismark.cov.gz\n",
      "Output directory:\t\t\t\t><\n",
      "Parent directory:\t\t\t\t>/Users/sr320/git-repos/nb-2019/P_generosa/<\n",
      "Genome directory:\t\t\t\t>/Volumes/Serine/wd/19-11-05/genome/<\n",
      "CX context:\t\t\t\t\tno (CpG context only, default)\n",
      "Pooling CpG top/bottom strand evidence:\t\tyes\n",
      "Genome coordinates used:\t\t\t1-based (default)\n",
      "GZIP compression:\t\t\t\tno\n",
      "Split by chromosome:\t\t\t\tno\n",
      "\n",
      "\n",
      "Now reading in and storing sequence information of the genome specified in: /Volumes/Serine/wd/19-11-05/genome/\n",
      "\n",
      "chr Scaffold_01 (89643857 bp)\n",
      "chr Scaffold_02 (69596280 bp)\n",
      "chr Scaffold_03 (57743597 bp)\n",
      "chr Scaffold_04 (65288255 bp)\n",
      "chr Scaffold_05 (67248332 bp)\n",
      "chr Scaffold_06 (61759565 bp)\n",
      "chr Scaffold_07 (43120122 bp)\n",
      "chr Scaffold_08 (61151155 bp)\n",
      "chr Scaffold_09 (38581958 bp)\n",
      "chr Scaffold_10 (53961475 bp)\n",
      "chr Scaffold_11 (51449921 bp)\n",
      "chr Scaffold_12 (50438331 bp)\n",
      "chr Scaffold_13 (44396874 bp)\n",
      "chr Scaffold_14 (45393038 bp)\n",
      "chr Scaffold_15 (47938513 bp)\n",
      "chr Scaffold_16 (31980953 bp)\n",
      "chr Scaffold_17 (34923512 bp)\n",
      "chr Scaffold_18 (27737463 bp)\n",
      "\n",
      "Stored sequence information of 18 chromosomes/scaffolds in total\n",
      "\n",
      "==============================================================================\n",
      "Methylation information will now be written into a genome-wide cytosine report\n",
      "==============================================================================\n",
      "\n",
      ">>> Writing genome-wide cytosine report to: /Volumes/Serine/wd/19-12-17/EPI-119_S31_L005.CpG_report.txt <<<\n",
      "\n",
      "Storing all covered cytosine positions for chromosome: Scaffold_01\n",
      "Writing cytosine report for chromosome Scaffold_01 (stored 1258095 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_02 (stored 981838 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_03 (stored 783744 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_04 (stored 903172 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_05 (stored 814296 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_06 (stored 793418 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_07 (stored 581495 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_08 (stored 816384 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_09 (stored 515848 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_10 (stored 677051 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_11 (stored 672248 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_12 (stored 707857 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_13 (stored 578501 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_14 (stored 726275 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_15 (stored 589938 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_16 (stored 466175 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_17 (stored 495298 different covered positions)\n",
      "Writing cytosine report for last chromosome Scaffold_18 (stored 357030 different covered positions)\n",
      "Finished writing out cytosine report for covered chromosomes (processed 18 chromosomes/scaffolds in total)\n",
      "\n",
      "Now processing chromosomes that were not covered by any methylation calls in the coverage file...\n",
      "All chromosomes in the genome were covered by at least some reads. coverage2cytosine processing complete.\n",
      "\n",
      "Now merging top and bottom strand CpGs into a single CG dinucleotide entity (reading from file >>/Volumes/Serine/wd/19-12-17/EPI-119_S31_L005.CpG_report.txt<<, in output directory '')\n",
      ">>> Writing a new coverage file with top and bottom strand CpG methylation evidence merged to /Volumes/Serine/wd/19-12-17/EPI-119_S31_L005.CpG_report.merged_CpG_evidence.cov <<<\n",
      "\n",
      "CpG merging complete. Good luck finding DMRs with bsseq!\n",
      "\n",
      "Summary of parameters for genome-wide cytosine report:\n",
      "==============================================================================\n",
      "Coverage infile:\t\t\t\t/Volumes/Serine/wd/19-12-17/bismark_coverage_files/EPI-120_S32_L005_R1_001_val_1_bismark_bt2_pe.deduplicated.bismark.cov.gz\n",
      "Output directory:\t\t\t\t><\n",
      "Parent directory:\t\t\t\t>/Users/sr320/git-repos/nb-2019/P_generosa/<\n",
      "Genome directory:\t\t\t\t>/Volumes/Serine/wd/19-11-05/genome/<\n",
      "CX context:\t\t\t\t\tno (CpG context only, default)\n",
      "Pooling CpG top/bottom strand evidence:\t\tyes\n",
      "Genome coordinates used:\t\t\t1-based (default)\n",
      "GZIP compression:\t\t\t\tno\n",
      "Split by chromosome:\t\t\t\tno\n",
      "\n",
      "\n",
      "Now reading in and storing sequence information of the genome specified in: /Volumes/Serine/wd/19-11-05/genome/\n",
      "\n",
      "chr Scaffold_01 (89643857 bp)\n",
      "chr Scaffold_02 (69596280 bp)\n",
      "chr Scaffold_03 (57743597 bp)\n",
      "chr Scaffold_04 (65288255 bp)\n",
      "chr Scaffold_05 (67248332 bp)\n",
      "chr Scaffold_06 (61759565 bp)\n",
      "chr Scaffold_07 (43120122 bp)\n",
      "chr Scaffold_08 (61151155 bp)\n",
      "chr Scaffold_09 (38581958 bp)\n",
      "chr Scaffold_10 (53961475 bp)\n",
      "chr Scaffold_11 (51449921 bp)\n",
      "chr Scaffold_12 (50438331 bp)\n",
      "chr Scaffold_13 (44396874 bp)\n",
      "chr Scaffold_14 (45393038 bp)\n",
      "chr Scaffold_15 (47938513 bp)\n",
      "chr Scaffold_16 (31980953 bp)\n",
      "chr Scaffold_17 (34923512 bp)\n",
      "chr Scaffold_18 (27737463 bp)\n",
      "\n",
      "Stored sequence information of 18 chromosomes/scaffolds in total\n",
      "\n",
      "==============================================================================\n",
      "Methylation information will now be written into a genome-wide cytosine report\n",
      "==============================================================================\n",
      "\n",
      ">>> Writing genome-wide cytosine report to: /Volumes/Serine/wd/19-12-17/EPI-120_S32_L005.CpG_report.txt <<<\n",
      "\n",
      "Storing all covered cytosine positions for chromosome: Scaffold_01\n",
      "Writing cytosine report for chromosome Scaffold_01 (stored 1149296 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_02 (stored 891788 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_03 (stored 709909 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_04 (stored 825490 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_05 (stored 734685 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_06 (stored 713899 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_07 (stored 529135 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_08 (stored 735663 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_09 (stored 466720 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_10 (stored 618795 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_11 (stored 604403 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_12 (stored 644350 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_13 (stored 526388 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_14 (stored 660854 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_15 (stored 532091 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_16 (stored 418026 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_17 (stored 452442 different covered positions)\n",
      "Writing cytosine report for last chromosome Scaffold_18 (stored 318699 different covered positions)\n",
      "Finished writing out cytosine report for covered chromosomes (processed 18 chromosomes/scaffolds in total)\n",
      "\n",
      "Now processing chromosomes that were not covered by any methylation calls in the coverage file...\n",
      "All chromosomes in the genome were covered by at least some reads. coverage2cytosine processing complete.\n",
      "\n",
      "Now merging top and bottom strand CpGs into a single CG dinucleotide entity (reading from file >>/Volumes/Serine/wd/19-12-17/EPI-120_S32_L005.CpG_report.txt<<, in output directory '')\n",
      ">>> Writing a new coverage file with top and bottom strand CpG methylation evidence merged to /Volumes/Serine/wd/19-12-17/EPI-120_S32_L005.CpG_report.merged_CpG_evidence.cov <<<\n",
      "\n",
      "CpG merging complete. Good luck finding DMRs with bsseq!\n",
      "\n",
      "Summary of parameters for genome-wide cytosine report:\n",
      "==============================================================================\n",
      "Coverage infile:\t\t\t\t/Volumes/Serine/wd/19-12-17/bismark_coverage_files/EPI-127_S33_L005_R1_001_val_1_bismark_bt2_pe.deduplicated.bismark.cov.gz\n",
      "Output directory:\t\t\t\t><\n",
      "Parent directory:\t\t\t\t>/Users/sr320/git-repos/nb-2019/P_generosa/<\n",
      "Genome directory:\t\t\t\t>/Volumes/Serine/wd/19-11-05/genome/<\n",
      "CX context:\t\t\t\t\tno (CpG context only, default)\n",
      "Pooling CpG top/bottom strand evidence:\t\tyes\n",
      "Genome coordinates used:\t\t\t1-based (default)\n",
      "GZIP compression:\t\t\t\tno\n",
      "Split by chromosome:\t\t\t\tno\n",
      "\n",
      "\n",
      "Now reading in and storing sequence information of the genome specified in: /Volumes/Serine/wd/19-11-05/genome/\n",
      "\n",
      "chr Scaffold_01 (89643857 bp)\n",
      "chr Scaffold_02 (69596280 bp)\n",
      "chr Scaffold_03 (57743597 bp)\n",
      "chr Scaffold_04 (65288255 bp)\n",
      "chr Scaffold_05 (67248332 bp)\n",
      "chr Scaffold_06 (61759565 bp)\n",
      "chr Scaffold_07 (43120122 bp)\n",
      "chr Scaffold_08 (61151155 bp)\n",
      "chr Scaffold_09 (38581958 bp)\n",
      "chr Scaffold_10 (53961475 bp)\n",
      "chr Scaffold_11 (51449921 bp)\n",
      "chr Scaffold_12 (50438331 bp)\n",
      "chr Scaffold_13 (44396874 bp)\n",
      "chr Scaffold_14 (45393038 bp)\n",
      "chr Scaffold_15 (47938513 bp)\n",
      "chr Scaffold_16 (31980953 bp)\n",
      "chr Scaffold_17 (34923512 bp)\n",
      "chr Scaffold_18 (27737463 bp)\n",
      "\n",
      "Stored sequence information of 18 chromosomes/scaffolds in total\n",
      "\n",
      "==============================================================================\n",
      "Methylation information will now be written into a genome-wide cytosine report\n",
      "==============================================================================\n",
      "\n",
      ">>> Writing genome-wide cytosine report to: /Volumes/Serine/wd/19-12-17/EPI-127_S33_L005.CpG_report.txt <<<\n",
      "\n",
      "Storing all covered cytosine positions for chromosome: Scaffold_01\n",
      "Writing cytosine report for chromosome Scaffold_01 (stored 1080767 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_02 (stored 835656 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_03 (stored 669892 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_04 (stored 769248 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_05 (stored 695838 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_06 (stored 674370 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_07 (stored 497398 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_08 (stored 693054 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_09 (stored 441134 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_10 (stored 576963 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_11 (stored 571834 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_12 (stored 603117 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_13 (stored 492545 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_14 (stored 624704 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_15 (stored 499341 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_16 (stored 393815 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_17 (stored 418916 different covered positions)\n",
      "Writing cytosine report for last chromosome Scaffold_18 (stored 298019 different covered positions)\n",
      "Finished writing out cytosine report for covered chromosomes (processed 18 chromosomes/scaffolds in total)\n",
      "\n",
      "Now processing chromosomes that were not covered by any methylation calls in the coverage file...\n",
      "All chromosomes in the genome were covered by at least some reads. coverage2cytosine processing complete.\n",
      "\n",
      "Now merging top and bottom strand CpGs into a single CG dinucleotide entity (reading from file >>/Volumes/Serine/wd/19-12-17/EPI-127_S33_L005.CpG_report.txt<<, in output directory '')\n",
      ">>> Writing a new coverage file with top and bottom strand CpG methylation evidence merged to /Volumes/Serine/wd/19-12-17/EPI-127_S33_L005.CpG_report.merged_CpG_evidence.cov <<<\n",
      "\n",
      "CpG merging complete. Good luck finding DMRs with bsseq!\n",
      "\n",
      "Summary of parameters for genome-wide cytosine report:\n",
      "==============================================================================\n",
      "Coverage infile:\t\t\t\t/Volumes/Serine/wd/19-12-17/bismark_coverage_files/EPI-128_S34_L005_R1_001_val_1_bismark_bt2_pe.deduplicated.bismark.cov.gz\n",
      "Output directory:\t\t\t\t><\n",
      "Parent directory:\t\t\t\t>/Users/sr320/git-repos/nb-2019/P_generosa/<\n",
      "Genome directory:\t\t\t\t>/Volumes/Serine/wd/19-11-05/genome/<\n",
      "CX context:\t\t\t\t\tno (CpG context only, default)\n",
      "Pooling CpG top/bottom strand evidence:\t\tyes\n",
      "Genome coordinates used:\t\t\t1-based (default)\n",
      "GZIP compression:\t\t\t\tno\n",
      "Split by chromosome:\t\t\t\tno\n",
      "\n",
      "\n",
      "Now reading in and storing sequence information of the genome specified in: /Volumes/Serine/wd/19-11-05/genome/\n",
      "\n",
      "chr Scaffold_01 (89643857 bp)\n",
      "chr Scaffold_02 (69596280 bp)\n",
      "chr Scaffold_03 (57743597 bp)\n",
      "chr Scaffold_04 (65288255 bp)\n",
      "chr Scaffold_05 (67248332 bp)\n",
      "chr Scaffold_06 (61759565 bp)\n",
      "chr Scaffold_07 (43120122 bp)\n",
      "chr Scaffold_08 (61151155 bp)\n",
      "chr Scaffold_09 (38581958 bp)\n",
      "chr Scaffold_10 (53961475 bp)\n",
      "chr Scaffold_11 (51449921 bp)\n",
      "chr Scaffold_12 (50438331 bp)\n",
      "chr Scaffold_13 (44396874 bp)\n",
      "chr Scaffold_14 (45393038 bp)\n",
      "chr Scaffold_15 (47938513 bp)\n",
      "chr Scaffold_16 (31980953 bp)\n",
      "chr Scaffold_17 (34923512 bp)\n",
      "chr Scaffold_18 (27737463 bp)\n",
      "\n",
      "Stored sequence information of 18 chromosomes/scaffolds in total\n",
      "\n",
      "==============================================================================\n",
      "Methylation information will now be written into a genome-wide cytosine report\n",
      "==============================================================================\n",
      "\n",
      ">>> Writing genome-wide cytosine report to: /Volumes/Serine/wd/19-12-17/EPI-128_S34_L005.CpG_report.txt <<<\n",
      "\n",
      "Storing all covered cytosine positions for chromosome: Scaffold_01\n",
      "Writing cytosine report for chromosome Scaffold_01 (stored 1118299 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_02 (stored 868235 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_03 (stored 697636 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_04 (stored 794861 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_05 (stored 718042 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_06 (stored 698463 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_07 (stored 516544 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_08 (stored 718157 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_09 (stored 454963 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_10 (stored 598576 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_11 (stored 587298 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_12 (stored 625777 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_13 (stored 511044 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_14 (stored 646813 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_15 (stored 519831 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_16 (stored 408189 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_17 (stored 436717 different covered positions)\n",
      "Writing cytosine report for last chromosome Scaffold_18 (stored 309215 different covered positions)\n",
      "Finished writing out cytosine report for covered chromosomes (processed 18 chromosomes/scaffolds in total)\n",
      "\n",
      "Now processing chromosomes that were not covered by any methylation calls in the coverage file...\n",
      "All chromosomes in the genome were covered by at least some reads. coverage2cytosine processing complete.\n",
      "\n",
      "Now merging top and bottom strand CpGs into a single CG dinucleotide entity (reading from file >>/Volumes/Serine/wd/19-12-17/EPI-128_S34_L005.CpG_report.txt<<, in output directory '')\n",
      ">>> Writing a new coverage file with top and bottom strand CpG methylation evidence merged to /Volumes/Serine/wd/19-12-17/EPI-128_S34_L005.CpG_report.merged_CpG_evidence.cov <<<\n",
      "\n",
      "CpG merging complete. Good luck finding DMRs with bsseq!\n",
      "\n",
      "Summary of parameters for genome-wide cytosine report:\n",
      "==============================================================================\n",
      "Coverage infile:\t\t\t\t/Volumes/Serine/wd/19-12-17/bismark_coverage_files/EPI-135_S35_L005_R1_001_val_1_bismark_bt2_pe.deduplicated.bismark.cov.gz\n",
      "Output directory:\t\t\t\t><\n",
      "Parent directory:\t\t\t\t>/Users/sr320/git-repos/nb-2019/P_generosa/<\n",
      "Genome directory:\t\t\t\t>/Volumes/Serine/wd/19-11-05/genome/<\n",
      "CX context:\t\t\t\t\tno (CpG context only, default)\n",
      "Pooling CpG top/bottom strand evidence:\t\tyes\n",
      "Genome coordinates used:\t\t\t1-based (default)\n",
      "GZIP compression:\t\t\t\tno\n",
      "Split by chromosome:\t\t\t\tno\n",
      "\n",
      "\n",
      "Now reading in and storing sequence information of the genome specified in: /Volumes/Serine/wd/19-11-05/genome/\n",
      "\n",
      "chr Scaffold_01 (89643857 bp)\n",
      "chr Scaffold_02 (69596280 bp)\n",
      "chr Scaffold_03 (57743597 bp)\n",
      "chr Scaffold_04 (65288255 bp)\n",
      "chr Scaffold_05 (67248332 bp)\n",
      "chr Scaffold_06 (61759565 bp)\n",
      "chr Scaffold_07 (43120122 bp)\n",
      "chr Scaffold_08 (61151155 bp)\n",
      "chr Scaffold_09 (38581958 bp)\n",
      "chr Scaffold_10 (53961475 bp)\n",
      "chr Scaffold_11 (51449921 bp)\n",
      "chr Scaffold_12 (50438331 bp)\n",
      "chr Scaffold_13 (44396874 bp)\n",
      "chr Scaffold_14 (45393038 bp)\n",
      "chr Scaffold_15 (47938513 bp)\n",
      "chr Scaffold_16 (31980953 bp)\n",
      "chr Scaffold_17 (34923512 bp)\n",
      "chr Scaffold_18 (27737463 bp)\n",
      "\n",
      "Stored sequence information of 18 chromosomes/scaffolds in total\n",
      "\n",
      "==============================================================================\n",
      "Methylation information will now be written into a genome-wide cytosine report\n",
      "==============================================================================\n",
      "\n",
      ">>> Writing genome-wide cytosine report to: /Volumes/Serine/wd/19-12-17/EPI-135_S35_L005.CpG_report.txt <<<\n",
      "\n",
      "Storing all covered cytosine positions for chromosome: Scaffold_01\n",
      "Writing cytosine report for chromosome Scaffold_01 (stored 1111554 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_02 (stored 861610 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_03 (stored 696231 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_04 (stored 791967 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_05 (stored 710008 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_06 (stored 693184 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_07 (stored 512744 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_08 (stored 717754 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_09 (stored 453428 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_10 (stored 598852 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_11 (stored 583083 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_12 (stored 623176 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_13 (stored 508375 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_14 (stored 640525 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_15 (stored 514190 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_16 (stored 405937 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_17 (stored 432890 different covered positions)\n",
      "Writing cytosine report for last chromosome Scaffold_18 (stored 307670 different covered positions)\n",
      "Finished writing out cytosine report for covered chromosomes (processed 18 chromosomes/scaffolds in total)\n",
      "\n",
      "Now processing chromosomes that were not covered by any methylation calls in the coverage file...\n",
      "All chromosomes in the genome were covered by at least some reads. coverage2cytosine processing complete.\n",
      "\n",
      "Now merging top and bottom strand CpGs into a single CG dinucleotide entity (reading from file >>/Volumes/Serine/wd/19-12-17/EPI-135_S35_L005.CpG_report.txt<<, in output directory '')\n",
      ">>> Writing a new coverage file with top and bottom strand CpG methylation evidence merged to /Volumes/Serine/wd/19-12-17/EPI-135_S35_L005.CpG_report.merged_CpG_evidence.cov <<<\n",
      "\n",
      "CpG merging complete. Good luck finding DMRs with bsseq!\n",
      "\n",
      "Summary of parameters for genome-wide cytosine report:\n",
      "==============================================================================\n",
      "Coverage infile:\t\t\t\t/Volumes/Serine/wd/19-12-17/bismark_coverage_files/EPI-136_S36_L005_R1_001_val_1_bismark_bt2_pe.deduplicated.bismark.cov.gz\n",
      "Output directory:\t\t\t\t><\n",
      "Parent directory:\t\t\t\t>/Users/sr320/git-repos/nb-2019/P_generosa/<\n",
      "Genome directory:\t\t\t\t>/Volumes/Serine/wd/19-11-05/genome/<\n",
      "CX context:\t\t\t\t\tno (CpG context only, default)\n",
      "Pooling CpG top/bottom strand evidence:\t\tyes\n",
      "Genome coordinates used:\t\t\t1-based (default)\n",
      "GZIP compression:\t\t\t\tno\n",
      "Split by chromosome:\t\t\t\tno\n",
      "\n",
      "\n",
      "Now reading in and storing sequence information of the genome specified in: /Volumes/Serine/wd/19-11-05/genome/\n",
      "\n",
      "chr Scaffold_01 (89643857 bp)\n",
      "chr Scaffold_02 (69596280 bp)\n",
      "chr Scaffold_03 (57743597 bp)\n",
      "chr Scaffold_04 (65288255 bp)\n",
      "chr Scaffold_05 (67248332 bp)\n",
      "chr Scaffold_06 (61759565 bp)\n",
      "chr Scaffold_07 (43120122 bp)\n",
      "chr Scaffold_08 (61151155 bp)\n",
      "chr Scaffold_09 (38581958 bp)\n",
      "chr Scaffold_10 (53961475 bp)\n",
      "chr Scaffold_11 (51449921 bp)\n",
      "chr Scaffold_12 (50438331 bp)\n",
      "chr Scaffold_13 (44396874 bp)\n",
      "chr Scaffold_14 (45393038 bp)\n",
      "chr Scaffold_15 (47938513 bp)\n",
      "chr Scaffold_16 (31980953 bp)\n",
      "chr Scaffold_17 (34923512 bp)\n",
      "chr Scaffold_18 (27737463 bp)\n",
      "\n",
      "Stored sequence information of 18 chromosomes/scaffolds in total\n",
      "\n",
      "==============================================================================\n",
      "Methylation information will now be written into a genome-wide cytosine report\n",
      "==============================================================================\n",
      "\n",
      ">>> Writing genome-wide cytosine report to: /Volumes/Serine/wd/19-12-17/EPI-136_S36_L005.CpG_report.txt <<<\n",
      "\n",
      "Storing all covered cytosine positions for chromosome: Scaffold_01\n",
      "Writing cytosine report for chromosome Scaffold_01 (stored 1138865 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_02 (stored 879482 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_03 (stored 710165 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_04 (stored 808271 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_05 (stored 725471 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_06 (stored 714051 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_07 (stored 521796 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_08 (stored 728520 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_09 (stored 460988 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_10 (stored 609333 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_11 (stored 598730 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_12 (stored 634410 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_13 (stored 519292 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_14 (stored 654163 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_15 (stored 521834 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_16 (stored 415638 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_17 (stored 443714 different covered positions)\n",
      "Writing cytosine report for last chromosome Scaffold_18 (stored 314317 different covered positions)\n",
      "Finished writing out cytosine report for covered chromosomes (processed 18 chromosomes/scaffolds in total)\n",
      "\n",
      "Now processing chromosomes that were not covered by any methylation calls in the coverage file...\n",
      "All chromosomes in the genome were covered by at least some reads. coverage2cytosine processing complete.\n",
      "\n",
      "Now merging top and bottom strand CpGs into a single CG dinucleotide entity (reading from file >>/Volumes/Serine/wd/19-12-17/EPI-136_S36_L005.CpG_report.txt<<, in output directory '')\n",
      ">>> Writing a new coverage file with top and bottom strand CpG methylation evidence merged to /Volumes/Serine/wd/19-12-17/EPI-136_S36_L005.CpG_report.merged_CpG_evidence.cov <<<\n",
      "\n",
      "CpG merging complete. Good luck finding DMRs with bsseq!\n",
      "\n",
      "Summary of parameters for genome-wide cytosine report:\n",
      "==============================================================================\n",
      "Coverage infile:\t\t\t\t/Volumes/Serine/wd/19-12-17/bismark_coverage_files/EPI-143_S37_L005_R1_001_val_1_bismark_bt2_pe.deduplicated.bismark.cov.gz\n",
      "Output directory:\t\t\t\t><\n",
      "Parent directory:\t\t\t\t>/Users/sr320/git-repos/nb-2019/P_generosa/<\n",
      "Genome directory:\t\t\t\t>/Volumes/Serine/wd/19-11-05/genome/<\n",
      "CX context:\t\t\t\t\tno (CpG context only, default)\n",
      "Pooling CpG top/bottom strand evidence:\t\tyes\n",
      "Genome coordinates used:\t\t\t1-based (default)\n",
      "GZIP compression:\t\t\t\tno\n",
      "Split by chromosome:\t\t\t\tno\n",
      "\n",
      "\n",
      "Now reading in and storing sequence information of the genome specified in: /Volumes/Serine/wd/19-11-05/genome/\n",
      "\n",
      "chr Scaffold_01 (89643857 bp)\n",
      "chr Scaffold_02 (69596280 bp)\n",
      "chr Scaffold_03 (57743597 bp)\n",
      "chr Scaffold_04 (65288255 bp)\n",
      "chr Scaffold_05 (67248332 bp)\n",
      "chr Scaffold_06 (61759565 bp)\n",
      "chr Scaffold_07 (43120122 bp)\n",
      "chr Scaffold_08 (61151155 bp)\n",
      "chr Scaffold_09 (38581958 bp)\n",
      "chr Scaffold_10 (53961475 bp)\n",
      "chr Scaffold_11 (51449921 bp)\n",
      "chr Scaffold_12 (50438331 bp)\n",
      "chr Scaffold_13 (44396874 bp)\n",
      "chr Scaffold_14 (45393038 bp)\n",
      "chr Scaffold_15 (47938513 bp)\n",
      "chr Scaffold_16 (31980953 bp)\n",
      "chr Scaffold_17 (34923512 bp)\n",
      "chr Scaffold_18 (27737463 bp)\n",
      "\n",
      "Stored sequence information of 18 chromosomes/scaffolds in total\n",
      "\n",
      "==============================================================================\n",
      "Methylation information will now be written into a genome-wide cytosine report\n",
      "==============================================================================\n",
      "\n",
      ">>> Writing genome-wide cytosine report to: /Volumes/Serine/wd/19-12-17/EPI-143_S37_L005.CpG_report.txt <<<\n",
      "\n",
      "Storing all covered cytosine positions for chromosome: Scaffold_01\n",
      "Writing cytosine report for chromosome Scaffold_01 (stored 1031704 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_02 (stored 802134 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_03 (stored 645547 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_04 (stored 733058 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_05 (stored 658967 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_06 (stored 644780 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_07 (stored 476265 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_08 (stored 665879 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_09 (stored 423164 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_10 (stored 553777 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_11 (stored 544486 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_12 (stored 579307 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_13 (stored 471307 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_14 (stored 598154 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_15 (stored 478612 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_16 (stored 375442 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_17 (stored 405105 different covered positions)\n",
      "Writing cytosine report for last chromosome Scaffold_18 (stored 287266 different covered positions)\n",
      "Finished writing out cytosine report for covered chromosomes (processed 18 chromosomes/scaffolds in total)\n",
      "\n",
      "Now processing chromosomes that were not covered by any methylation calls in the coverage file...\n",
      "All chromosomes in the genome were covered by at least some reads. coverage2cytosine processing complete.\n",
      "\n",
      "Now merging top and bottom strand CpGs into a single CG dinucleotide entity (reading from file >>/Volumes/Serine/wd/19-12-17/EPI-143_S37_L005.CpG_report.txt<<, in output directory '')\n",
      ">>> Writing a new coverage file with top and bottom strand CpG methylation evidence merged to /Volumes/Serine/wd/19-12-17/EPI-143_S37_L005.CpG_report.merged_CpG_evidence.cov <<<\n",
      "\n",
      "CpG merging complete. Good luck finding DMRs with bsseq!\n",
      "\n",
      "Summary of parameters for genome-wide cytosine report:\n",
      "==============================================================================\n",
      "Coverage infile:\t\t\t\t/Volumes/Serine/wd/19-12-17/bismark_coverage_files/EPI-145_S38_L005_R1_001_val_1_bismark_bt2_pe.deduplicated.bismark.cov.gz\n",
      "Output directory:\t\t\t\t><\n",
      "Parent directory:\t\t\t\t>/Users/sr320/git-repos/nb-2019/P_generosa/<\n",
      "Genome directory:\t\t\t\t>/Volumes/Serine/wd/19-11-05/genome/<\n",
      "CX context:\t\t\t\t\tno (CpG context only, default)\n",
      "Pooling CpG top/bottom strand evidence:\t\tyes\n",
      "Genome coordinates used:\t\t\t1-based (default)\n",
      "GZIP compression:\t\t\t\tno\n",
      "Split by chromosome:\t\t\t\tno\n",
      "\n",
      "\n",
      "Now reading in and storing sequence information of the genome specified in: /Volumes/Serine/wd/19-11-05/genome/\n",
      "\n",
      "chr Scaffold_01 (89643857 bp)\n",
      "chr Scaffold_02 (69596280 bp)\n",
      "chr Scaffold_03 (57743597 bp)\n",
      "chr Scaffold_04 (65288255 bp)\n",
      "chr Scaffold_05 (67248332 bp)\n",
      "chr Scaffold_06 (61759565 bp)\n",
      "chr Scaffold_07 (43120122 bp)\n",
      "chr Scaffold_08 (61151155 bp)\n",
      "chr Scaffold_09 (38581958 bp)\n",
      "chr Scaffold_10 (53961475 bp)\n",
      "chr Scaffold_11 (51449921 bp)\n",
      "chr Scaffold_12 (50438331 bp)\n",
      "chr Scaffold_13 (44396874 bp)\n",
      "chr Scaffold_14 (45393038 bp)\n",
      "chr Scaffold_15 (47938513 bp)\n",
      "chr Scaffold_16 (31980953 bp)\n",
      "chr Scaffold_17 (34923512 bp)\n",
      "chr Scaffold_18 (27737463 bp)\n",
      "\n",
      "Stored sequence information of 18 chromosomes/scaffolds in total\n",
      "\n",
      "==============================================================================\n",
      "Methylation information will now be written into a genome-wide cytosine report\n",
      "==============================================================================\n",
      "\n",
      ">>> Writing genome-wide cytosine report to: /Volumes/Serine/wd/19-12-17/EPI-145_S38_L005.CpG_report.txt <<<\n",
      "\n",
      "Storing all covered cytosine positions for chromosome: Scaffold_01\n",
      "Writing cytosine report for chromosome Scaffold_01 (stored 1123355 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_02 (stored 869315 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_03 (stored 694748 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_04 (stored 795166 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_05 (stored 718802 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_06 (stored 703924 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_07 (stored 517083 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_08 (stored 720656 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_09 (stored 456431 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_10 (stored 601856 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_11 (stored 595335 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_12 (stored 628128 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_13 (stored 514060 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_14 (stored 640967 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_15 (stored 519907 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_16 (stored 406366 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_17 (stored 438128 different covered positions)\n",
      "Writing cytosine report for last chromosome Scaffold_18 (stored 313701 different covered positions)\n",
      "Finished writing out cytosine report for covered chromosomes (processed 18 chromosomes/scaffolds in total)\n",
      "\n",
      "Now processing chromosomes that were not covered by any methylation calls in the coverage file...\n",
      "All chromosomes in the genome were covered by at least some reads. coverage2cytosine processing complete.\n",
      "\n",
      "Now merging top and bottom strand CpGs into a single CG dinucleotide entity (reading from file >>/Volumes/Serine/wd/19-12-17/EPI-145_S38_L005.CpG_report.txt<<, in output directory '')\n",
      ">>> Writing a new coverage file with top and bottom strand CpG methylation evidence merged to /Volumes/Serine/wd/19-12-17/EPI-145_S38_L005.CpG_report.merged_CpG_evidence.cov <<<\n",
      "\n",
      "CpG merging complete. Good luck finding DMRs with bsseq!\n",
      "\n",
      "Summary of parameters for genome-wide cytosine report:\n",
      "==============================================================================\n",
      "Coverage infile:\t\t\t\t/Volumes/Serine/wd/19-12-17/bismark_coverage_files/EPI-151_S2_L002_R1_001_val_1_bismark_bt2_pe.deduplicated.bismark.cov.gz\n",
      "Output directory:\t\t\t\t><\n",
      "Parent directory:\t\t\t\t>/Users/sr320/git-repos/nb-2019/P_generosa/<\n",
      "Genome directory:\t\t\t\t>/Volumes/Serine/wd/19-11-05/genome/<\n",
      "CX context:\t\t\t\t\tno (CpG context only, default)\n",
      "Pooling CpG top/bottom strand evidence:\t\tyes\n",
      "Genome coordinates used:\t\t\t1-based (default)\n",
      "GZIP compression:\t\t\t\tno\n",
      "Split by chromosome:\t\t\t\tno\n",
      "\n",
      "\n",
      "Now reading in and storing sequence information of the genome specified in: /Volumes/Serine/wd/19-11-05/genome/\n",
      "\n",
      "chr Scaffold_01 (89643857 bp)\n",
      "chr Scaffold_02 (69596280 bp)\n",
      "chr Scaffold_03 (57743597 bp)\n",
      "chr Scaffold_04 (65288255 bp)\n",
      "chr Scaffold_05 (67248332 bp)\n",
      "chr Scaffold_06 (61759565 bp)\n",
      "chr Scaffold_07 (43120122 bp)\n",
      "chr Scaffold_08 (61151155 bp)\n",
      "chr Scaffold_09 (38581958 bp)\n",
      "chr Scaffold_10 (53961475 bp)\n",
      "chr Scaffold_11 (51449921 bp)\n",
      "chr Scaffold_12 (50438331 bp)\n",
      "chr Scaffold_13 (44396874 bp)\n",
      "chr Scaffold_14 (45393038 bp)\n",
      "chr Scaffold_15 (47938513 bp)\n",
      "chr Scaffold_16 (31980953 bp)\n",
      "chr Scaffold_17 (34923512 bp)\n",
      "chr Scaffold_18 (27737463 bp)\n",
      "\n",
      "Stored sequence information of 18 chromosomes/scaffolds in total\n",
      "\n",
      "==============================================================================\n",
      "Methylation information will now be written into a genome-wide cytosine report\n",
      "==============================================================================\n",
      "\n",
      ">>> Writing genome-wide cytosine report to: /Volumes/Serine/wd/19-12-17/EPI-151_S2_L002.CpG_report.txt <<<\n",
      "\n",
      "Storing all covered cytosine positions for chromosome: Scaffold_01\n",
      "Writing cytosine report for chromosome Scaffold_01 (stored 917488 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_02 (stored 710836 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_03 (stored 572506 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_04 (stored 651179 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_05 (stored 586577 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_06 (stored 571817 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_07 (stored 421052 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_08 (stored 589920 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_09 (stored 373543 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_10 (stored 493551 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_11 (stored 477687 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_12 (stored 512479 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_13 (stored 417542 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_14 (stored 534622 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_15 (stored 426534 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_16 (stored 331754 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_17 (stored 359835 different covered positions)\n",
      "Writing cytosine report for last chromosome Scaffold_18 (stored 260112 different covered positions)\n",
      "Finished writing out cytosine report for covered chromosomes (processed 18 chromosomes/scaffolds in total)\n",
      "\n",
      "Now processing chromosomes that were not covered by any methylation calls in the coverage file...\n",
      "All chromosomes in the genome were covered by at least some reads. coverage2cytosine processing complete.\n",
      "\n",
      "Now merging top and bottom strand CpGs into a single CG dinucleotide entity (reading from file >>/Volumes/Serine/wd/19-12-17/EPI-151_S2_L002.CpG_report.txt<<, in output directory '')\n",
      ">>> Writing a new coverage file with top and bottom strand CpG methylation evidence merged to /Volumes/Serine/wd/19-12-17/EPI-151_S2_L002.CpG_report.merged_CpG_evidence.cov <<<\n",
      "\n",
      "CpG merging complete. Good luck finding DMRs with bsseq!\n",
      "\n",
      "Summary of parameters for genome-wide cytosine report:\n",
      "==============================================================================\n",
      "Coverage infile:\t\t\t\t/Volumes/Serine/wd/19-12-17/bismark_coverage_files/EPI-152_S3_L002_R1_001_val_1_bismark_bt2_pe.deduplicated.bismark.cov.gz\n",
      "Output directory:\t\t\t\t><\n",
      "Parent directory:\t\t\t\t>/Users/sr320/git-repos/nb-2019/P_generosa/<\n",
      "Genome directory:\t\t\t\t>/Volumes/Serine/wd/19-11-05/genome/<\n",
      "CX context:\t\t\t\t\tno (CpG context only, default)\n",
      "Pooling CpG top/bottom strand evidence:\t\tyes\n",
      "Genome coordinates used:\t\t\t1-based (default)\n",
      "GZIP compression:\t\t\t\tno\n",
      "Split by chromosome:\t\t\t\tno\n",
      "\n",
      "\n",
      "Now reading in and storing sequence information of the genome specified in: /Volumes/Serine/wd/19-11-05/genome/\n",
      "\n",
      "chr Scaffold_01 (89643857 bp)\n",
      "chr Scaffold_02 (69596280 bp)\n",
      "chr Scaffold_03 (57743597 bp)\n",
      "chr Scaffold_04 (65288255 bp)\n",
      "chr Scaffold_05 (67248332 bp)\n",
      "chr Scaffold_06 (61759565 bp)\n",
      "chr Scaffold_07 (43120122 bp)\n",
      "chr Scaffold_08 (61151155 bp)\n",
      "chr Scaffold_09 (38581958 bp)\n",
      "chr Scaffold_10 (53961475 bp)\n",
      "chr Scaffold_11 (51449921 bp)\n",
      "chr Scaffold_12 (50438331 bp)\n",
      "chr Scaffold_13 (44396874 bp)\n",
      "chr Scaffold_14 (45393038 bp)\n",
      "chr Scaffold_15 (47938513 bp)\n",
      "chr Scaffold_16 (31980953 bp)\n",
      "chr Scaffold_17 (34923512 bp)\n",
      "chr Scaffold_18 (27737463 bp)\n",
      "\n",
      "Stored sequence information of 18 chromosomes/scaffolds in total\n",
      "\n",
      "==============================================================================\n",
      "Methylation information will now be written into a genome-wide cytosine report\n",
      "==============================================================================\n",
      "\n",
      ">>> Writing genome-wide cytosine report to: /Volumes/Serine/wd/19-12-17/EPI-152_S3_L002.CpG_report.txt <<<\n",
      "\n",
      "Storing all covered cytosine positions for chromosome: Scaffold_01\n",
      "Writing cytosine report for chromosome Scaffold_01 (stored 985275 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_02 (stored 761753 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_03 (stored 612427 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_04 (stored 692270 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_05 (stored 628856 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_06 (stored 611275 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_07 (stored 456942 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_08 (stored 635587 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_09 (stored 399396 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_10 (stored 531151 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_11 (stored 510000 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_12 (stored 552500 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_13 (stored 451232 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_14 (stored 569697 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_15 (stored 455064 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_16 (stored 355309 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_17 (stored 384776 different covered positions)\n",
      "Writing cytosine report for last chromosome Scaffold_18 (stored 274441 different covered positions)\n",
      "Finished writing out cytosine report for covered chromosomes (processed 18 chromosomes/scaffolds in total)\n",
      "\n",
      "Now processing chromosomes that were not covered by any methylation calls in the coverage file...\n",
      "All chromosomes in the genome were covered by at least some reads. coverage2cytosine processing complete.\n",
      "\n",
      "Now merging top and bottom strand CpGs into a single CG dinucleotide entity (reading from file >>/Volumes/Serine/wd/19-12-17/EPI-152_S3_L002.CpG_report.txt<<, in output directory '')\n",
      ">>> Writing a new coverage file with top and bottom strand CpG methylation evidence merged to /Volumes/Serine/wd/19-12-17/EPI-152_S3_L002.CpG_report.merged_CpG_evidence.cov <<<\n",
      "\n",
      "CpG merging complete. Good luck finding DMRs with bsseq!\n",
      "\n",
      "Summary of parameters for genome-wide cytosine report:\n",
      "==============================================================================\n",
      "Coverage infile:\t\t\t\t/Volumes/Serine/wd/19-12-17/bismark_coverage_files/EPI-153_S4_L002_R1_001_val_1_bismark_bt2_pe.deduplicated.bismark.cov.gz\n",
      "Output directory:\t\t\t\t><\n",
      "Parent directory:\t\t\t\t>/Users/sr320/git-repos/nb-2019/P_generosa/<\n",
      "Genome directory:\t\t\t\t>/Volumes/Serine/wd/19-11-05/genome/<\n",
      "CX context:\t\t\t\t\tno (CpG context only, default)\n",
      "Pooling CpG top/bottom strand evidence:\t\tyes\n",
      "Genome coordinates used:\t\t\t1-based (default)\n",
      "GZIP compression:\t\t\t\tno\n",
      "Split by chromosome:\t\t\t\tno\n",
      "\n",
      "\n",
      "Now reading in and storing sequence information of the genome specified in: /Volumes/Serine/wd/19-11-05/genome/\n",
      "\n",
      "chr Scaffold_01 (89643857 bp)\n",
      "chr Scaffold_02 (69596280 bp)\n",
      "chr Scaffold_03 (57743597 bp)\n",
      "chr Scaffold_04 (65288255 bp)\n",
      "chr Scaffold_05 (67248332 bp)\n",
      "chr Scaffold_06 (61759565 bp)\n",
      "chr Scaffold_07 (43120122 bp)\n",
      "chr Scaffold_08 (61151155 bp)\n",
      "chr Scaffold_09 (38581958 bp)\n",
      "chr Scaffold_10 (53961475 bp)\n",
      "chr Scaffold_11 (51449921 bp)\n",
      "chr Scaffold_12 (50438331 bp)\n",
      "chr Scaffold_13 (44396874 bp)\n",
      "chr Scaffold_14 (45393038 bp)\n",
      "chr Scaffold_15 (47938513 bp)\n",
      "chr Scaffold_16 (31980953 bp)\n",
      "chr Scaffold_17 (34923512 bp)\n",
      "chr Scaffold_18 (27737463 bp)\n",
      "\n",
      "Stored sequence information of 18 chromosomes/scaffolds in total\n",
      "\n",
      "==============================================================================\n",
      "Methylation information will now be written into a genome-wide cytosine report\n",
      "==============================================================================\n",
      "\n",
      ">>> Writing genome-wide cytosine report to: /Volumes/Serine/wd/19-12-17/EPI-153_S4_L002.CpG_report.txt <<<\n",
      "\n",
      "Storing all covered cytosine positions for chromosome: Scaffold_01\n",
      "Writing cytosine report for chromosome Scaffold_01 (stored 823252 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_02 (stored 634980 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_03 (stored 510329 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_04 (stored 585349 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_05 (stored 522095 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_06 (stored 513299 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_07 (stored 378483 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_08 (stored 528072 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_09 (stored 332036 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_10 (stored 441562 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_11 (stored 429418 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_12 (stored 459135 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_13 (stored 375203 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_14 (stored 472706 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_15 (stored 379642 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_16 (stored 297137 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_17 (stored 321938 different covered positions)\n",
      "Writing cytosine report for last chromosome Scaffold_18 (stored 232067 different covered positions)\n",
      "Finished writing out cytosine report for covered chromosomes (processed 18 chromosomes/scaffolds in total)\n",
      "\n",
      "Now processing chromosomes that were not covered by any methylation calls in the coverage file...\n",
      "All chromosomes in the genome were covered by at least some reads. coverage2cytosine processing complete.\n",
      "\n",
      "Now merging top and bottom strand CpGs into a single CG dinucleotide entity (reading from file >>/Volumes/Serine/wd/19-12-17/EPI-153_S4_L002.CpG_report.txt<<, in output directory '')\n",
      ">>> Writing a new coverage file with top and bottom strand CpG methylation evidence merged to /Volumes/Serine/wd/19-12-17/EPI-153_S4_L002.CpG_report.merged_CpG_evidence.cov <<<\n",
      "\n",
      "CpG merging complete. Good luck finding DMRs with bsseq!\n",
      "\n",
      "Summary of parameters for genome-wide cytosine report:\n",
      "==============================================================================\n",
      "Coverage infile:\t\t\t\t/Volumes/Serine/wd/19-12-17/bismark_coverage_files/EPI-154_S5_L002_R1_001_val_1_bismark_bt2_pe.deduplicated.bismark.cov.gz\n",
      "Output directory:\t\t\t\t><\n",
      "Parent directory:\t\t\t\t>/Users/sr320/git-repos/nb-2019/P_generosa/<\n",
      "Genome directory:\t\t\t\t>/Volumes/Serine/wd/19-11-05/genome/<\n",
      "CX context:\t\t\t\t\tno (CpG context only, default)\n",
      "Pooling CpG top/bottom strand evidence:\t\tyes\n",
      "Genome coordinates used:\t\t\t1-based (default)\n",
      "GZIP compression:\t\t\t\tno\n",
      "Split by chromosome:\t\t\t\tno\n",
      "\n",
      "\n",
      "Now reading in and storing sequence information of the genome specified in: /Volumes/Serine/wd/19-11-05/genome/\n",
      "\n",
      "chr Scaffold_01 (89643857 bp)\n",
      "chr Scaffold_02 (69596280 bp)\n",
      "chr Scaffold_03 (57743597 bp)\n",
      "chr Scaffold_04 (65288255 bp)\n",
      "chr Scaffold_05 (67248332 bp)\n",
      "chr Scaffold_06 (61759565 bp)\n",
      "chr Scaffold_07 (43120122 bp)\n",
      "chr Scaffold_08 (61151155 bp)\n",
      "chr Scaffold_09 (38581958 bp)\n",
      "chr Scaffold_10 (53961475 bp)\n",
      "chr Scaffold_11 (51449921 bp)\n",
      "chr Scaffold_12 (50438331 bp)\n",
      "chr Scaffold_13 (44396874 bp)\n",
      "chr Scaffold_14 (45393038 bp)\n",
      "chr Scaffold_15 (47938513 bp)\n",
      "chr Scaffold_16 (31980953 bp)\n",
      "chr Scaffold_17 (34923512 bp)\n",
      "chr Scaffold_18 (27737463 bp)\n",
      "\n",
      "Stored sequence information of 18 chromosomes/scaffolds in total\n",
      "\n",
      "==============================================================================\n",
      "Methylation information will now be written into a genome-wide cytosine report\n",
      "==============================================================================\n",
      "\n",
      ">>> Writing genome-wide cytosine report to: /Volumes/Serine/wd/19-12-17/EPI-154_S5_L002.CpG_report.txt <<<\n",
      "\n",
      "Storing all covered cytosine positions for chromosome: Scaffold_01\n",
      "Writing cytosine report for chromosome Scaffold_01 (stored 888035 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_02 (stored 690016 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_03 (stored 550880 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_04 (stored 625712 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_05 (stored 567151 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_06 (stored 554398 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_07 (stored 410273 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_08 (stored 568429 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_09 (stored 361707 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_10 (stored 480747 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_11 (stored 464292 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_12 (stored 494758 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_13 (stored 404268 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_14 (stored 507274 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_15 (stored 409208 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_16 (stored 322155 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_17 (stored 346467 different covered positions)\n",
      "Writing cytosine report for last chromosome Scaffold_18 (stored 253569 different covered positions)\n",
      "Finished writing out cytosine report for covered chromosomes (processed 18 chromosomes/scaffolds in total)\n",
      "\n",
      "Now processing chromosomes that were not covered by any methylation calls in the coverage file...\n",
      "All chromosomes in the genome were covered by at least some reads. coverage2cytosine processing complete.\n",
      "\n",
      "Now merging top and bottom strand CpGs into a single CG dinucleotide entity (reading from file >>/Volumes/Serine/wd/19-12-17/EPI-154_S5_L002.CpG_report.txt<<, in output directory '')\n",
      ">>> Writing a new coverage file with top and bottom strand CpG methylation evidence merged to /Volumes/Serine/wd/19-12-17/EPI-154_S5_L002.CpG_report.merged_CpG_evidence.cov <<<\n",
      "\n",
      "CpG merging complete. Good luck finding DMRs with bsseq!\n",
      "\n",
      "Summary of parameters for genome-wide cytosine report:\n",
      "==============================================================================\n",
      "Coverage infile:\t\t\t\t/Volumes/Serine/wd/19-12-17/bismark_coverage_files/EPI-159_S6_L002_R1_001_val_1_bismark_bt2_pe.deduplicated.bismark.cov.gz\n",
      "Output directory:\t\t\t\t><\n",
      "Parent directory:\t\t\t\t>/Users/sr320/git-repos/nb-2019/P_generosa/<\n",
      "Genome directory:\t\t\t\t>/Volumes/Serine/wd/19-11-05/genome/<\n",
      "CX context:\t\t\t\t\tno (CpG context only, default)\n",
      "Pooling CpG top/bottom strand evidence:\t\tyes\n",
      "Genome coordinates used:\t\t\t1-based (default)\n",
      "GZIP compression:\t\t\t\tno\n",
      "Split by chromosome:\t\t\t\tno\n",
      "\n",
      "\n",
      "Now reading in and storing sequence information of the genome specified in: /Volumes/Serine/wd/19-11-05/genome/\n",
      "\n",
      "chr Scaffold_01 (89643857 bp)\n",
      "chr Scaffold_02 (69596280 bp)\n",
      "chr Scaffold_03 (57743597 bp)\n",
      "chr Scaffold_04 (65288255 bp)\n",
      "chr Scaffold_05 (67248332 bp)\n",
      "chr Scaffold_06 (61759565 bp)\n",
      "chr Scaffold_07 (43120122 bp)\n",
      "chr Scaffold_08 (61151155 bp)\n",
      "chr Scaffold_09 (38581958 bp)\n",
      "chr Scaffold_10 (53961475 bp)\n",
      "chr Scaffold_11 (51449921 bp)\n",
      "chr Scaffold_12 (50438331 bp)\n",
      "chr Scaffold_13 (44396874 bp)\n",
      "chr Scaffold_14 (45393038 bp)\n",
      "chr Scaffold_15 (47938513 bp)\n",
      "chr Scaffold_16 (31980953 bp)\n",
      "chr Scaffold_17 (34923512 bp)\n",
      "chr Scaffold_18 (27737463 bp)\n",
      "\n",
      "Stored sequence information of 18 chromosomes/scaffolds in total\n",
      "\n",
      "==============================================================================\n",
      "Methylation information will now be written into a genome-wide cytosine report\n",
      "==============================================================================\n",
      "\n",
      ">>> Writing genome-wide cytosine report to: /Volumes/Serine/wd/19-12-17/EPI-159_S6_L002.CpG_report.txt <<<\n",
      "\n",
      "Storing all covered cytosine positions for chromosome: Scaffold_01\n",
      "Writing cytosine report for chromosome Scaffold_01 (stored 696991 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_02 (stored 538435 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_03 (stored 435628 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_04 (stored 493156 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_05 (stored 444835 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_06 (stored 437174 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_07 (stored 322082 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_08 (stored 447375 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_09 (stored 281339 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_10 (stored 376613 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_11 (stored 362371 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_12 (stored 387040 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_13 (stored 315456 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_14 (stored 407522 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_15 (stored 320771 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_16 (stored 249595 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_17 (stored 271640 different covered positions)\n",
      "Writing cytosine report for last chromosome Scaffold_18 (stored 197802 different covered positions)\n",
      "Finished writing out cytosine report for covered chromosomes (processed 18 chromosomes/scaffolds in total)\n",
      "\n",
      "Now processing chromosomes that were not covered by any methylation calls in the coverage file...\n",
      "All chromosomes in the genome were covered by at least some reads. coverage2cytosine processing complete.\n",
      "\n",
      "Now merging top and bottom strand CpGs into a single CG dinucleotide entity (reading from file >>/Volumes/Serine/wd/19-12-17/EPI-159_S6_L002.CpG_report.txt<<, in output directory '')\n",
      ">>> Writing a new coverage file with top and bottom strand CpG methylation evidence merged to /Volumes/Serine/wd/19-12-17/EPI-159_S6_L002.CpG_report.merged_CpG_evidence.cov <<<\n",
      "\n",
      "CpG merging complete. Good luck finding DMRs with bsseq!\n",
      "\n",
      "Summary of parameters for genome-wide cytosine report:\n",
      "==============================================================================\n",
      "Coverage infile:\t\t\t\t/Volumes/Serine/wd/19-12-17/bismark_coverage_files/EPI-160_S7_L002_R1_001_val_1_bismark_bt2_pe.deduplicated.bismark.cov.gz\n",
      "Output directory:\t\t\t\t><\n",
      "Parent directory:\t\t\t\t>/Users/sr320/git-repos/nb-2019/P_generosa/<\n",
      "Genome directory:\t\t\t\t>/Volumes/Serine/wd/19-11-05/genome/<\n",
      "CX context:\t\t\t\t\tno (CpG context only, default)\n",
      "Pooling CpG top/bottom strand evidence:\t\tyes\n",
      "Genome coordinates used:\t\t\t1-based (default)\n",
      "GZIP compression:\t\t\t\tno\n",
      "Split by chromosome:\t\t\t\tno\n",
      "\n",
      "\n",
      "Now reading in and storing sequence information of the genome specified in: /Volumes/Serine/wd/19-11-05/genome/\n",
      "\n",
      "chr Scaffold_01 (89643857 bp)\n",
      "chr Scaffold_02 (69596280 bp)\n",
      "chr Scaffold_03 (57743597 bp)\n",
      "chr Scaffold_04 (65288255 bp)\n",
      "chr Scaffold_05 (67248332 bp)\n",
      "chr Scaffold_06 (61759565 bp)\n",
      "chr Scaffold_07 (43120122 bp)\n",
      "chr Scaffold_08 (61151155 bp)\n",
      "chr Scaffold_09 (38581958 bp)\n",
      "chr Scaffold_10 (53961475 bp)\n",
      "chr Scaffold_11 (51449921 bp)\n",
      "chr Scaffold_12 (50438331 bp)\n",
      "chr Scaffold_13 (44396874 bp)\n",
      "chr Scaffold_14 (45393038 bp)\n",
      "chr Scaffold_15 (47938513 bp)\n",
      "chr Scaffold_16 (31980953 bp)\n",
      "chr Scaffold_17 (34923512 bp)\n",
      "chr Scaffold_18 (27737463 bp)\n",
      "\n",
      "Stored sequence information of 18 chromosomes/scaffolds in total\n",
      "\n",
      "==============================================================================\n",
      "Methylation information will now be written into a genome-wide cytosine report\n",
      "==============================================================================\n",
      "\n",
      ">>> Writing genome-wide cytosine report to: /Volumes/Serine/wd/19-12-17/EPI-160_S7_L002.CpG_report.txt <<<\n",
      "\n",
      "Storing all covered cytosine positions for chromosome: Scaffold_01\n",
      "Writing cytosine report for chromosome Scaffold_01 (stored 983464 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_02 (stored 762039 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_03 (stored 612215 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_04 (stored 695096 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_05 (stored 633032 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_06 (stored 619295 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_07 (stored 452038 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_08 (stored 630025 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_09 (stored 399644 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_10 (stored 528350 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_11 (stored 514303 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_12 (stored 549748 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_13 (stored 448936 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_14 (stored 568370 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_15 (stored 453690 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_16 (stored 356275 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_17 (stored 386758 different covered positions)\n",
      "Writing cytosine report for last chromosome Scaffold_18 (stored 276622 different covered positions)\n",
      "Finished writing out cytosine report for covered chromosomes (processed 18 chromosomes/scaffolds in total)\n",
      "\n",
      "Now processing chromosomes that were not covered by any methylation calls in the coverage file...\n",
      "All chromosomes in the genome were covered by at least some reads. coverage2cytosine processing complete.\n",
      "\n",
      "Now merging top and bottom strand CpGs into a single CG dinucleotide entity (reading from file >>/Volumes/Serine/wd/19-12-17/EPI-160_S7_L002.CpG_report.txt<<, in output directory '')\n",
      ">>> Writing a new coverage file with top and bottom strand CpG methylation evidence merged to /Volumes/Serine/wd/19-12-17/EPI-160_S7_L002.CpG_report.merged_CpG_evidence.cov <<<\n",
      "\n",
      "CpG merging complete. Good luck finding DMRs with bsseq!\n",
      "\n",
      "Summary of parameters for genome-wide cytosine report:\n",
      "==============================================================================\n",
      "Coverage infile:\t\t\t\t/Volumes/Serine/wd/19-12-17/bismark_coverage_files/EPI-161_S8_L002_R1_001_val_1_bismark_bt2_pe.deduplicated.bismark.cov.gz\n",
      "Output directory:\t\t\t\t><\n",
      "Parent directory:\t\t\t\t>/Users/sr320/git-repos/nb-2019/P_generosa/<\n",
      "Genome directory:\t\t\t\t>/Volumes/Serine/wd/19-11-05/genome/<\n",
      "CX context:\t\t\t\t\tno (CpG context only, default)\n",
      "Pooling CpG top/bottom strand evidence:\t\tyes\n",
      "Genome coordinates used:\t\t\t1-based (default)\n",
      "GZIP compression:\t\t\t\tno\n",
      "Split by chromosome:\t\t\t\tno\n",
      "\n",
      "\n",
      "Now reading in and storing sequence information of the genome specified in: /Volumes/Serine/wd/19-11-05/genome/\n",
      "\n",
      "chr Scaffold_01 (89643857 bp)\n",
      "chr Scaffold_02 (69596280 bp)\n",
      "chr Scaffold_03 (57743597 bp)\n",
      "chr Scaffold_04 (65288255 bp)\n",
      "chr Scaffold_05 (67248332 bp)\n",
      "chr Scaffold_06 (61759565 bp)\n",
      "chr Scaffold_07 (43120122 bp)\n",
      "chr Scaffold_08 (61151155 bp)\n",
      "chr Scaffold_09 (38581958 bp)\n",
      "chr Scaffold_10 (53961475 bp)\n",
      "chr Scaffold_11 (51449921 bp)\n",
      "chr Scaffold_12 (50438331 bp)\n",
      "chr Scaffold_13 (44396874 bp)\n",
      "chr Scaffold_14 (45393038 bp)\n",
      "chr Scaffold_15 (47938513 bp)\n",
      "chr Scaffold_16 (31980953 bp)\n",
      "chr Scaffold_17 (34923512 bp)\n",
      "chr Scaffold_18 (27737463 bp)\n",
      "\n",
      "Stored sequence information of 18 chromosomes/scaffolds in total\n",
      "\n",
      "==============================================================================\n",
      "Methylation information will now be written into a genome-wide cytosine report\n",
      "==============================================================================\n",
      "\n",
      ">>> Writing genome-wide cytosine report to: /Volumes/Serine/wd/19-12-17/EPI-161_S8_L002.CpG_report.txt <<<\n",
      "\n",
      "Storing all covered cytosine positions for chromosome: Scaffold_01\n",
      "Writing cytosine report for chromosome Scaffold_01 (stored 839695 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_02 (stored 653856 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_03 (stored 521097 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_04 (stored 600273 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_05 (stored 535499 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_06 (stored 527366 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_07 (stored 386794 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_08 (stored 539081 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_09 (stored 343247 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_10 (stored 448148 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_11 (stored 441994 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_12 (stored 469099 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_13 (stored 381538 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_14 (stored 484678 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_15 (stored 387699 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_16 (stored 303897 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_17 (stored 327884 different covered positions)\n",
      "Writing cytosine report for last chromosome Scaffold_18 (stored 234600 different covered positions)\n",
      "Finished writing out cytosine report for covered chromosomes (processed 18 chromosomes/scaffolds in total)\n",
      "\n",
      "Now processing chromosomes that were not covered by any methylation calls in the coverage file...\n",
      "All chromosomes in the genome were covered by at least some reads. coverage2cytosine processing complete.\n",
      "\n",
      "Now merging top and bottom strand CpGs into a single CG dinucleotide entity (reading from file >>/Volumes/Serine/wd/19-12-17/EPI-161_S8_L002.CpG_report.txt<<, in output directory '')\n",
      ">>> Writing a new coverage file with top and bottom strand CpG methylation evidence merged to /Volumes/Serine/wd/19-12-17/EPI-161_S8_L002.CpG_report.merged_CpG_evidence.cov <<<\n",
      "\n",
      "CpG merging complete. Good luck finding DMRs with bsseq!\n",
      "\n",
      "Summary of parameters for genome-wide cytosine report:\n",
      "==============================================================================\n",
      "Coverage infile:\t\t\t\t/Volumes/Serine/wd/19-12-17/bismark_coverage_files/EPI-162_S9_L002_R1_001_val_1_bismark_bt2_pe.deduplicated.bismark.cov.gz\n",
      "Output directory:\t\t\t\t><\n",
      "Parent directory:\t\t\t\t>/Users/sr320/git-repos/nb-2019/P_generosa/<\n",
      "Genome directory:\t\t\t\t>/Volumes/Serine/wd/19-11-05/genome/<\n",
      "CX context:\t\t\t\t\tno (CpG context only, default)\n",
      "Pooling CpG top/bottom strand evidence:\t\tyes\n",
      "Genome coordinates used:\t\t\t1-based (default)\n",
      "GZIP compression:\t\t\t\tno\n",
      "Split by chromosome:\t\t\t\tno\n",
      "\n",
      "\n",
      "Now reading in and storing sequence information of the genome specified in: /Volumes/Serine/wd/19-11-05/genome/\n",
      "\n",
      "chr Scaffold_01 (89643857 bp)\n",
      "chr Scaffold_02 (69596280 bp)\n",
      "chr Scaffold_03 (57743597 bp)\n",
      "chr Scaffold_04 (65288255 bp)\n",
      "chr Scaffold_05 (67248332 bp)\n",
      "chr Scaffold_06 (61759565 bp)\n",
      "chr Scaffold_07 (43120122 bp)\n",
      "chr Scaffold_08 (61151155 bp)\n",
      "chr Scaffold_09 (38581958 bp)\n",
      "chr Scaffold_10 (53961475 bp)\n",
      "chr Scaffold_11 (51449921 bp)\n",
      "chr Scaffold_12 (50438331 bp)\n",
      "chr Scaffold_13 (44396874 bp)\n",
      "chr Scaffold_14 (45393038 bp)\n",
      "chr Scaffold_15 (47938513 bp)\n",
      "chr Scaffold_16 (31980953 bp)\n",
      "chr Scaffold_17 (34923512 bp)\n",
      "chr Scaffold_18 (27737463 bp)\n",
      "\n",
      "Stored sequence information of 18 chromosomes/scaffolds in total\n",
      "\n",
      "==============================================================================\n",
      "Methylation information will now be written into a genome-wide cytosine report\n",
      "==============================================================================\n",
      "\n",
      ">>> Writing genome-wide cytosine report to: /Volumes/Serine/wd/19-12-17/EPI-162_S9_L002.CpG_report.txt <<<\n",
      "\n",
      "Storing all covered cytosine positions for chromosome: Scaffold_01\n",
      "Writing cytosine report for chromosome Scaffold_01 (stored 824811 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_02 (stored 636674 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_03 (stored 511422 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_04 (stored 588916 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_05 (stored 523360 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_06 (stored 513804 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_07 (stored 381151 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_08 (stored 528499 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_09 (stored 333536 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_10 (stored 439322 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_11 (stored 433963 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_12 (stored 460897 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_13 (stored 373712 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_14 (stored 475525 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_15 (stored 380650 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_16 (stored 296799 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_17 (stored 322633 different covered positions)\n",
      "Writing cytosine report for last chromosome Scaffold_18 (stored 231606 different covered positions)\n",
      "Finished writing out cytosine report for covered chromosomes (processed 18 chromosomes/scaffolds in total)\n",
      "\n",
      "Now processing chromosomes that were not covered by any methylation calls in the coverage file...\n",
      "All chromosomes in the genome were covered by at least some reads. coverage2cytosine processing complete.\n",
      "\n",
      "Now merging top and bottom strand CpGs into a single CG dinucleotide entity (reading from file >>/Volumes/Serine/wd/19-12-17/EPI-162_S9_L002.CpG_report.txt<<, in output directory '')\n",
      ">>> Writing a new coverage file with top and bottom strand CpG methylation evidence merged to /Volumes/Serine/wd/19-12-17/EPI-162_S9_L002.CpG_report.merged_CpG_evidence.cov <<<\n",
      "\n",
      "CpG merging complete. Good luck finding DMRs with bsseq!\n",
      "\n",
      "Summary of parameters for genome-wide cytosine report:\n",
      "==============================================================================\n",
      "Coverage infile:\t\t\t\t/Volumes/Serine/wd/19-12-17/bismark_coverage_files/EPI-167_S10_L002_R1_001_val_1_bismark_bt2_pe.deduplicated.bismark.cov.gz\n",
      "Output directory:\t\t\t\t><\n",
      "Parent directory:\t\t\t\t>/Users/sr320/git-repos/nb-2019/P_generosa/<\n",
      "Genome directory:\t\t\t\t>/Volumes/Serine/wd/19-11-05/genome/<\n",
      "CX context:\t\t\t\t\tno (CpG context only, default)\n",
      "Pooling CpG top/bottom strand evidence:\t\tyes\n",
      "Genome coordinates used:\t\t\t1-based (default)\n",
      "GZIP compression:\t\t\t\tno\n",
      "Split by chromosome:\t\t\t\tno\n",
      "\n",
      "\n",
      "Now reading in and storing sequence information of the genome specified in: /Volumes/Serine/wd/19-11-05/genome/\n",
      "\n",
      "chr Scaffold_01 (89643857 bp)\n",
      "chr Scaffold_02 (69596280 bp)\n",
      "chr Scaffold_03 (57743597 bp)\n",
      "chr Scaffold_04 (65288255 bp)\n",
      "chr Scaffold_05 (67248332 bp)\n",
      "chr Scaffold_06 (61759565 bp)\n",
      "chr Scaffold_07 (43120122 bp)\n",
      "chr Scaffold_08 (61151155 bp)\n",
      "chr Scaffold_09 (38581958 bp)\n",
      "chr Scaffold_10 (53961475 bp)\n",
      "chr Scaffold_11 (51449921 bp)\n",
      "chr Scaffold_12 (50438331 bp)\n",
      "chr Scaffold_13 (44396874 bp)\n",
      "chr Scaffold_14 (45393038 bp)\n",
      "chr Scaffold_15 (47938513 bp)\n",
      "chr Scaffold_16 (31980953 bp)\n",
      "chr Scaffold_17 (34923512 bp)\n",
      "chr Scaffold_18 (27737463 bp)\n",
      "\n",
      "Stored sequence information of 18 chromosomes/scaffolds in total\n",
      "\n",
      "==============================================================================\n",
      "Methylation information will now be written into a genome-wide cytosine report\n",
      "==============================================================================\n",
      "\n",
      ">>> Writing genome-wide cytosine report to: /Volumes/Serine/wd/19-12-17/EPI-167_S10_L002.CpG_report.txt <<<\n",
      "\n",
      "Storing all covered cytosine positions for chromosome: Scaffold_01\n",
      "Writing cytosine report for chromosome Scaffold_01 (stored 1018116 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_02 (stored 785397 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_03 (stored 626447 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_04 (stored 711392 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_05 (stored 640091 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_06 (stored 628486 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_07 (stored 460858 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_08 (stored 643157 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_09 (stored 405664 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_10 (stored 546922 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_11 (stored 523529 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_12 (stored 568951 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_13 (stored 458151 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_14 (stored 575423 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_15 (stored 458729 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_16 (stored 365934 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_17 (stored 389376 different covered positions)\n",
      "Writing cytosine report for last chromosome Scaffold_18 (stored 281952 different covered positions)\n",
      "Finished writing out cytosine report for covered chromosomes (processed 18 chromosomes/scaffolds in total)\n",
      "\n",
      "Now processing chromosomes that were not covered by any methylation calls in the coverage file...\n",
      "All chromosomes in the genome were covered by at least some reads. coverage2cytosine processing complete.\n",
      "\n",
      "Now merging top and bottom strand CpGs into a single CG dinucleotide entity (reading from file >>/Volumes/Serine/wd/19-12-17/EPI-167_S10_L002.CpG_report.txt<<, in output directory '')\n",
      ">>> Writing a new coverage file with top and bottom strand CpG methylation evidence merged to /Volumes/Serine/wd/19-12-17/EPI-167_S10_L002.CpG_report.merged_CpG_evidence.cov <<<\n",
      "\n",
      "CpG merging complete. Good luck finding DMRs with bsseq!\n",
      "\n",
      "Summary of parameters for genome-wide cytosine report:\n",
      "==============================================================================\n",
      "Coverage infile:\t\t\t\t/Volumes/Serine/wd/19-12-17/bismark_coverage_files/EPI-168_S11_L002_R1_001_val_1_bismark_bt2_pe.deduplicated.bismark.cov.gz\n",
      "Output directory:\t\t\t\t><\n",
      "Parent directory:\t\t\t\t>/Users/sr320/git-repos/nb-2019/P_generosa/<\n",
      "Genome directory:\t\t\t\t>/Volumes/Serine/wd/19-11-05/genome/<\n",
      "CX context:\t\t\t\t\tno (CpG context only, default)\n",
      "Pooling CpG top/bottom strand evidence:\t\tyes\n",
      "Genome coordinates used:\t\t\t1-based (default)\n",
      "GZIP compression:\t\t\t\tno\n",
      "Split by chromosome:\t\t\t\tno\n",
      "\n",
      "\n",
      "Now reading in and storing sequence information of the genome specified in: /Volumes/Serine/wd/19-11-05/genome/\n",
      "\n",
      "chr Scaffold_01 (89643857 bp)\n",
      "chr Scaffold_02 (69596280 bp)\n",
      "chr Scaffold_03 (57743597 bp)\n",
      "chr Scaffold_04 (65288255 bp)\n",
      "chr Scaffold_05 (67248332 bp)\n",
      "chr Scaffold_06 (61759565 bp)\n",
      "chr Scaffold_07 (43120122 bp)\n",
      "chr Scaffold_08 (61151155 bp)\n",
      "chr Scaffold_09 (38581958 bp)\n",
      "chr Scaffold_10 (53961475 bp)\n",
      "chr Scaffold_11 (51449921 bp)\n",
      "chr Scaffold_12 (50438331 bp)\n",
      "chr Scaffold_13 (44396874 bp)\n",
      "chr Scaffold_14 (45393038 bp)\n",
      "chr Scaffold_15 (47938513 bp)\n",
      "chr Scaffold_16 (31980953 bp)\n",
      "chr Scaffold_17 (34923512 bp)\n",
      "chr Scaffold_18 (27737463 bp)\n",
      "\n",
      "Stored sequence information of 18 chromosomes/scaffolds in total\n",
      "\n",
      "==============================================================================\n",
      "Methylation information will now be written into a genome-wide cytosine report\n",
      "==============================================================================\n",
      "\n",
      ">>> Writing genome-wide cytosine report to: /Volumes/Serine/wd/19-12-17/EPI-168_S11_L002.CpG_report.txt <<<\n",
      "\n",
      "Storing all covered cytosine positions for chromosome: Scaffold_01\n",
      "Writing cytosine report for chromosome Scaffold_01 (stored 851833 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_02 (stored 658637 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_03 (stored 532651 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_04 (stored 607722 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_05 (stored 541913 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_06 (stored 533405 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_07 (stored 395443 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_08 (stored 550033 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_09 (stored 348188 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_10 (stored 457276 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_11 (stored 445413 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_12 (stored 475085 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_13 (stored 388593 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_14 (stored 497461 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_15 (stored 395878 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_16 (stored 306998 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_17 (stored 335908 different covered positions)\n",
      "Writing cytosine report for last chromosome Scaffold_18 (stored 241347 different covered positions)\n",
      "Finished writing out cytosine report for covered chromosomes (processed 18 chromosomes/scaffolds in total)\n",
      "\n",
      "Now processing chromosomes that were not covered by any methylation calls in the coverage file...\n",
      "All chromosomes in the genome were covered by at least some reads. coverage2cytosine processing complete.\n",
      "\n",
      "Now merging top and bottom strand CpGs into a single CG dinucleotide entity (reading from file >>/Volumes/Serine/wd/19-12-17/EPI-168_S11_L002.CpG_report.txt<<, in output directory '')\n",
      ">>> Writing a new coverage file with top and bottom strand CpG methylation evidence merged to /Volumes/Serine/wd/19-12-17/EPI-168_S11_L002.CpG_report.merged_CpG_evidence.cov <<<\n",
      "\n",
      "CpG merging complete. Good luck finding DMRs with bsseq!\n",
      "\n",
      "Summary of parameters for genome-wide cytosine report:\n",
      "==============================================================================\n",
      "Coverage infile:\t\t\t\t/Volumes/Serine/wd/19-12-17/bismark_coverage_files/EPI-169_S12_L002_R1_001_val_1_bismark_bt2_pe.deduplicated.bismark.cov.gz\n",
      "Output directory:\t\t\t\t><\n",
      "Parent directory:\t\t\t\t>/Users/sr320/git-repos/nb-2019/P_generosa/<\n",
      "Genome directory:\t\t\t\t>/Volumes/Serine/wd/19-11-05/genome/<\n",
      "CX context:\t\t\t\t\tno (CpG context only, default)\n",
      "Pooling CpG top/bottom strand evidence:\t\tyes\n",
      "Genome coordinates used:\t\t\t1-based (default)\n",
      "GZIP compression:\t\t\t\tno\n",
      "Split by chromosome:\t\t\t\tno\n",
      "\n",
      "\n",
      "Now reading in and storing sequence information of the genome specified in: /Volumes/Serine/wd/19-11-05/genome/\n",
      "\n",
      "chr Scaffold_01 (89643857 bp)\n",
      "chr Scaffold_02 (69596280 bp)\n",
      "chr Scaffold_03 (57743597 bp)\n",
      "chr Scaffold_04 (65288255 bp)\n",
      "chr Scaffold_05 (67248332 bp)\n",
      "chr Scaffold_06 (61759565 bp)\n",
      "chr Scaffold_07 (43120122 bp)\n",
      "chr Scaffold_08 (61151155 bp)\n",
      "chr Scaffold_09 (38581958 bp)\n",
      "chr Scaffold_10 (53961475 bp)\n",
      "chr Scaffold_11 (51449921 bp)\n",
      "chr Scaffold_12 (50438331 bp)\n",
      "chr Scaffold_13 (44396874 bp)\n",
      "chr Scaffold_14 (45393038 bp)\n",
      "chr Scaffold_15 (47938513 bp)\n",
      "chr Scaffold_16 (31980953 bp)\n",
      "chr Scaffold_17 (34923512 bp)\n",
      "chr Scaffold_18 (27737463 bp)\n",
      "\n",
      "Stored sequence information of 18 chromosomes/scaffolds in total\n",
      "\n",
      "==============================================================================\n",
      "Methylation information will now be written into a genome-wide cytosine report\n",
      "==============================================================================\n",
      "\n",
      ">>> Writing genome-wide cytosine report to: /Volumes/Serine/wd/19-12-17/EPI-169_S12_L002.CpG_report.txt <<<\n",
      "\n",
      "Storing all covered cytosine positions for chromosome: Scaffold_01\n",
      "Writing cytosine report for chromosome Scaffold_01 (stored 813911 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_02 (stored 625817 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_03 (stored 501452 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_04 (stored 563569 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_05 (stored 514149 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_06 (stored 500743 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_07 (stored 370520 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_08 (stored 515777 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_09 (stored 326970 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_10 (stored 436873 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_11 (stored 418853 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_12 (stored 449813 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_13 (stored 363957 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_14 (stored 466678 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_15 (stored 368345 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_16 (stored 291196 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_17 (stored 313484 different covered positions)\n",
      "Writing cytosine report for last chromosome Scaffold_18 (stored 227693 different covered positions)\n",
      "Finished writing out cytosine report for covered chromosomes (processed 18 chromosomes/scaffolds in total)\n",
      "\n",
      "Now processing chromosomes that were not covered by any methylation calls in the coverage file...\n",
      "All chromosomes in the genome were covered by at least some reads. coverage2cytosine processing complete.\n",
      "\n",
      "Now merging top and bottom strand CpGs into a single CG dinucleotide entity (reading from file >>/Volumes/Serine/wd/19-12-17/EPI-169_S12_L002.CpG_report.txt<<, in output directory '')\n",
      ">>> Writing a new coverage file with top and bottom strand CpG methylation evidence merged to /Volumes/Serine/wd/19-12-17/EPI-169_S12_L002.CpG_report.merged_CpG_evidence.cov <<<\n",
      "\n",
      "CpG merging complete. Good luck finding DMRs with bsseq!\n",
      "\n",
      "Summary of parameters for genome-wide cytosine report:\n",
      "==============================================================================\n",
      "Coverage infile:\t\t\t\t/Volumes/Serine/wd/19-12-17/bismark_coverage_files/EPI-170_S13_L002_R1_001_val_1_bismark_bt2_pe.deduplicated.bismark.cov.gz\n",
      "Output directory:\t\t\t\t><\n",
      "Parent directory:\t\t\t\t>/Users/sr320/git-repos/nb-2019/P_generosa/<\n",
      "Genome directory:\t\t\t\t>/Volumes/Serine/wd/19-11-05/genome/<\n",
      "CX context:\t\t\t\t\tno (CpG context only, default)\n",
      "Pooling CpG top/bottom strand evidence:\t\tyes\n",
      "Genome coordinates used:\t\t\t1-based (default)\n",
      "GZIP compression:\t\t\t\tno\n",
      "Split by chromosome:\t\t\t\tno\n",
      "\n",
      "\n",
      "Now reading in and storing sequence information of the genome specified in: /Volumes/Serine/wd/19-11-05/genome/\n",
      "\n",
      "chr Scaffold_01 (89643857 bp)\n",
      "chr Scaffold_02 (69596280 bp)\n",
      "chr Scaffold_03 (57743597 bp)\n",
      "chr Scaffold_04 (65288255 bp)\n",
      "chr Scaffold_05 (67248332 bp)\n",
      "chr Scaffold_06 (61759565 bp)\n",
      "chr Scaffold_07 (43120122 bp)\n",
      "chr Scaffold_08 (61151155 bp)\n",
      "chr Scaffold_09 (38581958 bp)\n",
      "chr Scaffold_10 (53961475 bp)\n",
      "chr Scaffold_11 (51449921 bp)\n",
      "chr Scaffold_12 (50438331 bp)\n",
      "chr Scaffold_13 (44396874 bp)\n",
      "chr Scaffold_14 (45393038 bp)\n",
      "chr Scaffold_15 (47938513 bp)\n",
      "chr Scaffold_16 (31980953 bp)\n",
      "chr Scaffold_17 (34923512 bp)\n",
      "chr Scaffold_18 (27737463 bp)\n",
      "\n",
      "Stored sequence information of 18 chromosomes/scaffolds in total\n",
      "\n",
      "==============================================================================\n",
      "Methylation information will now be written into a genome-wide cytosine report\n",
      "==============================================================================\n",
      "\n",
      ">>> Writing genome-wide cytosine report to: /Volumes/Serine/wd/19-12-17/EPI-170_S13_L002.CpG_report.txt <<<\n",
      "\n",
      "Storing all covered cytosine positions for chromosome: Scaffold_01\n",
      "Writing cytosine report for chromosome Scaffold_01 (stored 922019 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_02 (stored 718685 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_03 (stored 579664 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_04 (stored 667234 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_05 (stored 598437 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_06 (stored 575309 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_07 (stored 428688 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_08 (stored 595151 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_09 (stored 374691 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_10 (stored 496883 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_11 (stored 490605 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_12 (stored 517070 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_13 (stored 424042 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_14 (stored 539467 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_15 (stored 429896 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_16 (stored 335247 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_17 (stored 366582 different covered positions)\n",
      "Writing cytosine report for last chromosome Scaffold_18 (stored 267188 different covered positions)\n",
      "Finished writing out cytosine report for covered chromosomes (processed 18 chromosomes/scaffolds in total)\n",
      "\n",
      "Now processing chromosomes that were not covered by any methylation calls in the coverage file...\n",
      "All chromosomes in the genome were covered by at least some reads. coverage2cytosine processing complete.\n",
      "\n",
      "Now merging top and bottom strand CpGs into a single CG dinucleotide entity (reading from file >>/Volumes/Serine/wd/19-12-17/EPI-170_S13_L002.CpG_report.txt<<, in output directory '')\n",
      ">>> Writing a new coverage file with top and bottom strand CpG methylation evidence merged to /Volumes/Serine/wd/19-12-17/EPI-170_S13_L002.CpG_report.merged_CpG_evidence.cov <<<\n",
      "\n",
      "CpG merging complete. Good luck finding DMRs with bsseq!\n",
      "\n",
      "Summary of parameters for genome-wide cytosine report:\n",
      "==============================================================================\n",
      "Coverage infile:\t\t\t\t/Volumes/Serine/wd/19-12-17/bismark_coverage_files/EPI-175_S14_L003_R1_001_val_1_bismark_bt2_pe.deduplicated.bismark.cov.gz\n",
      "Output directory:\t\t\t\t><\n",
      "Parent directory:\t\t\t\t>/Users/sr320/git-repos/nb-2019/P_generosa/<\n",
      "Genome directory:\t\t\t\t>/Volumes/Serine/wd/19-11-05/genome/<\n",
      "CX context:\t\t\t\t\tno (CpG context only, default)\n",
      "Pooling CpG top/bottom strand evidence:\t\tyes\n",
      "Genome coordinates used:\t\t\t1-based (default)\n",
      "GZIP compression:\t\t\t\tno\n",
      "Split by chromosome:\t\t\t\tno\n",
      "\n",
      "\n",
      "Now reading in and storing sequence information of the genome specified in: /Volumes/Serine/wd/19-11-05/genome/\n",
      "\n",
      "chr Scaffold_01 (89643857 bp)\n",
      "chr Scaffold_02 (69596280 bp)\n",
      "chr Scaffold_03 (57743597 bp)\n",
      "chr Scaffold_04 (65288255 bp)\n",
      "chr Scaffold_05 (67248332 bp)\n",
      "chr Scaffold_06 (61759565 bp)\n",
      "chr Scaffold_07 (43120122 bp)\n",
      "chr Scaffold_08 (61151155 bp)\n",
      "chr Scaffold_09 (38581958 bp)\n",
      "chr Scaffold_10 (53961475 bp)\n",
      "chr Scaffold_11 (51449921 bp)\n",
      "chr Scaffold_12 (50438331 bp)\n",
      "chr Scaffold_13 (44396874 bp)\n",
      "chr Scaffold_14 (45393038 bp)\n",
      "chr Scaffold_15 (47938513 bp)\n",
      "chr Scaffold_16 (31980953 bp)\n",
      "chr Scaffold_17 (34923512 bp)\n",
      "chr Scaffold_18 (27737463 bp)\n",
      "\n",
      "Stored sequence information of 18 chromosomes/scaffolds in total\n",
      "\n",
      "==============================================================================\n",
      "Methylation information will now be written into a genome-wide cytosine report\n",
      "==============================================================================\n",
      "\n",
      ">>> Writing genome-wide cytosine report to: /Volumes/Serine/wd/19-12-17/EPI-175_S14_L003.CpG_report.txt <<<\n",
      "\n",
      "Storing all covered cytosine positions for chromosome: Scaffold_01\n",
      "Writing cytosine report for chromosome Scaffold_01 (stored 988407 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_02 (stored 764177 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_03 (stored 611384 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_04 (stored 690495 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_05 (stored 630755 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_06 (stored 615274 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_07 (stored 451643 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_08 (stored 633856 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_09 (stored 399555 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_10 (stored 533542 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_11 (stored 512287 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_12 (stored 553228 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_13 (stored 449652 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_14 (stored 567070 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_15 (stored 454946 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_16 (stored 358216 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_17 (stored 381508 different covered positions)\n",
      "Writing cytosine report for last chromosome Scaffold_18 (stored 279672 different covered positions)\n",
      "Finished writing out cytosine report for covered chromosomes (processed 18 chromosomes/scaffolds in total)\n",
      "\n",
      "Now processing chromosomes that were not covered by any methylation calls in the coverage file...\n",
      "All chromosomes in the genome were covered by at least some reads. coverage2cytosine processing complete.\n",
      "\n",
      "Now merging top and bottom strand CpGs into a single CG dinucleotide entity (reading from file >>/Volumes/Serine/wd/19-12-17/EPI-175_S14_L003.CpG_report.txt<<, in output directory '')\n",
      ">>> Writing a new coverage file with top and bottom strand CpG methylation evidence merged to /Volumes/Serine/wd/19-12-17/EPI-175_S14_L003.CpG_report.merged_CpG_evidence.cov <<<\n",
      "\n",
      "CpG merging complete. Good luck finding DMRs with bsseq!\n",
      "\n",
      "Summary of parameters for genome-wide cytosine report:\n",
      "==============================================================================\n",
      "Coverage infile:\t\t\t\t/Volumes/Serine/wd/19-12-17/bismark_coverage_files/EPI-176_S15_L003_R1_001_val_1_bismark_bt2_pe.deduplicated.bismark.cov.gz\n",
      "Output directory:\t\t\t\t><\n",
      "Parent directory:\t\t\t\t>/Users/sr320/git-repos/nb-2019/P_generosa/<\n",
      "Genome directory:\t\t\t\t>/Volumes/Serine/wd/19-11-05/genome/<\n",
      "CX context:\t\t\t\t\tno (CpG context only, default)\n",
      "Pooling CpG top/bottom strand evidence:\t\tyes\n",
      "Genome coordinates used:\t\t\t1-based (default)\n",
      "GZIP compression:\t\t\t\tno\n",
      "Split by chromosome:\t\t\t\tno\n",
      "\n",
      "\n",
      "Now reading in and storing sequence information of the genome specified in: /Volumes/Serine/wd/19-11-05/genome/\n",
      "\n",
      "chr Scaffold_01 (89643857 bp)\n",
      "chr Scaffold_02 (69596280 bp)\n",
      "chr Scaffold_03 (57743597 bp)\n",
      "chr Scaffold_04 (65288255 bp)\n",
      "chr Scaffold_05 (67248332 bp)\n",
      "chr Scaffold_06 (61759565 bp)\n",
      "chr Scaffold_07 (43120122 bp)\n",
      "chr Scaffold_08 (61151155 bp)\n",
      "chr Scaffold_09 (38581958 bp)\n",
      "chr Scaffold_10 (53961475 bp)\n",
      "chr Scaffold_11 (51449921 bp)\n",
      "chr Scaffold_12 (50438331 bp)\n",
      "chr Scaffold_13 (44396874 bp)\n",
      "chr Scaffold_14 (45393038 bp)\n",
      "chr Scaffold_15 (47938513 bp)\n",
      "chr Scaffold_16 (31980953 bp)\n",
      "chr Scaffold_17 (34923512 bp)\n",
      "chr Scaffold_18 (27737463 bp)\n",
      "\n",
      "Stored sequence information of 18 chromosomes/scaffolds in total\n",
      "\n",
      "==============================================================================\n",
      "Methylation information will now be written into a genome-wide cytosine report\n",
      "==============================================================================\n",
      "\n",
      ">>> Writing genome-wide cytosine report to: /Volumes/Serine/wd/19-12-17/EPI-176_S15_L003.CpG_report.txt <<<\n",
      "\n",
      "Storing all covered cytosine positions for chromosome: Scaffold_01\n",
      "Writing cytosine report for chromosome Scaffold_01 (stored 1159032 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_02 (stored 900348 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_03 (stored 718766 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_04 (stored 828066 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_05 (stored 741726 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_06 (stored 717732 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_07 (stored 538566 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_08 (stored 744939 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_09 (stored 469032 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_10 (stored 621815 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_11 (stored 611769 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_12 (stored 651006 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_13 (stored 527137 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_14 (stored 661544 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_15 (stored 538992 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_16 (stored 423563 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_17 (stored 451365 different covered positions)\n",
      "Writing cytosine report for last chromosome Scaffold_18 (stored 322183 different covered positions)\n",
      "Finished writing out cytosine report for covered chromosomes (processed 18 chromosomes/scaffolds in total)\n",
      "\n",
      "Now processing chromosomes that were not covered by any methylation calls in the coverage file...\n",
      "All chromosomes in the genome were covered by at least some reads. coverage2cytosine processing complete.\n",
      "\n",
      "Now merging top and bottom strand CpGs into a single CG dinucleotide entity (reading from file >>/Volumes/Serine/wd/19-12-17/EPI-176_S15_L003.CpG_report.txt<<, in output directory '')\n",
      ">>> Writing a new coverage file with top and bottom strand CpG methylation evidence merged to /Volumes/Serine/wd/19-12-17/EPI-176_S15_L003.CpG_report.merged_CpG_evidence.cov <<<\n",
      "\n",
      "CpG merging complete. Good luck finding DMRs with bsseq!\n",
      "\n",
      "Summary of parameters for genome-wide cytosine report:\n",
      "==============================================================================\n",
      "Coverage infile:\t\t\t\t/Volumes/Serine/wd/19-12-17/bismark_coverage_files/EPI-181_S16_L003_R1_001_val_1_bismark_bt2_pe.deduplicated.bismark.cov.gz\n",
      "Output directory:\t\t\t\t><\n",
      "Parent directory:\t\t\t\t>/Users/sr320/git-repos/nb-2019/P_generosa/<\n",
      "Genome directory:\t\t\t\t>/Volumes/Serine/wd/19-11-05/genome/<\n",
      "CX context:\t\t\t\t\tno (CpG context only, default)\n",
      "Pooling CpG top/bottom strand evidence:\t\tyes\n",
      "Genome coordinates used:\t\t\t1-based (default)\n",
      "GZIP compression:\t\t\t\tno\n",
      "Split by chromosome:\t\t\t\tno\n",
      "\n",
      "\n",
      "Now reading in and storing sequence information of the genome specified in: /Volumes/Serine/wd/19-11-05/genome/\n",
      "\n",
      "chr Scaffold_01 (89643857 bp)\n",
      "chr Scaffold_02 (69596280 bp)\n",
      "chr Scaffold_03 (57743597 bp)\n",
      "chr Scaffold_04 (65288255 bp)\n",
      "chr Scaffold_05 (67248332 bp)\n",
      "chr Scaffold_06 (61759565 bp)\n",
      "chr Scaffold_07 (43120122 bp)\n",
      "chr Scaffold_08 (61151155 bp)\n",
      "chr Scaffold_09 (38581958 bp)\n",
      "chr Scaffold_10 (53961475 bp)\n",
      "chr Scaffold_11 (51449921 bp)\n",
      "chr Scaffold_12 (50438331 bp)\n",
      "chr Scaffold_13 (44396874 bp)\n",
      "chr Scaffold_14 (45393038 bp)\n",
      "chr Scaffold_15 (47938513 bp)\n",
      "chr Scaffold_16 (31980953 bp)\n",
      "chr Scaffold_17 (34923512 bp)\n",
      "chr Scaffold_18 (27737463 bp)\n",
      "\n",
      "Stored sequence information of 18 chromosomes/scaffolds in total\n",
      "\n",
      "==============================================================================\n",
      "Methylation information will now be written into a genome-wide cytosine report\n",
      "==============================================================================\n",
      "\n",
      ">>> Writing genome-wide cytosine report to: /Volumes/Serine/wd/19-12-17/EPI-181_S16_L003.CpG_report.txt <<<\n",
      "\n",
      "Storing all covered cytosine positions for chromosome: Scaffold_01\n",
      "Writing cytosine report for chromosome Scaffold_01 (stored 998161 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_02 (stored 768701 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_03 (stored 619104 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_04 (stored 699129 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_05 (stored 637275 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_06 (stored 615659 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_07 (stored 457724 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_08 (stored 638718 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_09 (stored 402123 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_10 (stored 536706 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_11 (stored 513651 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_12 (stored 555522 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_13 (stored 454680 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_14 (stored 574243 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_15 (stored 456918 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_16 (stored 360109 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_17 (stored 387428 different covered positions)\n",
      "Writing cytosine report for last chromosome Scaffold_18 (stored 279820 different covered positions)\n",
      "Finished writing out cytosine report for covered chromosomes (processed 18 chromosomes/scaffolds in total)\n",
      "\n",
      "Now processing chromosomes that were not covered by any methylation calls in the coverage file...\n",
      "All chromosomes in the genome were covered by at least some reads. coverage2cytosine processing complete.\n",
      "\n",
      "Now merging top and bottom strand CpGs into a single CG dinucleotide entity (reading from file >>/Volumes/Serine/wd/19-12-17/EPI-181_S16_L003.CpG_report.txt<<, in output directory '')\n",
      ">>> Writing a new coverage file with top and bottom strand CpG methylation evidence merged to /Volumes/Serine/wd/19-12-17/EPI-181_S16_L003.CpG_report.merged_CpG_evidence.cov <<<\n",
      "\n",
      "CpG merging complete. Good luck finding DMRs with bsseq!\n",
      "\n",
      "Summary of parameters for genome-wide cytosine report:\n",
      "==============================================================================\n",
      "Coverage infile:\t\t\t\t/Volumes/Serine/wd/19-12-17/bismark_coverage_files/EPI-182_S17_L003_R1_001_val_1_bismark_bt2_pe.deduplicated.bismark.cov.gz\n",
      "Output directory:\t\t\t\t><\n",
      "Parent directory:\t\t\t\t>/Users/sr320/git-repos/nb-2019/P_generosa/<\n",
      "Genome directory:\t\t\t\t>/Volumes/Serine/wd/19-11-05/genome/<\n",
      "CX context:\t\t\t\t\tno (CpG context only, default)\n",
      "Pooling CpG top/bottom strand evidence:\t\tyes\n",
      "Genome coordinates used:\t\t\t1-based (default)\n",
      "GZIP compression:\t\t\t\tno\n",
      "Split by chromosome:\t\t\t\tno\n",
      "\n",
      "\n",
      "Now reading in and storing sequence information of the genome specified in: /Volumes/Serine/wd/19-11-05/genome/\n",
      "\n",
      "chr Scaffold_01 (89643857 bp)\n",
      "chr Scaffold_02 (69596280 bp)\n",
      "chr Scaffold_03 (57743597 bp)\n",
      "chr Scaffold_04 (65288255 bp)\n",
      "chr Scaffold_05 (67248332 bp)\n",
      "chr Scaffold_06 (61759565 bp)\n",
      "chr Scaffold_07 (43120122 bp)\n",
      "chr Scaffold_08 (61151155 bp)\n",
      "chr Scaffold_09 (38581958 bp)\n",
      "chr Scaffold_10 (53961475 bp)\n",
      "chr Scaffold_11 (51449921 bp)\n",
      "chr Scaffold_12 (50438331 bp)\n",
      "chr Scaffold_13 (44396874 bp)\n",
      "chr Scaffold_14 (45393038 bp)\n",
      "chr Scaffold_15 (47938513 bp)\n",
      "chr Scaffold_16 (31980953 bp)\n",
      "chr Scaffold_17 (34923512 bp)\n",
      "chr Scaffold_18 (27737463 bp)\n",
      "\n",
      "Stored sequence information of 18 chromosomes/scaffolds in total\n",
      "\n",
      "==============================================================================\n",
      "Methylation information will now be written into a genome-wide cytosine report\n",
      "==============================================================================\n",
      "\n",
      ">>> Writing genome-wide cytosine report to: /Volumes/Serine/wd/19-12-17/EPI-182_S17_L003.CpG_report.txt <<<\n",
      "\n",
      "Storing all covered cytosine positions for chromosome: Scaffold_01\n",
      "Writing cytosine report for chromosome Scaffold_01 (stored 1151279 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_02 (stored 888521 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_03 (stored 711485 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_04 (stored 820744 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_05 (stored 734933 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_06 (stored 718122 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_07 (stored 529592 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_08 (stored 737582 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_09 (stored 463641 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_10 (stored 616317 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_11 (stored 599860 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_12 (stored 642837 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_13 (stored 520472 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_14 (stored 652934 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_15 (stored 530515 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_16 (stored 414750 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_17 (stored 445036 different covered positions)\n",
      "Writing cytosine report for last chromosome Scaffold_18 (stored 319418 different covered positions)\n",
      "Finished writing out cytosine report for covered chromosomes (processed 18 chromosomes/scaffolds in total)\n",
      "\n",
      "Now processing chromosomes that were not covered by any methylation calls in the coverage file...\n",
      "All chromosomes in the genome were covered by at least some reads. coverage2cytosine processing complete.\n",
      "\n",
      "Now merging top and bottom strand CpGs into a single CG dinucleotide entity (reading from file >>/Volumes/Serine/wd/19-12-17/EPI-182_S17_L003.CpG_report.txt<<, in output directory '')\n",
      ">>> Writing a new coverage file with top and bottom strand CpG methylation evidence merged to /Volumes/Serine/wd/19-12-17/EPI-182_S17_L003.CpG_report.merged_CpG_evidence.cov <<<\n",
      "\n",
      "CpG merging complete. Good luck finding DMRs with bsseq!\n",
      "\n",
      "Summary of parameters for genome-wide cytosine report:\n",
      "==============================================================================\n",
      "Coverage infile:\t\t\t\t/Volumes/Serine/wd/19-12-17/bismark_coverage_files/EPI-184_S18_L003_R1_001_val_1_bismark_bt2_pe.deduplicated.bismark.cov.gz\n",
      "Output directory:\t\t\t\t><\n",
      "Parent directory:\t\t\t\t>/Users/sr320/git-repos/nb-2019/P_generosa/<\n",
      "Genome directory:\t\t\t\t>/Volumes/Serine/wd/19-11-05/genome/<\n",
      "CX context:\t\t\t\t\tno (CpG context only, default)\n",
      "Pooling CpG top/bottom strand evidence:\t\tyes\n",
      "Genome coordinates used:\t\t\t1-based (default)\n",
      "GZIP compression:\t\t\t\tno\n",
      "Split by chromosome:\t\t\t\tno\n",
      "\n",
      "\n",
      "Now reading in and storing sequence information of the genome specified in: /Volumes/Serine/wd/19-11-05/genome/\n",
      "\n",
      "chr Scaffold_01 (89643857 bp)\n",
      "chr Scaffold_02 (69596280 bp)\n",
      "chr Scaffold_03 (57743597 bp)\n",
      "chr Scaffold_04 (65288255 bp)\n",
      "chr Scaffold_05 (67248332 bp)\n",
      "chr Scaffold_06 (61759565 bp)\n",
      "chr Scaffold_07 (43120122 bp)\n",
      "chr Scaffold_08 (61151155 bp)\n",
      "chr Scaffold_09 (38581958 bp)\n",
      "chr Scaffold_10 (53961475 bp)\n",
      "chr Scaffold_11 (51449921 bp)\n",
      "chr Scaffold_12 (50438331 bp)\n",
      "chr Scaffold_13 (44396874 bp)\n",
      "chr Scaffold_14 (45393038 bp)\n",
      "chr Scaffold_15 (47938513 bp)\n",
      "chr Scaffold_16 (31980953 bp)\n",
      "chr Scaffold_17 (34923512 bp)\n",
      "chr Scaffold_18 (27737463 bp)\n",
      "\n",
      "Stored sequence information of 18 chromosomes/scaffolds in total\n",
      "\n",
      "==============================================================================\n",
      "Methylation information will now be written into a genome-wide cytosine report\n",
      "==============================================================================\n",
      "\n",
      ">>> Writing genome-wide cytosine report to: /Volumes/Serine/wd/19-12-17/EPI-184_S18_L003.CpG_report.txt <<<\n",
      "\n",
      "Storing all covered cytosine positions for chromosome: Scaffold_01\n",
      "Writing cytosine report for chromosome Scaffold_01 (stored 848319 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_02 (stored 655019 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_03 (stored 517902 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_04 (stored 584845 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_05 (stored 533780 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_06 (stored 516401 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_07 (stored 381508 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_08 (stored 535872 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_09 (stored 337659 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_10 (stored 458185 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_11 (stored 432639 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_12 (stored 474384 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_13 (stored 381630 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_14 (stored 492486 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_15 (stored 379007 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_16 (stored 304134 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_17 (stored 322647 different covered positions)\n",
      "Writing cytosine report for last chromosome Scaffold_18 (stored 237367 different covered positions)\n",
      "Finished writing out cytosine report for covered chromosomes (processed 18 chromosomes/scaffolds in total)\n",
      "\n",
      "Now processing chromosomes that were not covered by any methylation calls in the coverage file...\n",
      "All chromosomes in the genome were covered by at least some reads. coverage2cytosine processing complete.\n",
      "\n",
      "Now merging top and bottom strand CpGs into a single CG dinucleotide entity (reading from file >>/Volumes/Serine/wd/19-12-17/EPI-184_S18_L003.CpG_report.txt<<, in output directory '')\n",
      ">>> Writing a new coverage file with top and bottom strand CpG methylation evidence merged to /Volumes/Serine/wd/19-12-17/EPI-184_S18_L003.CpG_report.merged_CpG_evidence.cov <<<\n",
      "\n",
      "CpG merging complete. Good luck finding DMRs with bsseq!\n",
      "\n",
      "Summary of parameters for genome-wide cytosine report:\n",
      "==============================================================================\n",
      "Coverage infile:\t\t\t\t/Volumes/Serine/wd/19-12-17/bismark_coverage_files/EPI-185_S19_L003_R1_001_val_1_bismark_bt2_pe.deduplicated.bismark.cov.gz\n",
      "Output directory:\t\t\t\t><\n",
      "Parent directory:\t\t\t\t>/Users/sr320/git-repos/nb-2019/P_generosa/<\n",
      "Genome directory:\t\t\t\t>/Volumes/Serine/wd/19-11-05/genome/<\n",
      "CX context:\t\t\t\t\tno (CpG context only, default)\n",
      "Pooling CpG top/bottom strand evidence:\t\tyes\n",
      "Genome coordinates used:\t\t\t1-based (default)\n",
      "GZIP compression:\t\t\t\tno\n",
      "Split by chromosome:\t\t\t\tno\n",
      "\n",
      "\n",
      "Now reading in and storing sequence information of the genome specified in: /Volumes/Serine/wd/19-11-05/genome/\n",
      "\n",
      "chr Scaffold_01 (89643857 bp)\n",
      "chr Scaffold_02 (69596280 bp)\n",
      "chr Scaffold_03 (57743597 bp)\n",
      "chr Scaffold_04 (65288255 bp)\n",
      "chr Scaffold_05 (67248332 bp)\n",
      "chr Scaffold_06 (61759565 bp)\n",
      "chr Scaffold_07 (43120122 bp)\n",
      "chr Scaffold_08 (61151155 bp)\n",
      "chr Scaffold_09 (38581958 bp)\n",
      "chr Scaffold_10 (53961475 bp)\n",
      "chr Scaffold_11 (51449921 bp)\n",
      "chr Scaffold_12 (50438331 bp)\n",
      "chr Scaffold_13 (44396874 bp)\n",
      "chr Scaffold_14 (45393038 bp)\n",
      "chr Scaffold_15 (47938513 bp)\n",
      "chr Scaffold_16 (31980953 bp)\n",
      "chr Scaffold_17 (34923512 bp)\n",
      "chr Scaffold_18 (27737463 bp)\n",
      "\n",
      "Stored sequence information of 18 chromosomes/scaffolds in total\n",
      "\n",
      "==============================================================================\n",
      "Methylation information will now be written into a genome-wide cytosine report\n",
      "==============================================================================\n",
      "\n",
      ">>> Writing genome-wide cytosine report to: /Volumes/Serine/wd/19-12-17/EPI-185_S19_L003.CpG_report.txt <<<\n",
      "\n",
      "Storing all covered cytosine positions for chromosome: Scaffold_01\n",
      "Writing cytosine report for chromosome Scaffold_01 (stored 484179 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_02 (stored 373957 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_03 (stored 300673 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_04 (stored 342166 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_05 (stored 305901 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_06 (stored 302013 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_07 (stored 223663 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_08 (stored 310818 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_09 (stored 196060 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_10 (stored 260328 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_11 (stored 251300 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_12 (stored 268719 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_13 (stored 218854 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_14 (stored 287116 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_15 (stored 223166 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_16 (stored 170762 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_17 (stored 190139 different covered positions)\n",
      "Writing cytosine report for last chromosome Scaffold_18 (stored 137016 different covered positions)\n",
      "Finished writing out cytosine report for covered chromosomes (processed 18 chromosomes/scaffolds in total)\n",
      "\n",
      "Now processing chromosomes that were not covered by any methylation calls in the coverage file...\n",
      "All chromosomes in the genome were covered by at least some reads. coverage2cytosine processing complete.\n",
      "\n",
      "Now merging top and bottom strand CpGs into a single CG dinucleotide entity (reading from file >>/Volumes/Serine/wd/19-12-17/EPI-185_S19_L003.CpG_report.txt<<, in output directory '')\n",
      ">>> Writing a new coverage file with top and bottom strand CpG methylation evidence merged to /Volumes/Serine/wd/19-12-17/EPI-185_S19_L003.CpG_report.merged_CpG_evidence.cov <<<\n",
      "\n",
      "CpG merging complete. Good luck finding DMRs with bsseq!\n",
      "\n",
      "Summary of parameters for genome-wide cytosine report:\n",
      "==============================================================================\n",
      "Coverage infile:\t\t\t\t/Volumes/Serine/wd/19-12-17/bismark_coverage_files/EPI-187_S20_L003_R1_001_val_1_bismark_bt2_pe.deduplicated.bismark.cov.gz\n",
      "Output directory:\t\t\t\t><\n",
      "Parent directory:\t\t\t\t>/Users/sr320/git-repos/nb-2019/P_generosa/<\n",
      "Genome directory:\t\t\t\t>/Volumes/Serine/wd/19-11-05/genome/<\n",
      "CX context:\t\t\t\t\tno (CpG context only, default)\n",
      "Pooling CpG top/bottom strand evidence:\t\tyes\n",
      "Genome coordinates used:\t\t\t1-based (default)\n",
      "GZIP compression:\t\t\t\tno\n",
      "Split by chromosome:\t\t\t\tno\n",
      "\n",
      "\n",
      "Now reading in and storing sequence information of the genome specified in: /Volumes/Serine/wd/19-11-05/genome/\n",
      "\n",
      "chr Scaffold_01 (89643857 bp)\n",
      "chr Scaffold_02 (69596280 bp)\n",
      "chr Scaffold_03 (57743597 bp)\n",
      "chr Scaffold_04 (65288255 bp)\n",
      "chr Scaffold_05 (67248332 bp)\n",
      "chr Scaffold_06 (61759565 bp)\n",
      "chr Scaffold_07 (43120122 bp)\n",
      "chr Scaffold_08 (61151155 bp)\n",
      "chr Scaffold_09 (38581958 bp)\n",
      "chr Scaffold_10 (53961475 bp)\n",
      "chr Scaffold_11 (51449921 bp)\n",
      "chr Scaffold_12 (50438331 bp)\n",
      "chr Scaffold_13 (44396874 bp)\n",
      "chr Scaffold_14 (45393038 bp)\n",
      "chr Scaffold_15 (47938513 bp)\n",
      "chr Scaffold_16 (31980953 bp)\n",
      "chr Scaffold_17 (34923512 bp)\n",
      "chr Scaffold_18 (27737463 bp)\n",
      "\n",
      "Stored sequence information of 18 chromosomes/scaffolds in total\n",
      "\n",
      "==============================================================================\n",
      "Methylation information will now be written into a genome-wide cytosine report\n",
      "==============================================================================\n",
      "\n",
      ">>> Writing genome-wide cytosine report to: /Volumes/Serine/wd/19-12-17/EPI-187_S20_L003.CpG_report.txt <<<\n",
      "\n",
      "Storing all covered cytosine positions for chromosome: Scaffold_01\n",
      "Writing cytosine report for chromosome Scaffold_01 (stored 881972 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_02 (stored 676395 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_03 (stored 537813 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_04 (stored 606694 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_05 (stored 553812 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_06 (stored 536520 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_07 (stored 398885 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_08 (stored 561750 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_09 (stored 348321 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_10 (stored 480620 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_11 (stored 446585 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_12 (stored 490597 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_13 (stored 397847 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_14 (stored 501399 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_15 (stored 393666 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_16 (stored 310045 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_17 (stored 333037 different covered positions)\n",
      "Writing cytosine report for last chromosome Scaffold_18 (stored 245945 different covered positions)\n",
      "Finished writing out cytosine report for covered chromosomes (processed 18 chromosomes/scaffolds in total)\n",
      "\n",
      "Now processing chromosomes that were not covered by any methylation calls in the coverage file...\n",
      "All chromosomes in the genome were covered by at least some reads. coverage2cytosine processing complete.\n",
      "\n",
      "Now merging top and bottom strand CpGs into a single CG dinucleotide entity (reading from file >>/Volumes/Serine/wd/19-12-17/EPI-187_S20_L003.CpG_report.txt<<, in output directory '')\n",
      ">>> Writing a new coverage file with top and bottom strand CpG methylation evidence merged to /Volumes/Serine/wd/19-12-17/EPI-187_S20_L003.CpG_report.merged_CpG_evidence.cov <<<\n",
      "\n",
      "CpG merging complete. Good luck finding DMRs with bsseq!\n",
      "\n",
      "Summary of parameters for genome-wide cytosine report:\n",
      "==============================================================================\n",
      "Coverage infile:\t\t\t\t/Volumes/Serine/wd/19-12-17/bismark_coverage_files/EPI-188_S21_L003_R1_001_val_1_bismark_bt2_pe.deduplicated.bismark.cov.gz\n",
      "Output directory:\t\t\t\t><\n",
      "Parent directory:\t\t\t\t>/Users/sr320/git-repos/nb-2019/P_generosa/<\n",
      "Genome directory:\t\t\t\t>/Volumes/Serine/wd/19-11-05/genome/<\n",
      "CX context:\t\t\t\t\tno (CpG context only, default)\n",
      "Pooling CpG top/bottom strand evidence:\t\tyes\n",
      "Genome coordinates used:\t\t\t1-based (default)\n",
      "GZIP compression:\t\t\t\tno\n",
      "Split by chromosome:\t\t\t\tno\n",
      "\n",
      "\n",
      "Now reading in and storing sequence information of the genome specified in: /Volumes/Serine/wd/19-11-05/genome/\n",
      "\n",
      "chr Scaffold_01 (89643857 bp)\n",
      "chr Scaffold_02 (69596280 bp)\n",
      "chr Scaffold_03 (57743597 bp)\n",
      "chr Scaffold_04 (65288255 bp)\n",
      "chr Scaffold_05 (67248332 bp)\n",
      "chr Scaffold_06 (61759565 bp)\n",
      "chr Scaffold_07 (43120122 bp)\n",
      "chr Scaffold_08 (61151155 bp)\n",
      "chr Scaffold_09 (38581958 bp)\n",
      "chr Scaffold_10 (53961475 bp)\n",
      "chr Scaffold_11 (51449921 bp)\n",
      "chr Scaffold_12 (50438331 bp)\n",
      "chr Scaffold_13 (44396874 bp)\n",
      "chr Scaffold_14 (45393038 bp)\n",
      "chr Scaffold_15 (47938513 bp)\n",
      "chr Scaffold_16 (31980953 bp)\n",
      "chr Scaffold_17 (34923512 bp)\n",
      "chr Scaffold_18 (27737463 bp)\n",
      "\n",
      "Stored sequence information of 18 chromosomes/scaffolds in total\n",
      "\n",
      "==============================================================================\n",
      "Methylation information will now be written into a genome-wide cytosine report\n",
      "==============================================================================\n",
      "\n",
      ">>> Writing genome-wide cytosine report to: /Volumes/Serine/wd/19-12-17/EPI-188_S21_L003.CpG_report.txt <<<\n",
      "\n",
      "Storing all covered cytosine positions for chromosome: Scaffold_01\n",
      "Writing cytosine report for chromosome Scaffold_01 (stored 798150 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_02 (stored 615466 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_03 (stored 495784 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_04 (stored 567021 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_05 (stored 511133 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_06 (stored 497142 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_07 (stored 363280 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_08 (stored 511105 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_09 (stored 322414 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_10 (stored 427587 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_11 (stored 412612 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_12 (stored 442809 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_13 (stored 360304 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_14 (stored 459543 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_15 (stored 363610 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_16 (stored 288904 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_17 (stored 311855 different covered positions)\n",
      "Writing cytosine report for last chromosome Scaffold_18 (stored 224087 different covered positions)\n",
      "Finished writing out cytosine report for covered chromosomes (processed 18 chromosomes/scaffolds in total)\n",
      "\n",
      "Now processing chromosomes that were not covered by any methylation calls in the coverage file...\n",
      "All chromosomes in the genome were covered by at least some reads. coverage2cytosine processing complete.\n",
      "\n",
      "Now merging top and bottom strand CpGs into a single CG dinucleotide entity (reading from file >>/Volumes/Serine/wd/19-12-17/EPI-188_S21_L003.CpG_report.txt<<, in output directory '')\n",
      ">>> Writing a new coverage file with top and bottom strand CpG methylation evidence merged to /Volumes/Serine/wd/19-12-17/EPI-188_S21_L003.CpG_report.merged_CpG_evidence.cov <<<\n",
      "\n",
      "CpG merging complete. Good luck finding DMRs with bsseq!\n",
      "\n",
      "Summary of parameters for genome-wide cytosine report:\n",
      "==============================================================================\n",
      "Coverage infile:\t\t\t\t/Volumes/Serine/wd/19-12-17/bismark_coverage_files/EPI-193_S22_L003_R1_001_val_1_bismark_bt2_pe.deduplicated.bismark.cov.gz\n",
      "Output directory:\t\t\t\t><\n",
      "Parent directory:\t\t\t\t>/Users/sr320/git-repos/nb-2019/P_generosa/<\n",
      "Genome directory:\t\t\t\t>/Volumes/Serine/wd/19-11-05/genome/<\n",
      "CX context:\t\t\t\t\tno (CpG context only, default)\n",
      "Pooling CpG top/bottom strand evidence:\t\tyes\n",
      "Genome coordinates used:\t\t\t1-based (default)\n",
      "GZIP compression:\t\t\t\tno\n",
      "Split by chromosome:\t\t\t\tno\n",
      "\n",
      "\n",
      "Now reading in and storing sequence information of the genome specified in: /Volumes/Serine/wd/19-11-05/genome/\n",
      "\n",
      "chr Scaffold_01 (89643857 bp)\n",
      "chr Scaffold_02 (69596280 bp)\n",
      "chr Scaffold_03 (57743597 bp)\n",
      "chr Scaffold_04 (65288255 bp)\n",
      "chr Scaffold_05 (67248332 bp)\n",
      "chr Scaffold_06 (61759565 bp)\n",
      "chr Scaffold_07 (43120122 bp)\n",
      "chr Scaffold_08 (61151155 bp)\n",
      "chr Scaffold_09 (38581958 bp)\n",
      "chr Scaffold_10 (53961475 bp)\n",
      "chr Scaffold_11 (51449921 bp)\n",
      "chr Scaffold_12 (50438331 bp)\n",
      "chr Scaffold_13 (44396874 bp)\n",
      "chr Scaffold_14 (45393038 bp)\n",
      "chr Scaffold_15 (47938513 bp)\n",
      "chr Scaffold_16 (31980953 bp)\n",
      "chr Scaffold_17 (34923512 bp)\n",
      "chr Scaffold_18 (27737463 bp)\n",
      "\n",
      "Stored sequence information of 18 chromosomes/scaffolds in total\n",
      "\n",
      "==============================================================================\n",
      "Methylation information will now be written into a genome-wide cytosine report\n",
      "==============================================================================\n",
      "\n",
      ">>> Writing genome-wide cytosine report to: /Volumes/Serine/wd/19-12-17/EPI-193_S22_L003.CpG_report.txt <<<\n",
      "\n",
      "Storing all covered cytosine positions for chromosome: Scaffold_01\n",
      "Writing cytosine report for chromosome Scaffold_01 (stored 940810 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_02 (stored 725522 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_03 (stored 581439 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_04 (stored 659857 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_05 (stored 597631 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_06 (stored 583141 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_07 (stored 431107 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_08 (stored 601439 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_09 (stored 379198 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_10 (stored 504234 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_11 (stored 488112 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_12 (stored 524874 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_13 (stored 426238 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_14 (stored 543391 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_15 (stored 430378 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_16 (stored 339210 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_17 (stored 365463 different covered positions)\n",
      "Writing cytosine report for last chromosome Scaffold_18 (stored 262264 different covered positions)\n",
      "Finished writing out cytosine report for covered chromosomes (processed 18 chromosomes/scaffolds in total)\n",
      "\n",
      "Now processing chromosomes that were not covered by any methylation calls in the coverage file...\n",
      "All chromosomes in the genome were covered by at least some reads. coverage2cytosine processing complete.\n",
      "\n",
      "Now merging top and bottom strand CpGs into a single CG dinucleotide entity (reading from file >>/Volumes/Serine/wd/19-12-17/EPI-193_S22_L003.CpG_report.txt<<, in output directory '')\n",
      ">>> Writing a new coverage file with top and bottom strand CpG methylation evidence merged to /Volumes/Serine/wd/19-12-17/EPI-193_S22_L003.CpG_report.merged_CpG_evidence.cov <<<\n",
      "\n",
      "CpG merging complete. Good luck finding DMRs with bsseq!\n",
      "\n",
      "Summary of parameters for genome-wide cytosine report:\n",
      "==============================================================================\n",
      "Coverage infile:\t\t\t\t/Volumes/Serine/wd/19-12-17/bismark_coverage_files/EPI-194_S23_L003_R1_001_val_1_bismark_bt2_pe.deduplicated.bismark.cov.gz\n",
      "Output directory:\t\t\t\t><\n",
      "Parent directory:\t\t\t\t>/Users/sr320/git-repos/nb-2019/P_generosa/<\n",
      "Genome directory:\t\t\t\t>/Volumes/Serine/wd/19-11-05/genome/<\n",
      "CX context:\t\t\t\t\tno (CpG context only, default)\n",
      "Pooling CpG top/bottom strand evidence:\t\tyes\n",
      "Genome coordinates used:\t\t\t1-based (default)\n",
      "GZIP compression:\t\t\t\tno\n",
      "Split by chromosome:\t\t\t\tno\n",
      "\n",
      "\n",
      "Now reading in and storing sequence information of the genome specified in: /Volumes/Serine/wd/19-11-05/genome/\n",
      "\n",
      "chr Scaffold_01 (89643857 bp)\n",
      "chr Scaffold_02 (69596280 bp)\n",
      "chr Scaffold_03 (57743597 bp)\n",
      "chr Scaffold_04 (65288255 bp)\n",
      "chr Scaffold_05 (67248332 bp)\n",
      "chr Scaffold_06 (61759565 bp)\n",
      "chr Scaffold_07 (43120122 bp)\n",
      "chr Scaffold_08 (61151155 bp)\n",
      "chr Scaffold_09 (38581958 bp)\n",
      "chr Scaffold_10 (53961475 bp)\n",
      "chr Scaffold_11 (51449921 bp)\n",
      "chr Scaffold_12 (50438331 bp)\n",
      "chr Scaffold_13 (44396874 bp)\n",
      "chr Scaffold_14 (45393038 bp)\n",
      "chr Scaffold_15 (47938513 bp)\n",
      "chr Scaffold_16 (31980953 bp)\n",
      "chr Scaffold_17 (34923512 bp)\n",
      "chr Scaffold_18 (27737463 bp)\n",
      "\n",
      "Stored sequence information of 18 chromosomes/scaffolds in total\n",
      "\n",
      "==============================================================================\n",
      "Methylation information will now be written into a genome-wide cytosine report\n",
      "==============================================================================\n",
      "\n",
      ">>> Writing genome-wide cytosine report to: /Volumes/Serine/wd/19-12-17/EPI-194_S23_L003.CpG_report.txt <<<\n",
      "\n",
      "Storing all covered cytosine positions for chromosome: Scaffold_01\n",
      "Writing cytosine report for chromosome Scaffold_01 (stored 1062314 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_02 (stored 821406 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_03 (stored 659527 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_04 (stored 749875 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_05 (stored 678861 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_06 (stored 660722 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_07 (stored 485774 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_08 (stored 682495 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_09 (stored 428876 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_10 (stored 576099 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_11 (stored 553837 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_12 (stored 593795 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_13 (stored 485470 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_14 (stored 609850 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_15 (stored 485152 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_16 (stored 384247 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_17 (stored 411437 different covered positions)\n",
      "Writing cytosine report for last chromosome Scaffold_18 (stored 297624 different covered positions)\n",
      "Finished writing out cytosine report for covered chromosomes (processed 18 chromosomes/scaffolds in total)\n",
      "\n",
      "Now processing chromosomes that were not covered by any methylation calls in the coverage file...\n",
      "All chromosomes in the genome were covered by at least some reads. coverage2cytosine processing complete.\n",
      "\n",
      "Now merging top and bottom strand CpGs into a single CG dinucleotide entity (reading from file >>/Volumes/Serine/wd/19-12-17/EPI-194_S23_L003.CpG_report.txt<<, in output directory '')\n",
      ">>> Writing a new coverage file with top and bottom strand CpG methylation evidence merged to /Volumes/Serine/wd/19-12-17/EPI-194_S23_L003.CpG_report.merged_CpG_evidence.cov <<<\n",
      "\n",
      "CpG merging complete. Good luck finding DMRs with bsseq!\n",
      "\n",
      "Summary of parameters for genome-wide cytosine report:\n",
      "==============================================================================\n",
      "Coverage infile:\t\t\t\t/Volumes/Serine/wd/19-12-17/bismark_coverage_files/EPI-199_S24_L003_R1_001_val_1_bismark_bt2_pe.deduplicated.bismark.cov.gz\n",
      "Output directory:\t\t\t\t><\n",
      "Parent directory:\t\t\t\t>/Users/sr320/git-repos/nb-2019/P_generosa/<\n",
      "Genome directory:\t\t\t\t>/Volumes/Serine/wd/19-11-05/genome/<\n",
      "CX context:\t\t\t\t\tno (CpG context only, default)\n",
      "Pooling CpG top/bottom strand evidence:\t\tyes\n",
      "Genome coordinates used:\t\t\t1-based (default)\n",
      "GZIP compression:\t\t\t\tno\n",
      "Split by chromosome:\t\t\t\tno\n",
      "\n",
      "\n",
      "Now reading in and storing sequence information of the genome specified in: /Volumes/Serine/wd/19-11-05/genome/\n",
      "\n",
      "chr Scaffold_01 (89643857 bp)\n",
      "chr Scaffold_02 (69596280 bp)\n",
      "chr Scaffold_03 (57743597 bp)\n",
      "chr Scaffold_04 (65288255 bp)\n",
      "chr Scaffold_05 (67248332 bp)\n",
      "chr Scaffold_06 (61759565 bp)\n",
      "chr Scaffold_07 (43120122 bp)\n",
      "chr Scaffold_08 (61151155 bp)\n",
      "chr Scaffold_09 (38581958 bp)\n",
      "chr Scaffold_10 (53961475 bp)\n",
      "chr Scaffold_11 (51449921 bp)\n",
      "chr Scaffold_12 (50438331 bp)\n",
      "chr Scaffold_13 (44396874 bp)\n",
      "chr Scaffold_14 (45393038 bp)\n",
      "chr Scaffold_15 (47938513 bp)\n",
      "chr Scaffold_16 (31980953 bp)\n",
      "chr Scaffold_17 (34923512 bp)\n",
      "chr Scaffold_18 (27737463 bp)\n",
      "\n",
      "Stored sequence information of 18 chromosomes/scaffolds in total\n",
      "\n",
      "==============================================================================\n",
      "Methylation information will now be written into a genome-wide cytosine report\n",
      "==============================================================================\n",
      "\n",
      ">>> Writing genome-wide cytosine report to: /Volumes/Serine/wd/19-12-17/EPI-199_S24_L003.CpG_report.txt <<<\n",
      "\n",
      "Storing all covered cytosine positions for chromosome: Scaffold_01\n",
      "Writing cytosine report for chromosome Scaffold_01 (stored 638642 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_02 (stored 489136 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_03 (stored 397405 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_04 (stored 447812 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_05 (stored 402771 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_06 (stored 391983 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_07 (stored 293167 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_08 (stored 410189 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_09 (stored 258109 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_10 (stored 343796 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_11 (stored 328259 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_12 (stored 353616 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_13 (stored 290825 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_14 (stored 371946 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_15 (stored 290202 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_16 (stored 227350 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_17 (stored 246994 different covered positions)\n",
      "Writing cytosine report for last chromosome Scaffold_18 (stored 179607 different covered positions)\n",
      "Finished writing out cytosine report for covered chromosomes (processed 18 chromosomes/scaffolds in total)\n",
      "\n",
      "Now processing chromosomes that were not covered by any methylation calls in the coverage file...\n",
      "All chromosomes in the genome were covered by at least some reads. coverage2cytosine processing complete.\n",
      "\n",
      "Now merging top and bottom strand CpGs into a single CG dinucleotide entity (reading from file >>/Volumes/Serine/wd/19-12-17/EPI-199_S24_L003.CpG_report.txt<<, in output directory '')\n",
      ">>> Writing a new coverage file with top and bottom strand CpG methylation evidence merged to /Volumes/Serine/wd/19-12-17/EPI-199_S24_L003.CpG_report.merged_CpG_evidence.cov <<<\n",
      "\n",
      "CpG merging complete. Good luck finding DMRs with bsseq!\n",
      "\n",
      "Summary of parameters for genome-wide cytosine report:\n",
      "==============================================================================\n",
      "Coverage infile:\t\t\t\t/Volumes/Serine/wd/19-12-17/bismark_coverage_files/EPI-200_S25_L003_R1_001_val_1_bismark_bt2_pe.deduplicated.bismark.cov.gz\n",
      "Output directory:\t\t\t\t><\n",
      "Parent directory:\t\t\t\t>/Users/sr320/git-repos/nb-2019/P_generosa/<\n",
      "Genome directory:\t\t\t\t>/Volumes/Serine/wd/19-11-05/genome/<\n",
      "CX context:\t\t\t\t\tno (CpG context only, default)\n",
      "Pooling CpG top/bottom strand evidence:\t\tyes\n",
      "Genome coordinates used:\t\t\t1-based (default)\n",
      "GZIP compression:\t\t\t\tno\n",
      "Split by chromosome:\t\t\t\tno\n",
      "\n",
      "\n",
      "Now reading in and storing sequence information of the genome specified in: /Volumes/Serine/wd/19-11-05/genome/\n",
      "\n",
      "chr Scaffold_01 (89643857 bp)\n",
      "chr Scaffold_02 (69596280 bp)\n",
      "chr Scaffold_03 (57743597 bp)\n",
      "chr Scaffold_04 (65288255 bp)\n",
      "chr Scaffold_05 (67248332 bp)\n",
      "chr Scaffold_06 (61759565 bp)\n",
      "chr Scaffold_07 (43120122 bp)\n",
      "chr Scaffold_08 (61151155 bp)\n",
      "chr Scaffold_09 (38581958 bp)\n",
      "chr Scaffold_10 (53961475 bp)\n",
      "chr Scaffold_11 (51449921 bp)\n",
      "chr Scaffold_12 (50438331 bp)\n",
      "chr Scaffold_13 (44396874 bp)\n",
      "chr Scaffold_14 (45393038 bp)\n",
      "chr Scaffold_15 (47938513 bp)\n",
      "chr Scaffold_16 (31980953 bp)\n",
      "chr Scaffold_17 (34923512 bp)\n",
      "chr Scaffold_18 (27737463 bp)\n",
      "\n",
      "Stored sequence information of 18 chromosomes/scaffolds in total\n",
      "\n",
      "==============================================================================\n",
      "Methylation information will now be written into a genome-wide cytosine report\n",
      "==============================================================================\n",
      "\n",
      ">>> Writing genome-wide cytosine report to: /Volumes/Serine/wd/19-12-17/EPI-200_S25_L003.CpG_report.txt <<<\n",
      "\n",
      "Storing all covered cytosine positions for chromosome: Scaffold_01\n",
      "Writing cytosine report for chromosome Scaffold_01 (stored 844120 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_02 (stored 649218 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_03 (stored 524205 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_04 (stored 593654 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_05 (stored 537636 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_06 (stored 521971 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_07 (stored 388534 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_08 (stored 540152 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_09 (stored 337515 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_10 (stored 456011 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_11 (stored 433925 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_12 (stored 470310 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_13 (stored 383574 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_14 (stored 486062 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_15 (stored 384431 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_16 (stored 301035 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_17 (stored 324842 different covered positions)\n",
      "Writing cytosine report for last chromosome Scaffold_18 (stored 234051 different covered positions)\n",
      "Finished writing out cytosine report for covered chromosomes (processed 18 chromosomes/scaffolds in total)\n",
      "\n",
      "Now processing chromosomes that were not covered by any methylation calls in the coverage file...\n",
      "All chromosomes in the genome were covered by at least some reads. coverage2cytosine processing complete.\n",
      "\n",
      "Now merging top and bottom strand CpGs into a single CG dinucleotide entity (reading from file >>/Volumes/Serine/wd/19-12-17/EPI-200_S25_L003.CpG_report.txt<<, in output directory '')\n",
      ">>> Writing a new coverage file with top and bottom strand CpG methylation evidence merged to /Volumes/Serine/wd/19-12-17/EPI-200_S25_L003.CpG_report.merged_CpG_evidence.cov <<<\n",
      "\n",
      "CpG merging complete. Good luck finding DMRs with bsseq!\n",
      "\n",
      "Summary of parameters for genome-wide cytosine report:\n",
      "==============================================================================\n",
      "Coverage infile:\t\t\t\t/Volumes/Serine/wd/19-12-17/bismark_coverage_files/EPI-205_S26_L004_R1_001_val_1_bismark_bt2_pe.deduplicated.bismark.cov.gz\n",
      "Output directory:\t\t\t\t><\n",
      "Parent directory:\t\t\t\t>/Users/sr320/git-repos/nb-2019/P_generosa/<\n",
      "Genome directory:\t\t\t\t>/Volumes/Serine/wd/19-11-05/genome/<\n",
      "CX context:\t\t\t\t\tno (CpG context only, default)\n",
      "Pooling CpG top/bottom strand evidence:\t\tyes\n",
      "Genome coordinates used:\t\t\t1-based (default)\n",
      "GZIP compression:\t\t\t\tno\n",
      "Split by chromosome:\t\t\t\tno\n",
      "\n",
      "\n",
      "Now reading in and storing sequence information of the genome specified in: /Volumes/Serine/wd/19-11-05/genome/\n",
      "\n",
      "chr Scaffold_01 (89643857 bp)\n",
      "chr Scaffold_02 (69596280 bp)\n",
      "chr Scaffold_03 (57743597 bp)\n",
      "chr Scaffold_04 (65288255 bp)\n",
      "chr Scaffold_05 (67248332 bp)\n",
      "chr Scaffold_06 (61759565 bp)\n",
      "chr Scaffold_07 (43120122 bp)\n",
      "chr Scaffold_08 (61151155 bp)\n",
      "chr Scaffold_09 (38581958 bp)\n",
      "chr Scaffold_10 (53961475 bp)\n",
      "chr Scaffold_11 (51449921 bp)\n",
      "chr Scaffold_12 (50438331 bp)\n",
      "chr Scaffold_13 (44396874 bp)\n",
      "chr Scaffold_14 (45393038 bp)\n",
      "chr Scaffold_15 (47938513 bp)\n",
      "chr Scaffold_16 (31980953 bp)\n",
      "chr Scaffold_17 (34923512 bp)\n",
      "chr Scaffold_18 (27737463 bp)\n",
      "\n",
      "Stored sequence information of 18 chromosomes/scaffolds in total\n",
      "\n",
      "==============================================================================\n",
      "Methylation information will now be written into a genome-wide cytosine report\n",
      "==============================================================================\n",
      "\n",
      ">>> Writing genome-wide cytosine report to: /Volumes/Serine/wd/19-12-17/EPI-205_S26_L004.CpG_report.txt <<<\n",
      "\n",
      "Storing all covered cytosine positions for chromosome: Scaffold_01\n",
      "Writing cytosine report for chromosome Scaffold_01 (stored 633868 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_02 (stored 489365 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_03 (stored 392582 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_04 (stored 447283 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_05 (stored 405765 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_06 (stored 389583 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_07 (stored 288420 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_08 (stored 405302 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_09 (stored 255438 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_10 (stored 343287 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_11 (stored 326515 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_12 (stored 353084 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_13 (stored 287684 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_14 (stored 367743 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_15 (stored 289319 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_16 (stored 227685 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_17 (stored 246330 different covered positions)\n",
      "Writing cytosine report for last chromosome Scaffold_18 (stored 178113 different covered positions)\n",
      "Finished writing out cytosine report for covered chromosomes (processed 18 chromosomes/scaffolds in total)\n",
      "\n",
      "Now processing chromosomes that were not covered by any methylation calls in the coverage file...\n",
      "All chromosomes in the genome were covered by at least some reads. coverage2cytosine processing complete.\n",
      "\n",
      "Now merging top and bottom strand CpGs into a single CG dinucleotide entity (reading from file >>/Volumes/Serine/wd/19-12-17/EPI-205_S26_L004.CpG_report.txt<<, in output directory '')\n",
      ">>> Writing a new coverage file with top and bottom strand CpG methylation evidence merged to /Volumes/Serine/wd/19-12-17/EPI-205_S26_L004.CpG_report.merged_CpG_evidence.cov <<<\n",
      "\n",
      "CpG merging complete. Good luck finding DMRs with bsseq!\n",
      "\n",
      "Summary of parameters for genome-wide cytosine report:\n",
      "==============================================================================\n",
      "Coverage infile:\t\t\t\t/Volumes/Serine/wd/19-12-17/bismark_coverage_files/EPI-206_S27_L004_R1_001_val_1_bismark_bt2_pe.deduplicated.bismark.cov.gz\n",
      "Output directory:\t\t\t\t><\n",
      "Parent directory:\t\t\t\t>/Users/sr320/git-repos/nb-2019/P_generosa/<\n",
      "Genome directory:\t\t\t\t>/Volumes/Serine/wd/19-11-05/genome/<\n",
      "CX context:\t\t\t\t\tno (CpG context only, default)\n",
      "Pooling CpG top/bottom strand evidence:\t\tyes\n",
      "Genome coordinates used:\t\t\t1-based (default)\n",
      "GZIP compression:\t\t\t\tno\n",
      "Split by chromosome:\t\t\t\tno\n",
      "\n",
      "\n",
      "Now reading in and storing sequence information of the genome specified in: /Volumes/Serine/wd/19-11-05/genome/\n",
      "\n",
      "chr Scaffold_01 (89643857 bp)\n",
      "chr Scaffold_02 (69596280 bp)\n",
      "chr Scaffold_03 (57743597 bp)\n",
      "chr Scaffold_04 (65288255 bp)\n",
      "chr Scaffold_05 (67248332 bp)\n",
      "chr Scaffold_06 (61759565 bp)\n",
      "chr Scaffold_07 (43120122 bp)\n",
      "chr Scaffold_08 (61151155 bp)\n",
      "chr Scaffold_09 (38581958 bp)\n",
      "chr Scaffold_10 (53961475 bp)\n",
      "chr Scaffold_11 (51449921 bp)\n",
      "chr Scaffold_12 (50438331 bp)\n",
      "chr Scaffold_13 (44396874 bp)\n",
      "chr Scaffold_14 (45393038 bp)\n",
      "chr Scaffold_15 (47938513 bp)\n",
      "chr Scaffold_16 (31980953 bp)\n",
      "chr Scaffold_17 (34923512 bp)\n",
      "chr Scaffold_18 (27737463 bp)\n",
      "\n",
      "Stored sequence information of 18 chromosomes/scaffolds in total\n",
      "\n",
      "==============================================================================\n",
      "Methylation information will now be written into a genome-wide cytosine report\n",
      "==============================================================================\n",
      "\n",
      ">>> Writing genome-wide cytosine report to: /Volumes/Serine/wd/19-12-17/EPI-206_S27_L004.CpG_report.txt <<<\n",
      "\n",
      "Storing all covered cytosine positions for chromosome: Scaffold_01\n",
      "Writing cytosine report for chromosome Scaffold_01 (stored 819271 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_02 (stored 633748 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_03 (stored 508285 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_04 (stored 573892 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_05 (stored 523880 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_06 (stored 502290 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_07 (stored 369655 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_08 (stored 524936 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_09 (stored 329216 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_10 (stored 446368 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_11 (stored 418828 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_12 (stored 460621 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_13 (stored 370383 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_14 (stored 477303 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_15 (stored 369033 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_16 (stored 292288 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_17 (stored 315950 different covered positions)\n",
      "Writing cytosine report for last chromosome Scaffold_18 (stored 228844 different covered positions)\n",
      "Finished writing out cytosine report for covered chromosomes (processed 18 chromosomes/scaffolds in total)\n",
      "\n",
      "Now processing chromosomes that were not covered by any methylation calls in the coverage file...\n",
      "All chromosomes in the genome were covered by at least some reads. coverage2cytosine processing complete.\n",
      "\n",
      "Now merging top and bottom strand CpGs into a single CG dinucleotide entity (reading from file >>/Volumes/Serine/wd/19-12-17/EPI-206_S27_L004.CpG_report.txt<<, in output directory '')\n",
      ">>> Writing a new coverage file with top and bottom strand CpG methylation evidence merged to /Volumes/Serine/wd/19-12-17/EPI-206_S27_L004.CpG_report.merged_CpG_evidence.cov <<<\n",
      "\n",
      "CpG merging complete. Good luck finding DMRs with bsseq!\n",
      "\n",
      "Summary of parameters for genome-wide cytosine report:\n",
      "==============================================================================\n",
      "Coverage infile:\t\t\t\t/Volumes/Serine/wd/19-12-17/bismark_coverage_files/EPI-208_S28_L004_R1_001_val_1_bismark_bt2_pe.deduplicated.bismark.cov.gz\n",
      "Output directory:\t\t\t\t><\n",
      "Parent directory:\t\t\t\t>/Users/sr320/git-repos/nb-2019/P_generosa/<\n",
      "Genome directory:\t\t\t\t>/Volumes/Serine/wd/19-11-05/genome/<\n",
      "CX context:\t\t\t\t\tno (CpG context only, default)\n",
      "Pooling CpG top/bottom strand evidence:\t\tyes\n",
      "Genome coordinates used:\t\t\t1-based (default)\n",
      "GZIP compression:\t\t\t\tno\n",
      "Split by chromosome:\t\t\t\tno\n",
      "\n",
      "\n",
      "Now reading in and storing sequence information of the genome specified in: /Volumes/Serine/wd/19-11-05/genome/\n",
      "\n",
      "chr Scaffold_01 (89643857 bp)\n",
      "chr Scaffold_02 (69596280 bp)\n",
      "chr Scaffold_03 (57743597 bp)\n",
      "chr Scaffold_04 (65288255 bp)\n",
      "chr Scaffold_05 (67248332 bp)\n",
      "chr Scaffold_06 (61759565 bp)\n",
      "chr Scaffold_07 (43120122 bp)\n",
      "chr Scaffold_08 (61151155 bp)\n",
      "chr Scaffold_09 (38581958 bp)\n",
      "chr Scaffold_10 (53961475 bp)\n",
      "chr Scaffold_11 (51449921 bp)\n",
      "chr Scaffold_12 (50438331 bp)\n",
      "chr Scaffold_13 (44396874 bp)\n",
      "chr Scaffold_14 (45393038 bp)\n",
      "chr Scaffold_15 (47938513 bp)\n",
      "chr Scaffold_16 (31980953 bp)\n",
      "chr Scaffold_17 (34923512 bp)\n",
      "chr Scaffold_18 (27737463 bp)\n",
      "\n",
      "Stored sequence information of 18 chromosomes/scaffolds in total\n",
      "\n",
      "==============================================================================\n",
      "Methylation information will now be written into a genome-wide cytosine report\n",
      "==============================================================================\n",
      "\n",
      ">>> Writing genome-wide cytosine report to: /Volumes/Serine/wd/19-12-17/EPI-208_S28_L004.CpG_report.txt <<<\n",
      "\n",
      "Storing all covered cytosine positions for chromosome: Scaffold_01\n",
      "Writing cytosine report for chromosome Scaffold_01 (stored 959798 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_02 (stored 739451 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_03 (stored 593722 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_04 (stored 682325 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_05 (stored 613105 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_06 (stored 592973 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_07 (stored 438640 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_08 (stored 613937 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_09 (stored 386665 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_10 (stored 513649 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_11 (stored 501028 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_12 (stored 535057 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_13 (stored 434593 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_14 (stored 553280 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_15 (stored 438881 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_16 (stored 348769 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_17 (stored 372849 different covered positions)\n",
      "Writing cytosine report for last chromosome Scaffold_18 (stored 275459 different covered positions)\n",
      "Finished writing out cytosine report for covered chromosomes (processed 18 chromosomes/scaffolds in total)\n",
      "\n",
      "Now processing chromosomes that were not covered by any methylation calls in the coverage file...\n",
      "All chromosomes in the genome were covered by at least some reads. coverage2cytosine processing complete.\n",
      "\n",
      "Now merging top and bottom strand CpGs into a single CG dinucleotide entity (reading from file >>/Volumes/Serine/wd/19-12-17/EPI-208_S28_L004.CpG_report.txt<<, in output directory '')\n",
      ">>> Writing a new coverage file with top and bottom strand CpG methylation evidence merged to /Volumes/Serine/wd/19-12-17/EPI-208_S28_L004.CpG_report.merged_CpG_evidence.cov <<<\n",
      "\n",
      "CpG merging complete. Good luck finding DMRs with bsseq!\n",
      "\n",
      "Summary of parameters for genome-wide cytosine report:\n",
      "==============================================================================\n",
      "Coverage infile:\t\t\t\t/Volumes/Serine/wd/19-12-17/bismark_coverage_files/EPI-209_S29_L004_R1_001_val_1_bismark_bt2_pe.deduplicated.bismark.cov.gz\n",
      "Output directory:\t\t\t\t><\n",
      "Parent directory:\t\t\t\t>/Users/sr320/git-repos/nb-2019/P_generosa/<\n",
      "Genome directory:\t\t\t\t>/Volumes/Serine/wd/19-11-05/genome/<\n",
      "CX context:\t\t\t\t\tno (CpG context only, default)\n",
      "Pooling CpG top/bottom strand evidence:\t\tyes\n",
      "Genome coordinates used:\t\t\t1-based (default)\n",
      "GZIP compression:\t\t\t\tno\n",
      "Split by chromosome:\t\t\t\tno\n",
      "\n",
      "\n",
      "Now reading in and storing sequence information of the genome specified in: /Volumes/Serine/wd/19-11-05/genome/\n",
      "\n",
      "chr Scaffold_01 (89643857 bp)\n",
      "chr Scaffold_02 (69596280 bp)\n",
      "chr Scaffold_03 (57743597 bp)\n",
      "chr Scaffold_04 (65288255 bp)\n",
      "chr Scaffold_05 (67248332 bp)\n",
      "chr Scaffold_06 (61759565 bp)\n",
      "chr Scaffold_07 (43120122 bp)\n",
      "chr Scaffold_08 (61151155 bp)\n",
      "chr Scaffold_09 (38581958 bp)\n",
      "chr Scaffold_10 (53961475 bp)\n",
      "chr Scaffold_11 (51449921 bp)\n",
      "chr Scaffold_12 (50438331 bp)\n",
      "chr Scaffold_13 (44396874 bp)\n",
      "chr Scaffold_14 (45393038 bp)\n",
      "chr Scaffold_15 (47938513 bp)\n",
      "chr Scaffold_16 (31980953 bp)\n",
      "chr Scaffold_17 (34923512 bp)\n",
      "chr Scaffold_18 (27737463 bp)\n",
      "\n",
      "Stored sequence information of 18 chromosomes/scaffolds in total\n",
      "\n",
      "==============================================================================\n",
      "Methylation information will now be written into a genome-wide cytosine report\n",
      "==============================================================================\n",
      "\n",
      ">>> Writing genome-wide cytosine report to: /Volumes/Serine/wd/19-12-17/EPI-209_S29_L004.CpG_report.txt <<<\n",
      "\n",
      "Storing all covered cytosine positions for chromosome: Scaffold_01\n",
      "Writing cytosine report for chromosome Scaffold_01 (stored 991820 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_02 (stored 765645 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_03 (stored 617532 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_04 (stored 698975 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_05 (stored 636316 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_06 (stored 613630 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_07 (stored 453739 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_08 (stored 635411 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_09 (stored 402034 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_10 (stored 535118 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_11 (stored 517565 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_12 (stored 552793 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_13 (stored 454112 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_14 (stored 568648 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_15 (stored 453748 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_16 (stored 356583 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_17 (stored 384548 different covered positions)\n",
      "Writing cytosine report for last chromosome Scaffold_18 (stored 278130 different covered positions)\n",
      "Finished writing out cytosine report for covered chromosomes (processed 18 chromosomes/scaffolds in total)\n",
      "\n",
      "Now processing chromosomes that were not covered by any methylation calls in the coverage file...\n",
      "All chromosomes in the genome were covered by at least some reads. coverage2cytosine processing complete.\n",
      "\n",
      "Now merging top and bottom strand CpGs into a single CG dinucleotide entity (reading from file >>/Volumes/Serine/wd/19-12-17/EPI-209_S29_L004.CpG_report.txt<<, in output directory '')\n",
      ">>> Writing a new coverage file with top and bottom strand CpG methylation evidence merged to /Volumes/Serine/wd/19-12-17/EPI-209_S29_L004.CpG_report.merged_CpG_evidence.cov <<<\n",
      "\n",
      "CpG merging complete. Good luck finding DMRs with bsseq!\n",
      "\n",
      "Summary of parameters for genome-wide cytosine report:\n",
      "==============================================================================\n",
      "Coverage infile:\t\t\t\t/Volumes/Serine/wd/19-12-17/bismark_coverage_files/EPI-214_S30_L004_R1_001_val_1_bismark_bt2_pe.deduplicated.bismark.cov.gz\n",
      "Output directory:\t\t\t\t><\n",
      "Parent directory:\t\t\t\t>/Users/sr320/git-repos/nb-2019/P_generosa/<\n",
      "Genome directory:\t\t\t\t>/Volumes/Serine/wd/19-11-05/genome/<\n",
      "CX context:\t\t\t\t\tno (CpG context only, default)\n",
      "Pooling CpG top/bottom strand evidence:\t\tyes\n",
      "Genome coordinates used:\t\t\t1-based (default)\n",
      "GZIP compression:\t\t\t\tno\n",
      "Split by chromosome:\t\t\t\tno\n",
      "\n",
      "\n",
      "Now reading in and storing sequence information of the genome specified in: /Volumes/Serine/wd/19-11-05/genome/\n",
      "\n",
      "chr Scaffold_01 (89643857 bp)\n",
      "chr Scaffold_02 (69596280 bp)\n",
      "chr Scaffold_03 (57743597 bp)\n",
      "chr Scaffold_04 (65288255 bp)\n",
      "chr Scaffold_05 (67248332 bp)\n",
      "chr Scaffold_06 (61759565 bp)\n",
      "chr Scaffold_07 (43120122 bp)\n",
      "chr Scaffold_08 (61151155 bp)\n",
      "chr Scaffold_09 (38581958 bp)\n",
      "chr Scaffold_10 (53961475 bp)\n",
      "chr Scaffold_11 (51449921 bp)\n",
      "chr Scaffold_12 (50438331 bp)\n",
      "chr Scaffold_13 (44396874 bp)\n",
      "chr Scaffold_14 (45393038 bp)\n",
      "chr Scaffold_15 (47938513 bp)\n",
      "chr Scaffold_16 (31980953 bp)\n",
      "chr Scaffold_17 (34923512 bp)\n",
      "chr Scaffold_18 (27737463 bp)\n",
      "\n",
      "Stored sequence information of 18 chromosomes/scaffolds in total\n",
      "\n",
      "==============================================================================\n",
      "Methylation information will now be written into a genome-wide cytosine report\n",
      "==============================================================================\n",
      "\n",
      ">>> Writing genome-wide cytosine report to: /Volumes/Serine/wd/19-12-17/EPI-214_S30_L004.CpG_report.txt <<<\n",
      "\n",
      "Storing all covered cytosine positions for chromosome: Scaffold_01\n",
      "Writing cytosine report for chromosome Scaffold_01 (stored 1040752 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_02 (stored 804070 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_03 (stored 644840 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_04 (stored 732741 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_05 (stored 667086 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_06 (stored 647371 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_07 (stored 471837 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_08 (stored 665764 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_09 (stored 417522 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_10 (stored 557644 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_11 (stored 541195 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_12 (stored 582086 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_13 (stored 470372 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_14 (stored 592504 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_15 (stored 475296 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_16 (stored 375326 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_17 (stored 402364 different covered positions)\n",
      "Writing cytosine report for last chromosome Scaffold_18 (stored 290665 different covered positions)\n",
      "Finished writing out cytosine report for covered chromosomes (processed 18 chromosomes/scaffolds in total)\n",
      "\n",
      "Now processing chromosomes that were not covered by any methylation calls in the coverage file...\n",
      "All chromosomes in the genome were covered by at least some reads. coverage2cytosine processing complete.\n",
      "\n",
      "Now merging top and bottom strand CpGs into a single CG dinucleotide entity (reading from file >>/Volumes/Serine/wd/19-12-17/EPI-214_S30_L004.CpG_report.txt<<, in output directory '')\n",
      ">>> Writing a new coverage file with top and bottom strand CpG methylation evidence merged to /Volumes/Serine/wd/19-12-17/EPI-214_S30_L004.CpG_report.merged_CpG_evidence.cov <<<\n",
      "\n",
      "CpG merging complete. Good luck finding DMRs with bsseq!\n",
      "\n",
      "Summary of parameters for genome-wide cytosine report:\n",
      "==============================================================================\n",
      "Coverage infile:\t\t\t\t/Volumes/Serine/wd/19-12-17/bismark_coverage_files/EPI-215_S31_L004_R1_001_val_1_bismark_bt2_pe.deduplicated.bismark.cov.gz\n",
      "Output directory:\t\t\t\t><\n",
      "Parent directory:\t\t\t\t>/Users/sr320/git-repos/nb-2019/P_generosa/<\n",
      "Genome directory:\t\t\t\t>/Volumes/Serine/wd/19-11-05/genome/<\n",
      "CX context:\t\t\t\t\tno (CpG context only, default)\n",
      "Pooling CpG top/bottom strand evidence:\t\tyes\n",
      "Genome coordinates used:\t\t\t1-based (default)\n",
      "GZIP compression:\t\t\t\tno\n",
      "Split by chromosome:\t\t\t\tno\n",
      "\n",
      "\n",
      "Now reading in and storing sequence information of the genome specified in: /Volumes/Serine/wd/19-11-05/genome/\n",
      "\n",
      "chr Scaffold_01 (89643857 bp)\n",
      "chr Scaffold_02 (69596280 bp)\n",
      "chr Scaffold_03 (57743597 bp)\n",
      "chr Scaffold_04 (65288255 bp)\n",
      "chr Scaffold_05 (67248332 bp)\n",
      "chr Scaffold_06 (61759565 bp)\n",
      "chr Scaffold_07 (43120122 bp)\n",
      "chr Scaffold_08 (61151155 bp)\n",
      "chr Scaffold_09 (38581958 bp)\n",
      "chr Scaffold_10 (53961475 bp)\n",
      "chr Scaffold_11 (51449921 bp)\n",
      "chr Scaffold_12 (50438331 bp)\n",
      "chr Scaffold_13 (44396874 bp)\n",
      "chr Scaffold_14 (45393038 bp)\n",
      "chr Scaffold_15 (47938513 bp)\n",
      "chr Scaffold_16 (31980953 bp)\n",
      "chr Scaffold_17 (34923512 bp)\n",
      "chr Scaffold_18 (27737463 bp)\n",
      "\n",
      "Stored sequence information of 18 chromosomes/scaffolds in total\n",
      "\n",
      "==============================================================================\n",
      "Methylation information will now be written into a genome-wide cytosine report\n",
      "==============================================================================\n",
      "\n",
      ">>> Writing genome-wide cytosine report to: /Volumes/Serine/wd/19-12-17/EPI-215_S31_L004.CpG_report.txt <<<\n",
      "\n",
      "Storing all covered cytosine positions for chromosome: Scaffold_01\n",
      "Writing cytosine report for chromosome Scaffold_01 (stored 787457 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_02 (stored 605843 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_03 (stored 488162 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_04 (stored 548026 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_05 (stored 498521 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_06 (stored 483934 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_07 (stored 360961 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_08 (stored 504654 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_09 (stored 317439 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_10 (stored 425813 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_11 (stored 405567 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_12 (stored 433779 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_13 (stored 356256 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_14 (stored 452628 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_15 (stored 360234 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_16 (stored 278919 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_17 (stored 303620 different covered positions)\n",
      "Writing cytosine report for last chromosome Scaffold_18 (stored 219955 different covered positions)\n",
      "Finished writing out cytosine report for covered chromosomes (processed 18 chromosomes/scaffolds in total)\n",
      "\n",
      "Now processing chromosomes that were not covered by any methylation calls in the coverage file...\n",
      "All chromosomes in the genome were covered by at least some reads. coverage2cytosine processing complete.\n",
      "\n",
      "Now merging top and bottom strand CpGs into a single CG dinucleotide entity (reading from file >>/Volumes/Serine/wd/19-12-17/EPI-215_S31_L004.CpG_report.txt<<, in output directory '')\n",
      ">>> Writing a new coverage file with top and bottom strand CpG methylation evidence merged to /Volumes/Serine/wd/19-12-17/EPI-215_S31_L004.CpG_report.merged_CpG_evidence.cov <<<\n",
      "\n",
      "CpG merging complete. Good luck finding DMRs with bsseq!\n",
      "\n",
      "Summary of parameters for genome-wide cytosine report:\n",
      "==============================================================================\n",
      "Coverage infile:\t\t\t\t/Volumes/Serine/wd/19-12-17/bismark_coverage_files/EPI-220_S32_L004_R1_001_val_1_bismark_bt2_pe.deduplicated.bismark.cov.gz\n",
      "Output directory:\t\t\t\t><\n",
      "Parent directory:\t\t\t\t>/Users/sr320/git-repos/nb-2019/P_generosa/<\n",
      "Genome directory:\t\t\t\t>/Volumes/Serine/wd/19-11-05/genome/<\n",
      "CX context:\t\t\t\t\tno (CpG context only, default)\n",
      "Pooling CpG top/bottom strand evidence:\t\tyes\n",
      "Genome coordinates used:\t\t\t1-based (default)\n",
      "GZIP compression:\t\t\t\tno\n",
      "Split by chromosome:\t\t\t\tno\n",
      "\n",
      "\n",
      "Now reading in and storing sequence information of the genome specified in: /Volumes/Serine/wd/19-11-05/genome/\n",
      "\n",
      "chr Scaffold_01 (89643857 bp)\n",
      "chr Scaffold_02 (69596280 bp)\n",
      "chr Scaffold_03 (57743597 bp)\n",
      "chr Scaffold_04 (65288255 bp)\n",
      "chr Scaffold_05 (67248332 bp)\n",
      "chr Scaffold_06 (61759565 bp)\n",
      "chr Scaffold_07 (43120122 bp)\n",
      "chr Scaffold_08 (61151155 bp)\n",
      "chr Scaffold_09 (38581958 bp)\n",
      "chr Scaffold_10 (53961475 bp)\n",
      "chr Scaffold_11 (51449921 bp)\n",
      "chr Scaffold_12 (50438331 bp)\n",
      "chr Scaffold_13 (44396874 bp)\n",
      "chr Scaffold_14 (45393038 bp)\n",
      "chr Scaffold_15 (47938513 bp)\n",
      "chr Scaffold_16 (31980953 bp)\n",
      "chr Scaffold_17 (34923512 bp)\n",
      "chr Scaffold_18 (27737463 bp)\n",
      "\n",
      "Stored sequence information of 18 chromosomes/scaffolds in total\n",
      "\n",
      "==============================================================================\n",
      "Methylation information will now be written into a genome-wide cytosine report\n",
      "==============================================================================\n",
      "\n",
      ">>> Writing genome-wide cytosine report to: /Volumes/Serine/wd/19-12-17/EPI-220_S32_L004.CpG_report.txt <<<\n",
      "\n",
      "Storing all covered cytosine positions for chromosome: Scaffold_01\n",
      "Writing cytosine report for chromosome Scaffold_01 (stored 651275 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_02 (stored 502417 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_03 (stored 404651 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_04 (stored 455894 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_05 (stored 410467 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_06 (stored 403915 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_07 (stored 296192 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_08 (stored 413495 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_09 (stored 262974 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_10 (stored 351401 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_11 (stored 335396 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_12 (stored 363128 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_13 (stored 295032 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_14 (stored 377557 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_15 (stored 294735 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_16 (stored 233828 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_17 (stored 252190 different covered positions)\n",
      "Writing cytosine report for last chromosome Scaffold_18 (stored 181839 different covered positions)\n",
      "Finished writing out cytosine report for covered chromosomes (processed 18 chromosomes/scaffolds in total)\n",
      "\n",
      "Now processing chromosomes that were not covered by any methylation calls in the coverage file...\n",
      "All chromosomes in the genome were covered by at least some reads. coverage2cytosine processing complete.\n",
      "\n",
      "Now merging top and bottom strand CpGs into a single CG dinucleotide entity (reading from file >>/Volumes/Serine/wd/19-12-17/EPI-220_S32_L004.CpG_report.txt<<, in output directory '')\n",
      ">>> Writing a new coverage file with top and bottom strand CpG methylation evidence merged to /Volumes/Serine/wd/19-12-17/EPI-220_S32_L004.CpG_report.merged_CpG_evidence.cov <<<\n",
      "\n",
      "CpG merging complete. Good luck finding DMRs with bsseq!\n",
      "\n",
      "Summary of parameters for genome-wide cytosine report:\n",
      "==============================================================================\n",
      "Coverage infile:\t\t\t\t/Volumes/Serine/wd/19-12-17/bismark_coverage_files/EPI-221_S33_L004_R1_001_val_1_bismark_bt2_pe.deduplicated.bismark.cov.gz\n",
      "Output directory:\t\t\t\t><\n",
      "Parent directory:\t\t\t\t>/Users/sr320/git-repos/nb-2019/P_generosa/<\n",
      "Genome directory:\t\t\t\t>/Volumes/Serine/wd/19-11-05/genome/<\n",
      "CX context:\t\t\t\t\tno (CpG context only, default)\n",
      "Pooling CpG top/bottom strand evidence:\t\tyes\n",
      "Genome coordinates used:\t\t\t1-based (default)\n",
      "GZIP compression:\t\t\t\tno\n",
      "Split by chromosome:\t\t\t\tno\n",
      "\n",
      "\n",
      "Now reading in and storing sequence information of the genome specified in: /Volumes/Serine/wd/19-11-05/genome/\n",
      "\n",
      "chr Scaffold_01 (89643857 bp)\n",
      "chr Scaffold_02 (69596280 bp)\n",
      "chr Scaffold_03 (57743597 bp)\n",
      "chr Scaffold_04 (65288255 bp)\n",
      "chr Scaffold_05 (67248332 bp)\n",
      "chr Scaffold_06 (61759565 bp)\n",
      "chr Scaffold_07 (43120122 bp)\n",
      "chr Scaffold_08 (61151155 bp)\n",
      "chr Scaffold_09 (38581958 bp)\n",
      "chr Scaffold_10 (53961475 bp)\n",
      "chr Scaffold_11 (51449921 bp)\n",
      "chr Scaffold_12 (50438331 bp)\n",
      "chr Scaffold_13 (44396874 bp)\n",
      "chr Scaffold_14 (45393038 bp)\n",
      "chr Scaffold_15 (47938513 bp)\n",
      "chr Scaffold_16 (31980953 bp)\n",
      "chr Scaffold_17 (34923512 bp)\n",
      "chr Scaffold_18 (27737463 bp)\n",
      "\n",
      "Stored sequence information of 18 chromosomes/scaffolds in total\n",
      "\n",
      "==============================================================================\n",
      "Methylation information will now be written into a genome-wide cytosine report\n",
      "==============================================================================\n",
      "\n",
      ">>> Writing genome-wide cytosine report to: /Volumes/Serine/wd/19-12-17/EPI-221_S33_L004.CpG_report.txt <<<\n",
      "\n",
      "Storing all covered cytosine positions for chromosome: Scaffold_01\n",
      "Writing cytosine report for chromosome Scaffold_01 (stored 694072 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_02 (stored 530785 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_03 (stored 427521 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_04 (stored 488137 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_05 (stored 439980 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_06 (stored 428529 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_07 (stored 317419 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_08 (stored 443343 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_09 (stored 276962 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_10 (stored 372833 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_11 (stored 358735 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_12 (stored 386741 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_13 (stored 313721 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_14 (stored 404708 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_15 (stored 315587 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_16 (stored 247440 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_17 (stored 268934 different covered positions)\n",
      "Writing cytosine report for last chromosome Scaffold_18 (stored 193081 different covered positions)\n",
      "Finished writing out cytosine report for covered chromosomes (processed 18 chromosomes/scaffolds in total)\n",
      "\n",
      "Now processing chromosomes that were not covered by any methylation calls in the coverage file...\n",
      "All chromosomes in the genome were covered by at least some reads. coverage2cytosine processing complete.\n",
      "\n",
      "Now merging top and bottom strand CpGs into a single CG dinucleotide entity (reading from file >>/Volumes/Serine/wd/19-12-17/EPI-221_S33_L004.CpG_report.txt<<, in output directory '')\n",
      ">>> Writing a new coverage file with top and bottom strand CpG methylation evidence merged to /Volumes/Serine/wd/19-12-17/EPI-221_S33_L004.CpG_report.merged_CpG_evidence.cov <<<\n",
      "\n",
      "CpG merging complete. Good luck finding DMRs with bsseq!\n",
      "\n",
      "Summary of parameters for genome-wide cytosine report:\n",
      "==============================================================================\n",
      "Coverage infile:\t\t\t\t/Volumes/Serine/wd/19-12-17/bismark_coverage_files/EPI-226_S34_L004_R1_001_val_1_bismark_bt2_pe.deduplicated.bismark.cov.gz\n",
      "Output directory:\t\t\t\t><\n",
      "Parent directory:\t\t\t\t>/Users/sr320/git-repos/nb-2019/P_generosa/<\n",
      "Genome directory:\t\t\t\t>/Volumes/Serine/wd/19-11-05/genome/<\n",
      "CX context:\t\t\t\t\tno (CpG context only, default)\n",
      "Pooling CpG top/bottom strand evidence:\t\tyes\n",
      "Genome coordinates used:\t\t\t1-based (default)\n",
      "GZIP compression:\t\t\t\tno\n",
      "Split by chromosome:\t\t\t\tno\n",
      "\n",
      "\n",
      "Now reading in and storing sequence information of the genome specified in: /Volumes/Serine/wd/19-11-05/genome/\n",
      "\n",
      "chr Scaffold_01 (89643857 bp)\n",
      "chr Scaffold_02 (69596280 bp)\n",
      "chr Scaffold_03 (57743597 bp)\n",
      "chr Scaffold_04 (65288255 bp)\n",
      "chr Scaffold_05 (67248332 bp)\n",
      "chr Scaffold_06 (61759565 bp)\n",
      "chr Scaffold_07 (43120122 bp)\n",
      "chr Scaffold_08 (61151155 bp)\n",
      "chr Scaffold_09 (38581958 bp)\n",
      "chr Scaffold_10 (53961475 bp)\n",
      "chr Scaffold_11 (51449921 bp)\n",
      "chr Scaffold_12 (50438331 bp)\n",
      "chr Scaffold_13 (44396874 bp)\n",
      "chr Scaffold_14 (45393038 bp)\n",
      "chr Scaffold_15 (47938513 bp)\n",
      "chr Scaffold_16 (31980953 bp)\n",
      "chr Scaffold_17 (34923512 bp)\n",
      "chr Scaffold_18 (27737463 bp)\n",
      "\n",
      "Stored sequence information of 18 chromosomes/scaffolds in total\n",
      "\n",
      "==============================================================================\n",
      "Methylation information will now be written into a genome-wide cytosine report\n",
      "==============================================================================\n",
      "\n",
      ">>> Writing genome-wide cytosine report to: /Volumes/Serine/wd/19-12-17/EPI-226_S34_L004.CpG_report.txt <<<\n",
      "\n",
      "Storing all covered cytosine positions for chromosome: Scaffold_01\n",
      "Writing cytosine report for chromosome Scaffold_01 (stored 557259 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_02 (stored 430644 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_03 (stored 345154 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_04 (stored 389519 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_05 (stored 353048 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_06 (stored 342963 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_07 (stored 254345 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_08 (stored 355825 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_09 (stored 225260 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_10 (stored 299356 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_11 (stored 287375 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_12 (stored 309429 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_13 (stored 253934 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_14 (stored 321961 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_15 (stored 251222 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_16 (stored 197802 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_17 (stored 217183 different covered positions)\n",
      "Writing cytosine report for last chromosome Scaffold_18 (stored 154555 different covered positions)\n",
      "Finished writing out cytosine report for covered chromosomes (processed 18 chromosomes/scaffolds in total)\n",
      "\n",
      "Now processing chromosomes that were not covered by any methylation calls in the coverage file...\n",
      "All chromosomes in the genome were covered by at least some reads. coverage2cytosine processing complete.\n",
      "\n",
      "Now merging top and bottom strand CpGs into a single CG dinucleotide entity (reading from file >>/Volumes/Serine/wd/19-12-17/EPI-226_S34_L004.CpG_report.txt<<, in output directory '')\n",
      ">>> Writing a new coverage file with top and bottom strand CpG methylation evidence merged to /Volumes/Serine/wd/19-12-17/EPI-226_S34_L004.CpG_report.merged_CpG_evidence.cov <<<\n",
      "\n",
      "CpG merging complete. Good luck finding DMRs with bsseq!\n",
      "\n",
      "Summary of parameters for genome-wide cytosine report:\n",
      "==============================================================================\n",
      "Coverage infile:\t\t\t\t/Volumes/Serine/wd/19-12-17/bismark_coverage_files/EPI-227_S35_L004_R1_001_val_1_bismark_bt2_pe.deduplicated.bismark.cov.gz\n",
      "Output directory:\t\t\t\t><\n",
      "Parent directory:\t\t\t\t>/Users/sr320/git-repos/nb-2019/P_generosa/<\n",
      "Genome directory:\t\t\t\t>/Volumes/Serine/wd/19-11-05/genome/<\n",
      "CX context:\t\t\t\t\tno (CpG context only, default)\n",
      "Pooling CpG top/bottom strand evidence:\t\tyes\n",
      "Genome coordinates used:\t\t\t1-based (default)\n",
      "GZIP compression:\t\t\t\tno\n",
      "Split by chromosome:\t\t\t\tno\n",
      "\n",
      "\n",
      "Now reading in and storing sequence information of the genome specified in: /Volumes/Serine/wd/19-11-05/genome/\n",
      "\n",
      "chr Scaffold_01 (89643857 bp)\n",
      "chr Scaffold_02 (69596280 bp)\n",
      "chr Scaffold_03 (57743597 bp)\n",
      "chr Scaffold_04 (65288255 bp)\n",
      "chr Scaffold_05 (67248332 bp)\n",
      "chr Scaffold_06 (61759565 bp)\n",
      "chr Scaffold_07 (43120122 bp)\n",
      "chr Scaffold_08 (61151155 bp)\n",
      "chr Scaffold_09 (38581958 bp)\n",
      "chr Scaffold_10 (53961475 bp)\n",
      "chr Scaffold_11 (51449921 bp)\n",
      "chr Scaffold_12 (50438331 bp)\n",
      "chr Scaffold_13 (44396874 bp)\n",
      "chr Scaffold_14 (45393038 bp)\n",
      "chr Scaffold_15 (47938513 bp)\n",
      "chr Scaffold_16 (31980953 bp)\n",
      "chr Scaffold_17 (34923512 bp)\n",
      "chr Scaffold_18 (27737463 bp)\n",
      "\n",
      "Stored sequence information of 18 chromosomes/scaffolds in total\n",
      "\n",
      "==============================================================================\n",
      "Methylation information will now be written into a genome-wide cytosine report\n",
      "==============================================================================\n",
      "\n",
      ">>> Writing genome-wide cytosine report to: /Volumes/Serine/wd/19-12-17/EPI-227_S35_L004.CpG_report.txt <<<\n",
      "\n",
      "Storing all covered cytosine positions for chromosome: Scaffold_01\n",
      "Writing cytosine report for chromosome Scaffold_01 (stored 639892 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_02 (stored 494526 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_03 (stored 398276 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_04 (stored 450713 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_05 (stored 404972 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_06 (stored 401346 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_07 (stored 292955 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_08 (stored 411968 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_09 (stored 258569 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_10 (stored 345000 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_11 (stored 334987 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_12 (stored 356896 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_13 (stored 291728 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_14 (stored 376146 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_15 (stored 290620 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_16 (stored 228970 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_17 (stored 250136 different covered positions)\n",
      "Writing cytosine report for last chromosome Scaffold_18 (stored 179827 different covered positions)\n",
      "Finished writing out cytosine report for covered chromosomes (processed 18 chromosomes/scaffolds in total)\n",
      "\n",
      "Now processing chromosomes that were not covered by any methylation calls in the coverage file...\n",
      "All chromosomes in the genome were covered by at least some reads. coverage2cytosine processing complete.\n",
      "\n",
      "Now merging top and bottom strand CpGs into a single CG dinucleotide entity (reading from file >>/Volumes/Serine/wd/19-12-17/EPI-227_S35_L004.CpG_report.txt<<, in output directory '')\n",
      ">>> Writing a new coverage file with top and bottom strand CpG methylation evidence merged to /Volumes/Serine/wd/19-12-17/EPI-227_S35_L004.CpG_report.merged_CpG_evidence.cov <<<\n",
      "\n",
      "CpG merging complete. Good luck finding DMRs with bsseq!\n",
      "\n",
      "Summary of parameters for genome-wide cytosine report:\n",
      "==============================================================================\n",
      "Coverage infile:\t\t\t\t/Volumes/Serine/wd/19-12-17/bismark_coverage_files/EPI-229_S36_L004_R1_001_val_1_bismark_bt2_pe.deduplicated.bismark.cov.gz\n",
      "Output directory:\t\t\t\t><\n",
      "Parent directory:\t\t\t\t>/Users/sr320/git-repos/nb-2019/P_generosa/<\n",
      "Genome directory:\t\t\t\t>/Volumes/Serine/wd/19-11-05/genome/<\n",
      "CX context:\t\t\t\t\tno (CpG context only, default)\n",
      "Pooling CpG top/bottom strand evidence:\t\tyes\n",
      "Genome coordinates used:\t\t\t1-based (default)\n",
      "GZIP compression:\t\t\t\tno\n",
      "Split by chromosome:\t\t\t\tno\n",
      "\n",
      "\n",
      "Now reading in and storing sequence information of the genome specified in: /Volumes/Serine/wd/19-11-05/genome/\n",
      "\n",
      "chr Scaffold_01 (89643857 bp)\n",
      "chr Scaffold_02 (69596280 bp)\n",
      "chr Scaffold_03 (57743597 bp)\n",
      "chr Scaffold_04 (65288255 bp)\n",
      "chr Scaffold_05 (67248332 bp)\n",
      "chr Scaffold_06 (61759565 bp)\n",
      "chr Scaffold_07 (43120122 bp)\n",
      "chr Scaffold_08 (61151155 bp)\n",
      "chr Scaffold_09 (38581958 bp)\n",
      "chr Scaffold_10 (53961475 bp)\n",
      "chr Scaffold_11 (51449921 bp)\n",
      "chr Scaffold_12 (50438331 bp)\n",
      "chr Scaffold_13 (44396874 bp)\n",
      "chr Scaffold_14 (45393038 bp)\n",
      "chr Scaffold_15 (47938513 bp)\n",
      "chr Scaffold_16 (31980953 bp)\n",
      "chr Scaffold_17 (34923512 bp)\n",
      "chr Scaffold_18 (27737463 bp)\n",
      "\n",
      "Stored sequence information of 18 chromosomes/scaffolds in total\n",
      "\n",
      "==============================================================================\n",
      "Methylation information will now be written into a genome-wide cytosine report\n",
      "==============================================================================\n",
      "\n",
      ">>> Writing genome-wide cytosine report to: /Volumes/Serine/wd/19-12-17/EPI-229_S36_L004.CpG_report.txt <<<\n",
      "\n",
      "Storing all covered cytosine positions for chromosome: Scaffold_01\n",
      "Writing cytosine report for chromosome Scaffold_01 (stored 793018 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_02 (stored 612412 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_03 (stored 491510 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_04 (stored 554188 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_05 (stored 503590 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_06 (stored 489180 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_07 (stored 359190 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_08 (stored 507185 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_09 (stored 316121 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_10 (stored 428924 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_11 (stored 408830 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_12 (stored 441795 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_13 (stored 358005 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_14 (stored 459699 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_15 (stored 360378 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_16 (stored 284460 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_17 (stored 306547 different covered positions)\n",
      "Writing cytosine report for last chromosome Scaffold_18 (stored 223890 different covered positions)\n",
      "Finished writing out cytosine report for covered chromosomes (processed 18 chromosomes/scaffolds in total)\n",
      "\n",
      "Now processing chromosomes that were not covered by any methylation calls in the coverage file...\n",
      "All chromosomes in the genome were covered by at least some reads. coverage2cytosine processing complete.\n",
      "\n",
      "Now merging top and bottom strand CpGs into a single CG dinucleotide entity (reading from file >>/Volumes/Serine/wd/19-12-17/EPI-229_S36_L004.CpG_report.txt<<, in output directory '')\n",
      ">>> Writing a new coverage file with top and bottom strand CpG methylation evidence merged to /Volumes/Serine/wd/19-12-17/EPI-229_S36_L004.CpG_report.merged_CpG_evidence.cov <<<\n",
      "\n",
      "CpG merging complete. Good luck finding DMRs with bsseq!\n",
      "\n",
      "Summary of parameters for genome-wide cytosine report:\n",
      "==============================================================================\n",
      "Coverage infile:\t\t\t\t/Volumes/Serine/wd/19-12-17/bismark_coverage_files/EPI-230_S37_L004_R1_001_val_1_bismark_bt2_pe.deduplicated.bismark.cov.gz\n",
      "Output directory:\t\t\t\t><\n",
      "Parent directory:\t\t\t\t>/Users/sr320/git-repos/nb-2019/P_generosa/<\n",
      "Genome directory:\t\t\t\t>/Volumes/Serine/wd/19-11-05/genome/<\n",
      "CX context:\t\t\t\t\tno (CpG context only, default)\n",
      "Pooling CpG top/bottom strand evidence:\t\tyes\n",
      "Genome coordinates used:\t\t\t1-based (default)\n",
      "GZIP compression:\t\t\t\tno\n",
      "Split by chromosome:\t\t\t\tno\n",
      "\n",
      "\n",
      "Now reading in and storing sequence information of the genome specified in: /Volumes/Serine/wd/19-11-05/genome/\n",
      "\n",
      "chr Scaffold_01 (89643857 bp)\n",
      "chr Scaffold_02 (69596280 bp)\n",
      "chr Scaffold_03 (57743597 bp)\n",
      "chr Scaffold_04 (65288255 bp)\n",
      "chr Scaffold_05 (67248332 bp)\n",
      "chr Scaffold_06 (61759565 bp)\n",
      "chr Scaffold_07 (43120122 bp)\n",
      "chr Scaffold_08 (61151155 bp)\n",
      "chr Scaffold_09 (38581958 bp)\n",
      "chr Scaffold_10 (53961475 bp)\n",
      "chr Scaffold_11 (51449921 bp)\n",
      "chr Scaffold_12 (50438331 bp)\n",
      "chr Scaffold_13 (44396874 bp)\n",
      "chr Scaffold_14 (45393038 bp)\n",
      "chr Scaffold_15 (47938513 bp)\n",
      "chr Scaffold_16 (31980953 bp)\n",
      "chr Scaffold_17 (34923512 bp)\n",
      "chr Scaffold_18 (27737463 bp)\n",
      "\n",
      "Stored sequence information of 18 chromosomes/scaffolds in total\n",
      "\n",
      "==============================================================================\n",
      "Methylation information will now be written into a genome-wide cytosine report\n",
      "==============================================================================\n",
      "\n",
      ">>> Writing genome-wide cytosine report to: /Volumes/Serine/wd/19-12-17/EPI-230_S37_L004.CpG_report.txt <<<\n",
      "\n",
      "Storing all covered cytosine positions for chromosome: Scaffold_01\n",
      "Writing cytosine report for chromosome Scaffold_01 (stored 884635 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_02 (stored 682411 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_03 (stored 552029 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_04 (stored 629842 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_05 (stored 565750 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_06 (stored 551130 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_07 (stored 407491 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_08 (stored 570448 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_09 (stored 357899 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_10 (stored 475053 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_11 (stored 466137 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_12 (stored 495323 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_13 (stored 402587 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_14 (stored 514149 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_15 (stored 409282 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_16 (stored 321136 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_17 (stored 346471 different covered positions)\n",
      "Writing cytosine report for last chromosome Scaffold_18 (stored 247340 different covered positions)\n",
      "Finished writing out cytosine report for covered chromosomes (processed 18 chromosomes/scaffolds in total)\n",
      "\n",
      "Now processing chromosomes that were not covered by any methylation calls in the coverage file...\n",
      "All chromosomes in the genome were covered by at least some reads. coverage2cytosine processing complete.\n",
      "\n",
      "Now merging top and bottom strand CpGs into a single CG dinucleotide entity (reading from file >>/Volumes/Serine/wd/19-12-17/EPI-230_S37_L004.CpG_report.txt<<, in output directory '')\n",
      ">>> Writing a new coverage file with top and bottom strand CpG methylation evidence merged to /Volumes/Serine/wd/19-12-17/EPI-230_S37_L004.CpG_report.merged_CpG_evidence.cov <<<\n",
      "\n",
      "CpG merging complete. Good luck finding DMRs with bsseq!\n",
      "\n",
      "Summary of parameters for genome-wide cytosine report:\n",
      "==============================================================================\n",
      "Coverage infile:\t\t\t\t/Volumes/Serine/wd/19-12-17/bismark_coverage_files/EPI-41_S38_L005_R1_001_val_1_bismark_bt2_pe.deduplicated.bismark.cov.gz\n",
      "Output directory:\t\t\t\t><\n",
      "Parent directory:\t\t\t\t>/Users/sr320/git-repos/nb-2019/P_generosa/<\n",
      "Genome directory:\t\t\t\t>/Volumes/Serine/wd/19-11-05/genome/<\n",
      "CX context:\t\t\t\t\tno (CpG context only, default)\n",
      "Pooling CpG top/bottom strand evidence:\t\tyes\n",
      "Genome coordinates used:\t\t\t1-based (default)\n",
      "GZIP compression:\t\t\t\tno\n",
      "Split by chromosome:\t\t\t\tno\n",
      "\n",
      "\n",
      "Now reading in and storing sequence information of the genome specified in: /Volumes/Serine/wd/19-11-05/genome/\n",
      "\n",
      "chr Scaffold_01 (89643857 bp)\n",
      "chr Scaffold_02 (69596280 bp)\n",
      "chr Scaffold_03 (57743597 bp)\n",
      "chr Scaffold_04 (65288255 bp)\n",
      "chr Scaffold_05 (67248332 bp)\n",
      "chr Scaffold_06 (61759565 bp)\n",
      "chr Scaffold_07 (43120122 bp)\n",
      "chr Scaffold_08 (61151155 bp)\n",
      "chr Scaffold_09 (38581958 bp)\n",
      "chr Scaffold_10 (53961475 bp)\n",
      "chr Scaffold_11 (51449921 bp)\n",
      "chr Scaffold_12 (50438331 bp)\n",
      "chr Scaffold_13 (44396874 bp)\n",
      "chr Scaffold_14 (45393038 bp)\n",
      "chr Scaffold_15 (47938513 bp)\n",
      "chr Scaffold_16 (31980953 bp)\n",
      "chr Scaffold_17 (34923512 bp)\n",
      "chr Scaffold_18 (27737463 bp)\n",
      "\n",
      "Stored sequence information of 18 chromosomes/scaffolds in total\n",
      "\n",
      "==============================================================================\n",
      "Methylation information will now be written into a genome-wide cytosine report\n",
      "==============================================================================\n",
      "\n",
      ">>> Writing genome-wide cytosine report to: /Volumes/Serine/wd/19-12-17/EPI-41_S38_L005.CpG_report.txt <<<\n",
      "\n",
      "Storing all covered cytosine positions for chromosome: Scaffold_01\n",
      "Writing cytosine report for chromosome Scaffold_01 (stored 818143 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_02 (stored 632253 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_03 (stored 509202 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_04 (stored 578985 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_05 (stored 522764 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_06 (stored 511518 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_07 (stored 377164 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_08 (stored 528289 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_09 (stored 330752 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_10 (stored 442308 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_11 (stored 427749 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_12 (stored 456164 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_13 (stored 370653 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_14 (stored 476452 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_15 (stored 374465 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_16 (stored 293334 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_17 (stored 320271 different covered positions)\n",
      "Writing cytosine report for last chromosome Scaffold_18 (stored 232542 different covered positions)\n",
      "Finished writing out cytosine report for covered chromosomes (processed 18 chromosomes/scaffolds in total)\n",
      "\n",
      "Now processing chromosomes that were not covered by any methylation calls in the coverage file...\n",
      "All chromosomes in the genome were covered by at least some reads. coverage2cytosine processing complete.\n",
      "\n",
      "Now merging top and bottom strand CpGs into a single CG dinucleotide entity (reading from file >>/Volumes/Serine/wd/19-12-17/EPI-41_S38_L005.CpG_report.txt<<, in output directory '')\n",
      ">>> Writing a new coverage file with top and bottom strand CpG methylation evidence merged to /Volumes/Serine/wd/19-12-17/EPI-41_S38_L005.CpG_report.merged_CpG_evidence.cov <<<\n",
      "\n",
      "CpG merging complete. Good luck finding DMRs with bsseq!\n",
      "\n",
      "Summary of parameters for genome-wide cytosine report:\n",
      "==============================================================================\n",
      "Coverage infile:\t\t\t\t/Volumes/Serine/wd/19-12-17/bismark_coverage_files/EPI-42_S39_L005_R1_001_val_1_bismark_bt2_pe.deduplicated.bismark.cov.gz\n",
      "Output directory:\t\t\t\t><\n",
      "Parent directory:\t\t\t\t>/Users/sr320/git-repos/nb-2019/P_generosa/<\n",
      "Genome directory:\t\t\t\t>/Volumes/Serine/wd/19-11-05/genome/<\n",
      "CX context:\t\t\t\t\tno (CpG context only, default)\n",
      "Pooling CpG top/bottom strand evidence:\t\tyes\n",
      "Genome coordinates used:\t\t\t1-based (default)\n",
      "GZIP compression:\t\t\t\tno\n",
      "Split by chromosome:\t\t\t\tno\n",
      "\n",
      "\n",
      "Now reading in and storing sequence information of the genome specified in: /Volumes/Serine/wd/19-11-05/genome/\n",
      "\n",
      "chr Scaffold_01 (89643857 bp)\n",
      "chr Scaffold_02 (69596280 bp)\n",
      "chr Scaffold_03 (57743597 bp)\n",
      "chr Scaffold_04 (65288255 bp)\n",
      "chr Scaffold_05 (67248332 bp)\n",
      "chr Scaffold_06 (61759565 bp)\n",
      "chr Scaffold_07 (43120122 bp)\n",
      "chr Scaffold_08 (61151155 bp)\n",
      "chr Scaffold_09 (38581958 bp)\n",
      "chr Scaffold_10 (53961475 bp)\n",
      "chr Scaffold_11 (51449921 bp)\n",
      "chr Scaffold_12 (50438331 bp)\n",
      "chr Scaffold_13 (44396874 bp)\n",
      "chr Scaffold_14 (45393038 bp)\n",
      "chr Scaffold_15 (47938513 bp)\n",
      "chr Scaffold_16 (31980953 bp)\n",
      "chr Scaffold_17 (34923512 bp)\n",
      "chr Scaffold_18 (27737463 bp)\n",
      "\n",
      "Stored sequence information of 18 chromosomes/scaffolds in total\n",
      "\n",
      "==============================================================================\n",
      "Methylation information will now be written into a genome-wide cytosine report\n",
      "==============================================================================\n",
      "\n",
      ">>> Writing genome-wide cytosine report to: /Volumes/Serine/wd/19-12-17/EPI-42_S39_L005.CpG_report.txt <<<\n",
      "\n",
      "Storing all covered cytosine positions for chromosome: Scaffold_01\n",
      "Writing cytosine report for chromosome Scaffold_01 (stored 884520 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_02 (stored 681912 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_03 (stored 549373 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_04 (stored 624791 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_05 (stored 570838 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_06 (stored 550131 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_07 (stored 405945 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_08 (stored 571687 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_09 (stored 359550 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_10 (stored 478814 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_11 (stored 461898 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_12 (stored 493727 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_13 (stored 402130 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_14 (stored 507384 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_15 (stored 405569 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_16 (stored 316393 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_17 (stored 342993 different covered positions)\n",
      "Writing cytosine report for last chromosome Scaffold_18 (stored 248931 different covered positions)\n",
      "Finished writing out cytosine report for covered chromosomes (processed 18 chromosomes/scaffolds in total)\n",
      "\n",
      "Now processing chromosomes that were not covered by any methylation calls in the coverage file...\n",
      "All chromosomes in the genome were covered by at least some reads. coverage2cytosine processing complete.\n",
      "\n",
      "Now merging top and bottom strand CpGs into a single CG dinucleotide entity (reading from file >>/Volumes/Serine/wd/19-12-17/EPI-42_S39_L005.CpG_report.txt<<, in output directory '')\n",
      ">>> Writing a new coverage file with top and bottom strand CpG methylation evidence merged to /Volumes/Serine/wd/19-12-17/EPI-42_S39_L005.CpG_report.merged_CpG_evidence.cov <<<\n",
      "\n",
      "CpG merging complete. Good luck finding DMRs with bsseq!\n",
      "\n",
      "Summary of parameters for genome-wide cytosine report:\n",
      "==============================================================================\n",
      "Coverage infile:\t\t\t\t/Volumes/Serine/wd/19-12-17/bismark_coverage_files/EPI-43_S40_L005_R1_001_val_1_bismark_bt2_pe.deduplicated.bismark.cov.gz\n",
      "Output directory:\t\t\t\t><\n",
      "Parent directory:\t\t\t\t>/Users/sr320/git-repos/nb-2019/P_generosa/<\n",
      "Genome directory:\t\t\t\t>/Volumes/Serine/wd/19-11-05/genome/<\n",
      "CX context:\t\t\t\t\tno (CpG context only, default)\n",
      "Pooling CpG top/bottom strand evidence:\t\tyes\n",
      "Genome coordinates used:\t\t\t1-based (default)\n",
      "GZIP compression:\t\t\t\tno\n",
      "Split by chromosome:\t\t\t\tno\n",
      "\n",
      "\n",
      "Now reading in and storing sequence information of the genome specified in: /Volumes/Serine/wd/19-11-05/genome/\n",
      "\n",
      "chr Scaffold_01 (89643857 bp)\n",
      "chr Scaffold_02 (69596280 bp)\n",
      "chr Scaffold_03 (57743597 bp)\n",
      "chr Scaffold_04 (65288255 bp)\n",
      "chr Scaffold_05 (67248332 bp)\n",
      "chr Scaffold_06 (61759565 bp)\n",
      "chr Scaffold_07 (43120122 bp)\n",
      "chr Scaffold_08 (61151155 bp)\n",
      "chr Scaffold_09 (38581958 bp)\n",
      "chr Scaffold_10 (53961475 bp)\n",
      "chr Scaffold_11 (51449921 bp)\n",
      "chr Scaffold_12 (50438331 bp)\n",
      "chr Scaffold_13 (44396874 bp)\n",
      "chr Scaffold_14 (45393038 bp)\n",
      "chr Scaffold_15 (47938513 bp)\n",
      "chr Scaffold_16 (31980953 bp)\n",
      "chr Scaffold_17 (34923512 bp)\n",
      "chr Scaffold_18 (27737463 bp)\n",
      "\n",
      "Stored sequence information of 18 chromosomes/scaffolds in total\n",
      "\n",
      "==============================================================================\n",
      "Methylation information will now be written into a genome-wide cytosine report\n",
      "==============================================================================\n",
      "\n",
      ">>> Writing genome-wide cytosine report to: /Volumes/Serine/wd/19-12-17/EPI-43_S40_L005.CpG_report.txt <<<\n",
      "\n",
      "Storing all covered cytosine positions for chromosome: Scaffold_01\n",
      "Writing cytosine report for chromosome Scaffold_01 (stored 814989 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_02 (stored 627668 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_03 (stored 507139 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_04 (stored 575366 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_05 (stored 518695 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_06 (stored 503899 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_07 (stored 372483 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_08 (stored 523135 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_09 (stored 328660 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_10 (stored 435901 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_11 (stored 424382 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_12 (stored 452889 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_13 (stored 370802 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_14 (stored 470545 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_15 (stored 368403 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_16 (stored 292391 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_17 (stored 318579 different covered positions)\n",
      "Writing cytosine report for last chromosome Scaffold_18 (stored 226235 different covered positions)\n",
      "Finished writing out cytosine report for covered chromosomes (processed 18 chromosomes/scaffolds in total)\n",
      "\n",
      "Now processing chromosomes that were not covered by any methylation calls in the coverage file...\n",
      "All chromosomes in the genome were covered by at least some reads. coverage2cytosine processing complete.\n",
      "\n",
      "Now merging top and bottom strand CpGs into a single CG dinucleotide entity (reading from file >>/Volumes/Serine/wd/19-12-17/EPI-43_S40_L005.CpG_report.txt<<, in output directory '')\n",
      ">>> Writing a new coverage file with top and bottom strand CpG methylation evidence merged to /Volumes/Serine/wd/19-12-17/EPI-43_S40_L005.CpG_report.merged_CpG_evidence.cov <<<\n",
      "\n",
      "CpG merging complete. Good luck finding DMRs with bsseq!\n",
      "\n",
      "Summary of parameters for genome-wide cytosine report:\n",
      "==============================================================================\n",
      "Coverage infile:\t\t\t\t/Volumes/Serine/wd/19-12-17/bismark_coverage_files/EPI-44_S41_L005_R1_001_val_1_bismark_bt2_pe.deduplicated.bismark.cov.gz\n",
      "Output directory:\t\t\t\t><\n",
      "Parent directory:\t\t\t\t>/Users/sr320/git-repos/nb-2019/P_generosa/<\n",
      "Genome directory:\t\t\t\t>/Volumes/Serine/wd/19-11-05/genome/<\n",
      "CX context:\t\t\t\t\tno (CpG context only, default)\n",
      "Pooling CpG top/bottom strand evidence:\t\tyes\n",
      "Genome coordinates used:\t\t\t1-based (default)\n",
      "GZIP compression:\t\t\t\tno\n",
      "Split by chromosome:\t\t\t\tno\n",
      "\n",
      "\n",
      "Now reading in and storing sequence information of the genome specified in: /Volumes/Serine/wd/19-11-05/genome/\n",
      "\n",
      "chr Scaffold_01 (89643857 bp)\n",
      "chr Scaffold_02 (69596280 bp)\n",
      "chr Scaffold_03 (57743597 bp)\n",
      "chr Scaffold_04 (65288255 bp)\n",
      "chr Scaffold_05 (67248332 bp)\n",
      "chr Scaffold_06 (61759565 bp)\n",
      "chr Scaffold_07 (43120122 bp)\n",
      "chr Scaffold_08 (61151155 bp)\n",
      "chr Scaffold_09 (38581958 bp)\n",
      "chr Scaffold_10 (53961475 bp)\n",
      "chr Scaffold_11 (51449921 bp)\n",
      "chr Scaffold_12 (50438331 bp)\n",
      "chr Scaffold_13 (44396874 bp)\n",
      "chr Scaffold_14 (45393038 bp)\n",
      "chr Scaffold_15 (47938513 bp)\n",
      "chr Scaffold_16 (31980953 bp)\n",
      "chr Scaffold_17 (34923512 bp)\n",
      "chr Scaffold_18 (27737463 bp)\n",
      "\n",
      "Stored sequence information of 18 chromosomes/scaffolds in total\n",
      "\n",
      "==============================================================================\n",
      "Methylation information will now be written into a genome-wide cytosine report\n",
      "==============================================================================\n",
      "\n",
      ">>> Writing genome-wide cytosine report to: /Volumes/Serine/wd/19-12-17/EPI-44_S41_L005.CpG_report.txt <<<\n",
      "\n",
      "Storing all covered cytosine positions for chromosome: Scaffold_01\n",
      "Writing cytosine report for chromosome Scaffold_01 (stored 588975 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_02 (stored 453991 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_03 (stored 365994 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_04 (stored 416373 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_05 (stored 373321 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_06 (stored 363497 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_07 (stored 273122 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_08 (stored 377813 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_09 (stored 236744 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_10 (stored 316503 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_11 (stored 305652 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_12 (stored 327312 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_13 (stored 268339 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_14 (stored 341289 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_15 (stored 267283 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_16 (stored 210410 different covered positions)\n",
      "Writing cytosine report for chromosome Scaffold_17 (stored 229577 different covered positions)\n",
      "Writing cytosine report for last chromosome Scaffold_18 (stored 164350 different covered positions)\n",
      "Finished writing out cytosine report for covered chromosomes (processed 18 chromosomes/scaffolds in total)\n",
      "\n",
      "Now processing chromosomes that were not covered by any methylation calls in the coverage file...\n",
      "All chromosomes in the genome were covered by at least some reads. coverage2cytosine processing complete.\n",
      "\n",
      "Now merging top and bottom strand CpGs into a single CG dinucleotide entity (reading from file >>/Volumes/Serine/wd/19-12-17/EPI-44_S41_L005.CpG_report.txt<<, in output directory '')\n",
      ">>> Writing a new coverage file with top and bottom strand CpG methylation evidence merged to /Volumes/Serine/wd/19-12-17/EPI-44_S41_L005.CpG_report.merged_CpG_evidence.cov <<<\n",
      "\n",
      "CpG merging complete. Good luck finding DMRs with bsseq!\n",
      "\n"
     ]
    }
   ],
   "source": [
    "!find /Volumes/Serine/wd/19-12-17/bismark_coverage_files/*cov.gz \\\n",
    "| xargs basename -s _R1_001_val_1_bismark_bt2_pe.deduplicated.bismark.cov.gz \\\n",
    "| xargs -I{} /Applications/bioinfo/Bismark_v0.20.0/coverage2cytosine \\\n",
    "--genome_folder /Volumes/Serine/wd/19-11-05/genome/ \\\n",
    "-o /Volumes/Serine/wd/19-12-17/{} \\\n",
    "--merge_CpG \\\n",
    "/Volumes/Serine/wd/19-12-17/bismark_coverage_files/{}_R1_001_val_1_bismark_bt2_pe.deduplicated.bismark.cov.gz"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {
    "collapsed": false,
    "deletable": true,
    "editable": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "total 6055968\r\n",
      "-rw-r--r--  1 sr320  staff     0B Nov  5 20:18 PgALL.CpG_report.merged_CpG_evidence.cov\r\n",
      "-rw-r--r--  1 sr320  staff   1.0G Nov  5 20:18 PgALL.CpG_report.txt\r\n",
      "-rw-r--r--@ 1 sr320  staff   1.9G Nov  5 19:10 Pg_val_1_bismark_bt2_pe.deduplicated.bismark.cov\r\n",
      "drwxr-xr-x  5 sr320  staff   170B Nov  5 20:07 \u001b[34mgenome\u001b[m\u001b[m\r\n",
      "drwxr-xr-x  2 sr320  staff    68B Nov  5 20:09 \u001b[34mv074\u001b[m\u001b[m\r\n"
     ]
    }
   ],
   "source": [
    "!ls -lh /Volumes/Serine/wd/19-11-05/ "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {
    "collapsed": false,
    "deletable": true,
    "editable": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Scaffold_01\t4\t+\t0\t0\tCG\tCGC\r\n",
      "Scaffold_01\t5\t-\t0\t0\tCG\tCGC\r\n",
      "Scaffold_01\t18\t+\t0\t0\tCG\tCGG\r\n",
      "Scaffold_01\t19\t-\t0\t0\tCG\tCGG\r\n",
      "Scaffold_01\t54\t+\t0\t0\tCG\tCGT\r\n",
      "Scaffold_01\t55\t-\t0\t0\tCG\tCGG\r\n",
      "Scaffold_01\t72\t+\t0\t0\tCG\tCGA\r\n",
      "Scaffold_01\t73\t-\t0\t0\tCG\tCGC\r\n",
      "Scaffold_01\t96\t+\t0\t0\tCG\tCGC\r\n",
      "Scaffold_01\t97\t-\t0\t0\tCG\tCGG\r\n"
     ]
    }
   ],
   "source": [
    "!head /Volumes/Serine/wd/19-11-05/PgALL.CpG_report.txt\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {
    "collapsed": false,
    "deletable": true,
    "editable": true,
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Summary of parameters for genome-wide cytosine report:\n",
      "==============================================================================\n",
      "Coverage infile:\t\t\t\t/Volumes/Serine/wd/19-11-05/Pg_val_1_bismark_bt2_pe.deduplicated.bismark.cov\n",
      "Output directory:\t\t\t\t><\n",
      "Parent directory:\t\t\t\t>/Users/sr320/git-repos/nb-2019/P_generosa/<\n",
      "Genome directory:\t\t\t\t>/Volumes/Serine/wd/19-11-05/v074/<\n",
      "CX context:\t\t\t\t\tno (CpG context only, default)\n",
      "Pooling CpG top/bottom strand evidence:\t\tyes\n",
      "Genome coordinates used:\t\t\t1-based (default)\n",
      "GZIP compression:\t\t\t\tno\n",
      "Split by chromosome:\t\t\t\tno\n",
      "\n",
      "\n",
      "Now reading in and storing sequence information of the genome specified in: /Volumes/Serine/wd/19-11-05/v074/\n",
      "\n",
      "chr PGA_scaffold1__77_contigs__length_89643857 (89643857 bp)\n",
      "chr PGA_scaffold2__36_contigs__length_69596280 (69596280 bp)\n",
      "chr PGA_scaffold3__111_contigs__length_57743597 (57743597 bp)\n",
      "chr PGA_scaffold4__129_contigs__length_65288255 (65288255 bp)\n",
      "chr PGA_scaffold5__109_contigs__length_67248332 (67248332 bp)\n",
      "chr PGA_scaffold6__104_contigs__length_61759565 (61759565 bp)\n",
      "chr PGA_scaffold7__69_contigs__length_43120122 (43120122 bp)\n",
      "chr PGA_scaffold8__63_contigs__length_61151155 (61151155 bp)\n",
      "chr PGA_scaffold9__45_contigs__length_38581958 (38581958 bp)\n",
      "chr PGA_scaffold10__49_contigs__length_53961475 (53961475 bp)\n",
      "chr PGA_scaffold11__79_contigs__length_51449921 (51449921 bp)\n",
      "chr PGA_scaffold12__71_contigs__length_50438331 (50438331 bp)\n",
      "chr PGA_scaffold13__52_contigs__length_44396874 (44396874 bp)\n",
      "chr PGA_scaffold14__91_contigs__length_45393038 (45393038 bp)\n",
      "chr PGA_scaffold15__101_contigs__length_47938513 (47938513 bp)\n",
      "chr PGA_scaffold16__33_contigs__length_31980953 (31980953 bp)\n",
      "chr PGA_scaffold17__51_contigs__length_34923512 (34923512 bp)\n",
      "chr PGA_scaffold18__69_contigs__length_27737463 (27737463 bp)\n",
      "\n",
      "Stored sequence information of 18 chromosomes/scaffolds in total\n",
      "\n",
      "==============================================================================\n",
      "Methylation information will now be written into a genome-wide cytosine report\n",
      "==============================================================================\n",
      "\n",
      ">>> Writing genome-wide cytosine report to: /Volumes/Serine/wd/19-11-05/PgALL.74.CpG_report.txt <<<\n",
      "\n",
      "Storing all covered cytosine positions for chromosome: PGA_scaffold1__77_contigs__length_89643857\n",
      "Writing cytosine report for chromosome PGA_scaffold1__77_contigs__length_89643857 (stored 2673624 different covered positions)\n",
      "Writing cytosine report for chromosome PGA_scaffold2__36_contigs__length_69596280 (stored 2102612 different covered positions)\n",
      "Writing cytosine report for chromosome PGA_scaffold3__111_contigs__length_57743597 (stored 1720926 different covered positions)\n",
      "Writing cytosine report for chromosome PGA_scaffold4__129_contigs__length_65288255 (stored 2016884 different covered positions)\n",
      "Writing cytosine report for chromosome PGA_scaffold5__109_contigs__length_67248332 (stored 1946117 different covered positions)\n",
      "Writing cytosine report for chromosome PGA_scaffold6__104_contigs__length_61759565 (stored 1765092 different covered positions)\n",
      "Writing cytosine report for chromosome PGA_scaffold7__69_contigs__length_43120122 (stored 1288185 different covered positions)\n",
      "Writing cytosine report for chromosome PGA_scaffold8__63_contigs__length_61151155 (stored 1780075 different covered positions)\n",
      "Writing cytosine report for chromosome PGA_scaffold9__45_contigs__length_38581958 (stored 1136392 different covered positions)\n",
      "Writing cytosine report for chromosome PGA_scaffold10__49_contigs__length_53961475 (stored 1498339 different covered positions)\n",
      "Writing cytosine report for chromosome PGA_scaffold11__79_contigs__length_51449921 (stored 1532901 different covered positions)\n",
      "Writing cytosine report for chromosome PGA_scaffold12__71_contigs__length_50438331 (stored 1529835 different covered positions)\n",
      "Writing cytosine report for chromosome PGA_scaffold13__52_contigs__length_44396874 (stored 1285005 different covered positions)\n",
      "Writing cytosine report for chromosome PGA_scaffold14__91_contigs__length_45393038 (stored 1503871 different covered positions)\n",
      "Writing cytosine report for chromosome PGA_scaffold15__101_contigs__length_47938513 (stored 1391061 different covered positions)\n",
      "Writing cytosine report for chromosome PGA_scaffold16__33_contigs__length_31980953 (stored 1020688 different covered positions)\n",
      "Writing cytosine report for chromosome PGA_scaffold17__51_contigs__length_34923512 (stored 1071550 different covered positions)\n",
      "Writing cytosine report for last chromosome PGA_scaffold18__69_contigs__length_27737463 (stored 818843 different covered positions)\n",
      "Finished writing out cytosine report for covered chromosomes (processed 18 chromosomes/scaffolds in total)\n",
      "\n",
      "Now processing chromosomes that were not covered by any methylation calls in the coverage file...\n",
      "All chromosomes in the genome were covered by at least some reads. coverage2cytosine processing complete.\n",
      "\n",
      "Now merging top and bottom strand CpGs into a single CG dinucleotide entity (reading from file >>/Volumes/Serine/wd/19-11-05/PgALL.74.CpG_report.txt<<, in output directory '')\n",
      ">>> Writing a new coverage file with top and bottom strand CpG methylation evidence merged to /Volumes/Serine/wd/19-11-05/PgALL.74.CpG_report.merged_CpG_evidence.cov <<<\n",
      "\n",
      "CpG merging complete. Good luck finding DMRs with bsseq!\n",
      "\n"
     ]
    }
   ],
   "source": [
    "!/Applications/bioinfo/Bismark_v0.20.0/coverage2cytosine \\\n",
    "--genome_folder /Volumes/Serine/wd/19-11-05/v074 \\\n",
    "-o /Volumes/Serine/wd/19-11-05/PgALL.74 \\\n",
    "--merge_CpG \\\n",
    "/Volumes/Serine/wd/19-11-05/Pg_val_1_bismark_bt2_pe.deduplicated.bismark.cov"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {
    "collapsed": false,
    "deletable": true,
    "editable": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "total 12338208\r\n",
      "-rw-r--r--  1 sr320  staff   1.1G Nov  5 21:00 PgALL.74.CpG_report.merged_CpG_evidence.cov\r\n",
      "-rw-r--r--  1 sr320  staff   1.9G Nov  5 20:58 PgALL.74.CpG_report.txt\r\n",
      "-rw-r--r--  1 sr320  staff     0B Nov  5 20:18 PgALL.CpG_report.merged_CpG_evidence.cov\r\n",
      "-rw-r--r--  1 sr320  staff   1.0G Nov  5 20:18 PgALL.CpG_report.txt\r\n",
      "-rw-r--r--@ 1 sr320  staff   1.9G Nov  5 19:10 Pg_val_1_bismark_bt2_pe.deduplicated.bismark.cov\r\n",
      "drwxr-xr-x  5 sr320  staff   170B Nov  5 20:07 \u001b[34mgenome\u001b[m\u001b[m\r\n",
      "drwxr-xr-x  4 sr320  staff   136B Nov  5 20:20 \u001b[34mv074\u001b[m\u001b[m\r\n"
     ]
    }
   ],
   "source": [
    "!ls -lh /Volumes/Serine/wd/19-11-05/ "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "metadata": {
    "collapsed": false,
    "deletable": true,
    "editable": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PGA_scaffold1__77_contigs__length_89643857\t4\t5\t50.000000\t1\t1\r\n",
      "PGA_scaffold1__77_contigs__length_89643857\t18\t19\t11.111111\t1\t8\r\n",
      "PGA_scaffold1__77_contigs__length_89643857\t54\t55\t3.125000\t1\t31\r\n",
      "PGA_scaffold1__77_contigs__length_89643857\t72\t73\t2.777778\t1\t35\r\n",
      "PGA_scaffold1__77_contigs__length_89643857\t96\t97\t2.040816\t1\t48\r\n",
      "PGA_scaffold1__77_contigs__length_89643857\t119\t120\t0.000000\t0\t48\r\n",
      "PGA_scaffold1__77_contigs__length_89643857\t193\t194\t0.000000\t0\t43\r\n",
      "PGA_scaffold1__77_contigs__length_89643857\t202\t203\t0.000000\t0\t48\r\n",
      "PGA_scaffold1__77_contigs__length_89643857\t209\t210\t0.000000\t0\t54\r\n",
      "PGA_scaffold1__77_contigs__length_89643857\t213\t214\t0.000000\t0\t57\r\n"
     ]
    }
   ],
   "source": [
    "!head /Volumes/Serine/wd/19-11-05/PgALL.74.CpG_report.merged_CpG_evidence.cov"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "metadata": {
    "collapsed": false,
    "deletable": true,
    "editable": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PGA_scaffold1__77_contigs__length_89643857 54 55 3.125000 1 31\r\n",
      "PGA_scaffold1__77_contigs__length_89643857 72 73 2.777778 1 35\r\n",
      "PGA_scaffold1__77_contigs__length_89643857 96 97 2.040816 1 48\r\n",
      "PGA_scaffold1__77_contigs__length_89643857 119 120 0.000000 0 48\r\n",
      "PGA_scaffold1__77_contigs__length_89643857 193 194 0.000000 0 43\r\n",
      "PGA_scaffold1__77_contigs__length_89643857 202 203 0.000000 0 48\r\n",
      "PGA_scaffold1__77_contigs__length_89643857 209 210 0.000000 0 54\r\n",
      "PGA_scaffold1__77_contigs__length_89643857 213 214 0.000000 0 57\r\n",
      "PGA_scaffold1__77_contigs__length_89643857 221 222 0.000000 0 62\r\n",
      "PGA_scaffold1__77_contigs__length_89643857 238 239 1.315789 1 75\r\n",
      "awk: write error on /dev/stdout\r\n",
      " input record number 605, file \r\n",
      " source line number 1\r\n",
      "cat: stdout: Broken pipe\r\n"
     ]
    }
   ],
   "source": [
    "!cat /Volumes/Serine/wd/19-11-05/PgALL.74.CpG_report.merged_CpG_evidence.cov \\\n",
    "| awk '{if ($5+$6 >= 10) { print $1, $2, $3, $4, $5, $6 }}' | head"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "metadata": {
    "collapsed": false,
    "deletable": true,
    "editable": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PGA_scaffold10__49_contigs__length_53961475\n",
      "PGA_scaffold11__79_contigs__length_51449921\n",
      "PGA_scaffold12__71_contigs__length_50438331\n",
      "PGA_scaffold13__52_contigs__length_44396874\n",
      "PGA_scaffold14__91_contigs__length_45393038\n",
      "PGA_scaffold15__101_contigs__length_47938513\n",
      "PGA_scaffold16__33_contigs__length_31980953\n",
      "PGA_scaffold17__51_contigs__length_34923512\n",
      "PGA_scaffold18__69_contigs__length_27737463\n",
      "PGA_scaffold1__77_contigs__length_89643857\n",
      "PGA_scaffold2__36_contigs__length_69596280\n",
      "PGA_scaffold3__111_contigs__length_57743597\n",
      "PGA_scaffold4__129_contigs__length_65288255\n",
      "PGA_scaffold5__109_contigs__length_67248332\n",
      "PGA_scaffold6__104_contigs__length_61759565\n",
      "PGA_scaffold7__69_contigs__length_43120122\n",
      "PGA_scaffold8__63_contigs__length_61151155\n",
      "PGA_scaffold9__45_contigs__length_38581958\n"
     ]
    }
   ],
   "source": [
    "!cat /Volumes/Serine/wd/19-11-05/PgALL.74.CpG_report.merged_CpG_evidence.cov | cut -f 1 | sort | uniq"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "metadata": {
    "collapsed": false,
    "deletable": true,
    "editable": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Scaffold_01\n",
      "Scaffold_02\n",
      "Scaffold_03\n",
      "Scaffold_04\n",
      "Scaffold_05\n",
      "Scaffold_06\n",
      "Scaffold_07\n",
      "Scaffold_08\n",
      "Scaffold_09\n",
      "Scaffold_10\n",
      "Scaffold_11\n",
      "Scaffold_12\n",
      "Scaffold_13\n",
      "Scaffold_14\n",
      "Scaffold_15\n",
      "Scaffold_16\n",
      "Scaffold_17\n",
      "Scaffold_18\n"
     ]
    }
   ],
   "source": [
    "!cat /Volumes/Serine/wd/19-11-05/PgALL.74.CpG_report.merged_CpG_evidence.cov \\\n",
    "| sed 's/PGA_scaffold10__49_contigs__length_53961475/Scaffold_10/g' \\\n",
    "| sed 's/PGA_scaffold11__79_contigs__length_51449921/Scaffold_11/g' \\\n",
    "| sed 's/PGA_scaffold12__71_contigs__length_50438331/Scaffold_12/g' \\\n",
    "| sed 's/PGA_scaffold13__52_contigs__length_44396874/Scaffold_13/g' \\\n",
    "| sed 's/PGA_scaffold14__91_contigs__length_45393038/Scaffold_14/g' \\\n",
    "| sed 's/PGA_scaffold15__101_contigs__length_47938513/Scaffold_15/g' \\\n",
    "| sed 's/PGA_scaffold16__33_contigs__length_31980953/Scaffold_16/g' \\\n",
    "| sed 's/PGA_scaffold17__51_contigs__length_34923512/Scaffold_17/g' \\\n",
    "| sed 's/PGA_scaffold18__69_contigs__length_27737463/Scaffold_18/g' \\\n",
    "| sed 's/PGA_scaffold1__77_contigs__length_89643857/Scaffold_01/g' \\\n",
    "| sed 's/PGA_scaffold2__36_contigs__length_69596280/Scaffold_02/g' \\\n",
    "| sed 's/PGA_scaffold3__111_contigs__length_57743597/Scaffold_03/g' \\\n",
    "| sed 's/PGA_scaffold4__129_contigs__length_65288255/Scaffold_04/g' \\\n",
    "| sed 's/PGA_scaffold5__109_contigs__length_67248332/Scaffold_05/g' \\\n",
    "| sed 's/PGA_scaffold6__104_contigs__length_61759565/Scaffold_06/g' \\\n",
    "| sed 's/PGA_scaffold7__69_contigs__length_43120122/Scaffold_07/g' \\\n",
    "| sed 's/PGA_scaffold8__63_contigs__length_61151155/Scaffold_08/g' \\\n",
    "| sed 's/PGA_scaffold9__45_contigs__length_38581958/Scaffold_09/g' \\\n",
    "| cut -f 1 | sort | uniq"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 32,
   "metadata": {
    "collapsed": false,
    "deletable": true,
    "editable": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Scaffold_01 \t 54 \t 55 \t 3.125000\r\n",
      "Scaffold_01 \t 72 \t 73 \t 2.777778\r\n",
      "Scaffold_01 \t 96 \t 97 \t 2.040816\r\n",
      "Scaffold_01 \t 119 \t 120 \t 0.000000\r\n",
      "Scaffold_01 \t 193 \t 194 \t 0.000000\r\n",
      "Scaffold_01 \t 202 \t 203 \t 0.000000\r\n",
      "Scaffold_01 \t 209 \t 210 \t 0.000000\r\n",
      "Scaffold_01 \t 213 \t 214 \t 0.000000\r\n",
      "Scaffold_01 \t 221 \t 222 \t 0.000000\r\n",
      "Scaffold_01 \t 238 \t 239 \t 1.315789\r\n",
      "awk: write error on /dev/stdout\r\n",
      " input record number 1533, file \r\n",
      " source line number 1\r\n",
      "sed: stdout: Broken pipe\r\n",
      "sed: stdout: Broken pipe\r\n",
      "sed: stdout: Broken pipe\r\n",
      "sed: stdout: Broken pipe\r\n",
      "sed: stdout: Broken pipe\r\n",
      "sed: stdout: Broken pipe\r\n",
      "sed: stdout: Broken pipe\r\n",
      "sed: stdout: Broken pipe\r\n",
      "sed: stdout: Broken pipe\r\n",
      "sed: stdout: Broken pipe\r\n",
      "sed: stdout: Broken pipe\r\n",
      "sed: stdout: Broken pipe\r\n",
      "sed: stdout: Broken pipe\r\n",
      "sed: stdout: Broken pipe\r\n",
      "sed: stdout: Broken pipe\r\n",
      "sed: stdout: Broken pipe\r\n",
      "sed: stdout: Broken pipe\r\n",
      "sed: stdout: Broken pipe\r\n",
      "cat: stdout: Broken pipe\r\n"
     ]
    }
   ],
   "source": [
    "!cat /Volumes/Serine/wd/19-11-05/PgALL.74.CpG_report.merged_CpG_evidence.cov \\\n",
    "| sed 's/PGA_scaffold10__49_contigs__length_53961475/Scaffold_10/g' \\\n",
    "| sed 's/PGA_scaffold11__79_contigs__length_51449921/Scaffold_11/g' \\\n",
    "| sed 's/PGA_scaffold12__71_contigs__length_50438331/Scaffold_12/g' \\\n",
    "| sed 's/PGA_scaffold13__52_contigs__length_44396874/Scaffold_13/g' \\\n",
    "| sed 's/PGA_scaffold14__91_contigs__length_45393038/Scaffold_14/g' \\\n",
    "| sed 's/PGA_scaffold15__101_contigs__length_47938513/Scaffold_15/g' \\\n",
    "| sed 's/PGA_scaffold16__33_contigs__length_31980953/Scaffold_16/g' \\\n",
    "| sed 's/PGA_scaffold17__51_contigs__length_34923512/Scaffold_17/g' \\\n",
    "| sed 's/PGA_scaffold18__69_contigs__length_27737463/Scaffold_18/g' \\\n",
    "| sed 's/PGA_scaffold1__77_contigs__length_89643857/Scaffold_01/g' \\\n",
    "| sed 's/PGA_scaffold2__36_contigs__length_69596280/Scaffold_02/g' \\\n",
    "| sed 's/PGA_scaffold3__111_contigs__length_57743597/Scaffold_03/g' \\\n",
    "| sed 's/PGA_scaffold4__129_contigs__length_65288255/Scaffold_04/g' \\\n",
    "| sed 's/PGA_scaffold5__109_contigs__length_67248332/Scaffold_05/g' \\\n",
    "| sed 's/PGA_scaffold6__104_contigs__length_61759565/Scaffold_06/g' \\\n",
    "| sed 's/PGA_scaffold7__69_contigs__length_43120122/Scaffold_07/g' \\\n",
    "| sed 's/PGA_scaffold8__63_contigs__length_61151155/Scaffold_08/g' \\\n",
    "| sed 's/PGA_scaffold9__45_contigs__length_38581958/Scaffold_09/g' \\\n",
    "| awk '{if ($5+$6 >= 10) { print $1, \"\\t\", $2, \"\\t\", $3, \"\\t\", $4}}' | head"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 33,
   "metadata": {
    "collapsed": true,
    "deletable": true,
    "editable": true
   },
   "outputs": [],
   "source": [
    "!cat /Volumes/Serine/wd/19-11-05/PgALL.74.CpG_report.merged_CpG_evidence.cov \\\n",
    "| sed 's/PGA_scaffold10__49_contigs__length_53961475/Scaffold_10/g' \\\n",
    "| sed 's/PGA_scaffold11__79_contigs__length_51449921/Scaffold_11/g' \\\n",
    "| sed 's/PGA_scaffold12__71_contigs__length_50438331/Scaffold_12/g' \\\n",
    "| sed 's/PGA_scaffold13__52_contigs__length_44396874/Scaffold_13/g' \\\n",
    "| sed 's/PGA_scaffold14__91_contigs__length_45393038/Scaffold_14/g' \\\n",
    "| sed 's/PGA_scaffold15__101_contigs__length_47938513/Scaffold_15/g' \\\n",
    "| sed 's/PGA_scaffold16__33_contigs__length_31980953/Scaffold_16/g' \\\n",
    "| sed 's/PGA_scaffold17__51_contigs__length_34923512/Scaffold_17/g' \\\n",
    "| sed 's/PGA_scaffold18__69_contigs__length_27737463/Scaffold_18/g' \\\n",
    "| sed 's/PGA_scaffold1__77_contigs__length_89643857/Scaffold_01/g' \\\n",
    "| sed 's/PGA_scaffold2__36_contigs__length_69596280/Scaffold_02/g' \\\n",
    "| sed 's/PGA_scaffold3__111_contigs__length_57743597/Scaffold_03/g' \\\n",
    "| sed 's/PGA_scaffold4__129_contigs__length_65288255/Scaffold_04/g' \\\n",
    "| sed 's/PGA_scaffold5__109_contigs__length_67248332/Scaffold_05/g' \\\n",
    "| sed 's/PGA_scaffold6__104_contigs__length_61759565/Scaffold_06/g' \\\n",
    "| sed 's/PGA_scaffold7__69_contigs__length_43120122/Scaffold_07/g' \\\n",
    "| sed 's/PGA_scaffold8__63_contigs__length_61151155/Scaffold_08/g' \\\n",
    "| sed 's/PGA_scaffold9__45_contigs__length_38581958/Scaffold_09/g' \\\n",
    "| awk '{if ($5+$6 >= 10) { print $1, \"\\t\", $2, \"\\t\", $3, \"\\t\", $4}}' \\\n",
    "> analyses/PgALL.bed"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2019-11-13T15:25:21.360448Z",
     "start_time": "2019-11-13T15:25:21.198470Z"
    },
    "collapsed": false,
    "deletable": true,
    "editable": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "0225_igv_session.xml\r\n",
      "0817_GO-annot.tab\r\n",
      "0821_GO-annot.tab\r\n",
      "1023_GO-annot.tab\r\n",
      "50p_intersect_exon_u.txt\r\n",
      "50p_intersect_intron_u.txt\r\n",
      "50p_intersect_rm_u.txt\r\n",
      "5xbedgraph.zip\r\n",
      "EPI-103_S27_L005_R1_001_val_1_bismark_bt2_pe.deduplicated.bismark.cov\r\n",
      "EPI-103_S27_L005_R1_001_val_1_bismark_bt2_pe.deduplicated.bismark.cov.destrand.10x.bedgraph\r\n",
      "EPI-103_S27_L005_R1_001_val_1_bismark_bt2_pe.deduplicated.bismark.cov.destrand.5x.bedgraph\r\n",
      "EPI-103_S27_L005_R1_001_val_1_bismark_bt2_pe.deduplicated.bismark.cov_10x.bedgraph\r\n",
      "EPI-104_S28_L005_R1_001_val_1_bismark_bt2_pe.deduplicated.bismark.cov\r\n",
      "EPI-104_S28_L005_R1_001_val_1_bismark_bt2_pe.deduplicated.bismark.cov.destrand.10x.bedgraph\r\n",
      "EPI-104_S28_L005_R1_001_val_1_bismark_bt2_pe.deduplicated.bismark.cov.destrand.5x.bedgraph\r\n",
      "EPI-104_S28_L005_R1_001_val_1_bismark_bt2_pe.deduplicated.bismark.cov_10x.bedgraph\r\n",
      "EPI-111_S29_L005_R1_001_val_1_bismark_bt2_pe.deduplicated.bismark.cov\r\n",
      "EPI-111_S29_L005_R1_001_val_1_bismark_bt2_pe.deduplicated.bismark.cov.destrand.10x.bedgraph\r\n",
      "EPI-111_S29_L005_R1_001_val_1_bismark_bt2_pe.deduplicated.bismark.cov.destrand.5x.bedgraph\r\n",
      "EPI-111_S29_L005_R1_001_val_1_bismark_bt2_pe.deduplicated.bismark.cov_10x.bedgraph\r\n",
      "EPI-113_S30_L005_R1_001_val_1_bismark_bt2_pe.deduplicated.bismark.cov\r\n",
      "EPI-113_S30_L005_R1_001_val_1_bismark_bt2_pe.deduplicated.bismark.cov.destrand.10x.bedgraph\r\n",
      "EPI-113_S30_L005_R1_001_val_1_bismark_bt2_pe.deduplicated.bismark.cov.destrand.5x.bedgraph\r\n",
      "EPI-113_S30_L005_R1_001_val_1_bismark_bt2_pe.deduplicated.bismark.cov_10x.bedgraph\r\n",
      "EPI-119_S31_L005_R1_001_val_1_bismark_bt2_pe.deduplicated.bismark.cov\r\n",
      "EPI-119_S31_L005_R1_001_val_1_bismark_bt2_pe.deduplicated.bismark.cov.destrand.10x.bedgraph\r\n",
      "EPI-119_S31_L005_R1_001_val_1_bismark_bt2_pe.deduplicated.bismark.cov.destrand.5x.bedgraph\r\n",
      "EPI-119_S31_L005_R1_001_val_1_bismark_bt2_pe.deduplicated.bismark.cov_10x.bedgraph\r\n",
      "EPI-120_S32_L005_R1_001_val_1_bismark_bt2_pe.deduplicated.bismark.cov\r\n",
      "EPI-120_S32_L005_R1_001_val_1_bismark_bt2_pe.deduplicated.bismark.cov.destrand.10x.bedgraph\r\n",
      "EPI-120_S32_L005_R1_001_val_1_bismark_bt2_pe.deduplicated.bismark.cov.destrand.5x.bedgraph\r\n",
      "EPI-120_S32_L005_R1_001_val_1_bismark_bt2_pe.deduplicated.bismark.cov_10x.bedgraph\r\n",
      "EPI-127_S33_L005_R1_001_val_1_bismark_bt2_pe.deduplicated.bismark.cov\r\n",
      "EPI-127_S33_L005_R1_001_val_1_bismark_bt2_pe.deduplicated.bismark.cov.destrand.10x.bedgraph\r\n",
      "EPI-127_S33_L005_R1_001_val_1_bismark_bt2_pe.deduplicated.bismark.cov.destrand.5x.bedgraph\r\n",
      "EPI-127_S33_L005_R1_001_val_1_bismark_bt2_pe.deduplicated.bismark.cov_10x.bedgraph\r\n",
      "EPI-128_S34_L005_R1_001_val_1_bismark_bt2_pe.deduplicated.bismark.cov\r\n",
      "EPI-128_S34_L005_R1_001_val_1_bismark_bt2_pe.deduplicated.bismark.cov.destrand.10x.bedgraph\r\n",
      "EPI-128_S34_L005_R1_001_val_1_bismark_bt2_pe.deduplicated.bismark.cov.destrand.5x.bedgraph\r\n",
      "EPI-128_S34_L005_R1_001_val_1_bismark_bt2_pe.deduplicated.bismark.cov_10x.bedgraph\r\n",
      "EPI-135_S35_L005_R1_001_val_1_bismark_bt2_pe.deduplicated.bismark.cov\r\n",
      "EPI-135_S35_L005_R1_001_val_1_bismark_bt2_pe.deduplicated.bismark.cov.destrand.10x.bedgraph\r\n",
      "EPI-135_S35_L005_R1_001_val_1_bismark_bt2_pe.deduplicated.bismark.cov.destrand.5x.bedgraph\r\n",
      "EPI-135_S35_L005_R1_001_val_1_bismark_bt2_pe.deduplicated.bismark.cov_10x.bedgraph\r\n",
      "EPI-136_S36_L005_R1_001_val_1_bismark_bt2_pe.deduplicated.bismark.cov\r\n",
      "EPI-136_S36_L005_R1_001_val_1_bismark_bt2_pe.deduplicated.bismark.cov.destrand.10x.bedgraph\r\n",
      "EPI-136_S36_L005_R1_001_val_1_bismark_bt2_pe.deduplicated.bismark.cov.destrand.5x.bedgraph\r\n",
      "EPI-136_S36_L005_R1_001_val_1_bismark_bt2_pe.deduplicated.bismark.cov_10x.bedgraph\r\n",
      "EPI-143_S37_L005_R1_001_val_1_bismark_bt2_pe.deduplicated.bismark.cov\r\n",
      "EPI-143_S37_L005_R1_001_val_1_bismark_bt2_pe.deduplicated.bismark.cov.destrand.10x.bedgraph\r\n",
      "EPI-143_S37_L005_R1_001_val_1_bismark_bt2_pe.deduplicated.bismark.cov.destrand.5x.bedgraph\r\n",
      "EPI-143_S37_L005_R1_001_val_1_bismark_bt2_pe.deduplicated.bismark.cov_10x.bedgraph\r\n",
      "EPI-145_S38_L005_R1_001_val_1_bismark_bt2_pe.deduplicated.bismark.cov\r\n",
      "EPI-145_S38_L005_R1_001_val_1_bismark_bt2_pe.deduplicated.bismark.cov.destrand\r\n",
      "EPI-145_S38_L005_R1_001_val_1_bismark_bt2_pe.deduplicated.bismark.cov.destrand.10x.bedgraph\r\n",
      "EPI-145_S38_L005_R1_001_val_1_bismark_bt2_pe.deduplicated.bismark.cov.destrand.5x.bedgraph\r\n",
      "EPI-151_S2_L002_R1_001_val_1_bismark_bt2_pe.deduplicated.bismark.cov\r\n",
      "EPI-151_S2_L002_R1_001_val_1_bismark_bt2_pe.deduplicated.bismark.cov.destrand.10x.bedgraph\r\n",
      "EPI-151_S2_L002_R1_001_val_1_bismark_bt2_pe.deduplicated.bismark.cov.destrand.5x.bedgraph\r\n",
      "EPI-151_S2_L002_R1_001_val_1_bismark_bt2_pe.deduplicated.bismark.cov_10x.bedgraph\r\n",
      "EPI-152_S3_L002_R1_001_val_1_bismark_bt2_pe.deduplicated.bismark.cov\r\n",
      "EPI-152_S3_L002_R1_001_val_1_bismark_bt2_pe.deduplicated.bismark.cov.destrand.10x.bedgraph\r\n",
      "EPI-152_S3_L002_R1_001_val_1_bismark_bt2_pe.deduplicated.bismark.cov.destrand.5x.bedgraph\r\n",
      "EPI-152_S3_L002_R1_001_val_1_bismark_bt2_pe.deduplicated.bismark.cov_10x.bedgraph\r\n",
      "EPI-153_S4_L002_R1_001_val_1_bismark_bt2_pe.deduplicated.bismark.cov\r\n",
      "EPI-153_S4_L002_R1_001_val_1_bismark_bt2_pe.deduplicated.bismark.cov.destrand.10x.bedgraph\r\n",
      "EPI-153_S4_L002_R1_001_val_1_bismark_bt2_pe.deduplicated.bismark.cov.destrand.5x.bedgraph\r\n",
      "EPI-153_S4_L002_R1_001_val_1_bismark_bt2_pe.deduplicated.bismark.cov_10x.bedgraph\r\n",
      "EPI-154_S5_L002_R1_001_val_1_bismark_bt2_pe.deduplicated.bismark.cov\r\n",
      "EPI-154_S5_L002_R1_001_val_1_bismark_bt2_pe.deduplicated.bismark.cov.destrand.10x.bedgraph\r\n",
      "EPI-154_S5_L002_R1_001_val_1_bismark_bt2_pe.deduplicated.bismark.cov.destrand.5x.bedgraph\r\n",
      "EPI-154_S5_L002_R1_001_val_1_bismark_bt2_pe.deduplicated.bismark.cov_10x.bedgraph\r\n",
      "EPI-159_S6_L002_R1_001_val_1_bismark_bt2_pe.deduplicated.bismark.cov\r\n",
      "EPI-159_S6_L002_R1_001_val_1_bismark_bt2_pe.deduplicated.bismark.cov.destrand.10x.bedgraph\r\n",
      "EPI-159_S6_L002_R1_001_val_1_bismark_bt2_pe.deduplicated.bismark.cov.destrand.5x.bedgraph\r\n",
      "EPI-159_S6_L002_R1_001_val_1_bismark_bt2_pe.deduplicated.bismark.cov_10x.bedgraph\r\n",
      "EPI-160_S7_L002_R1_001_val_1_bismark_bt2_pe.deduplicated.bismark.cov\r\n",
      "EPI-160_S7_L002_R1_001_val_1_bismark_bt2_pe.deduplicated.bismark.cov.destrand.10x.bedgraph\r\n",
      "EPI-160_S7_L002_R1_001_val_1_bismark_bt2_pe.deduplicated.bismark.cov.destrand.5x.bedgraph\r\n",
      "EPI-160_S7_L002_R1_001_val_1_bismark_bt2_pe.deduplicated.bismark.cov_10x.bedgraph\r\n",
      "EPI-161_S8_L002_R1_001_val_1_bismark_bt2_pe.deduplicated.bismark.cov\r\n",
      "EPI-161_S8_L002_R1_001_val_1_bismark_bt2_pe.deduplicated.bismark.cov.destrand.10x.bedgraph\r\n",
      "EPI-161_S8_L002_R1_001_val_1_bismark_bt2_pe.deduplicated.bismark.cov.destrand.5x.bedgraph\r\n",
      "EPI-161_S8_L002_R1_001_val_1_bismark_bt2_pe.deduplicated.bismark.cov_10x.bedgraph\r\n",
      "EPI-162_S9_L002_R1_001_val_1_bismark_bt2_pe.deduplicated.bismark.cov\r\n",
      "EPI-162_S9_L002_R1_001_val_1_bismark_bt2_pe.deduplicated.bismark.cov.destrand.10x.bedgraph\r\n",
      "EPI-162_S9_L002_R1_001_val_1_bismark_bt2_pe.deduplicated.bismark.cov.destrand.5x.bedgraph\r\n",
      "EPI-162_S9_L002_R1_001_val_1_bismark_bt2_pe.deduplicated.bismark.cov_10x.bedgraph\r\n",
      "EPI-167_S10_L002_R1_001_val_1_bismark_bt2_pe.deduplicated.bismark.cov\r\n",
      "EPI-167_S10_L002_R1_001_val_1_bismark_bt2_pe.deduplicated.bismark.cov.destrand.10x.bedgraph\r\n",
      "EPI-167_S10_L002_R1_001_val_1_bismark_bt2_pe.deduplicated.bismark.cov.destrand.5x.bedgraph\r\n",
      "EPI-167_S10_L002_R1_001_val_1_bismark_bt2_pe.deduplicated.bismark.cov_10x.bedgraph\r\n",
      "EPI-168_S11_L002_R1_001_val_1_bismark_bt2_pe.deduplicated.bismark.cov\r\n",
      "EPI-168_S11_L002_R1_001_val_1_bismark_bt2_pe.deduplicated.bismark.cov.destrand.10x.bedgraph\r\n",
      "EPI-168_S11_L002_R1_001_val_1_bismark_bt2_pe.deduplicated.bismark.cov.destrand.5x.bedgraph\r\n",
      "EPI-168_S11_L002_R1_001_val_1_bismark_bt2_pe.deduplicated.bismark.cov_10x.bedgraph\r\n",
      "EPI-169_S12_L002_R1_001_val_1_bismark_bt2_pe.deduplicated.bismark.cov\r\n",
      "EPI-169_S12_L002_R1_001_val_1_bismark_bt2_pe.deduplicated.bismark.cov.destrand.10x.bedgraph\r\n",
      "EPI-169_S12_L002_R1_001_val_1_bismark_bt2_pe.deduplicated.bismark.cov.destrand.5x.bedgraph\r\n",
      "EPI-169_S12_L002_R1_001_val_1_bismark_bt2_pe.deduplicated.bismark.cov_10x.bedgraph\r\n",
      "EPI-170_S13_L002_R1_001_val_1_bismark_bt2_pe.deduplicated.bismark.cov\r\n",
      "EPI-170_S13_L002_R1_001_val_1_bismark_bt2_pe.deduplicated.bismark.cov.destrand.10x.bedgraph\r\n",
      "EPI-170_S13_L002_R1_001_val_1_bismark_bt2_pe.deduplicated.bismark.cov.destrand.5x.bedgraph\r\n",
      "EPI-170_S13_L002_R1_001_val_1_bismark_bt2_pe.deduplicated.bismark.cov_10x.bedgraph\r\n",
      "EPI-175_S14_L003_R1_001_val_1_bismark_bt2_pe.deduplicated.bismark.cov\r\n",
      "EPI-175_S14_L003_R1_001_val_1_bismark_bt2_pe.deduplicated.bismark.cov.destrand.10x.bedgraph\r\n",
      "EPI-175_S14_L003_R1_001_val_1_bismark_bt2_pe.deduplicated.bismark.cov.destrand.5x.bedgraph\r\n",
      "EPI-175_S14_L003_R1_001_val_1_bismark_bt2_pe.deduplicated.bismark.cov_10x.bedgraph\r\n",
      "EPI-176_S15_L003_R1_001_val_1_bismark_bt2_pe.deduplicated.bismark.cov\r\n",
      "EPI-176_S15_L003_R1_001_val_1_bismark_bt2_pe.deduplicated.bismark.cov.destrand.10x.bedgraph\r\n",
      "EPI-176_S15_L003_R1_001_val_1_bismark_bt2_pe.deduplicated.bismark.cov.destrand.5x.bedgraph\r\n",
      "EPI-176_S15_L003_R1_001_val_1_bismark_bt2_pe.deduplicated.bismark.cov_10x.bedgraph\r\n",
      "EPI-181_S16_L003_R1_001_val_1_bismark_bt2_pe.deduplicated.bismark.cov\r\n",
      "EPI-181_S16_L003_R1_001_val_1_bismark_bt2_pe.deduplicated.bismark.cov.destrand.10x.bedgraph\r\n",
      "EPI-181_S16_L003_R1_001_val_1_bismark_bt2_pe.deduplicated.bismark.cov.destrand.5x.bedgraph\r\n",
      "EPI-181_S16_L003_R1_001_val_1_bismark_bt2_pe.deduplicated.bismark.cov_10x.bedgraph\r\n",
      "EPI-182_S17_L003_R1_001_val_1_bismark_bt2_pe.deduplicated.bismark.cov\r\n",
      "EPI-182_S17_L003_R1_001_val_1_bismark_bt2_pe.deduplicated.bismark.cov.destrand.10x.bedgraph\r\n",
      "EPI-182_S17_L003_R1_001_val_1_bismark_bt2_pe.deduplicated.bismark.cov.destrand.5x.bedgraph\r\n",
      "EPI-182_S17_L003_R1_001_val_1_bismark_bt2_pe.deduplicated.bismark.cov_10x.bedgraph\r\n",
      "EPI-184_S18_L003_R1_001_val_1_bismark_bt2_pe.deduplicated.bismark.cov\r\n",
      "EPI-184_S18_L003_R1_001_val_1_bismark_bt2_pe.deduplicated.bismark.cov.destrand.10x.bedgraph\r\n",
      "EPI-184_S18_L003_R1_001_val_1_bismark_bt2_pe.deduplicated.bismark.cov.destrand.5x.bedgraph\r\n",
      "EPI-184_S18_L003_R1_001_val_1_bismark_bt2_pe.deduplicated.bismark.cov_10x.bedgraph\r\n",
      "EPI-185_S19_L003_R1_001_val_1_bismark_bt2_pe.deduplicated.bismark.cov\r\n",
      "EPI-185_S19_L003_R1_001_val_1_bismark_bt2_pe.deduplicated.bismark.cov.destrand.10x.bedgraph\r\n",
      "EPI-185_S19_L003_R1_001_val_1_bismark_bt2_pe.deduplicated.bismark.cov.destrand.5x.bedgraph\r\n",
      "EPI-185_S19_L003_R1_001_val_1_bismark_bt2_pe.deduplicated.bismark.cov_10x.bedgraph\r\n",
      "EPI-187_S20_L003_R1_001_val_1_bismark_bt2_pe.deduplicated.bismark.cov\r\n",
      "EPI-187_S20_L003_R1_001_val_1_bismark_bt2_pe.deduplicated.bismark.cov.destrand.10x.bedgraph\r\n",
      "EPI-187_S20_L003_R1_001_val_1_bismark_bt2_pe.deduplicated.bismark.cov.destrand.5x.bedgraph\r\n",
      "EPI-187_S20_L003_R1_001_val_1_bismark_bt2_pe.deduplicated.bismark.cov_10x.bedgraph\r\n",
      "EPI-188_S21_L003_R1_001_val_1_bismark_bt2_pe.deduplicated.bismark.cov\r\n",
      "EPI-188_S21_L003_R1_001_val_1_bismark_bt2_pe.deduplicated.bismark.cov.destrand.10x.bedgraph\r\n",
      "EPI-188_S21_L003_R1_001_val_1_bismark_bt2_pe.deduplicated.bismark.cov.destrand.5x.bedgraph\r\n",
      "EPI-188_S21_L003_R1_001_val_1_bismark_bt2_pe.deduplicated.bismark.cov_10x.bedgraph\r\n",
      "EPI-193_S22_L003_R1_001_val_1_bismark_bt2_pe.deduplicated.bismark.cov\r\n",
      "EPI-193_S22_L003_R1_001_val_1_bismark_bt2_pe.deduplicated.bismark.cov.destrand.10x.bedgraph\r\n",
      "EPI-193_S22_L003_R1_001_val_1_bismark_bt2_pe.deduplicated.bismark.cov.destrand.5x.bedgraph\r\n",
      "EPI-193_S22_L003_R1_001_val_1_bismark_bt2_pe.deduplicated.bismark.cov_10x.bedgraph\r\n",
      "EPI-194_S23_L003_R1_001_val_1_bismark_bt2_pe.deduplicated.bismark.cov\r\n",
      "EPI-194_S23_L003_R1_001_val_1_bismark_bt2_pe.deduplicated.bismark.cov.destrand.10x.bedgraph\r\n",
      "EPI-194_S23_L003_R1_001_val_1_bismark_bt2_pe.deduplicated.bismark.cov.destrand.5x.bedgraph\r\n",
      "EPI-194_S23_L003_R1_001_val_1_bismark_bt2_pe.deduplicated.bismark.cov_10x.bedgraph\r\n",
      "EPI-199_S24_L003_R1_001_val_1_bismark_bt2_pe.deduplicated.bismark.cov\r\n",
      "EPI-199_S24_L003_R1_001_val_1_bismark_bt2_pe.deduplicated.bismark.cov.destrand.10x.bedgraph\r\n",
      "EPI-199_S24_L003_R1_001_val_1_bismark_bt2_pe.deduplicated.bismark.cov.destrand.5x.bedgraph\r\n",
      "EPI-199_S24_L003_R1_001_val_1_bismark_bt2_pe.deduplicated.bismark.cov_10x.bedgraph\r\n",
      "EPI-200_S25_L003_R1_001_val_1_bismark_bt2_pe.deduplicated.bismark.cov\r\n",
      "EPI-200_S25_L003_R1_001_val_1_bismark_bt2_pe.deduplicated.bismark.cov.destrand.10x.bedgraph\r\n",
      "EPI-200_S25_L003_R1_001_val_1_bismark_bt2_pe.deduplicated.bismark.cov.destrand.5x.bedgraph\r\n",
      "EPI-200_S25_L003_R1_001_val_1_bismark_bt2_pe.deduplicated.bismark.cov_10x.bedgraph\r\n",
      "EPI-205_S26_L004_R1_001_val_1_bismark_bt2_pe.deduplicated.bismark.cov\r\n",
      "EPI-205_S26_L004_R1_001_val_1_bismark_bt2_pe.deduplicated.bismark.cov.destrand.10x.bedgraph\r\n",
      "EPI-205_S26_L004_R1_001_val_1_bismark_bt2_pe.deduplicated.bismark.cov.destrand.5x.bedgraph\r\n",
      "EPI-205_S26_L004_R1_001_val_1_bismark_bt2_pe.deduplicated.bismark.cov_10x.bedgraph\r\n",
      "EPI-206_S27_L004_R1_001_val_1_bismark_bt2_pe.deduplicated.bismark.cov\r\n",
      "EPI-206_S27_L004_R1_001_val_1_bismark_bt2_pe.deduplicated.bismark.cov.destrand.10x.bedgraph\r\n",
      "EPI-206_S27_L004_R1_001_val_1_bismark_bt2_pe.deduplicated.bismark.cov.destrand.5x.bedgraph\r\n",
      "EPI-206_S27_L004_R1_001_val_1_bismark_bt2_pe.deduplicated.bismark.cov_10x.bedgraph\r\n",
      "EPI-208_S28_L004_R1_001_val_1_bismark_bt2_pe.deduplicated.bismark.cov\r\n",
      "EPI-208_S28_L004_R1_001_val_1_bismark_bt2_pe.deduplicated.bismark.cov.destrand.10x.bedgraph\r\n",
      "EPI-208_S28_L004_R1_001_val_1_bismark_bt2_pe.deduplicated.bismark.cov.destrand.5x.bedgraph\r",
      "\r\n",
      "EPI-208_S28_L004_R1_001_val_1_bismark_bt2_pe.deduplicated.bismark.cov_10x.bedgraph\r\n",
      "EPI-209_S29_L004_R1_001_val_1_bismark_bt2_pe.deduplicated.bismark.cov\r\n",
      "EPI-209_S29_L004_R1_001_val_1_bismark_bt2_pe.deduplicated.bismark.cov.destrand.10x.bedgraph\r\n",
      "EPI-209_S29_L004_R1_001_val_1_bismark_bt2_pe.deduplicated.bismark.cov.destrand.5x.bedgraph\r\n",
      "EPI-209_S29_L004_R1_001_val_1_bismark_bt2_pe.deduplicated.bismark.cov_10x.bedgraph\r\n",
      "EPI-214_S30_L004_R1_001_val_1_bismark_bt2_pe.deduplicated.bismark.cov\r\n",
      "EPI-214_S30_L004_R1_001_val_1_bismark_bt2_pe.deduplicated.bismark.cov.destrand.10x.bedgraph\r\n",
      "EPI-214_S30_L004_R1_001_val_1_bismark_bt2_pe.deduplicated.bismark.cov.destrand.5x.bedgraph\r\n",
      "EPI-214_S30_L004_R1_001_val_1_bismark_bt2_pe.deduplicated.bismark.cov_10x.bedgraph\r\n",
      "EPI-215_S31_L004_R1_001_val_1_bismark_bt2_pe.deduplicated.bismark.cov\r\n",
      "EPI-215_S31_L004_R1_001_val_1_bismark_bt2_pe.deduplicated.bismark.cov.destrand.10x.bedgraph\r\n",
      "EPI-215_S31_L004_R1_001_val_1_bismark_bt2_pe.deduplicated.bismark.cov.destrand.5x.bedgraph\r\n",
      "EPI-215_S31_L004_R1_001_val_1_bismark_bt2_pe.deduplicated.bismark.cov_10x.bedgraph\r\n",
      "EPI-220_S32_L004_R1_001_val_1_bismark_bt2_pe.deduplicated.bismark.cov\r\n",
      "EPI-220_S32_L004_R1_001_val_1_bismark_bt2_pe.deduplicated.bismark.cov.destrand.10x.bedgraph\r\n",
      "EPI-220_S32_L004_R1_001_val_1_bismark_bt2_pe.deduplicated.bismark.cov.destrand.5x.bedgraph\r\n",
      "EPI-220_S32_L004_R1_001_val_1_bismark_bt2_pe.deduplicated.bismark.cov_10x.bedgraph\r\n",
      "EPI-221_S33_L004_R1_001_val_1_bismark_bt2_pe.deduplicated.bismark.cov\r\n",
      "EPI-221_S33_L004_R1_001_val_1_bismark_bt2_pe.deduplicated.bismark.cov.destrand.10x.bedgraph\r\n",
      "EPI-221_S33_L004_R1_001_val_1_bismark_bt2_pe.deduplicated.bismark.cov.destrand.5x.bedgraph\r\n",
      "EPI-221_S33_L004_R1_001_val_1_bismark_bt2_pe.deduplicated.bismark.cov_10x.bedgraph\r\n",
      "EPI-226_S34_L004_R1_001_val_1_bismark_bt2_pe.deduplicated.bismark.cov\r\n",
      "EPI-226_S34_L004_R1_001_val_1_bismark_bt2_pe.deduplicated.bismark.cov.destrand.10x.bedgraph\r\n",
      "EPI-226_S34_L004_R1_001_val_1_bismark_bt2_pe.deduplicated.bismark.cov.destrand.5x.bedgraph\r\n",
      "EPI-226_S34_L004_R1_001_val_1_bismark_bt2_pe.deduplicated.bismark.cov_10x.bedgraph\r\n",
      "EPI-227_S35_L004_R1_001_val_1_bismark_bt2_pe.deduplicated.bismark.cov\r\n",
      "EPI-227_S35_L004_R1_001_val_1_bismark_bt2_pe.deduplicated.bismark.cov.destrand.10x.bedgraph\r\n",
      "EPI-227_S35_L004_R1_001_val_1_bismark_bt2_pe.deduplicated.bismark.cov.destrand.5x.bedgraph\r\n",
      "EPI-227_S35_L004_R1_001_val_1_bismark_bt2_pe.deduplicated.bismark.cov_10x.bedgraph\r\n",
      "EPI-229_S36_L004_R1_001_val_1_bismark_bt2_pe.deduplicated.bismark.cov\r\n",
      "EPI-229_S36_L004_R1_001_val_1_bismark_bt2_pe.deduplicated.bismark.cov.destrand.10x.bedgraph\r\n",
      "EPI-229_S36_L004_R1_001_val_1_bismark_bt2_pe.deduplicated.bismark.cov.destrand.5x.bedgraph\r\n",
      "EPI-229_S36_L004_R1_001_val_1_bismark_bt2_pe.deduplicated.bismark.cov_10x.bedgraph\r\n",
      "EPI-230_S37_L004_R1_001_val_1_bismark_bt2_pe.deduplicated.bismark.cov\r\n",
      "EPI-230_S37_L004_R1_001_val_1_bismark_bt2_pe.deduplicated.bismark.cov.destrand.10x.bedgraph\r\n",
      "EPI-230_S37_L004_R1_001_val_1_bismark_bt2_pe.deduplicated.bismark.cov.destrand.5x.bedgraph\r\n",
      "EPI-230_S37_L004_R1_001_val_1_bismark_bt2_pe.deduplicated.bismark.cov_10x.bedgraph\r\n",
      "EPI-41_S38_L005_R1_001_val_1_bismark_bt2_pe.deduplicated.bismark.cov\r\n",
      "EPI-41_S38_L005_R1_001_val_1_bismark_bt2_pe.deduplicated.bismark.cov.destrand.10x.bedgraph\r\n",
      "EPI-41_S38_L005_R1_001_val_1_bismark_bt2_pe.deduplicated.bismark.cov.destrand.5x.bedgraph\r\n",
      "EPI-41_S38_L005_R1_001_val_1_bismark_bt2_pe.deduplicated.bismark.cov_10x.bedgraph\r\n",
      "EPI-42_S39_L005_R1_001_val_1_bismark_bt2_pe.deduplicated.bismark.cov\r\n",
      "EPI-42_S39_L005_R1_001_val_1_bismark_bt2_pe.deduplicated.bismark.cov.destrand.10x.bedgraph\r\n",
      "EPI-42_S39_L005_R1_001_val_1_bismark_bt2_pe.deduplicated.bismark.cov.destrand.5x.bedgraph\r\n",
      "EPI-42_S39_L005_R1_001_val_1_bismark_bt2_pe.deduplicated.bismark.cov_10x.bedgraph\r\n",
      "EPI-43_S40_L005_R1_001_val_1_bismark_bt2_pe.deduplicated.bismark.cov\r\n",
      "EPI-43_S40_L005_R1_001_val_1_bismark_bt2_pe.deduplicated.bismark.cov.destrand.10x.bedgraph\r\n",
      "EPI-43_S40_L005_R1_001_val_1_bismark_bt2_pe.deduplicated.bismark.cov.destrand.5x.bedgraph\r\n",
      "EPI-43_S40_L005_R1_001_val_1_bismark_bt2_pe.deduplicated.bismark.cov_10x.bedgraph\r\n",
      "EPI-44.2.destrand.bedgraph\r\n",
      "EPI-44.3.destrand.bedgraph\r\n",
      "EPI-44.4.destrand.bedgraph\r\n",
      "EPI-44.5.destrand.bedgraph\r\n",
      "EPI-44.destrand\r\n",
      "EPI-44.destrand.bedgraph\r\n",
      "EPI-44_S41_L005_R1_001_val_1_bismark_bt2_pe.deduplicated.bismark.cov\r\n",
      "EPI-44_S41_L005_R1_001_val_1_bismark_bt2_pe.deduplicated.bismark.cov.destrand.10x.bedgraph\r\n",
      "EPI-44_S41_L005_R1_001_val_1_bismark_bt2_pe.deduplicated.bismark.cov.destrand.5x.bedgraph\r\n",
      "EPI-44_S41_L005_R1_001_val_1_bismark_bt2_pe.deduplicated.bismark.cov_10x.bedgraph\r\n",
      "PGA_scaffold1_89643857.fa\r\n",
      "PGA_scaffold1__77_contigs__length_89643857.fa\r\n",
      "PGA_scaffold1_splits_2000.fasta\r\n",
      "Panopea-generosa-vv0.74.a3.exon.comp.bed\r\n",
      "Panopea-generosa-vv0.74.a3.exon.gff3.comp\r\n",
      "Pg-gensas-blastx.sorted\r\n",
      "Pg-gensas-blastx.tab\r\n",
      "Pg_genome.tab\r\n",
      "Pg_sc1-uniprot_blastx.tab\r\n",
      "Pg_val_destrand.10x.50p.bed\r\n",
      "Pg_val_destrand.10x.50p.bedgraph\r\n",
      "Pg_val_destrand.10x.bedgraph\r\n",
      "Pg_val_destrand.10x.bedgraph.tdf\r\n",
      "Pg_val_destrand.10x.sparse.bed\r\n",
      "Pg_val_destrand.10x.sparse.bedgraph\r\n",
      "Pg_val_destrand.10x.zero.bed\r\n",
      "Pg_val_destrand.10x.zero.bedgraph\r\n",
      "Pg_val_destrand.5x.50p.bed\r\n",
      "Pg_val_destrand.5x.50p.bed.idx\r\n",
      "Pg_val_destrand.5x.50p.bedgraph\r\n",
      "Pg_val_destrand.5x.bedgraph\r\n",
      "Pg_val_destrand.5x.sparse.bed\r\n",
      "Pg_val_destrand.5x.sparse.bedgraph\r\n",
      "Pg_val_destrand.5x.zero.bed\r\n",
      "Pg_val_destrand.5x.zero.bedgraph\r\n",
      "_test.1.bed\r\n",
      "_test.1.intron\r\n",
      "_test.1.tab\r\n",
      "_test.bedgraph\r\n",
      "_test.gene.1.bed\r\n",
      "_testout.merge.tab.CpG_report.merged_CpG_evidence.cov\r\n",
      "_testout.merge.tab.CpG_report.txt\r\n",
      "_testout.tab.CpG_report.txt\r\n",
      "_testv2.bedGraph\r\n",
      "blastp.outfmt6\r\n",
      "blastp_v074.outfmt6\r\n",
      "blastp_v074.sorted\r\n",
      "dml_d10.bed\r\n",
      "dmr.bed\r\n",
      "gene_blastp.col4\r\n",
      "gene_blastp.col5\r\n",
      "hp_blastp.col3\r\n",
      "hp_blastp.outfmt6\r\n",
      "hp_blastp.sorted\r\n",
      "hp_blastp_v074.col3\r\n",
      "hp_blastp_v074.outfmt6\r\n",
      "hp_blastp_v074.sorted\r\n",
      "myDiff50p_d10.tab\r\n",
      "myDiff75p_d10.tab\r\n",
      "myobj_145\r\n",
      "myobj_d10\r\n",
      "sparse_intersect_exon_u.txt\r\n",
      "sparse_intersect_intron_u.txt\r\n",
      "sparse_intersect_rm_u.txt\r\n",
      "\u001b[34mtest\u001b[m\u001b[m/\r\n",
      "testv3.bedGraph\r\n",
      "tmp.bed\r\n",
      "tmp_002_cov.txt\r\n",
      "uniprot-E123954914.sorted\r\n",
      "uniprot-yourlist_M2019.sorted\r\n",
      "uniprot-yourlist_M201908176746803381A1F0E0DB47453E0216320D908FE8J.tab\r\n",
      "uniprot-yourlist_M201908218471C63D39733769F8E060B506551E123954914.tab\r\n",
      "uniprot-yourlist_M201908218471C63D39733769F8E060B506551E123954914.tab.gz\r\n",
      "uniprot-yourlist_M20191023216DA2B77BFBD2E6699CA9B6D1C41EB24D82ED2.tab\r\n",
      "uniprot-yourlist_M2019_82ED2.sorted\r\n",
      "zero_intersect_exon_u.txt\r\n",
      "zero_intersect_intron_u.txt\r\n",
      "zero_intersect_rm_u.txt\r\n"
     ]
    }
   ],
   "source": [
    "ls analyses/"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2019-11-13T15:24:43.857110Z",
     "start_time": "2019-11-13T15:24:43.692760Z"
    },
    "collapsed": false,
    "deletable": true,
    "editable": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "ls: analyses/PgALL.bed: No such file or directory\r\n"
     ]
    }
   ],
   "source": [
    "!ls -l -h analyses/PgALL.bed"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "deletable": true,
    "editable": true
   },
   "source": [
    "https://d.pr/f/O5Dg5b"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 2",
   "language": "python",
   "name": "python2"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 2
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython2",
   "version": "2.7.12"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
